// @alfresco/adf-content-services/esm2015/lib/permission-manager/components/pop-over.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, Input, TemplateRef, ViewContainerRef } from '@angular/core';
import { ConnectionPositionPair, Overlay } from '@angular/cdk/overlay';
import { TemplatePortal } from '@angular/cdk/portal';
import { takeUntil } from 'rxjs/operators';
import { Subject } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/cdk/overlay';
export class PopOverDirective {
    constructor(element, overlay, vcr) {
        this.element = element;
        this.overlay = overlay;
        this.vcr = vcr;
        this.panelClass = 'adf-permission-pop-over';
        this._open = false;
        this.destroy$ = new Subject();
    }
    get open() {
        return this._open;
    }
    ngOnInit() {
        this.createOverlay();
    }
    ngAfterViewInit() {
        this.element.nativeElement.addEventListener('click', () => this.attachOverlay());
    }
    ngOnDestroy() {
        this.detachOverlay();
        this.destroy$.next();
        this.destroy$.complete();
    }
    createOverlay() {
        const scrollStrategy = this.overlay.scrollStrategies.reposition();
        const positionStrategy = this.overlay
            .position()
            .flexibleConnectedTo(this.target)
            .withPositions([
            new ConnectionPositionPair({ originX: 'start', originY: 'bottom' }, { overlayX: 'start', overlayY: 'top' }),
            new ConnectionPositionPair({ originX: 'start', originY: 'top' }, { overlayX: 'start', overlayY: 'bottom' })
        ])
            .withPush(false);
        this.overlayRef = this.overlay.create({
            positionStrategy,
            scrollStrategy,
            hasBackdrop: true,
            backdropClass: 'cdk-overlay-transparent-backdrop',
            panelClass: this.panelClass
        });
        this.overlayRef
            .backdropClick()
            .pipe(takeUntil(this.destroy$))
            .subscribe(() => {
            this._open = false;
            this.detachOverlay();
        });
    }
    attachOverlay() {
        if (!this.overlayRef.hasAttached()) {
            const periodSelectorPortal = new TemplatePortal(this.popOver, this.vcr);
            this.overlayRef.attach(periodSelectorPortal);
            this._open = true;
        }
    }
    detachOverlay() {
        if (this.overlayRef.hasAttached()) {
            this.overlayRef.detach();
        }
    }
}
PopOverDirective.ɵfac = function PopOverDirective_Factory(t) { return new (t || PopOverDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Overlay), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef)); };
PopOverDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: PopOverDirective, selectors: [["", "adf-pop-over", ""]], inputs: { panelClass: "panelClass", popOver: ["adf-pop-over", "popOver"], target: "target" }, exportAs: ["adfPopOver"] });
PopOverDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Overlay },
    { type: ViewContainerRef }
];
PopOverDirective.propDecorators = {
    popOver: [{ type: Input, args: ['adf-pop-over',] }],
    target: [{ type: Input }],
    panelClass: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PopOverDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-pop-over]',
                exportAs: 'adfPopOver'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc1.Overlay }, { type: ɵngcc0.ViewContainerRef }]; }, { panelClass: [{
            type: Input
        }], popOver: [{
            type: Input,
            args: ['adf-pop-over']
        }], target: [{
            type: Input
        }] }); })();
