// @alfresco/adf-core/esm2015/form/components/widgets/unknown/unknown.widget.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { WidgetComponent } from './../widget.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../../../services/form.service';
import * as ɵngcc2 from '@angular/material/list';
import * as ɵngcc3 from '@angular/material/icon';
export class UnknownWidgetComponent extends WidgetComponent {
    constructor(formService) {
        super(formService);
        this.formService = formService;
    }
}
UnknownWidgetComponent.ɵfac = function UnknownWidgetComponent_Factory(t) { return new (t || UnknownWidgetComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService)); };
UnknownWidgetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: UnknownWidgetComponent, selectors: [["unknown-widget"]], hostBindings: function UnknownWidgetComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function UnknownWidgetComponent_click_HostBindingHandler($event) { return ctx.event($event); })("blur", function UnknownWidgetComponent_blur_HostBindingHandler($event) { return ctx.event($event); })("change", function UnknownWidgetComponent_change_HostBindingHandler($event) { return ctx.event($event); })("focus", function UnknownWidgetComponent_focus_HostBindingHandler($event) { return ctx.event($event); })("focusin", function UnknownWidgetComponent_focusin_HostBindingHandler($event) { return ctx.event($event); })("focusout", function UnknownWidgetComponent_focusout_HostBindingHandler($event) { return ctx.event($event); })("input", function UnknownWidgetComponent_input_HostBindingHandler($event) { return ctx.event($event); })("invalid", function UnknownWidgetComponent_invalid_HostBindingHandler($event) { return ctx.event($event); })("select", function UnknownWidgetComponent_select_HostBindingHandler($event) { return ctx.event($event); });
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 6, vars: 1, consts: [[1, "adf-unknown-widget"], [1, "mat-24"], [1, "adf-unknown-text"]], template: function UnknownWidgetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "mat-list", 0);
        ɵngcc0.ɵɵelementStart(1, "mat-list-item");
        ɵngcc0.ɵɵelementStart(2, "mat-icon", 1);
        ɵngcc0.ɵɵtext(3, "error_outline");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "span", 2);
        ɵngcc0.ɵɵtext(5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(5);
        ɵngcc0.ɵɵtextInterpolate1("Unknown type: ", ctx.field.type, "");
    } }, directives: [ɵngcc2.MatList, ɵngcc2.MatListItem, ɵngcc3.MatIcon], styles: [".adf-unknown-text{color:red;margin-left:10px}.adf-unknown-widget{margin:42px}"], encapsulation: 2 });
UnknownWidgetComponent.ctorParameters = () => [
    { type: FormService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UnknownWidgetComponent, [{
        type: Component,
        args: [{
                selector: 'unknown-widget',
                template: `
            <mat-list class="adf-unknown-widget">
                <mat-list-item>
                     <mat-icon class="mat-24">error_outline</mat-icon>
                     <span class="adf-unknown-text">Unknown type: {{field.type}}</span>
                </mat-list-item>
            </mat-list>

    `,
                host: {
                    '(click)': 'event($event)',
                    '(blur)': 'event($event)',
                    '(change)': 'event($event)',
                    '(focus)': 'event($event)',
                    '(focusin)': 'event($event)',
                    '(focusout)': 'event($event)',
                    '(input)': 'event($event)',
                    '(invalid)': 'event($event)',
                    '(select)': 'event($event)'
                },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-unknown-text{color:red;margin-left:10px}.adf-unknown-widget{margin:42px}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormService }]; }, null); })();
