// @alfresco/adf-core/esm2015/mock/cookie.service.mock.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { CookieService } from '../services/cookie.service';
import * as ɵngcc0 from '@angular/core';
export class CookieServiceMock extends CookieService {
    isEnabled() {
        return true;
    }
    getItem(key) {
        return this[key] && this[key].data || null;
    }
    setItem(key, data, expiration, path) {
        this[key] = { data, expiration, path };
    }
    clear() {
        Object.keys(this).forEach((key) => {
            if (this.hasOwnProperty(key) && typeof (this[key]) !== 'function') {
                this[key] = undefined;
            }
        });
    }
}
CookieServiceMock.ɵfac = function CookieServiceMock_Factory(t) { return ɵCookieServiceMock_BaseFactory(t || CookieServiceMock); };
CookieServiceMock.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: CookieServiceMock, factory: CookieServiceMock.ɵfac });
const ɵCookieServiceMock_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(CookieServiceMock);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CookieServiceMock, [{
        type: Injectable
    }], null, null); })();
