// @alfresco/adf-content-services/esm2015/lib/search/components/search-filter-chips/search-facet-chip/search-facet-chip.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { ConfigurableFocusTrapFactory } from '@angular/cdk/a11y';
import { MatMenuTrigger } from '@angular/material/menu';
import { SearchFacetFieldComponent } from '../../search-facet-field/search-facet-field.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/cdk/a11y';
import * as ɵngcc2 from '@angular/material/chips';
import * as ɵngcc3 from '@angular/material/menu';
import * as ɵngcc4 from '@angular/common';
import * as ɵngcc5 from '@angular/material/icon';
import * as ɵngcc6 from '../search-filter-menu-card/search-filter-menu-card.component';
import * as ɵngcc7 from '../../search-facet-field/search-facet-field.component';
import * as ɵngcc8 from '@angular/material/button';
import * as ɵngcc9 from '@ngx-translate/core';

const _c0 = ["menuContainer"];
const _c1 = ["menuTrigger"];
function SearchFacetChipComponent_ng_container_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtext(1, ":");
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SearchFacetChipComponent_span_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 3);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const displayValue_r6 = ctx.ngIf;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" \u00A0 ", ɵngcc0.ɵɵpipeBind1(2, 1, displayValue_r6), " ");
} }
export class SearchFacetChipComponent {
    constructor(focusTrapFactory) {
        this.focusTrapFactory = focusTrapFactory;
    }
    onMenuOpen() {
        if (this.menuContainer && !this.focusTrap) {
            this.focusTrap = this.focusTrapFactory.create(this.menuContainer.nativeElement);
            this.focusTrap.focusInitialElement();
        }
    }
    onClosed() {
        this.focusTrap.destroy();
        this.focusTrap = null;
    }
    onRemove() {
        this.facetFieldComponent.reset();
        this.menuTrigger.closeMenu();
    }
    onApply() {
        this.facetFieldComponent.submitValues();
        this.menuTrigger.closeMenu();
    }
}
SearchFacetChipComponent.ɵfac = function SearchFacetChipComponent_Factory(t) { return new (t || SearchFacetChipComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ConfigurableFocusTrapFactory)); };
SearchFacetChipComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SearchFacetChipComponent, selectors: [["adf-search-facet-chip"]], viewQuery: function SearchFacetChipComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
        ɵngcc0.ɵɵviewQuery(_c1, true);
        ɵngcc0.ɵɵviewQuery(SearchFacetFieldComponent, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.menuContainer = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.menuTrigger = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.facetFieldComponent = _t.first);
    } }, inputs: { field: "field" }, decls: 32, vars: 29, consts: [["disableRipple", "", 1, "adf-search-filter-chip", 3, "matMenuTriggerFor", "onMenuOpen"], ["menuTrigger", "matMenuTrigger"], [1, "adf-search-filter-placeholder"], [1, "adf-search-filter-ellipsis"], [4, "ngIf"], ["class", "adf-search-filter-ellipsis", 4, "ngIf"], ["backdropClass", "adf-search-filter-chip-menu", 3, "closed"], ["menu", "matMenu"], ["menuContainer", ""], [3, "click", "keydown.tab", "close"], ["ngProjectAs", "filter-title", 5, ["filter-title"]], ["ngProjectAs", "filter-content", 5, ["filter-content"]], [3, "field"], ["facetField", ""], ["ngProjectAs", "filter-actions", 5, ["filter-actions"]], ["mat-flat-button", "", "color", "accent", "id", "cancel-filter-button", 1, "adf-search-action-button", 3, "click"], ["mat-flat-button", "", "color", "primary", "id", "apply-filter-button", 1, "adf-search-action-button", 3, "click"]], template: function SearchFacetChipComponent_Template(rf, ctx) { if (rf & 1) {
        const _r7 = ɵngcc0.ɵɵgetCurrentView();
        ɵngcc0.ɵɵelementStart(0, "mat-chip", 0, 1);
        ɵngcc0.ɵɵlistener("onMenuOpen", function SearchFacetChipComponent_Template_mat_chip_onMenuOpen_0_listener() { return ctx.onMenuOpen(); });
        ɵngcc0.ɵɵpipe(2, "async");
        ɵngcc0.ɵɵpipe(3, "async");
        ɵngcc0.ɵɵelementStart(4, "span", 2);
        ɵngcc0.ɵɵelementStart(5, "span", 3);
        ɵngcc0.ɵɵtext(6);
        ɵngcc0.ɵɵpipe(7, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(8, SearchFacetChipComponent_ng_container_8_Template, 2, 0, "ng-container", 4);
        ɵngcc0.ɵɵpipe(9, "async");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(10, SearchFacetChipComponent_span_10_Template, 3, 3, "span", 5);
        ɵngcc0.ɵɵpipe(11, "async");
        ɵngcc0.ɵɵelementStart(12, "mat-icon");
        ɵngcc0.ɵɵtext(13, "keyboard_arrow_down");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(14, "mat-menu", 6, 7);
        ɵngcc0.ɵɵlistener("closed", function SearchFacetChipComponent_Template_mat_menu_closed_14_listener() { return ctx.onClosed(); });
        ɵngcc0.ɵɵelementStart(16, "div", null, 8);
        ɵngcc0.ɵɵelementStart(18, "adf-search-filter-menu-card", 9);
        ɵngcc0.ɵɵlistener("click", function SearchFacetChipComponent_Template_adf_search_filter_menu_card_click_18_listener($event) { return $event.stopPropagation(); })("keydown.tab", function SearchFacetChipComponent_Template_adf_search_filter_menu_card_keydown_tab_18_listener($event) { return $event.stopPropagation(); })("close", function SearchFacetChipComponent_Template_adf_search_filter_menu_card_close_18_listener() { ɵngcc0.ɵɵrestoreView(_r7); const _r0 = ɵngcc0.ɵɵreference(1); return _r0.closeMenu(); });
        ɵngcc0.ɵɵelementContainerStart(19, 10);
        ɵngcc0.ɵɵtext(20);
        ɵngcc0.ɵɵpipe(21, "translate");
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵelementContainerStart(22, 11);
        ɵngcc0.ɵɵelement(23, "adf-search-facet-field", 12, 13);
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵelementContainerStart(25, 14);
        ɵngcc0.ɵɵelementStart(26, "button", 15);
        ɵngcc0.ɵɵlistener("click", function SearchFacetChipComponent_Template_button_click_26_listener() { return ctx.onRemove(); });
        ɵngcc0.ɵɵtext(27);
        ɵngcc0.ɵɵpipe(28, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(29, "button", 16);
        ɵngcc0.ɵɵlistener("click", function SearchFacetChipComponent_Template_button_click_29_listener() { return ctx.onApply(); });
        ɵngcc0.ɵɵtext(30);
        ɵngcc0.ɵɵpipe(31, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        const _r0 = ɵngcc0.ɵɵreference(1);
        const _r3 = ɵngcc0.ɵɵreference(15);
        const _r5 = ɵngcc0.ɵɵreference(24);
        ɵngcc0.ɵɵclassProp("adf-search-toggle-chip", ɵngcc0.ɵɵpipeBind1(2, 13, _r5.displayValue$) || _r0.menuOpen);
        ɵngcc0.ɵɵproperty("matMenuTriggerFor", _r3);
        ɵngcc0.ɵɵattribute("data-automation-id", "search-filter-chip-" + ctx.field.label)("title", ɵngcc0.ɵɵpipeBind1(3, 15, _r5.displayValue$));
        ɵngcc0.ɵɵadvance(6);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(7, 17, ctx.field.label));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(9, 19, _r5.displayValue$));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(11, 21, _r5.displayValue$));
        ɵngcc0.ɵɵadvance(6);
        ɵngcc0.ɵɵattribute("data-automation-id", "search-field-" + ctx.field.label);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(21, 23, ctx.field.label), " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("field", ctx.field);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(28, 25, "SEARCH.FILTER.BUTTONS.REMOVE"), " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(31, 27, "SEARCH.FILTER.BUTTONS.APPLY"), " ");
    } }, directives: [ɵngcc2.MatChip, ɵngcc3.MatMenuTrigger, ɵngcc4.NgIf, ɵngcc5.MatIcon, ɵngcc3._MatMenu, ɵngcc6.SearchFilterMenuCardComponent, ɵngcc7.SearchFacetFieldComponent, ɵngcc8.MatButton], pipes: [ɵngcc4.AsyncPipe, ɵngcc9.TranslatePipe], encapsulation: 2 });
SearchFacetChipComponent.ctorParameters = () => [
    { type: ConfigurableFocusTrapFactory }
];
SearchFacetChipComponent.propDecorators = {
    field: [{ type: Input }],
    menuContainer: [{ type: ViewChild, args: ['menuContainer', { static: false },] }],
    menuTrigger: [{ type: ViewChild, args: ['menuTrigger', { static: false },] }],
    facetFieldComponent: [{ type: ViewChild, args: [SearchFacetFieldComponent, { static: false },] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SearchFacetChipComponent, [{
        type: Component,
        args: [{
                selector: 'adf-search-facet-chip',
                template: "<mat-chip [attr.data-automation-id]=\"'search-filter-chip-' + field.label\"\n          disableRipple\n          class=\"adf-search-filter-chip\"\n          [class.adf-search-toggle-chip]=\"(facetField.displayValue$ | async) || menuTrigger.menuOpen\"\n          [matMenuTriggerFor]=\"menu\"\n          (onMenuOpen)=\"onMenuOpen()\"\n          [attr.title]=\"facetField.displayValue$ | async\"\n          #menuTrigger=\"matMenuTrigger\">\n\n    <span class=\"adf-search-filter-placeholder\">\n        <span class=\"adf-search-filter-ellipsis\">{{ field.label | translate }}</span>\n        <ng-container *ngIf=\"facetField.displayValue$ | async\">:</ng-container>\n    </span>\n\n    <span class=\"adf-search-filter-ellipsis\" *ngIf=\"facetField.displayValue$ | async as displayValue\">\n        &nbsp; {{ displayValue | translate }}\n    </span>\n    <mat-icon>keyboard_arrow_down</mat-icon>\n</mat-chip>\n\n<mat-menu #menu=\"matMenu\" backdropClass=\"adf-search-filter-chip-menu\" (closed)=\"onClosed()\">\n    <div #menuContainer [attr.data-automation-id]=\"'search-field-' + field.label\">\n        <adf-search-filter-menu-card (click)=\"$event.stopPropagation()\"\n                                     (keydown.tab)=\"$event.stopPropagation();\"\n                                     (close)=\"menuTrigger.closeMenu()\">\n            <ng-container ngProjectAs=\"filter-title\">\n                {{ field.label | translate }}\n            </ng-container>\n            <ng-container ngProjectAs=\"filter-content\">\n                <adf-search-facet-field [field]=\"field\" #facetField></adf-search-facet-field>\n            </ng-container>\n            <ng-container ngProjectAs=\"filter-actions\">\n                <button mat-flat-button class=\"adf-search-action-button\" color=\"accent\" (click)=\"onRemove()\" id=\"cancel-filter-button\">\n                    {{ 'SEARCH.FILTER.BUTTONS.REMOVE' | translate }}\n                </button>\n                <button mat-flat-button class=\"adf-search-action-button\" color=\"primary\" (click)=\"onApply()\" id=\"apply-filter-button\">\n                    {{ 'SEARCH.FILTER.BUTTONS.APPLY' | translate }}\n                </button>\n            </ng-container>\n        </adf-search-filter-menu-card>\n    </div>\n</mat-menu>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.ConfigurableFocusTrapFactory }]; }, { field: [{
            type: Input
        }], menuContainer: [{
            type: ViewChild,
            args: ['menuContainer', { static: false }]
        }], menuTrigger: [{
            type: ViewChild,
            args: ['menuTrigger', { static: false }]
        }], facetFieldComponent: [{
            type: ViewChild,
            args: [SearchFacetFieldComponent, { static: false }]
        }] }); })();
