// @alfresco/adf-content-services/esm2015/lib/search/components/search-filter-chips/search-filter-chips.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Inject, Input, ViewEncapsulation } from '@angular/core';
import { SearchFacetFiltersService } from '../../services/search-facet-filters.service';
import { SEARCH_QUERY_SERVICE_TOKEN } from '../../search-query-service.token';
import { SearchQueryBuilderService } from '../../services/search-query-builder.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/search-facet-filters.service';
import * as ɵngcc2 from '@angular/material/chips';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from './search-widget-chip/search-widget-chip.component';
import * as ɵngcc5 from './search-facet-chip/search-facet-chip.component';
import * as ɵngcc6 from '../../services/search-query-builder.service';

function SearchFilterChipsComponent_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "adf-search-widget-chip", 3);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const category_r2 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("category", category_r2);
} }
function SearchFilterChipsComponent_ng_container_2_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "adf-search-facet-chip", 4);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const field_r4 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("field", field_r4);
    ɵngcc0.ɵɵattribute("data-automation-id", "search-fact-chip-" + field_r4.field);
} }
function SearchFilterChipsComponent_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, SearchFilterChipsComponent_ng_container_2_ng_container_1_Template, 2, 2, "ng-container", 1);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r1.facetFiltersService.responseFacets);
} }
export class SearchFilterChipsComponent {
    constructor(queryBuilder, facetFiltersService) {
        this.queryBuilder = queryBuilder;
        this.facetFiltersService = facetFiltersService;
        this.showContextFacets = true;
    }
}
SearchFilterChipsComponent.ɵfac = function SearchFilterChipsComponent_Factory(t) { return new (t || SearchFilterChipsComponent)(ɵngcc0.ɵɵdirectiveInject(SEARCH_QUERY_SERVICE_TOKEN), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.SearchFacetFiltersService)); };
SearchFilterChipsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SearchFilterChipsComponent, selectors: [["adf-search-filter-chips"]], inputs: { showContextFacets: "showContextFacets" }, decls: 3, vars: 2, consts: [["aria-orientation", "horizontal"], [4, "ngFor", "ngForOf"], [4, "ngIf"], [3, "category"], [3, "field"]], template: function SearchFilterChipsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "mat-chip-list", 0);
        ɵngcc0.ɵɵtemplate(1, SearchFilterChipsComponent_ng_container_1_Template, 2, 1, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(2, SearchFilterChipsComponent_ng_container_2_Template, 2, 1, "ng-container", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.queryBuilder.categories);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.facetFiltersService.responseFacets && ctx.showContextFacets);
    } }, directives: [ɵngcc2.MatChipList, ɵngcc3.NgForOf, ɵngcc3.NgIf, ɵngcc4.SearchWidgetChipComponent, ɵngcc5.SearchFacetChipComponent], styles: [".adf-search-filter-chip.mat-chip{background:var(--theme-unselected-chip-bg-color);border:2px solid transparent;max-width:320px;overflow:hidden;text-overflow:ellipsis;transition:border .5s ease-in-out}.adf-search-filter-chip.mat-chip:focus{color:unset}.adf-search-filter-chip.mat-chip.mat-standard-chip:after{background:var(--theme-unselected-chip-bg-color);color:unset}.adf-search-filter-chip.mat-chip.mat-chip-list-wrapper{margin:4px 6px}.adf-search-filter-chip.adf-search-toggle-chip{background:var(--theme-card-bg-color);border:2px solid var(--theme-accent-color)}.adf-search-filter-chip.adf-search-toggle-chip.mat-chip:after{background:unset}.adf-search-filter-chip .adf-search-filter-placeholder{color:var(--theme-disabled-text-color);flex:1 1 auto;white-space:nowrap}.adf-search-filter-chip .adf-search-filter-ellipsis{overflow:hidden;text-overflow:ellipsis;white-space:nowrap}.adf-search-filter-chip .mat-icon{padding-left:5px;padding-top:5px}.adf-search-filter-chip-menu+* .cdk-overlay-pane .mat-menu-panel{border-radius:12px;box-shadow:0 3px 1px -2px rgba(0,0,0,.2),0 2px 2px 0 rgba(0,0,0,.14),0 1px 5px 0 rgba(0,0,0,.12);min-width:320px}"], encapsulation: 2 });
SearchFilterChipsComponent.ctorParameters = () => [
    { type: SearchQueryBuilderService, decorators: [{ type: Inject, args: [SEARCH_QUERY_SERVICE_TOKEN,] }] },
    { type: SearchFacetFiltersService }
];
SearchFilterChipsComponent.propDecorators = {
    showContextFacets: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SearchFilterChipsComponent, [{
        type: Component,
        args: [{
                selector: 'adf-search-filter-chips',
                template: "<mat-chip-list aria-orientation=\"horizontal\">\n    <ng-container *ngFor=\"let category of queryBuilder.categories\">\n        <adf-search-widget-chip [category]=\"category\"></adf-search-widget-chip>\n    </ng-container>\n\n    <ng-container *ngIf=\"facetFiltersService.responseFacets && showContextFacets\">\n        <ng-container *ngFor=\"let field of facetFiltersService.responseFacets\">\n            <adf-search-facet-chip [field]=\"field\" [attr.data-automation-id]=\"'search-fact-chip-' + field.field\" ></adf-search-facet-chip>\n        </ng-container>\n    </ng-container>\n</mat-chip-list>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-search-filter-chip.mat-chip{background:var(--theme-unselected-chip-bg-color);border:2px solid transparent;max-width:320px;overflow:hidden;text-overflow:ellipsis;transition:border .5s ease-in-out}.adf-search-filter-chip.mat-chip:focus{color:unset}.adf-search-filter-chip.mat-chip.mat-standard-chip:after{background:var(--theme-unselected-chip-bg-color);color:unset}.adf-search-filter-chip.mat-chip.mat-chip-list-wrapper{margin:4px 6px}.adf-search-filter-chip.adf-search-toggle-chip{background:var(--theme-card-bg-color);border:2px solid var(--theme-accent-color)}.adf-search-filter-chip.adf-search-toggle-chip.mat-chip:after{background:unset}.adf-search-filter-chip .adf-search-filter-placeholder{color:var(--theme-disabled-text-color);flex:1 1 auto;white-space:nowrap}.adf-search-filter-chip .adf-search-filter-ellipsis{overflow:hidden;text-overflow:ellipsis;white-space:nowrap}.adf-search-filter-chip .mat-icon{padding-left:5px;padding-top:5px}.adf-search-filter-chip-menu+* .cdk-overlay-pane .mat-menu-panel{border-radius:12px;box-shadow:0 3px 1px -2px rgba(0,0,0,.2),0 2px 2px 0 rgba(0,0,0,.14),0 1px 5px 0 rgba(0,0,0,.12);min-width:320px}"]
            }]
    }], function () { return [{ type: ɵngcc6.SearchQueryBuilderService, decorators: [{
                type: Inject,
                args: [SEARCH_QUERY_SERVICE_TOKEN]
            }] }, { type: ɵngcc1.SearchFacetFiltersService }]; }, { showContextFacets: [{
            type: Input
        }] }); })();
