// @alfresco/adf-process-services/esm2015/lib/form/form.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { CoreModule } from '@alfresco/adf-core';
import { FormComponent } from './form.component';
import { StartFormComponent } from './start-form.component';
import { FormCustomOutcomesComponent } from './form-custom-outcomes.component';
import * as ɵngcc0 from '@angular/core';
export class FormModule {
}
FormModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FormModule });
FormModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FormModule_Factory(t) { return new (t || FormModule)(); }, imports: [[
            CoreModule,
            MaterialModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FormModule, { declarations: function () { return [FormComponent,
        StartFormComponent,
        FormCustomOutcomesComponent]; }, imports: function () { return [CoreModule,
        MaterialModule]; }, exports: function () { return [FormComponent,
        StartFormComponent,
        FormCustomOutcomesComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FormModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CoreModule,
                    MaterialModule
                ],
                declarations: [
                    FormComponent,
                    StartFormComponent,
                    FormCustomOutcomesComponent
                ],
                exports: [
                    FormComponent,
                    StartFormComponent,
                    FormCustomOutcomesComponent
                ]
            }]
    }], null, null); })();
