// @alfresco/adf-core/esm2015/notifications/notification-history.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { PipeModule } from './../pipes/pipe.module';
import { NotificationHistoryComponent } from './components/notification-history.component';
import { TranslateModule } from '@ngx-translate/core';
import { NotificationIconPipe } from './pipes/notification-icon.pipe';
import { PaginationModule } from './../pagination/pagination.module';
import * as ɵngcc0 from '@angular/core';
export class NotificationHistoryModule {
}
NotificationHistoryModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NotificationHistoryModule });
NotificationHistoryModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NotificationHistoryModule_Factory(t) { return new (t || NotificationHistoryModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            TranslateModule,
            PipeModule,
            PaginationModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NotificationHistoryModule, { declarations: function () { return [NotificationHistoryComponent,
        NotificationIconPipe]; }, imports: function () { return [CommonModule,
        MaterialModule,
        TranslateModule,
        PipeModule,
        PaginationModule]; }, exports: function () { return [NotificationHistoryComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NotificationHistoryModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    TranslateModule,
                    PipeModule,
                    PaginationModule
                ],
                declarations: [
                    NotificationHistoryComponent,
                    NotificationIconPipe
                ],
                exports: [
                    NotificationHistoryComponent
                ]
            }]
    }], null, null); })();
