// @alfresco/adf-process-services-cloud/esm2015/lib/form/components/widgets/properties-viewer/properties-viewer-wrapper/properties-viewer-wrapper.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { NodesApiService } from '@alfresco/adf-core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@alfresco/adf-content-services';
import * as ɵngcc4 from '@angular/material/progress-spinner';

function PropertiesViewerWrapperComponent_adf_content_metadata_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-content-metadata", 2);
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("displayDefaultProperties", ctx_r0.displayDefaultProperties)("expanded", ctx_r0.expanded)("node", ctx_r0.node)("preset", ctx_r0.preset)("displayEmpty", ctx_r0.displayEmpty)("editable", ctx_r0.editable)("multi", ctx_r0.multi)("displayAspect", ctx_r0.displayAspect)("copyToClipboardAction", ctx_r0.copyToClipboardAction)("useChipsForMultiValueProperty", ctx_r0.useChipsForMultiValueProperty);
} }
function PropertiesViewerWrapperComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 3);
    ɵngcc0.ɵɵelement(1, "mat-progress-spinner");
    ɵngcc0.ɵɵelementEnd();
} }
export class PropertiesViewerWrapperComponent {
    constructor(nodesApiService) {
        this.nodesApiService = nodesApiService;
        this.loading = true;
        this.displayAspect = null;
        this.nodeContentLoaded = new EventEmitter();
    }
    ngOnChanges(changes) {
        if (changes && changes['nodeId'] && changes['nodeId'].currentValue && !changes['nodeId'].isFirstChange()) {
            this.getNode(changes['nodeId'].currentValue);
        }
    }
    ngOnInit() {
        if (this.nodeId) {
            this.getNode(this.nodeId);
        }
    }
    getNode(nodeId) {
        this.loading = true;
        this.nodesApiService.getNode(nodeId).subscribe(retrievedNode => {
            this.node = retrievedNode;
            this.loading = false;
            this.nodeContentLoaded.emit(retrievedNode);
        });
    }
}
PropertiesViewerWrapperComponent.ɵfac = function PropertiesViewerWrapperComponent_Factory(t) { return new (t || PropertiesViewerWrapperComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NodesApiService)); };
PropertiesViewerWrapperComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PropertiesViewerWrapperComponent, selectors: [["adf-properties-viewer-wrapper"]], inputs: { displayAspect: "displayAspect", nodeId: "nodeId", editable: "editable", displayEmpty: "displayEmpty", expanded: "expanded", multi: "multi", preset: "preset", displayDefaultProperties: "displayDefaultProperties", copyToClipboardAction: "copyToClipboardAction", useChipsForMultiValueProperty: "useChipsForMultiValueProperty" }, outputs: { nodeContentLoaded: "nodeContentLoaded" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 3, vars: 2, consts: [[3, "displayDefaultProperties", "expanded", "node", "preset", "displayEmpty", "editable", "multi", "displayAspect", "copyToClipboardAction", "useChipsForMultiValueProperty", 4, "ngIf", "ngIfElse"], ["loadingTemplate", ""], [3, "displayDefaultProperties", "expanded", "node", "preset", "displayEmpty", "editable", "multi", "displayAspect", "copyToClipboardAction", "useChipsForMultiValueProperty"], [1, "adf-properties-viewer-wrapper-loading"]], template: function PropertiesViewerWrapperComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, PropertiesViewerWrapperComponent_adf_content_metadata_0_Template, 1, 10, "adf-content-metadata", 0);
        ɵngcc0.ɵɵtemplate(1, PropertiesViewerWrapperComponent_ng_template_1_Template, 2, 0, "ng-template", null, 1, ɵngcc0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        const _r1 = ɵngcc0.ɵɵreference(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.loading)("ngIfElse", _r1);
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.ContentMetadataComponent, ɵngcc4.MatProgressSpinner], encapsulation: 2 });
PropertiesViewerWrapperComponent.ctorParameters = () => [
    { type: NodesApiService }
];
PropertiesViewerWrapperComponent.propDecorators = {
    nodeId: [{ type: Input }],
    editable: [{ type: Input }],
    displayEmpty: [{ type: Input }],
    expanded: [{ type: Input }],
    multi: [{ type: Input }],
    preset: [{ type: Input }],
    displayDefaultProperties: [{ type: Input }],
    displayAspect: [{ type: Input }],
    copyToClipboardAction: [{ type: Input }],
    useChipsForMultiValueProperty: [{ type: Input }],
    nodeContentLoaded: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PropertiesViewerWrapperComponent, [{
        type: Component,
        args: [{
                selector: 'adf-properties-viewer-wrapper',
                template: "<adf-content-metadata [displayDefaultProperties]=\"displayDefaultProperties\" [expanded]=\"expanded\" [node]=\"node\"\n    [preset]=\"preset\" [displayEmpty]=\"displayEmpty\" [editable]=\"editable\" [multi]=\"multi\"\n    [displayAspect]=\"displayAspect\" [copyToClipboardAction]=\"copyToClipboardAction\"\n    [useChipsForMultiValueProperty]=\"useChipsForMultiValueProperty\" *ngIf=\"!loading; else loadingTemplate\">\n</adf-content-metadata>\n<ng-template #loadingTemplate>\n    <div class=\"adf-properties-viewer-wrapper-loading\">\n        <mat-progress-spinner></mat-progress-spinner>\n    </div>\n</ng-template>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.NodesApiService }]; }, { displayAspect: [{
            type: Input
        }], nodeContentLoaded: [{
            type: Output
        }], nodeId: [{
            type: Input
        }], editable: [{
            type: Input
        }], displayEmpty: [{
            type: Input
        }], expanded: [{
            type: Input
        }], multi: [{
            type: Input
        }], preset: [{
            type: Input
        }], displayDefaultProperties: [{
            type: Input
        }], copyToClipboardAction: [{
            type: Input
        }], useChipsForMultiValueProperty: [{
            type: Input
        }] }); })();
