// @alfresco/adf-content-services/esm2015/lib/search/components/search-widget-container/search-widget-container.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewChild, ViewContainerRef, ComponentFactoryResolver, Inject } from '@angular/core';
import { SearchFilterService } from '../../services/search-filter.service';
import { BaseQueryBuilderService } from '../../services/base-query-builder.service';
import { SEARCH_QUERY_SERVICE_TOKEN } from '../../search-query-service.token';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/search-filter.service';
import * as ɵngcc2 from '../../services/base-query-builder.service';

const _c0 = ["content"];
export class SearchWidgetContainerComponent {
    constructor(searchFilterService, queryBuilder, componentFactoryResolver) {
        this.searchFilterService = searchFilterService;
        this.queryBuilder = queryBuilder;
        this.componentFactoryResolver = componentFactoryResolver;
    }
    ngOnInit() {
        const componentType = this.searchFilterService.widgets[this.selector];
        if (componentType) {
            const factory = this.componentFactoryResolver.resolveComponentFactory(componentType);
            if (factory) {
                this.content.clear();
                this.componentRef = this.content.createComponent(factory, 0);
                this.setupWidget(this.componentRef);
            }
        }
    }
    ngOnChanges(changes) {
        var _a, _b;
        if (((_a = changes['value']) === null || _a === void 0 ? void 0 : _a.currentValue) && ((_b = this.componentRef) === null || _b === void 0 ? void 0 : _b.instance)) {
            this.componentRef.instance.isActive = true;
            this.componentRef.instance.setValue(changes['value'].currentValue);
        }
    }
    setupWidget(ref) {
        if (ref && ref.instance) {
            ref.instance.id = this.id;
            ref.instance.settings = Object.assign({}, this.settings);
            ref.instance.context = this.queryBuilder;
            if (this.value) {
                ref.instance.isActive = true;
                ref.instance.startValue = this.value;
            }
        }
    }
    ngOnDestroy() {
        if (this.componentRef) {
            this.componentRef.destroy();
            this.componentRef = null;
        }
    }
    applyInnerWidget() {
        this.componentRef.instance.submitValues();
    }
    setValue(currentValue) {
        this.componentRef.instance.isActive = true;
        this.componentRef.instance.setValue(currentValue);
    }
    hasValueSelected() {
        return this.componentRef.instance.hasValidValue();
    }
    getCurrentValue() {
        return this.componentRef.instance.getCurrentValue();
    }
    getDisplayValue() {
        var _a;
        if (!((_a = this.componentRef) === null || _a === void 0 ? void 0 : _a.instance)) {
            return null;
        }
        return this.componentRef.instance.displayValue$;
    }
    resetInnerWidget() {
        if (this.componentRef && this.componentRef.instance) {
            this.componentRef.instance.reset();
        }
    }
}
SearchWidgetContainerComponent.ɵfac = function SearchWidgetContainerComponent_Factory(t) { return new (t || SearchWidgetContainerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.SearchFilterService), ɵngcc0.ɵɵdirectiveInject(SEARCH_QUERY_SERVICE_TOKEN), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ComponentFactoryResolver)); };
SearchWidgetContainerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SearchWidgetContainerComponent, selectors: [["adf-search-widget-container"]], viewQuery: function SearchWidgetContainerComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(_c0, true, ViewContainerRef);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.content = _t.first);
    } }, inputs: { id: "id", selector: "selector", settings: "settings", config: "config", value: "value" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 2, vars: 0, consts: [["content", ""]], template: function SearchWidgetContainerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "div", null, 0);
    } }, encapsulation: 2 });
SearchWidgetContainerComponent.ctorParameters = () => [
    { type: SearchFilterService },
    { type: BaseQueryBuilderService, decorators: [{ type: Inject, args: [SEARCH_QUERY_SERVICE_TOKEN,] }] },
    { type: ComponentFactoryResolver }
];
SearchWidgetContainerComponent.propDecorators = {
    content: [{ type: ViewChild, args: ['content', { read: ViewContainerRef, static: true },] }],
    id: [{ type: Input }],
    selector: [{ type: Input }],
    settings: [{ type: Input }],
    config: [{ type: Input }],
    value: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SearchWidgetContainerComponent, [{
        type: Component,
        args: [{
                selector: 'adf-search-widget-container',
                template: '<div #content></div>'
            }]
    }], function () { return [{ type: ɵngcc1.SearchFilterService }, { type: ɵngcc2.BaseQueryBuilderService, decorators: [{
                type: Inject,
                args: [SEARCH_QUERY_SERVICE_TOKEN]
            }] }, { type: ɵngcc0.ComponentFactoryResolver }]; }, { content: [{
            type: ViewChild,
            args: ['content', { read: ViewContainerRef, static: true }]
        }], id: [{
            type: Input
        }], selector: [{
            type: Input
        }], settings: [{
            type: Input
        }], config: [{
            type: Input
        }], value: [{
            type: Input
        }] }); })();
