// @alfresco/adf-content-services/esm2015/lib/content-type/content-type.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatTableModule } from '@angular/material/table';
import { MatExpansionModule } from '@angular/material/expansion';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { TranslateModule } from '@ngx-translate/core';
import { MatDialogModule } from '@angular/material/dialog';
import { MatButtonModule } from '@angular/material/button';
import { ContentTypeDialogComponent } from './content-type-dialog.component';
import * as ɵngcc0 from '@angular/core';
export class ContentTypeModule {
}
ContentTypeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ContentTypeModule });
ContentTypeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ContentTypeModule_Factory(t) { return new (t || ContentTypeModule)(); }, imports: [[
            CommonModule,
            MatTableModule,
            MatExpansionModule,
            MatCheckboxModule,
            TranslateModule,
            MatDialogModule,
            MatButtonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ContentTypeModule, { declarations: function () { return [ContentTypeDialogComponent]; }, imports: function () { return [CommonModule,
        MatTableModule,
        MatExpansionModule,
        MatCheckboxModule,
        TranslateModule,
        MatDialogModule,
        MatButtonModule]; }, exports: function () { return [ContentTypeDialogComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContentTypeModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MatTableModule,
                    MatExpansionModule,
                    MatCheckboxModule,
                    TranslateModule,
                    MatDialogModule,
                    MatButtonModule
                ],
                exports: [
                    ContentTypeDialogComponent
                ],
                declarations: [
                    ContentTypeDialogComponent
                ]
            }]
    }], null, null); })();
