// @alfresco/adf-core/esm2015/form/components/widgets/document/document.widget.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { WidgetComponent } from './../widget.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../../../services/form.service';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../content/content.widget';

function DocumentWidgetComponent_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "adf-content", 1);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("id", ctx_r0.fileId)("showDocumentContent", true);
} }
export class DocumentWidgetComponent extends WidgetComponent {
    constructor(formService) {
        super(formService);
        this.formService = formService;
        this.fileId = null;
        this.hasFile = false;
    }
    ngOnInit() {
        if (this.field) {
            const file = this.field.value;
            if (file) {
                this.fileId = file.id;
                this.hasFile = true;
            }
            else {
                this.fileId = null;
                this.hasFile = false;
            }
        }
    }
}
DocumentWidgetComponent.ɵfac = function DocumentWidgetComponent_Factory(t) { return new (t || DocumentWidgetComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService)); };
DocumentWidgetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DocumentWidgetComponent, selectors: [["adf-form-document-widget"]], hostBindings: function DocumentWidgetComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function DocumentWidgetComponent_click_HostBindingHandler($event) { return ctx.event($event); })("blur", function DocumentWidgetComponent_blur_HostBindingHandler($event) { return ctx.event($event); })("change", function DocumentWidgetComponent_change_HostBindingHandler($event) { return ctx.event($event); })("focus", function DocumentWidgetComponent_focus_HostBindingHandler($event) { return ctx.event($event); })("focusin", function DocumentWidgetComponent_focusin_HostBindingHandler($event) { return ctx.event($event); })("focusout", function DocumentWidgetComponent_focusout_HostBindingHandler($event) { return ctx.event($event); })("input", function DocumentWidgetComponent_input_HostBindingHandler($event) { return ctx.event($event); })("invalid", function DocumentWidgetComponent_invalid_HostBindingHandler($event) { return ctx.event($event); })("select", function DocumentWidgetComponent_select_HostBindingHandler($event) { return ctx.event($event); });
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 2, vars: 4, consts: [[4, "ngIf"], [3, "id", "showDocumentContent"]], template: function DocumentWidgetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵtemplate(1, DocumentWidgetComponent_ng_container_1_Template, 2, 2, "ng-container", 0);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMapInterpolate1("adf-form-document-widget ", ctx.field.className, "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasFile);
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.ContentWidgetComponent], encapsulation: 2 });
DocumentWidgetComponent.ctorParameters = () => [
    { type: FormService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DocumentWidgetComponent, [{
        type: Component,
        args: [{
                selector: 'adf-form-document-widget',
                template: "<div class=\"adf-form-document-widget {{field.className}}\">\n    <ng-container *ngIf=\"hasFile\">\n        <adf-content [id]=\"fileId\" [showDocumentContent]=\"true\"></adf-content>\n    </ng-container>\n</div>\n",
                host: {
                    '(click)': 'event($event)',
                    '(blur)': 'event($event)',
                    '(change)': 'event($event)',
                    '(focus)': 'event($event)',
                    '(focusin)': 'event($event)',
                    '(focusout)': 'event($event)',
                    '(input)': 'event($event)',
                    '(invalid)': 'event($event)',
                    '(select)': 'event($event)'
                },
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.FormService }]; }, null); })();
