// @alfresco/adf-process-services-cloud/esm2015/lib/process/process-list/components/process-list-cloud.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, ContentChild, Output, EventEmitter, Input, ViewChild } from '@angular/core';
import { DataTableSchema, CustomEmptyContentTemplateDirective, AppConfigService, UserPreferencesService, UserPreferenceValues, CustomLoadingContentTemplateDirective, DataTableComponent } from '@alfresco/adf-core';
import { ProcessListCloudService } from '../services/process-list-cloud.service';
import { BehaviorSubject } from 'rxjs';
import { processCloudPresetsDefaultModel } from '../models/process-cloud-preset.model';
import { ProcessQueryCloudRequestModel } from '../models/process-cloud-query-request.model';
import { ProcessListCloudSortingModel } from '../models/process-list-sorting.model';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/process-list-cloud.service';
import * as ɵngcc2 from '@alfresco/adf-core';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@angular/material/progress-spinner';
import * as ɵngcc5 from '@ngx-translate/core';

function ProcessListCloudComponent_ng_template_3_mat_progress_spinner_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "mat-progress-spinner", 3);
} if (rf & 2) {
    ɵngcc0.ɵɵproperty("color", "primary")("mode", "indeterminate");
} }
function ProcessListCloudComponent_ng_template_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, ProcessListCloudComponent_ng_template_3_mat_progress_spinner_0_Template, 1, 2, "mat-progress-spinner", 2);
    ɵngcc0.ɵɵprojection(1);
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r1.customLoadingContent);
} }
function ProcessListCloudComponent_ng_template_5_adf_empty_content_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-empty-content", 5);
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵpipe(2, "translate");
} if (rf & 2) {
    ɵngcc0.ɵɵproperty("title", ɵngcc0.ɵɵpipeBind1(1, 2, "ADF_CLOUD_PROCESS_LIST.MESSAGES.TITLE"))("subtitle", ɵngcc0.ɵɵpipeBind1(2, 4, "ADF_CLOUD_PROCESS_LIST.MESSAGES.SUBTITLE"));
} }
function ProcessListCloudComponent_ng_template_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, ProcessListCloudComponent_ng_template_5_adf_empty_content_0_Template, 3, 6, "adf-empty-content", 4);
    ɵngcc0.ɵɵprojection(1, 1);
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r2.emptyCustomContent);
} }
const _c0 = [[["adf-custom-loading-content-template"]], [["adf-custom-empty-content-template"]]];
const _c1 = ["adf-custom-loading-content-template", "adf-custom-empty-content-template"];
export class ProcessListCloudComponent extends DataTableSchema {
    constructor(processListCloudService, appConfigService, userPreferences) {
        super(appConfigService, ProcessListCloudComponent.PRESET_KEY, processCloudPresetsDefaultModel);
        this.processListCloudService = processListCloudService;
        this.userPreferences = userPreferences;
        this.appName = '';
        this.initiator = '';
        this.id = '';
        this.name = '';
        this.processDefinitionId = '';
        this.processDefinitionName = '';
        this.processDefinitionKey = '';
        this.status = '';
        this.businessKey = '';
        this.startFrom = '';
        this.startTo = '';
        this.completedFrom = '';
        this.completedTo = '';
        this.completedDate = '';
        this.suspendedFrom = '';
        this.suspendedTo = '';
        this.selectionMode = 'single';
        this.multiselect = false;
        this.showActions = false;
        this.actionsPosition = 'right';
        this.stickyHeader = false;
        this.showContextMenu = false;
        this.rowClick = new EventEmitter();
        this.rowsSelected = new EventEmitter();
        this.showRowContextMenu = new EventEmitter();
        this.showRowActionsMenu = new EventEmitter();
        this.executeRowAction = new EventEmitter();
        this.error = new EventEmitter();
        this.success = new EventEmitter();
        this.skipCount = 0;
        this.isLoading = true;
        this.rows = [];
        this.defaultSorting = { key: 'startDate', direction: 'desc' };
        this.size = userPreferences.paginationSize;
        this.userPreferences.select(UserPreferenceValues.PaginationSize).subscribe((pageSize) => {
            this.size = pageSize;
        });
        this.pagination = new BehaviorSubject({
            maxItems: this.size,
            skipCount: 0,
            totalItems: 0
        });
    }
    ngAfterContentInit() {
        this.createDatatableSchema();
    }
    ngOnChanges(changes) {
        if (this.isPropertyChanged(changes, 'sorting')) {
            this.formatSorting(changes['sorting'].currentValue);
        }
        if (this.isAnyPropertyChanged(changes)) {
            this.reload();
        }
    }
    getCurrentId() {
        return this.currentInstanceId;
    }
    reload() {
        this.requestNode = this.createRequestNode();
        if (this.requestNode.appName || this.requestNode.appName === '') {
            this.load(this.requestNode);
        }
        else {
            this.rows = [];
        }
    }
    load(requestNode) {
        this.isLoading = true;
        this.processListCloudService.getProcessByRequest(requestNode).subscribe((processes) => {
            this.rows = processes.list.entries;
            this.success.emit(processes);
            this.isLoading = false;
            this.pagination.next(processes.list.pagination);
        }, (error) => {
            this.error.emit(error);
            this.isLoading = false;
        });
    }
    isAnyPropertyChanged(changes) {
        for (const property in changes) {
            if (this.isPropertyChanged(changes, property)) {
                return true;
            }
        }
        return false;
    }
    isPropertyChanged(changes, property) {
        return changes.hasOwnProperty(property);
    }
    isListEmpty() {
        return !this.rows || this.rows.length === 0;
    }
    resetPagination() {
        this.skipCount = 0;
        this.size = this.userPreferences.paginationSize;
        this.pagination.next({
            skipCount: 0,
            maxItems: this.size
        });
    }
    updatePagination(pagination) {
        this.size = pagination.maxItems;
        this.skipCount = pagination.skipCount;
        this.pagination.next(pagination);
        this.reload();
    }
    onSortingChanged(event) {
        this.setSorting(event.detail);
        this.formatSorting(this.sorting);
        this.reload();
    }
    onRowClick(item) {
        this.currentInstanceId = item.value.getValue('id');
        this.rowClick.emit(this.currentInstanceId);
    }
    onRowSelect(event) {
        this.selectedInstances = [...event.detail.selection];
        this.rowsSelected.emit(this.selectedInstances);
    }
    onRowUnselect(event) {
        this.selectedInstances = [...event.detail.selection];
        this.rowsSelected.emit(this.selectedInstances);
    }
    onRowKeyUp(event) {
        if (event.detail.keyboardEvent.key === 'Enter') {
            event.preventDefault();
            this.currentInstanceId = event.detail.row.getValue('id');
            this.rowClick.emit(this.currentInstanceId);
        }
    }
    onShowRowActionsMenu(event) {
        this.showRowActionsMenu.emit(event);
    }
    onShowRowContextMenu(event) {
        this.showRowContextMenu.emit(event);
    }
    onExecuteRowAction(row) {
        this.executeRowAction.emit(row);
    }
    createRequestNode() {
        const requestNode = {
            appName: this.appName,
            appVersion: this.getAppVersions(),
            maxItems: this.size,
            skipCount: this.skipCount,
            initiator: this.initiator,
            id: this.id,
            name: this.name,
            processDefinitionId: this.processDefinitionId,
            processDefinitionName: this.processDefinitionName,
            processDefinitionKey: this.processDefinitionKey,
            status: this.status,
            businessKey: this.businessKey,
            lastModifiedFrom: this.lastModifiedFrom,
            lastModifiedTo: this.lastModifiedTo,
            startFrom: this.startFrom,
            startTo: this.startTo,
            completedFrom: this.completedFrom,
            completedTo: this.completedTo,
            suspendedFrom: this.suspendedFrom,
            suspendedTo: this.suspendedTo,
            completedDate: this.completedDate,
            sorting: this.sorting
        };
        return new ProcessQueryCloudRequestModel(requestNode);
    }
    getAppVersions() {
        return this.appVersion instanceof Array ? this.appVersion.join(',') : (this.appVersion ? String(this.appVersion) : '');
    }
    setSorting(sortDetail) {
        const sorting = sortDetail ? {
            orderBy: sortDetail.key,
            direction: sortDetail.direction.toUpperCase()
        } : Object.assign({}, this.defaultSorting);
        this.sorting = [new ProcessListCloudSortingModel(sorting)];
    }
    formatSorting(sorting) {
        this.formattedSorting = this.isValidSorting(sorting) ? [
            sorting[0].orderBy,
            sorting[0].direction.toLocaleLowerCase()
        ] : null;
    }
    isValidSorting(sorting) {
        return sorting.length && sorting[0].orderBy && sorting[0].direction;
    }
}
ProcessListCloudComponent.ɵfac = function ProcessListCloudComponent_Factory(t) { return new (t || ProcessListCloudComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ProcessListCloudService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.AppConfigService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.UserPreferencesService)); };
ProcessListCloudComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ProcessListCloudComponent, selectors: [["adf-cloud-process-list"]], contentQueries: function ProcessListCloudComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, CustomEmptyContentTemplateDirective, true);
        ɵngcc0.ɵɵcontentQuery(dirIndex, CustomLoadingContentTemplateDirective, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.emptyCustomContent = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.customLoadingContent = _t.first);
    } }, viewQuery: function ProcessListCloudComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(DataTableComponent, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.dataTable = _t.first);
    } }, inputs: { appName: "appName", initiator: "initiator", id: "id", name: "name", processDefinitionId: "processDefinitionId", processDefinitionName: "processDefinitionName", processDefinitionKey: "processDefinitionKey", status: "status", businessKey: "businessKey", startFrom: "startFrom", startTo: "startTo", completedFrom: "completedFrom", completedTo: "completedTo", completedDate: "completedDate", suspendedFrom: "suspendedFrom", suspendedTo: "suspendedTo", selectionMode: "selectionMode", multiselect: "multiselect", showActions: "showActions", actionsPosition: "actionsPosition", stickyHeader: "stickyHeader", showContextMenu: "showContextMenu", sorting: "sorting", appVersion: "appVersion", lastModifiedFrom: "lastModifiedFrom", lastModifiedTo: "lastModifiedTo" }, outputs: { rowClick: "rowClick", rowsSelected: "rowsSelected", showRowContextMenu: "showRowContextMenu", showRowActionsMenu: "showRowActionsMenu", executeRowAction: "executeRowAction", error: "error", success: "success" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature, ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c1, decls: 6, vars: 10, consts: [[3, "rows", "columns", "stickyHeader", "loading", "sorting", "selectionMode", "multiselect", "actions", "actionsPosition", "contextMenu", "showRowActionsMenu", "showRowContextMenu", "executeRowAction", "rowClick", "row-select", "row-unselect", "row-keyup", "sorting-changed"], ["dataTable", ""], ["class", "adf-cloud-process-list-loading-margin", 3, "color", "mode", 4, "ngIf"], [1, "adf-cloud-process-list-loading-margin", 3, "color", "mode"], ["icon", "assessment", 3, "title", "subtitle", 4, "ngIf"], ["icon", "assessment", 3, "title", "subtitle"]], template: function ProcessListCloudComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵelementStart(0, "adf-datatable", 0, 1);
        ɵngcc0.ɵɵlistener("showRowActionsMenu", function ProcessListCloudComponent_Template_adf_datatable_showRowActionsMenu_0_listener($event) { return ctx.onShowRowActionsMenu($event); })("showRowContextMenu", function ProcessListCloudComponent_Template_adf_datatable_showRowContextMenu_0_listener($event) { return ctx.onShowRowContextMenu($event); })("executeRowAction", function ProcessListCloudComponent_Template_adf_datatable_executeRowAction_0_listener($event) { return ctx.onExecuteRowAction($event); })("rowClick", function ProcessListCloudComponent_Template_adf_datatable_rowClick_0_listener($event) { return ctx.onRowClick($event); })("row-select", function ProcessListCloudComponent_Template_adf_datatable_row_select_0_listener($event) { return ctx.onRowSelect($event); })("row-unselect", function ProcessListCloudComponent_Template_adf_datatable_row_unselect_0_listener($event) { return ctx.onRowUnselect($event); })("row-keyup", function ProcessListCloudComponent_Template_adf_datatable_row_keyup_0_listener($event) { return ctx.onRowKeyUp($event); })("sorting-changed", function ProcessListCloudComponent_Template_adf_datatable_sorting_changed_0_listener($event) { return ctx.onSortingChanged($event); });
        ɵngcc0.ɵɵelementStart(2, "adf-loading-content-template");
        ɵngcc0.ɵɵtemplate(3, ProcessListCloudComponent_ng_template_3_Template, 2, 1, "ng-template");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "adf-no-content-template");
        ɵngcc0.ɵɵtemplate(5, ProcessListCloudComponent_ng_template_5_Template, 2, 1, "ng-template");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("rows", ctx.rows)("columns", ctx.columns)("stickyHeader", ctx.stickyHeader)("loading", ctx.isLoading)("sorting", ctx.formattedSorting)("selectionMode", ctx.selectionMode)("multiselect", ctx.multiselect)("actions", ctx.showActions)("actionsPosition", ctx.actionsPosition)("contextMenu", ctx.showContextMenu);
    } }, directives: [ɵngcc2.DataTableComponent, ɵngcc2.LoadingContentTemplateDirective, ɵngcc2.NoContentTemplateDirective, ɵngcc3.NgIf, ɵngcc4.MatProgressSpinner, ɵngcc2.EmptyContentComponent], pipes: [ɵngcc5.TranslatePipe], styles: [".adf-cloud-process-list-loading-margin{margin-left:calc(50% - 50px);margin-right:calc(50% - 50px)}.adf-no-content-message{font-size:16px;font-weight:700;opacity:.54;text-align:center}"], encapsulation: 2 });
ProcessListCloudComponent.PRESET_KEY = 'adf-cloud-process-list.presets';
ProcessListCloudComponent.ctorParameters = () => [
    { type: ProcessListCloudService },
    { type: AppConfigService },
    { type: UserPreferencesService }
];
ProcessListCloudComponent.propDecorators = {
    dataTable: [{ type: ViewChild, args: [DataTableComponent,] }],
    emptyCustomContent: [{ type: ContentChild, args: [CustomEmptyContentTemplateDirective,] }],
    customLoadingContent: [{ type: ContentChild, args: [CustomLoadingContentTemplateDirective,] }],
    appName: [{ type: Input }],
    appVersion: [{ type: Input }],
    initiator: [{ type: Input }],
    id: [{ type: Input }],
    name: [{ type: Input }],
    processDefinitionId: [{ type: Input }],
    processDefinitionName: [{ type: Input }],
    processDefinitionKey: [{ type: Input }],
    status: [{ type: Input }],
    businessKey: [{ type: Input }],
    lastModifiedFrom: [{ type: Input }],
    lastModifiedTo: [{ type: Input }],
    startFrom: [{ type: Input }],
    startTo: [{ type: Input }],
    completedFrom: [{ type: Input }],
    completedTo: [{ type: Input }],
    completedDate: [{ type: Input }],
    suspendedFrom: [{ type: Input }],
    suspendedTo: [{ type: Input }],
    selectionMode: [{ type: Input }],
    multiselect: [{ type: Input }],
    sorting: [{ type: Input }],
    showActions: [{ type: Input }],
    actionsPosition: [{ type: Input }],
    stickyHeader: [{ type: Input }],
    showContextMenu: [{ type: Input }],
    rowClick: [{ type: Output }],
    rowsSelected: [{ type: Output }],
    showRowContextMenu: [{ type: Output }],
    showRowActionsMenu: [{ type: Output }],
    executeRowAction: [{ type: Output }],
    error: [{ type: Output }],
    success: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ProcessListCloudComponent, [{
        type: Component,
        args: [{
                selector: 'adf-cloud-process-list',
                template: "<adf-datatable #dataTable\n        [rows]=\"rows\"\n        [columns]=\"columns\"\n        [stickyHeader]=\"stickyHeader\"\n        [loading]=\"isLoading\"\n        [sorting]=\"formattedSorting\"\n        [selectionMode]=\"selectionMode\"\n        [multiselect]=\"multiselect\"\n        [actions]=\"showActions\"\n        [actionsPosition]=\"actionsPosition\"\n        [contextMenu]=\"showContextMenu\"\n        (showRowActionsMenu)=\"onShowRowActionsMenu($event)\"\n        (showRowContextMenu)=\"onShowRowContextMenu($event)\"\n        (executeRowAction)=\"onExecuteRowAction($event)\"\n        (rowClick)=\"onRowClick($any($event))\"\n        (row-select)=\"onRowSelect($any($event))\"\n        (row-unselect)=\"onRowUnselect($any($event))\"\n        (row-keyup)=\"onRowKeyUp($any($event))\"\n        (sorting-changed)=\"onSortingChanged($any($event))\">\n        <adf-loading-content-template>\n            <ng-template>\n                <mat-progress-spinner\n                    *ngIf=\"!customLoadingContent\"\n                    class=\"adf-cloud-process-list-loading-margin\"\n                    [color]=\"'primary'\"\n                    [mode]=\"'indeterminate'\">\n                </mat-progress-spinner>\n                <ng-content select=\"adf-custom-loading-content-template\"></ng-content>\n            </ng-template>\n        </adf-loading-content-template>\n        <adf-no-content-template>\n            <ng-template>\n                <adf-empty-content *ngIf=\"!emptyCustomContent\"\n                    icon=\"assessment\"\n                    [title]=\"'ADF_CLOUD_PROCESS_LIST.MESSAGES.TITLE' | translate\"\n                    [subtitle]=\"'ADF_CLOUD_PROCESS_LIST.MESSAGES.SUBTITLE'| translate\">\n                </adf-empty-content>\n                <ng-content select=\"adf-custom-empty-content-template\"></ng-content>\n            </ng-template>\n        </adf-no-content-template>\n</adf-datatable>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-cloud-process-list-loading-margin{margin-left:calc(50% - 50px);margin-right:calc(50% - 50px)}.adf-no-content-message{font-size:16px;font-weight:700;opacity:.54;text-align:center}"]
            }]
    }], function () { return [{ type: ɵngcc1.ProcessListCloudService }, { type: ɵngcc2.AppConfigService }, { type: ɵngcc2.UserPreferencesService }]; }, { appName: [{
            type: Input
        }], initiator: [{
            type: Input
        }], id: [{
            type: Input
        }], name: [{
            type: Input
        }], processDefinitionId: [{
            type: Input
        }], processDefinitionName: [{
            type: Input
        }], processDefinitionKey: [{
            type: Input
        }], status: [{
            type: Input
        }], businessKey: [{
            type: Input
        }], startFrom: [{
            type: Input
        }], startTo: [{
            type: Input
        }], completedFrom: [{
            type: Input
        }], completedTo: [{
            type: Input
        }], completedDate: [{
            type: Input
        }], suspendedFrom: [{
            type: Input
        }], suspendedTo: [{
            type: Input
        }], selectionMode: [{
            type: Input
        }], multiselect: [{
            type: Input
        }], showActions: [{
            type: Input
        }], actionsPosition: [{
            type: Input
        }], stickyHeader: [{
            type: Input
        }], showContextMenu: [{
            type: Input
        }], rowClick: [{
            type: Output
        }], rowsSelected: [{
            type: Output
        }], showRowContextMenu: [{
            type: Output
        }], showRowActionsMenu: [{
            type: Output
        }], executeRowAction: [{
            type: Output
        }], error: [{
            type: Output
        }], success: [{
            type: Output
        }], sorting: [{
            type: Input
        }], dataTable: [{
            type: ViewChild,
            args: [DataTableComponent]
        }], emptyCustomContent: [{
            type: ContentChild,
            args: [CustomEmptyContentTemplateDirective]
        }], customLoadingContent: [{
            type: ContentChild,
            args: [CustomLoadingContentTemplateDirective]
        }], appVersion: [{
            type: Input
        }], lastModifiedFrom: [{
            type: Input
        }], lastModifiedTo: [{
            type: Input
        }] }); })();
