// @alfresco/adf-process-services/esm2015/lib/people/components/people-search-field/people-search-field.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { TranslationService, PeopleProcessService } from '@alfresco/adf-core';
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { FormControl } from '@angular/forms';
import { debounceTime, switchMap } from 'rxjs/operators';
import { of } from 'rxjs';
import { getDisplayUser } from '../../helpers/get-display-user';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
import * as ɵngcc2 from '@angular/material/form-field';
import * as ɵngcc3 from '@angular/material/input';
import * as ɵngcc4 from '@angular/forms';
import * as ɵngcc5 from '@angular/common';
import * as ɵngcc6 from '../people-list/people-list.component';

function PeopleSearchFieldComponent_ng_container_2_div_1_ng_template_4_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 10);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const entry_r5 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r6 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r6.getInitialUserName(entry_r5.row.obj.firstName, entry_r5.row.obj.lastName), "");
} }
function PeopleSearchFieldComponent_ng_container_2_div_1_ng_template_4_img_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 11);
} if (rf & 2) {
    const entry_r5 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r7 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("alt", ctx_r7.getDisplayUser(entry_r5.row.obj.firstName, entry_r5.row.obj.lastName, " "))("src", ctx_r7.peopleProcessService.getUserImage(entry_r5.row.obj), ɵngcc0.ɵɵsanitizeUrl);
} }
function PeopleSearchFieldComponent_ng_container_2_div_1_ng_template_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, PeopleSearchFieldComponent_ng_container_2_div_1_ng_template_4_div_0_Template, 2, 1, "div", 8);
    ɵngcc0.ɵɵelementStart(1, "div");
    ɵngcc0.ɵɵtemplate(2, PeopleSearchFieldComponent_ng_container_2_div_1_ng_template_4_img_2_Template, 1, 2, "img", 9);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const entry_r5 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("ngIf", !entry_r5.row.obj.pictureId);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", entry_r5.row.obj.pictureId);
} }
function PeopleSearchFieldComponent_ng_container_2_div_1_ng_template_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 12);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const entry_r10 = ctx.$implicit;
    const ctx_r4 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵattribute("data-automation-id", "adf-people-full-name-" + ctx_r4.getDisplayUser(entry_r10.row.obj.firstName, entry_r10.row.obj.lastName, "-"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r4.getDisplayUser(entry_r10.row.obj.firstName, entry_r10.row.obj.lastName, " "));
} }
function PeopleSearchFieldComponent_ng_container_2_div_1_Template(rf, ctx) { if (rf & 1) {
    const _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵelementStart(1, "adf-people-list", 5);
    ɵngcc0.ɵɵlistener("clickRow", function PeopleSearchFieldComponent_ng_container_2_div_1_Template_adf_people_list_clickRow_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r12); const ctx_r11 = ɵngcc0.ɵɵnextContext(2); return ctx_r11.onRowClick($event); });
    ɵngcc0.ɵɵelementStart(2, "data-columns");
    ɵngcc0.ɵɵelementStart(3, "data-column", 6);
    ɵngcc0.ɵɵtemplate(4, PeopleSearchFieldComponent_ng_container_2_div_1_ng_template_4_Template, 3, 2, "ng-template");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(5, "data-column", 7);
    ɵngcc0.ɵɵtemplate(6, PeopleSearchFieldComponent_ng_container_2_div_1_ng_template_6_Template, 2, 2, "ng-template");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const users_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("users", users_r1);
} }
function PeopleSearchFieldComponent_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, PeopleSearchFieldComponent_ng_container_2_div_1_Template, 7, 1, "div", 3);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const users_r1 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", users_r1.length > 0);
} }
export class PeopleSearchFieldComponent {
    constructor(peopleProcessService, translationService) {
        this.peopleProcessService = peopleProcessService;
        this.translationService = translationService;
        this.rowClick = new EventEmitter();
        this.searchUser = new FormControl();
        this.defaultPlaceholder = 'ADF_TASK_LIST.PEOPLE.SEARCH_USER';
        this.users$ = this.searchUser.valueChanges
            .pipe(debounceTime(200), switchMap((searchWord) => {
            if (searchWord && searchWord.trim()) {
                return this.performSearch(searchWord);
            }
            else {
                return of([]);
            }
        }));
        this.defaultPlaceholder = this.translationService.instant(this.defaultPlaceholder);
    }
    reset() {
        this.searchUser.reset();
    }
    get searchPlaceholder() {
        return this.placeholder || this.defaultPlaceholder;
    }
    onRowClick(model) {
        this.rowClick.emit(model);
    }
    getDisplayUser(firstName, lastName, delimiter = '-') {
        return getDisplayUser(firstName, lastName, delimiter);
    }
    getInitialUserName(firstName, lastName) {
        firstName = (firstName !== null && firstName !== '' ? firstName[0] : '');
        lastName = (lastName !== null && lastName !== '' ? lastName[0] : '');
        return this.getDisplayUser(firstName, lastName, '');
    }
}
PeopleSearchFieldComponent.ɵfac = function PeopleSearchFieldComponent_Factory(t) { return new (t || PeopleSearchFieldComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.PeopleProcessService), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TranslationService)); };
PeopleSearchFieldComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PeopleSearchFieldComponent, selectors: [["adf-people-search-field"]], hostAttrs: [1, "adf-people-search-field"], inputs: { performSearch: "performSearch", placeholder: "placeholder" }, outputs: { rowClick: "rowClick" }, decls: 4, vars: 5, consts: [[1, "adf-search-text-container"], ["matInput", "", "type", "text", "id", "userSearchText", "data-automation-id", "adf-people-search-input", 3, "placeholder", "value", "formControl"], [4, "ngIf"], ["class", "adf-search-list-container", "id", "search-people-list", 4, "ngIf"], ["id", "search-people-list", 1, "adf-search-list-container"], [3, "users", "clickRow"], ["key", "firstName"], ["key", "email", 1, "adf-full-width"], ["class", "adf-people-pic", 4, "ngIf"], ["class", "adf-people-img", 3, "alt", "src", 4, "ngIf"], [1, "adf-people-pic"], [1, "adf-people-img", 3, "alt", "src"], [1, "adf-people-full-name"]], template: function PeopleSearchFieldComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "mat-form-field", 0);
        ɵngcc0.ɵɵelement(1, "input", 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(2, PeopleSearchFieldComponent_ng_container_2_Template, 2, 1, "ng-container", 2);
        ɵngcc0.ɵɵpipe(3, "async");
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵpropertyInterpolate("placeholder", ctx.searchPlaceholder);
        ɵngcc0.ɵɵproperty("formControl", ctx.searchUser);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(3, 3, ctx.users$));
    } }, directives: [ɵngcc2.MatFormField, ɵngcc3.MatInput, ɵngcc4.DefaultValueAccessor, ɵngcc4.NgControlStatus, ɵngcc4.FormControlDirective, ɵngcc5.NgIf, ɵngcc6.PeopleListComponent, ɵngcc1.DataColumnListComponent, ɵngcc1.DataColumnComponent], pipes: [ɵngcc5.AsyncPipe], styles: [".adf-people-search-field,.adf-people-search-field .adf-search-text-container{width:100%}.adf-people-search-field .adf-search-text-container input{line-height:normal}.adf-people-search-field .adf-search-list-container{max-height:152px;overflow-y:auto;width:100%}.adf-people-search-field adf-people-list ::ng-deep adf-datatable ::ng-deep thead{display:none}.adf-people-search-field .adf-people-pic{background:var(--theme-primary-color);border-radius:90%;color:#fff;font-size:18px;font-weight:bolder;padding:10px 5px;text-align:center;text-transform:uppercase;vertical-align:text-bottom;width:30px}.adf-people-search-field .adf-people-img{border-radius:90%;height:40px;vertical-align:middle;width:40px}"], encapsulation: 2 });
PeopleSearchFieldComponent.ctorParameters = () => [
    { type: PeopleProcessService },
    { type: TranslationService }
];
PeopleSearchFieldComponent.propDecorators = {
    performSearch: [{ type: Input }],
    placeholder: [{ type: Input }],
    rowClick: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PeopleSearchFieldComponent, [{
        type: Component,
        args: [{
                selector: 'adf-people-search-field',
                template: "<mat-form-field class=\"adf-search-text-container\">\n    <input\n        matInput\n        placeholder=\"{{ searchPlaceholder }}\"\n        type=\"text\"\n        id=\"userSearchText\"\n        [value]=\"\"\n        [formControl]=\"searchUser\"\n        data-automation-id=\"adf-people-search-input\">\n</mat-form-field>\n<ng-container *ngIf=\"users$ | async; let users\">\n    <div class=\"adf-search-list-container\" id=\"search-people-list\" *ngIf=\"users.length > 0\">\n        <adf-people-list [users]=\"users\" (clickRow)=\"onRowClick($event)\">\n            <data-columns>\n                <data-column key=\"firstName\">\n                    <ng-template let-entry=\"$implicit\">\n                            <div *ngIf=\"!entry.row.obj.pictureId\" class=\"adf-people-pic\">\n                                {{getInitialUserName(entry.row.obj.firstName, entry.row.obj.lastName)}}</div>\n                            <div>\n                                <img [alt]=\"getDisplayUser(entry.row.obj.firstName, entry.row.obj.lastName, ' ')\" *ngIf=\"entry.row.obj.pictureId\" class=\"adf-people-img\"\n                                    [src]=\"peopleProcessService.getUserImage(entry.row.obj)\"/>\n                            </div>\n                    </ng-template>\n                </data-column>\n                <data-column key=\"email\" class=\"adf-full-width\">\n                    <ng-template let-entry=\"$implicit\">\n                        <div [attr.data-automation-id]=\"'adf-people-full-name-'+ getDisplayUser(entry.row.obj.firstName, entry.row.obj.lastName, '-')\" class=\"adf-people-full-name\">{{ getDisplayUser(entry.row.obj.firstName, entry.row.obj.lastName, ' ') }}</div>\n                    </ng-template>\n                </data-column>\n            </data-columns>\n        </adf-people-list>\n    </div>\n</ng-container>\n",
                host: { 'class': 'adf-people-search-field' },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-people-search-field,.adf-people-search-field .adf-search-text-container{width:100%}.adf-people-search-field .adf-search-text-container input{line-height:normal}.adf-people-search-field .adf-search-list-container{max-height:152px;overflow-y:auto;width:100%}.adf-people-search-field adf-people-list ::ng-deep adf-datatable ::ng-deep thead{display:none}.adf-people-search-field .adf-people-pic{background:var(--theme-primary-color);border-radius:90%;color:#fff;font-size:18px;font-weight:bolder;padding:10px 5px;text-align:center;text-transform:uppercase;vertical-align:text-bottom;width:30px}.adf-people-search-field .adf-people-img{border-radius:90%;height:40px;vertical-align:middle;width:40px}"]
            }]
    }], function () { return [{ type: ɵngcc1.PeopleProcessService }, { type: ɵngcc1.TranslationService }]; }, { rowClick: [{
            type: Output
        }], performSearch: [{
            type: Input
        }], placeholder: [{
            type: Input
        }] }); })();
