// @alfresco/adf-content-services/esm2015/lib/aspect-list/aspect-list.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AspectListComponent } from './aspect-list.component';
import { MatTableModule } from '@angular/material/table';
import { MatExpansionModule } from '@angular/material/expansion';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { PipeModule } from '@alfresco/adf-core';
import { TranslateModule } from '@ngx-translate/core';
import { MatDialogModule } from '@angular/material/dialog';
import { AspectListDialogComponent } from './aspect-list-dialog.component';
import { MatButtonModule } from '@angular/material/button';
import { MatTooltipModule } from '@angular/material/tooltip';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/cdk/bidi';
import * as ɵngcc3 from '@angular/material/table';
import * as ɵngcc4 from '@angular/material/expansion';
import * as ɵngcc5 from '@angular/material/checkbox';
import * as ɵngcc6 from '@ngx-translate/core';
import * as ɵngcc7 from '@angular/material/dialog';
import * as ɵngcc8 from '@angular/material/button';
import * as ɵngcc9 from '@angular/material/tooltip';
import * as ɵngcc10 from '@angular/cdk/scrolling';
import * as ɵngcc11 from '@angular/material/progress-spinner';
import * as ɵngcc12 from '@alfresco/adf-core';
export class AspectListModule {
}
AspectListModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AspectListModule });
AspectListModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AspectListModule_Factory(t) { return new (t || AspectListModule)(); }, imports: [[
            CommonModule,
            MatTableModule,
            MatExpansionModule,
            MatCheckboxModule,
            PipeModule,
            TranslateModule,
            MatDialogModule,
            MatButtonModule,
            MatTooltipModule,
            MatProgressSpinnerModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AspectListModule, { declarations: function () { return [AspectListComponent,
        AspectListDialogComponent]; }, imports: function () { return [CommonModule,
        MatTableModule,
        MatExpansionModule,
        MatCheckboxModule,
        PipeModule,
        TranslateModule,
        MatDialogModule,
        MatButtonModule,
        MatTooltipModule,
        MatProgressSpinnerModule]; }, exports: function () { return [AspectListComponent,
        AspectListDialogComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AspectListModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MatTableModule,
                    MatExpansionModule,
                    MatCheckboxModule,
                    PipeModule,
                    TranslateModule,
                    MatDialogModule,
                    MatButtonModule,
                    MatTooltipModule,
                    MatProgressSpinnerModule
                ],
                exports: [
                    AspectListComponent,
                    AspectListDialogComponent
                ],
                declarations: [
                    AspectListComponent,
                    AspectListDialogComponent
                ]
            }]
    }], null, null); })();
ɵngcc0.ɵɵsetComponentScope(AspectListDialogComponent, [ɵngcc1.NgClass, ɵngcc1.NgComponentOutlet, ɵngcc1.NgForOf, ɵngcc1.NgIf, ɵngcc1.NgTemplateOutlet, ɵngcc1.NgStyle, ɵngcc1.NgSwitch, ɵngcc1.NgSwitchCase, ɵngcc1.NgSwitchDefault, ɵngcc1.NgPlural, ɵngcc1.NgPluralCase, ɵngcc2.Dir, ɵngcc3.MatTable, ɵngcc3.MatHeaderCellDef, ɵngcc3.MatHeaderRowDef, ɵngcc3.MatColumnDef, ɵngcc3.MatCellDef, ɵngcc3.MatRowDef, ɵngcc3.MatFooterCellDef, ɵngcc3.MatFooterRowDef, ɵngcc3.MatHeaderCell, ɵngcc3.MatCell, ɵngcc3.MatFooterCell, ɵngcc3.MatHeaderRow, ɵngcc3.MatRow, ɵngcc3.MatFooterRow, ɵngcc3.MatNoDataRow, ɵngcc3.MatTextColumn, ɵngcc4.MatAccordion, ɵngcc4.MatExpansionPanel, ɵngcc4.MatExpansionPanelActionRow, ɵngcc4.MatExpansionPanelHeader, ɵngcc4.MatExpansionPanelTitle, ɵngcc4.MatExpansionPanelDescription, ɵngcc4.MatExpansionPanelContent, ɵngcc5.MatCheckbox, ɵngcc5.MatCheckboxRequiredValidator, ɵngcc6.TranslateDirective, ɵngcc7.MatDialogContainer, ɵngcc7.MatDialogClose, ɵngcc7.MatDialogTitle, ɵngcc7.MatDialogContent, ɵngcc7.MatDialogActions, ɵngcc8.MatButton, ɵngcc8.MatAnchor, ɵngcc9.MatTooltip, ɵngcc9.TooltipComponent, ɵngcc10.CdkScrollable, ɵngcc11.MatProgressSpinner, ɵngcc11.MatSpinner, AspectListComponent,
    AspectListDialogComponent], [ɵngcc1.AsyncPipe, ɵngcc1.UpperCasePipe, ɵngcc1.LowerCasePipe, ɵngcc1.JsonPipe, ɵngcc1.SlicePipe, ɵngcc1.DecimalPipe, ɵngcc1.PercentPipe, ɵngcc1.TitleCasePipe, ɵngcc1.CurrencyPipe, ɵngcc1.DatePipe, ɵngcc1.I18nPluralPipe, ɵngcc1.I18nSelectPipe, ɵngcc1.KeyValuePipe, ɵngcc12.FileSizePipe, ɵngcc12.HighlightPipe, ɵngcc12.TimeAgoPipe, ɵngcc12.MimeTypeIconPipe, ɵngcc12.InitialUsernamePipe, ɵngcc12.FullNamePipe, ɵngcc12.NodeNameTooltipPipe, ɵngcc12.FormatSpacePipe, ɵngcc12.FileTypePipe, ɵngcc12.MultiValuePipe, ɵngcc12.LocalizedDatePipe, ɵngcc12.DecimalNumberPipe, ɵngcc12.LocalizedRolePipe, ɵngcc12.MomentDatePipe, ɵngcc12.MomentDateTimePipe, ɵngcc6.TranslatePipe]);
