// @alfresco/adf-content-services/esm2015/lib/aspect-list/node-aspect.service.js
import { Injectable } from '@angular/core';
import { AlfrescoApiService, CardViewUpdateService, NodesApiService } from '@alfresco/adf-core';
import { AspectListService } from './aspect-list.service';
import * as i0 from "@angular/core";
import * as i1 from "@alfresco/adf-core";
import * as i2 from "./aspect-list.service";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
import * as ɵngcc2 from './aspect-list.service';
export class NodeAspectService {
    constructor(alfrescoApiService, nodesApiService, aspectListService, cardViewUpdateService) {
        this.alfrescoApiService = alfrescoApiService;
        this.nodesApiService = nodesApiService;
        this.aspectListService = aspectListService;
        this.cardViewUpdateService = cardViewUpdateService;
    }
    updateNodeAspects(nodeId) {
        this.aspectListService.openAspectListDialog(nodeId).subscribe((aspectList) => {
            this.nodesApiService.updateNode(nodeId, { aspectNames: [...aspectList] }).subscribe((updatedNode) => {
                this.alfrescoApiService.nodeUpdated.next(updatedNode);
                this.cardViewUpdateService.updateNodeAspect(updatedNode);
            });
        });
    }
}
NodeAspectService.ɵfac = function NodeAspectService_Factory(t) { return new (t || NodeAspectService)(ɵngcc0.ɵɵinject(ɵngcc1.AlfrescoApiService), ɵngcc0.ɵɵinject(ɵngcc1.NodesApiService), ɵngcc0.ɵɵinject(ɵngcc2.AspectListService), ɵngcc0.ɵɵinject(ɵngcc1.CardViewUpdateService)); };
NodeAspectService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NodeAspectService_Factory() { return new NodeAspectService(i0.ɵɵinject(i1.AlfrescoApiService), i0.ɵɵinject(i1.NodesApiService), i0.ɵɵinject(i2.AspectListService), i0.ɵɵinject(i1.CardViewUpdateService)); }, token: NodeAspectService, providedIn: "root" });
NodeAspectService.ctorParameters = () => [
    { type: AlfrescoApiService },
    { type: NodesApiService },
    { type: AspectListService },
    { type: CardViewUpdateService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NodeAspectService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.AlfrescoApiService }, { type: ɵngcc1.NodesApiService }, { type: ɵngcc2.AspectListService }, { type: ɵngcc1.CardViewUpdateService }]; }, null); })();
