// @alfresco/adf-core/esm2015/viewer/components/unknown-format/unknown-format.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/icon';
import * as ɵngcc2 from '@ngx-translate/core';
export class UnknownFormatComponent {
}
UnknownFormatComponent.ɵfac = function UnknownFormatComponent_Factory(t) { return new (t || UnknownFormatComponent)(); };
UnknownFormatComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: UnknownFormatComponent, selectors: [["adf-viewer-unknown-format"]], decls: 7, vars: 3, consts: [[1, "adf-viewer__unknown-format-view"], [1, "icon"], [1, "adf-viewer__unknown-label"]], template: function UnknownFormatComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div");
        ɵngcc0.ɵɵelementStart(2, "mat-icon", 1);
        ɵngcc0.ɵɵtext(3, "error");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "div", 2);
        ɵngcc0.ɵɵtext(5);
        ɵngcc0.ɵɵpipe(6, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(5);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(6, 1, "ADF_VIEWER.UNKNOWN_FORMAT"));
    } }, directives: [ɵngcc1.MatIcon], pipes: [ɵngcc2.TranslatePipe], styles: [".adf-viewer__unknown-format-view{color:var(--theme-text-fg-color);display:flex;flex:1;flex-direction:column;height:90vh;justify-content:center;text-align:center}"], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UnknownFormatComponent, [{
        type: Component,
        args: [{
                selector: 'adf-viewer-unknown-format',
                template: "<div class=\"adf-viewer__unknown-format-view\">\n    <div>\n        <mat-icon class=\"icon\">error</mat-icon>\n        <div class=\"adf-viewer__unknown-label\">{{ 'ADF_VIEWER.UNKNOWN_FORMAT' | translate }}</div>\n    </div>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-viewer__unknown-format-view{color:var(--theme-text-fg-color);display:flex;flex:1;flex-direction:column;height:90vh;justify-content:center;text-align:center}"]
            }]
    }], null, null); })();
