// @alfresco/adf-core/esm2015/login/components/login-dialog-panel.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, ViewChild, Output, EventEmitter } from '@angular/core';
import { LoginComponent } from './login.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './login.component';
import * as ɵngcc2 from '../directives/login-header.directive';
import * as ɵngcc3 from '../directives/login-footer.directive';

const _c0 = ["adfLogin"];
function LoginDialogPanelComponent_ng_template_4_Template(rf, ctx) { }
function LoginDialogPanelComponent_ng_template_6_Template(rf, ctx) { }
export class LoginDialogPanelComponent {
    constructor() {
        this.success = new EventEmitter();
    }
    submitForm() {
        this.login.submit();
    }
    onLoginSuccess(event) {
        this.success.emit(event);
    }
    isValid() {
        return this.login && this.login.form ? this.login.form.valid : false;
    }
}
LoginDialogPanelComponent.ɵfac = function LoginDialogPanelComponent_Factory(t) { return new (t || LoginDialogPanelComponent)(); };
LoginDialogPanelComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: LoginDialogPanelComponent, selectors: [["adf-login-dialog-panel"]], viewQuery: function LoginDialogPanelComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.login = _t.first);
    } }, outputs: { success: "success" }, decls: 7, vars: 3, consts: [[1, "adf-panel-login-dialog-component", 3, "showRememberMe", "showLoginActions", "backgroundImageUrl", "success"], ["adfLogin", ""]], template: function LoginDialogPanelComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "adf-login", 0, 1);
        ɵngcc0.ɵɵlistener("success", function LoginDialogPanelComponent_Template_adf_login_success_1_listener($event) { return ctx.onLoginSuccess($event); });
        ɵngcc0.ɵɵelementStart(3, "adf-login-header");
        ɵngcc0.ɵɵtemplate(4, LoginDialogPanelComponent_ng_template_4_Template, 0, 0, "ng-template");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "adf-login-footer");
        ɵngcc0.ɵɵtemplate(6, LoginDialogPanelComponent_ng_template_6_Template, 0, 0, "ng-template");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("showRememberMe", false)("showLoginActions", false)("backgroundImageUrl", "");
    } }, directives: [ɵngcc1.LoginComponent, ɵngcc2.LoginHeaderDirective, ɵngcc3.LoginFooterDirective], styles: [".adf-panel-login-dialog-component .adf-login-content .adf-copyright,.adf-panel-login-dialog-component .adf-login-content .adf-login-button,.adf-panel-login-dialog-component .adf-login-content .adf-login__remember-me,.adf-panel-login-dialog-component .adf-login-content .mat-card-actions{display:none}"], encapsulation: 2 });
LoginDialogPanelComponent.propDecorators = {
    success: [{ type: Output }],
    login: [{ type: ViewChild, args: ['adfLogin', { static: true },] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LoginDialogPanelComponent, [{
        type: Component,
        args: [{
                selector: 'adf-login-dialog-panel',
                template: "<div>\n    <adf-login #adfLogin\n               class=\"adf-panel-login-dialog-component\"\n               [showRememberMe]=\"false\"\n               [showLoginActions]=\"false\"\n               [backgroundImageUrl]=\"''\"\n               (success)=\"onLoginSuccess($event)\">\n        <adf-login-header><ng-template></ng-template></adf-login-header>\n        <adf-login-footer><ng-template></ng-template></adf-login-footer>\n    </adf-login>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-panel-login-dialog-component .adf-login-content .adf-copyright,.adf-panel-login-dialog-component .adf-login-content .adf-login-button,.adf-panel-login-dialog-component .adf-login-content .adf-login__remember-me,.adf-panel-login-dialog-component .adf-login-content .mat-card-actions{display:none}"]
            }]
    }], function () { return []; }, { success: [{
            type: Output
        }], login: [{
            type: ViewChild,
            args: ['adfLogin', { static: true }]
        }] }); })();
