// @alfresco/adf-core/esm2015/form/components/widgets/json/json.widget.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { WidgetComponent } from './../widget.component';
import { MatDialog } from '@angular/material/dialog';
import { EditJsonDialogComponent } from '../../../../dialogs/edit-json/edit-json.dialog';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../../../services/form.service';
import * as ɵngcc2 from '@angular/material/dialog';
import * as ɵngcc3 from '@angular/material/button';
export class JsonWidgetComponent extends WidgetComponent {
    constructor(formService, dialog) {
        super(formService);
        this.formService = formService;
        this.dialog = dialog;
    }
    view() {
        const rawValue = this.field.value;
        const value = typeof rawValue === 'object'
            ? JSON.stringify(rawValue || {}, null, 2)
            : rawValue;
        const settings = {
            title: this.field.name,
            editable: false,
            value
        };
        this.dialog
            .open(EditJsonDialogComponent, {
            data: settings,
            minWidth: '50%',
            minHeight: '50%'
        });
    }
}
JsonWidgetComponent.ɵfac = function JsonWidgetComponent_Factory(t) { return new (t || JsonWidgetComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.MatDialog)); };
JsonWidgetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: JsonWidgetComponent, selectors: [["ng-component"]], hostBindings: function JsonWidgetComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function JsonWidgetComponent_click_HostBindingHandler($event) { return ctx.event($event); })("blur", function JsonWidgetComponent_blur_HostBindingHandler($event) { return ctx.event($event); })("change", function JsonWidgetComponent_change_HostBindingHandler($event) { return ctx.event($event); })("focus", function JsonWidgetComponent_focus_HostBindingHandler($event) { return ctx.event($event); })("focusin", function JsonWidgetComponent_focusin_HostBindingHandler($event) { return ctx.event($event); })("focusout", function JsonWidgetComponent_focusout_HostBindingHandler($event) { return ctx.event($event); })("input", function JsonWidgetComponent_input_HostBindingHandler($event) { return ctx.event($event); })("invalid", function JsonWidgetComponent_invalid_HostBindingHandler($event) { return ctx.event($event); })("select", function JsonWidgetComponent_select_HostBindingHandler($event) { return ctx.event($event); });
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 2, vars: 0, consts: [["mat-raised-button", "", "color", "primary", 3, "click"]], template: function JsonWidgetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "button", 0);
        ɵngcc0.ɵɵlistener("click", function JsonWidgetComponent_Template_button_click_0_listener() { return ctx.view(); });
        ɵngcc0.ɵɵtext(1, "json");
        ɵngcc0.ɵɵelementEnd();
    } }, directives: [ɵngcc3.MatButton], encapsulation: 2 });
JsonWidgetComponent.ctorParameters = () => [
    { type: FormService },
    { type: MatDialog }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(JsonWidgetComponent, [{
        type: Component,
        args: [{
                template: `
        <button mat-raised-button color="primary" (click)="view()">json</button>
    `,
                host: {
                    '(click)': 'event($event)',
                    '(blur)': 'event($event)',
                    '(change)': 'event($event)',
                    '(focus)': 'event($event)',
                    '(focusin)': 'event($event)',
                    '(focusout)': 'event($event)',
                    '(input)': 'event($event)',
                    '(invalid)': 'event($event)',
                    '(select)': 'event($event)'
                },
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.FormService }, { type: ɵngcc2.MatDialog }]; }, null); })();
