// @alfresco/adf-content-services/esm2015/lib/permission-manager/components/user-role-column/user-role-column.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/material/form-field';
import * as ɵngcc3 from '@angular/material/select';
import * as ɵngcc4 from '@angular/material/core';
import * as ɵngcc5 from '@ngx-translate/core';
import * as ɵngcc6 from '@alfresco/adf-core';

function UserRoleColumnComponent_mat_form_field_0_mat_option_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-option", 5);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "adfLocalizedRole");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const role_r3 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("value", role_r3.role);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 2, role_r3.label), " ");
} }
function UserRoleColumnComponent_mat_form_field_0_Template(rf, ctx) { if (rf & 1) {
    const _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "mat-form-field", 2);
    ɵngcc0.ɵɵelementStart(1, "mat-select", 3);
    ɵngcc0.ɵɵlistener("click", function UserRoleColumnComponent_mat_form_field_0_Template_mat_select_click_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r5); return $event.stopPropagation(); })("selectionChange", function UserRoleColumnComponent_mat_form_field_0_Template_mat_select_selectionChange_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r5); const ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.onRoleChanged($event.value); });
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵtemplate(3, UserRoleColumnComponent_mat_form_field_0_mat_option_3_Template, 3, 4, "mat-option", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("placeholder", ɵngcc0.ɵɵpipeBind1(2, 3, ctx_r0.placeholder))("value", ctx_r0.value);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.roles);
} }
function UserRoleColumnComponent_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 6);
    ɵngcc0.ɵɵpipe(1, "adfLocalizedRole");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "adfLocalizedRole");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("title", ɵngcc0.ɵɵpipeBind1(1, 2, ctx_r1.value));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(3, 4, ctx_r1.value), " ");
} }
export class UserRoleColumnComponent {
    constructor() {
        this.readonly = false;
        this.placeholder = 'PERMISSION_MANAGER.LABELS.SELECT-ROLE';
        this.roleChanged = new EventEmitter();
    }
    onRoleChanged(newRole) {
        this.value = newRole;
        this.roleChanged.emit(newRole);
    }
}
UserRoleColumnComponent.ɵfac = function UserRoleColumnComponent_Factory(t) { return new (t || UserRoleColumnComponent)(); };
UserRoleColumnComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: UserRoleColumnComponent, selectors: [["adf-user-role-column"]], hostAttrs: [1, "adf-user-role-column", "adf-datatable-content-cell", "adf-expand-cell-4"], inputs: { readonly: "readonly", placeholder: "placeholder", value: "value", roles: "roles" }, outputs: { roleChanged: "roleChanged" }, decls: 2, vars: 2, consts: [["floatLabel", "never", "class", "adf-role-selector-field", 4, "ngIf"], ["class", "adf-datatable-cell-value adf-readonly-role", 3, "title", 4, "ngIf"], ["floatLabel", "never", 1, "adf-role-selector-field"], [3, "placeholder", "value", "click", "selectionChange"], [3, "value", 4, "ngFor", "ngForOf"], [3, "value"], [1, "adf-datatable-cell-value", "adf-readonly-role", 3, "title"]], template: function UserRoleColumnComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, UserRoleColumnComponent_mat_form_field_0_Template, 4, 5, "mat-form-field", 0);
        ɵngcc0.ɵɵtemplate(1, UserRoleColumnComponent_span_1_Template, 4, 6, "span", 1);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", !ctx.readonly);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.readonly);
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.MatFormField, ɵngcc3.MatSelect, ɵngcc1.NgForOf, ɵngcc4.MatOption], pipes: [ɵngcc5.TranslatePipe, ɵngcc6.LocalizedRolePipe], styles: [".adf-role-selector-field[_ngcontent-%COMP%] {\n            width: 100%;\n            .mat-form-field {\n                width: 100%;\n                max-width: 200px;\n            }\n        }\n        .adf-readonly-role[_ngcontent-%COMP%] {\n            padding-left: 0 !important;\n        }"] });
UserRoleColumnComponent.propDecorators = {
    roles: [{ type: Input }],
    value: [{ type: Input }],
    readonly: [{ type: Input }],
    placeholder: [{ type: Input }],
    roleChanged: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UserRoleColumnComponent, [{
        type: Component,
        args: [{
                selector: 'adf-user-role-column',
                template: `
      <mat-form-field floatLabel="never" class="adf-role-selector-field" *ngIf="!readonly">
          <mat-select
              (click)="$event.stopPropagation()"
              [placeholder]="placeholder | translate"
              [value]="value"
              (selectionChange)="onRoleChanged($event.value)">
              <mat-option *ngFor="let role of roles" [value]="role.role">
                  {{ role.label | adfLocalizedRole }}
              </mat-option>
          </mat-select>
      </mat-form-field>

      <span class="adf-datatable-cell-value adf-readonly-role" [title]="value | adfLocalizedRole" *ngIf="readonly">
          {{value | adfLocalizedRole}}
      </span>
  `,
                host: { class: 'adf-user-role-column adf-datatable-content-cell adf-expand-cell-4' },
                styles: [`.adf-role-selector-field {
            width: 100%;
            .mat-form-field {
                width: 100%;
                max-width: 200px;
            }
        }
        .adf-readonly-role {
            padding-left: 0 !important;
        }
        `]
            }]
    }], function () { return []; }, { readonly: [{
            type: Input
        }], placeholder: [{
            type: Input
        }], roleChanged: [{
            type: Output
        }], value: [{
            type: Input
        }], roles: [{
            type: Input
        }] }); })();
