// @alfresco/adf-core/esm2015/about/about-server-settings/about-server-settings.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { AppConfigService, AppConfigValues } from '../../app-config/app-config.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../app-config/app-config.service';
import * as ɵngcc2 from '@angular/material/card';
import * as ɵngcc3 from '@ngx-translate/core';

const _c0 = function (a0) { return { value: a0 }; };
export class AboutServerSettingsComponent {
    constructor(appConfig) {
        this.appConfig = appConfig;
        this.ecmHost = '';
        this.bpmHost = '';
    }
    ngOnInit() {
        this.ecmHost = this.appConfig.get(AppConfigValues.ECMHOST);
        this.bpmHost = this.appConfig.get(AppConfigValues.BPMHOST);
    }
}
AboutServerSettingsComponent.ɵfac = function AboutServerSettingsComponent_Factory(t) { return new (t || AboutServerSettingsComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.AppConfigService)); };
AboutServerSettingsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AboutServerSettingsComponent, selectors: [["adf-about-server-settings"]], decls: 14, vars: 18, consts: [[1, "adf-github-link-container"], ["data-automation-id", "adf-about-setting-title"], ["data-automation-id", "adf-process-service-host"], ["data-automation-id", "adf-content-service-host"]], template: function AboutServerSettingsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "h3", 1);
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵpipe(3, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "small");
        ɵngcc0.ɵɵtext(5);
        ɵngcc0.ɵɵpipe(6, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "mat-card");
        ɵngcc0.ɵɵelementStart(8, "p", 2);
        ɵngcc0.ɵɵtext(9);
        ɵngcc0.ɵɵpipe(10, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "p", 3);
        ɵngcc0.ɵɵtext(12);
        ɵngcc0.ɵɵpipe(13, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 4, "ABOUT.SERVER_SETTINGS.TITLE"));
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(6, 6, "ABOUT.SERVER_SETTINGS.DESCRIPTION"));
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind2(10, 8, "ABOUT.SERVER_SETTINGS.PROCESS_SERVICE_HOST", ɵngcc0.ɵɵpureFunction1(14, _c0, ctx.bpmHost)), " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind2(13, 11, "ABOUT.SERVER_SETTINGS.CONTENT_SERVICE_HOST", ɵngcc0.ɵɵpureFunction1(16, _c0, ctx.ecmHost)), " ");
    } }, directives: [ɵngcc2.MatCard], pipes: [ɵngcc3.TranslatePipe], encapsulation: 2 });
AboutServerSettingsComponent.ctorParameters = () => [
    { type: AppConfigService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AboutServerSettingsComponent, [{
        type: Component,
        args: [{
                selector: 'adf-about-server-settings',
                template: "<div class=\"adf-github-link-container\">\n    <h3 data-automation-id=\"adf-about-setting-title\">{{ 'ABOUT.SERVER_SETTINGS.TITLE' | translate }}</h3>\n    <small>{{ 'ABOUT.SERVER_SETTINGS.DESCRIPTION' | translate }}</small>\n    <mat-card>\n        <p data-automation-id=\"adf-process-service-host\">\n            {{ 'ABOUT.SERVER_SETTINGS.PROCESS_SERVICE_HOST' | translate: {value: bpmHost} }}\n        </p>\n        <p data-automation-id=\"adf-content-service-host\">\n            {{ 'ABOUT.SERVER_SETTINGS.CONTENT_SERVICE_HOST' | translate: {value: ecmHost} }}\n        </p>\n    </mat-card>\n</div>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.AppConfigService }]; }, null); })();
