// @alfresco/adf-core/esm2015/card-view/components/base-card-view.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Input, Directive } from '@angular/core';
import { CardViewUpdateService } from '../services/card-view-update.service';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/card-view-update.service';
export class BaseCardView {
    constructor(cardViewUpdateService) {
        this.cardViewUpdateService = cardViewUpdateService;
        this.destroy$ = new Subject();
        this.cardViewUpdateService.updateItem$
            .pipe(takeUntil(this.destroy$))
            .subscribe((itemModel) => {
            if (this.property.key === itemModel.key) {
                this.property.value = itemModel.value;
            }
        });
    }
    ngOnDestroy() {
        this.destroy$.next(true);
        this.destroy$.complete();
    }
}
BaseCardView.ɵfac = function BaseCardView_Factory(t) { return new (t || BaseCardView)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CardViewUpdateService)); };
BaseCardView.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BaseCardView, inputs: { property: "property" } });
BaseCardView.ctorParameters = () => [
    { type: CardViewUpdateService }
];
BaseCardView.propDecorators = {
    property: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BaseCardView, [{
        type: Directive
    }], function () { return [{ type: ɵngcc1.CardViewUpdateService }]; }, { property: [{
            type: Input
        }] }); })();
