// @alfresco/adf-core/esm2015/viewer/components/media-player.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation, Output, EventEmitter } from '@angular/core';
import { ContentService } from '../../services/content.service';
import { ViewUtilService } from '../services/view-util.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/content.service';
import * as ɵngcc2 from '../services/view-util.service';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@angular/flex-layout/extended';

function MediaPlayerComponent_track_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "track", 3);
} if (rf & 2) {
    const track_r1 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("kind", track_r1.kind)("label", track_r1.label)("srclang", track_r1.srclang)("src", track_r1.src, ɵngcc0.ɵɵsanitizeUrl);
} }
const _c0 = function (a0) { return { "adf-audio-file": a0 }; };
export class MediaPlayerComponent {
    constructor(contentService, viewUtils) {
        this.contentService = contentService;
        this.viewUtils = viewUtils;
        this.tracks = [];
        this.error = new EventEmitter();
    }
    ngOnChanges(changes) {
        const blobFile = changes['blobFile'];
        const nodeId = changes['nodeId'];
        if (blobFile && blobFile.currentValue) {
            this.urlFile = this.contentService.createTrustedUrl(this.blobFile);
            return;
        }
        if (nodeId && nodeId.currentValue) {
            this.viewUtils.generateMediaTracks(this.nodeId).then((tracks) => this.tracks = tracks);
        }
        if (!this.urlFile && !this.blobFile) {
            throw new Error('Attribute urlFile or blobFile is required');
        }
    }
    onMediaPlayerError() {
        this.error.emit();
    }
}
MediaPlayerComponent.ɵfac = function MediaPlayerComponent_Factory(t) { return new (t || MediaPlayerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ContentService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.ViewUtilService)); };
MediaPlayerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MediaPlayerComponent, selectors: [["adf-media-player"]], hostAttrs: [1, "adf-media-player"], inputs: { tracks: "tracks", urlFile: "urlFile", blobFile: "blobFile", mimeType: "mimeType", nameFile: "nameFile", nodeId: "nodeId" }, outputs: { error: "error" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 3, vars: 6, consts: [["controls", "", 3, "ngClass"], [3, "src", "type", "error"], [3, "kind", "label", "srclang", "src", 4, "ngFor", "ngForOf"], [3, "kind", "label", "srclang", "src"]], template: function MediaPlayerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "video", 0);
        ɵngcc0.ɵɵelementStart(1, "source", 1);
        ɵngcc0.ɵɵlistener("error", function MediaPlayerComponent_Template_source_error_1_listener() { return ctx.onMediaPlayerError(); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(2, MediaPlayerComponent_track_2_Template, 1, 4, "track", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(4, _c0, ctx.mimeType && ctx.mimeType.startsWith("audio")));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("src", ctx.urlFile, ɵngcc0.ɵɵsanitizeUrl)("type", ctx.mimeType);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.tracks);
    } }, directives: [ɵngcc3.NgClass, ɵngcc4.DefaultClassDirective, ɵngcc3.NgForOf], styles: [".adf-media-player{display:flex}.adf-media-player video{display:flex;flex:1;max-height:90vh;max-width:100%}.adf-media-player video.adf-audio-file::-webkit-media-text-track-container{transform:translateY(-50%)!important}"], encapsulation: 2 });
MediaPlayerComponent.ctorParameters = () => [
    { type: ContentService },
    { type: ViewUtilService }
];
MediaPlayerComponent.propDecorators = {
    urlFile: [{ type: Input }],
    blobFile: [{ type: Input }],
    mimeType: [{ type: Input }],
    nameFile: [{ type: Input }],
    nodeId: [{ type: Input }],
    tracks: [{ type: Input }],
    error: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MediaPlayerComponent, [{
        type: Component,
        args: [{
                selector: 'adf-media-player',
                template: "<video controls [ngClass]=\"{'adf-audio-file': mimeType && mimeType.startsWith('audio')}\">\n    <source [src]=\"urlFile\" [type]=\"mimeType\" (error)=\"onMediaPlayerError()\"/>\n    <track *ngFor=\"let track of tracks\" [kind]=\"track.kind\" [label]=\"track.label\" [srclang]=\"track.srclang\" [src]=\"track.src\"/>\n</video>\n",
                host: { 'class': 'adf-media-player' },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-media-player{display:flex}.adf-media-player video{display:flex;flex:1;max-height:90vh;max-width:100%}.adf-media-player video.adf-audio-file::-webkit-media-text-track-container{transform:translateY(-50%)!important}"]
            }]
    }], function () { return [{ type: ɵngcc1.ContentService }, { type: ɵngcc2.ViewUtilService }]; }, { tracks: [{
            type: Input
        }], error: [{
            type: Output
        }], urlFile: [{
            type: Input
        }], blobFile: [{
            type: Input
        }], mimeType: [{
            type: Input
        }], nameFile: [{
            type: Input
        }], nodeId: [{
            type: Input
        }] }); })();
