// @alfresco/adf-core/esm2015/viewer/components/pdf-viewer-thumb.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ElementRef, Input, ViewEncapsulation } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/platform-browser';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@ngx-translate/core';

const _c0 = function (a0) { return { pageNum: a0 }; };
function PdfThumbComponent_ng_container_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "img", 1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵpipe(3, "translate");
    ɵngcc0.ɵɵpipe(4, "translate");
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const image_r1 = ctx.ngIf;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind2(3, 7, "ADF_VIEWER.SIDEBAR.THUMBNAILS.PAGE", ɵngcc0.ɵɵpureFunction1(13, _c0, ctx_r0.page.id)));
    ɵngcc0.ɵɵproperty("src", image_r1, ɵngcc0.ɵɵsanitizeUrl)("alt", ɵngcc0.ɵɵpipeBind2(2, 4, "ADF_VIEWER.SIDEBAR.THUMBNAILS.PAGE", ɵngcc0.ɵɵpureFunction1(15, _c0, ctx_r0.page.id)));
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind2(4, 10, "ADF_VIEWER.SIDEBAR.THUMBNAILS.PAGE", ɵngcc0.ɵɵpureFunction1(17, _c0, ctx_r0.page.id)));
} }
export class PdfThumbComponent {
    constructor(sanitizer, element) {
        this.sanitizer = sanitizer;
        this.element = element;
        this.page = null;
    }
    ngOnInit() {
        this.image$ = this.page.getPage().then((page) => this.getThumb(page));
    }
    focus() {
        this.element.nativeElement.focus();
    }
    getThumb(page) {
        const viewport = page.getViewport({ scale: 1 });
        const canvas = this.getCanvas();
        const scale = Math.min((canvas.height / viewport.height), (canvas.width / viewport.width));
        return page.render({
            canvasContext: canvas.getContext('2d'),
            viewport: page.getViewport({ scale: scale })
        }).promise.then(() => {
            const imageSource = canvas.toDataURL();
            return this.sanitizer.bypassSecurityTrustUrl(imageSource);
        });
    }
    getCanvas() {
        const canvas = document.createElement('canvas');
        canvas.width = this.page.getWidth();
        canvas.height = this.page.getHeight();
        return canvas;
    }
}
PdfThumbComponent.ɵfac = function PdfThumbComponent_Factory(t) { return new (t || PdfThumbComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DomSanitizer), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
PdfThumbComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PdfThumbComponent, selectors: [["adf-pdf-thumb"]], hostAttrs: ["tabindex", "0"], inputs: { page: "page" }, decls: 2, vars: 3, consts: [[4, "ngIf"], [3, "src", "alt", "title"]], template: function PdfThumbComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, PdfThumbComponent_ng_container_0_Template, 5, 19, "ng-container", 0);
        ɵngcc0.ɵɵpipe(1, "async");
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(1, 1, ctx.image$));
    } }, directives: [ɵngcc2.NgIf], pipes: [ɵngcc2.AsyncPipe, ɵngcc3.TranslatePipe], encapsulation: 2 });
PdfThumbComponent.ctorParameters = () => [
    { type: DomSanitizer },
    { type: ElementRef }
];
PdfThumbComponent.propDecorators = {
    page: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PdfThumbComponent, [{
        type: Component,
        args: [{
                selector: 'adf-pdf-thumb',
                template: "<ng-container *ngIf=\"image$ | async as image\">\n    <img [src]=\"image\" [alt]=\"'ADF_VIEWER.SIDEBAR.THUMBNAILS.PAGE' | translate: { pageNum: page.id }\"\n        title=\"{{ 'ADF_VIEWER.SIDEBAR.THUMBNAILS.PAGE' | translate: { pageNum: page.id } }}\"\n        [attr.aria-label]=\"'ADF_VIEWER.SIDEBAR.THUMBNAILS.PAGE' | translate: { pageNum: page.id }\">\n</ng-container>\n",
                encapsulation: ViewEncapsulation.None,
                host: { tabindex: '0' }
            }]
    }], function () { return [{ type: ɵngcc1.DomSanitizer }, { type: ɵngcc0.ElementRef }]; }, { page: [{
            type: Input
        }] }); })();
