// @alfresco/adf-core/esm2015/viewer/components/viewer-sidebar.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ChangeDetectionStrategy, Component, HostListener, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class ViewerSidebarComponent {
    onKeyDown(event) {
        event.cancelBubble = true;
    }
    onKeyUp(event) {
        event.cancelBubble = true;
    }
}
ViewerSidebarComponent.ɵfac = function ViewerSidebarComponent_Factory(t) { return new (t || ViewerSidebarComponent)(); };
ViewerSidebarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ViewerSidebarComponent, selectors: [["adf-viewer-sidebar"]], hostAttrs: [1, "adf-viewer-sidebar"], hostBindings: function ViewerSidebarComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("keydown", function ViewerSidebarComponent_keydown_HostBindingHandler($event) { return ctx.onKeyDown($event); })("keyup", function ViewerSidebarComponent_keyup_HostBindingHandler($event) { return ctx.onKeyUp($event); });
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function ViewerSidebarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2, changeDetection: 0 });
ViewerSidebarComponent.propDecorators = {
    onKeyDown: [{ type: HostListener, args: ['keydown', ['$event'],] }],
    onKeyUp: [{ type: HostListener, args: ['keyup', ['$event'],] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ViewerSidebarComponent, [{
        type: Component,
        args: [{
                selector: 'adf-viewer-sidebar',
                encapsulation: ViewEncapsulation.None,
                changeDetection: ChangeDetectionStrategy.OnPush,
                host: { class: 'adf-viewer-sidebar' },
                template: `<ng-content></ng-content>`
            }]
    }], null, { onKeyDown: [{
            type: HostListener,
            args: ['keydown', ['$event']]
        }], onKeyUp: [{
            type: HostListener,
            args: ['keyup', ['$event']]
        }] }); })();
