// @alfresco/adf-core/esm2015/viewer/components/viewer-open-with.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class ViewerOpenWithComponent {
}
ViewerOpenWithComponent.ɵfac = function ViewerOpenWithComponent_Factory(t) { return new (t || ViewerOpenWithComponent)(); };
ViewerOpenWithComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ViewerOpenWithComponent, selectors: [["adf-viewer-open-with"]], hostAttrs: [1, "adf-viewer-open-with"], ngContentSelectors: _c0, decls: 1, vars: 0, template: function ViewerOpenWithComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ViewerOpenWithComponent, [{
        type: Component,
        args: [{
                selector: 'adf-viewer-open-with',
                encapsulation: ViewEncapsulation.None,
                changeDetection: ChangeDetectionStrategy.OnPush,
                host: { class: 'adf-viewer-open-with' },
                template: `<ng-content></ng-content>`
            }]
    }], null, null); })();
