// @alfresco/adf-core/esm2015/app-config/app-config.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { AppConfigPipe } from './app-config.pipe';
import * as ɵngcc0 from '@angular/core';
export class AppConfigModule {
}
AppConfigModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AppConfigModule });
AppConfigModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AppConfigModule_Factory(t) { return new (t || AppConfigModule)(); }, imports: [[
            HttpClientModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AppConfigModule, { declarations: function () { return [AppConfigPipe]; }, imports: function () { return [HttpClientModule]; }, exports: function () { return [AppConfigPipe]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppConfigModule, [{
        type: NgModule,
        args: [{
                imports: [
                    HttpClientModule
                ],
                declarations: [
                    AppConfigPipe
                ],
                exports: [
                    AppConfigPipe
                ]
            }]
    }], null, null); })();
