// @alfresco/adf-core/esm2015/app-config/app-config.pipe.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Pipe } from '@angular/core';
import { AppConfigService } from './app-config.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './app-config.service';
export class AppConfigPipe {
    constructor(config) {
        this.config = config;
    }
    transform(value, fallback) {
        return this.config.get(value, fallback);
    }
}
AppConfigPipe.ɵfac = function AppConfigPipe_Factory(t) { return new (t || AppConfigPipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.AppConfigService)); };
AppConfigPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "adfAppConfig", type: AppConfigPipe, pure: true });
AppConfigPipe.ctorParameters = () => [
    { type: AppConfigService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppConfigPipe, [{
        type: Pipe,
        args: [{
                name: 'adfAppConfig'
            }]
    }], function () { return [{ type: ɵngcc1.AppConfigService }]; }, null); })();
