// @alfresco/adf-content-services/esm2015/lib/upload/pipes/file-upload-error.pipe.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Pipe } from '@angular/core';
import { TranslationService } from '@alfresco/adf-core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
export class FileUploadErrorPipe {
    constructor(translation) {
        this.translation = translation;
    }
    transform(errorCode) {
        return this.translation.instant(`FILE_UPLOAD.ERRORS.${errorCode || 'GENERIC'}`);
    }
}
FileUploadErrorPipe.ɵfac = function FileUploadErrorPipe_Factory(t) { return new (t || FileUploadErrorPipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TranslationService)); };
FileUploadErrorPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "adfFileUploadError", type: FileUploadErrorPipe, pure: true });
FileUploadErrorPipe.ctorParameters = () => [
    { type: TranslationService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FileUploadErrorPipe, [{
        type: Pipe,
        args: [{
                name: 'adfFileUploadError',
                pure: true
            }]
    }], function () { return [{ type: ɵngcc1.TranslationService }]; }, null); })();
