// @alfresco/adf-core/esm2015/settings/host-settings.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Output, ViewEncapsulation, Input } from '@angular/core';
import { Validators, FormBuilder, FormControl } from '@angular/forms';
import { AppConfigService, AppConfigValues } from '../app-config/app-config.service';
import { StorageService } from '../services/storage.service';
import { AlfrescoApiService } from '../services/alfresco-api.service';
import { ENTER } from '@angular/cdk/keycodes';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/forms';
import * as ɵngcc2 from '../services/storage.service';
import * as ɵngcc3 from '../services/alfresco-api.service';
import * as ɵngcc4 from '../app-config/app-config.service';
import * as ɵngcc5 from '@angular/material/toolbar';
import * as ɵngcc6 from '@angular/material/card';
import * as ɵngcc7 from '@angular/common';
import * as ɵngcc8 from '@angular/material/radio';
import * as ɵngcc9 from '@angular/material/button';
import * as ɵngcc10 from '@angular/material/form-field';
import * as ɵngcc11 from '@angular/material/select';
import * as ɵngcc12 from '@angular/material/core';
import * as ɵngcc13 from '@angular/material/input';
import * as ɵngcc14 from '@angular/material/slide-toggle';
import * as ɵngcc15 from '@ngx-translate/core';

function HostSettingsComponent_mat_form_field_7_mat_option_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-option", 14);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const provider_r6 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("value", provider_r6);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", provider_r6, " ");
} }
function HostSettingsComponent_mat_form_field_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-form-field");
    ɵngcc0.ɵɵelementStart(1, "mat-select", 12);
    ɵngcc0.ɵɵtemplate(2, HostSettingsComponent_mat_form_field_7_mat_option_2_Template, 2, 2, "mat-option", 13);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("formControl", ctx_r0.providersControl);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.providers);
} }
function HostSettingsComponent_ng_container_19_mat_error_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.NOT_VALID"), " ");
} }
function HostSettingsComponent_ng_container_19_mat_error_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.REQUIRED"), " ");
} }
function HostSettingsComponent_ng_container_19_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "mat-card-content");
    ɵngcc0.ɵɵelementStart(2, "mat-form-field", 15);
    ɵngcc0.ɵɵelementStart(3, "mat-label");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(6, "input", 16);
    ɵngcc0.ɵɵtemplate(7, HostSettingsComponent_ng_container_19_mat_error_7_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵtemplate(8, HostSettingsComponent_ng_container_19_mat_error_8_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(9, "p");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(5, 4, "CORE.HOST_SETTINGS.CS-HOST"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("formControl", ctx_r1.ecmHost);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.ecmHost.hasError("pattern"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.ecmHost.hasError("required"));
} }
function HostSettingsComponent_ng_container_20_mat_error_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.NOT_VALID"), " ");
} }
function HostSettingsComponent_ng_container_20_mat_error_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.REQUIRED"), " ");
} }
function HostSettingsComponent_ng_container_20_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "mat-card-content");
    ɵngcc0.ɵɵelementStart(2, "mat-form-field", 15);
    ɵngcc0.ɵɵelementStart(3, "mat-label");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(6, "input", 17);
    ɵngcc0.ɵɵtemplate(7, HostSettingsComponent_ng_container_20_mat_error_7_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵtemplate(8, HostSettingsComponent_ng_container_20_mat_error_8_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(5, 4, "CORE.HOST_SETTINGS.BP-HOST"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("formControl", ctx_r2.bpmHost);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r2.bpmHost.hasError("pattern"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r2.bpmHost.hasError("required"));
} }
function HostSettingsComponent_ng_container_21_mat_error_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.NOT_VALID"), " ");
} }
function HostSettingsComponent_ng_container_21_mat_error_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.REQUIRED"), " ");
} }
function HostSettingsComponent_ng_container_21_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "mat-card-content");
    ɵngcc0.ɵɵelementStart(2, "mat-form-field", 15);
    ɵngcc0.ɵɵelementStart(3, "mat-label");
    ɵngcc0.ɵɵtext(4, "Identity Host");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(5, "input", 18);
    ɵngcc0.ɵɵtemplate(6, HostSettingsComponent_ng_container_21_mat_error_6_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵtemplate(7, HostSettingsComponent_ng_container_21_mat_error_7_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(6);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r3.identityHost.hasError("pattern"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r3.identityHost.hasError("required"));
} }
function HostSettingsComponent_ng_container_22_mat_error_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.NOT_VALID"), " ");
} }
function HostSettingsComponent_ng_container_22_mat_error_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.REQUIRED"), " ");
} }
function HostSettingsComponent_ng_container_22_mat_error_13_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.REQUIRED"), " ");
} }
function HostSettingsComponent_ng_container_22_mat_error_20_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.REQUIRED"), " ");
} }
function HostSettingsComponent_ng_container_22_mat_error_35_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.HOST_SETTINGS.REQUIRED"), " ");
} }
function HostSettingsComponent_ng_container_22_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "div", 19);
    ɵngcc0.ɵɵelementStart(2, "mat-form-field", 15);
    ɵngcc0.ɵɵelementStart(3, "mat-label");
    ɵngcc0.ɵɵtext(4, "Auth Host");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(5, "input", 20);
    ɵngcc0.ɵɵtemplate(6, HostSettingsComponent_ng_container_22_mat_error_6_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵtemplate(7, HostSettingsComponent_ng_container_22_mat_error_7_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(8, "mat-form-field", 15);
    ɵngcc0.ɵɵelementStart(9, "mat-label");
    ɵngcc0.ɵɵtext(10);
    ɵngcc0.ɵɵpipe(11, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(12, "input", 21);
    ɵngcc0.ɵɵtemplate(13, HostSettingsComponent_ng_container_22_mat_error_13_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(14, "mat-form-field", 15);
    ɵngcc0.ɵɵelementStart(15, "mat-label");
    ɵngcc0.ɵɵtext(16);
    ɵngcc0.ɵɵpipe(17, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(18, "input", 22);
    ɵngcc0.ɵɵpipe(19, "translate");
    ɵngcc0.ɵɵtemplate(20, HostSettingsComponent_ng_container_22_mat_error_20_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(21, "label", 23);
    ɵngcc0.ɵɵtext(22);
    ɵngcc0.ɵɵpipe(23, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(24, "mat-slide-toggle", 24);
    ɵngcc0.ɵɵelementStart(25, "label", 25);
    ɵngcc0.ɵɵtext(26);
    ɵngcc0.ɵɵpipe(27, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(28, "mat-slide-toggle", 26);
    ɵngcc0.ɵɵelementStart(29, "mat-form-field", 15);
    ɵngcc0.ɵɵelementStart(30, "mat-label");
    ɵngcc0.ɵɵtext(31);
    ɵngcc0.ɵɵpipe(32, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(33, "input", 27);
    ɵngcc0.ɵɵpipe(34, "translate");
    ɵngcc0.ɵɵtemplate(35, HostSettingsComponent_ng_container_22_mat_error_35_Template, 3, 3, "mat-error", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(36, "mat-form-field", 15);
    ɵngcc0.ɵɵelementStart(37, "mat-label");
    ɵngcc0.ɵɵtext(38);
    ɵngcc0.ɵɵpipe(39, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(40, "input", 28);
    ɵngcc0.ɵɵpipe(41, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(42, "mat-form-field", 15);
    ɵngcc0.ɵɵelementStart(43, "mat-label");
    ɵngcc0.ɵɵtext(44);
    ɵngcc0.ɵɵpipe(45, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(46, "input", 29);
    ɵngcc0.ɵɵpipe(47, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(6);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r4.host.hasError("pattern"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r4.host.hasError("required"));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(11, 18, "CORE.HOST_SETTINGS.CLIENT"));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r4.clientId.hasError("required"));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(17, 20, "CORE.HOST_SETTINGS.SCOPE"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵpropertyInterpolate("name", ɵngcc0.ɵɵpipeBind1(19, 22, "CORE.HOST_SETTINGS.SCOPE"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r4.scope.hasError("required"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(23, 24, "CORE.HOST_SETTINGS.SILENT"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("color", "primary");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(27, 26, "CORE.HOST_SETTINGS.IMPLICIT-FLOW"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("color", "primary");
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(32, 28, "CORE.HOST_SETTINGS.REDIRECT"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵpropertyInterpolate("placeholder", ɵngcc0.ɵɵpipeBind1(34, 30, "CORE.HOST_SETTINGS.REDIRECT"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r4.redirectUri.hasError("required"));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(39, 32, "CORE.HOST_SETTINGS.REDIRECT_LOGOUT"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵpropertyInterpolate("placeholder", ɵngcc0.ɵɵpipeBind1(41, 34, "CORE.HOST_SETTINGS.REDIRECT_LOGOUT"));
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(45, 36, "CORE.HOST_SETTINGS.PUBLIC_URLS"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵpropertyInterpolate("placeholder", ɵngcc0.ɵɵpipeBind1(47, 38, "CORE.HOST_SETTINGS.PUBLIC_URLS"));
} }
export class HostSettingsComponent {
    constructor(formBuilder, storageService, alfrescoApiService, appConfig) {
        this.formBuilder = formBuilder;
        this.storageService = storageService;
        this.alfrescoApiService = alfrescoApiService;
        this.appConfig = appConfig;
        this.HOST_REGEX = '^(http|https):\/\/.*[^/]$';
        this.providers = ['BPM', 'ECM', 'ALL'];
        this.showSelectProviders = true;
        this.error = new EventEmitter();
        this.cancel = new EventEmitter();
        this.success = new EventEmitter();
    }
    ngOnInit() {
        if (this.providers.length === 1) {
            this.showSelectProviders = false;
        }
        const providerSelected = this.appConfig.get(AppConfigValues.PROVIDERS);
        const authType = this.appConfig.get(AppConfigValues.AUTHTYPE, 'BASIC');
        this.form = this.formBuilder.group({
            providersControl: [providerSelected, Validators.required],
            authType: authType
        });
        this.addFormGroups();
        if (authType === 'OAUTH') {
            this.addOAuthFormGroup();
            this.addIdentityHostFormControl();
        }
        this.form.get('authType').valueChanges.subscribe((value) => {
            if (value === 'BASIC') {
                this.form.removeControl('oauthConfig');
                this.form.removeControl('identityHost');
            }
            else {
                this.addOAuthFormGroup();
                this.addIdentityHostFormControl();
            }
        });
        this.providersControl.valueChanges.subscribe(() => {
            this.removeFormGroups();
            this.addFormGroups();
        });
    }
    removeFormGroups() {
        this.form.removeControl('bpmHost');
        this.form.removeControl('ecmHost');
    }
    addFormGroups() {
        this.addBPMFormControl();
        this.addECMFormControl();
    }
    addOAuthFormGroup() {
        const oauthFormGroup = this.createOAuthFormGroup();
        this.form.addControl('oauthConfig', oauthFormGroup);
    }
    addBPMFormControl() {
        if ((this.isBPM() || this.isALL() || this.isOAUTH()) && !this.bpmHost) {
            const bpmFormControl = this.createBPMFormControl();
            this.form.addControl('bpmHost', bpmFormControl);
        }
    }
    addIdentityHostFormControl() {
        const identityHostFormControl = this.createIdentityFormControl();
        this.form.addControl('identityHost', identityHostFormControl);
    }
    addECMFormControl() {
        if ((this.isECM() || this.isALL()) && !this.ecmHost) {
            const ecmFormControl = this.createECMFormControl();
            this.form.addControl('ecmHost', ecmFormControl);
        }
    }
    createOAuthFormGroup() {
        const oauth = this.appConfig.get(AppConfigValues.OAUTHCONFIG, {});
        return this.formBuilder.group({
            host: [oauth.host, [Validators.required, Validators.pattern(this.HOST_REGEX)]],
            clientId: [oauth.clientId, Validators.required],
            redirectUri: [oauth.redirectUri, Validators.required],
            redirectUriLogout: [oauth.redirectUriLogout],
            scope: [oauth.scope, Validators.required],
            secret: oauth.secret,
            silentLogin: oauth.silentLogin,
            implicitFlow: oauth.implicitFlow,
            publicUrls: [oauth.publicUrls]
        });
    }
    createBPMFormControl() {
        return new FormControl(this.appConfig.get(AppConfigValues.BPMHOST), [Validators.required, Validators.pattern(this.HOST_REGEX)]);
    }
    createIdentityFormControl() {
        return new FormControl(this.appConfig.get(AppConfigValues.IDENTITY_HOST), [Validators.required, Validators.pattern(this.HOST_REGEX)]);
    }
    createECMFormControl() {
        return new FormControl(this.appConfig.get(AppConfigValues.ECMHOST), [Validators.required, Validators.pattern(this.HOST_REGEX)]);
    }
    onCancel() {
        this.cancel.emit(true);
    }
    onSubmit(values) {
        this.storageService.setItem(AppConfigValues.PROVIDERS, values.providersControl);
        if (this.isBPM()) {
            this.saveBPMValues(values);
        }
        else if (this.isECM()) {
            this.saveECMValues(values);
        }
        else if (this.isALL()) {
            this.saveECMValues(values);
            this.saveBPMValues(values);
        }
        if (this.isOAUTH()) {
            this.saveOAuthValues(values);
        }
        this.storageService.setItem(AppConfigValues.AUTHTYPE, values.authType);
        this.alfrescoApiService.reset();
        this.alfrescoApiService.getInstance().invalidateSession();
        this.success.emit(true);
    }
    keyDownFunction(event) {
        if (event.keyCode === ENTER && this.form.valid) {
            this.onSubmit(this.form.value);
        }
    }
    saveOAuthValues(values) {
        if (values.oauthConfig.publicUrls && (typeof values.oauthConfig.publicUrls === 'string')) {
            values.oauthConfig.publicUrls = values.oauthConfig.publicUrls.split(',');
        }
        this.storageService.setItem(AppConfigValues.OAUTHCONFIG, JSON.stringify(values.oauthConfig));
        this.storageService.setItem(AppConfigValues.IDENTITY_HOST, values.identityHost);
    }
    saveBPMValues(values) {
        this.storageService.setItem(AppConfigValues.BPMHOST, values.bpmHost);
    }
    saveECMValues(values) {
        this.storageService.setItem(AppConfigValues.ECMHOST, values.ecmHost);
    }
    isBPM() {
        return this.providersControl.value === 'BPM';
    }
    isECM() {
        return this.providersControl.value === 'ECM';
    }
    isALL() {
        return this.providersControl.value === 'ALL';
    }
    isOAUTH() {
        return this.form.get('authType').value === 'OAUTH';
    }
    get providersControl() {
        return this.form.get('providersControl');
    }
    get bpmHost() {
        return this.form.get('bpmHost');
    }
    get ecmHost() {
        return this.form.get('ecmHost');
    }
    get host() {
        return this.oauthConfig.get('host');
    }
    get identityHost() {
        return this.form.get('identityHost');
    }
    get clientId() {
        return this.oauthConfig.get('clientId');
    }
    get scope() {
        return this.oauthConfig.get('scope');
    }
    get secretId() {
        return this.oauthConfig.get('secretId');
    }
    get implicitFlow() {
        return this.oauthConfig.get('implicitFlow');
    }
    get silentLogin() {
        return this.oauthConfig.get('silentLogin');
    }
    get redirectUri() {
        return this.oauthConfig.get('redirectUri');
    }
    get publicUrls() {
        return this.oauthConfig.get('publicUrls');
    }
    get redirectUriLogout() {
        return this.oauthConfig.get('redirectUriLogout');
    }
    get oauthConfig() {
        return this.form.get('oauthConfig');
    }
}
HostSettingsComponent.ɵfac = function HostSettingsComponent_Factory(t) { return new (t || HostSettingsComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormBuilder), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.StorageService), ɵngcc0.ɵɵdirectiveInject(ɵngcc3.AlfrescoApiService), ɵngcc0.ɵɵdirectiveInject(ɵngcc4.AppConfigService)); };
HostSettingsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: HostSettingsComponent, selectors: [["adf-host-settings"]], hostAttrs: [1, "adf-host-settings"], inputs: { providers: "providers" }, outputs: { error: "error", cancel: "cancel", success: "success" }, decls: 30, vars: 25, consts: [[1, "adf-setting-container"], ["color", "primary", 1, "adf-setting-toolbar"], [1, "adf-setting-card"], ["id", "host-form", 3, "formGroup", "submit", "keydown"], [4, "ngIf"], [1, "adf-authentication-type"], ["formControlName", "authType"], ["value", "BASIC"], ["value", "OAUTH"], [1, "adf-actions"], ["mat-button", "", "color", "primary", 3, "click"], ["type", "submit", "id", "host-button", "mat-raised-button", "", "color", "primary", "data-automation-id", "host-button", 1, "adf-login-button", 3, "disabled"], ["id", "adf-provider-selector", "placeholder", "Provider", 3, "formControl"], [3, "value", 4, "ngFor", "ngForOf"], [3, "value"], [1, "adf-full-width"], ["matInput", "", "data-automation-id", "ecmHost", "type", "text", "id", "ecmHost", "placeholder", "http(s)://host|ip:port(/path)", 3, "formControl"], ["matInput", "", "data-automation-id", "bpmHost", "type", "text", "id", "bpmHost", "placeholder", "http(s)://host|ip:port(/path)", 3, "formControl"], ["matInput", "", "name", "identityHost", "id", "identityHost", "formControlName", "identityHost", "placeholder", "http(s)://host|ip:port(/path)"], ["formGroupName", "oauthConfig"], ["matInput", "", "name", "host", "id", "oauthHost", "formControlName", "host", "placeholder", "http(s)://host|ip:port(/path)"], ["matInput", "", "name", "clientId", "id", "clientId", "formControlName", "clientId", "placeholder", "Client Id"], ["matInput", "", "formControlName", "scope", "placeholder", "Scope Id", 3, "name"], ["for", "silentLogin"], ["name", "silentLogin", "formControlName", "silentLogin", 1, "adf-full-width", 3, "color"], ["for", "implicitFlow"], ["name", "implicitFlow", "formControlName", "implicitFlow", 1, "adf-full-width", 3, "color"], ["matInput", "", "name", "redirectUri", "formControlName", "redirectUri", 3, "placeholder"], ["id", "logout-url", "matInput", "", "name", "redirectUriLogout", "formControlName", "redirectUriLogout", 3, "placeholder"], ["id", "public-url", "matInput", "", "name", "publicUrls", "formControlName", "publicUrls", 3, "placeholder"]], template: function HostSettingsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "mat-toolbar", 1);
        ɵngcc0.ɵɵelementStart(2, "h3");
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵpipe(4, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "mat-card", 2);
        ɵngcc0.ɵɵelementStart(6, "form", 3);
        ɵngcc0.ɵɵlistener("submit", function HostSettingsComponent_Template_form_submit_6_listener() { return ctx.onSubmit(ctx.form.value); })("keydown", function HostSettingsComponent_Template_form_keydown_6_listener($event) { return ctx.keyDownFunction($event); });
        ɵngcc0.ɵɵtemplate(7, HostSettingsComponent_mat_form_field_7_Template, 3, 2, "mat-form-field", 4);
        ɵngcc0.ɵɵelementStart(8, "div", 5);
        ɵngcc0.ɵɵelementStart(9, "div");
        ɵngcc0.ɵɵtext(10);
        ɵngcc0.ɵɵpipe(11, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(12, "mat-radio-group", 6);
        ɵngcc0.ɵɵelementStart(13, "mat-radio-button", 7);
        ɵngcc0.ɵɵtext(14);
        ɵngcc0.ɵɵpipe(15, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(16, "mat-radio-button", 8);
        ɵngcc0.ɵɵtext(17);
        ɵngcc0.ɵɵpipe(18, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(19, HostSettingsComponent_ng_container_19_Template, 10, 6, "ng-container", 4);
        ɵngcc0.ɵɵtemplate(20, HostSettingsComponent_ng_container_20_Template, 9, 6, "ng-container", 4);
        ɵngcc0.ɵɵtemplate(21, HostSettingsComponent_ng_container_21_Template, 8, 2, "ng-container", 4);
        ɵngcc0.ɵɵtemplate(22, HostSettingsComponent_ng_container_22_Template, 48, 40, "ng-container", 4);
        ɵngcc0.ɵɵelementStart(23, "mat-card-actions", 9);
        ɵngcc0.ɵɵelementStart(24, "button", 10);
        ɵngcc0.ɵɵlistener("click", function HostSettingsComponent_Template_button_click_24_listener() { return ctx.onCancel(); });
        ɵngcc0.ɵɵtext(25);
        ɵngcc0.ɵɵpipe(26, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(27, "button", 11);
        ɵngcc0.ɵɵtext(28);
        ɵngcc0.ɵɵpipe(29, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(4, 13, "CORE.HOST_SETTINGS.TITLE"));
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("formGroup", ctx.form);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showSelectProviders);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(11, 15, "CORE.HOST_SETTINGS.TYPE-AUTH"), " : ");
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate1("", ɵngcc0.ɵɵpipeBind1(15, 17, "CORE.HOST_SETTINGS.BASIC"), " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1("", ɵngcc0.ɵɵpipeBind1(18, 19, "CORE.HOST_SETTINGS.SSO"), " ");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isALL() || ctx.isECM());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isALL() || ctx.isBPM());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isOAUTH());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isOAUTH());
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(26, 21, "CORE.HOST_SETTINGS.BACK"), " ");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("disabled", !ctx.form.valid);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(29, 23, "CORE.HOST_SETTINGS.APPLY"), " ");
    } }, directives: [ɵngcc5.MatToolbar, ɵngcc6.MatCard, ɵngcc1.ɵangular_packages_forms_forms_y, ɵngcc1.NgControlStatusGroup, ɵngcc1.FormGroupDirective, ɵngcc7.NgIf, ɵngcc8.MatRadioGroup, ɵngcc1.NgControlStatus, ɵngcc1.FormControlName, ɵngcc8.MatRadioButton, ɵngcc6.MatCardActions, ɵngcc9.MatButton, ɵngcc10.MatFormField, ɵngcc11.MatSelect, ɵngcc1.FormControlDirective, ɵngcc7.NgForOf, ɵngcc12.MatOption, ɵngcc6.MatCardContent, ɵngcc10.MatLabel, ɵngcc13.MatInput, ɵngcc1.DefaultValueAccessor, ɵngcc10.MatError, ɵngcc1.FormGroupName, ɵngcc14.MatSlideToggle], pipes: [ɵngcc15.TranslatePipe], styles: [".adf-host-settings{align-items:center;display:flex;min-height:100%}.adf-host-settings .adf-authentication-type{margin-bottom:20px;margin-top:10px}.adf-host-settings .adf-setting-container{border-collapse:collapse;border-spacing:0;display:table;margin:0 auto;width:800px}.adf-host-settings .adf-setting-card-padding{display:table-cell;margin:0;vertical-align:middle;width:50%}.adf-host-settings .adf-settings-link-icon{margin-right:10px;position:relative;top:6px}.adf-host-settings .adf-settings-actions{display:flex;justify-content:flex-end}.adf-host-settings .adf-full-width{width:100%}"], encapsulation: 2 });
HostSettingsComponent.ctorParameters = () => [
    { type: FormBuilder },
    { type: StorageService },
    { type: AlfrescoApiService },
    { type: AppConfigService }
];
HostSettingsComponent.propDecorators = {
    providers: [{ type: Input }],
    error: [{ type: Output }],
    cancel: [{ type: Output }],
    success: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HostSettingsComponent, [{
        type: Component,
        args: [{
                selector: 'adf-host-settings',
                template: "<div class=\"adf-setting-container\">\n    <mat-toolbar color=\"primary\" class=\"adf-setting-toolbar\">\n        <h3>{{'CORE.HOST_SETTINGS.TITLE' | translate}}</h3>\n    </mat-toolbar>\n    <mat-card class=\"adf-setting-card\">\n        <form id=\"host-form\" [formGroup]=\"form\" (submit)=\"onSubmit(form.value)\" (keydown)=\"keyDownFunction($event)\">\n\n            <mat-form-field *ngIf=\"showSelectProviders\">\n                <mat-select  id=\"adf-provider-selector\" placeholder=\"Provider\" [formControl]=\"providersControl\">\n                    <mat-option *ngFor=\"let provider of providers\" [value]=\"provider\">\n                        {{ provider }}\n                    </mat-option>\n                </mat-select>\n            </mat-form-field>\n\n            <div class=\"adf-authentication-type\">\n                <div> {{'CORE.HOST_SETTINGS.TYPE-AUTH' | translate }} : </div>\n                <mat-radio-group formControlName=\"authType\" >\n                <mat-radio-button value=\"BASIC\">{{'CORE.HOST_SETTINGS.BASIC' | translate }}\n                </mat-radio-button>\n                <mat-radio-button value=\"OAUTH\">{{'CORE.HOST_SETTINGS.SSO' | translate }}\n                </mat-radio-button>\n            </mat-radio-group>\n            </div>\n\n            <ng-container *ngIf=\"isALL() || isECM()\">\n                <mat-card-content>\n                    <mat-form-field class=\"adf-full-width\">\n                        <mat-label>{{'CORE.HOST_SETTINGS.CS-HOST' | translate }}</mat-label>\n                        <input matInput [formControl]=\"ecmHost\" data-automation-id=\"ecmHost\" type=\"text\"\n                               id=\"ecmHost\" placeholder=\"http(s)://host|ip:port(/path)\">\n                        <mat-error *ngIf=\"ecmHost.hasError('pattern')\">\n                            {{ 'CORE.HOST_SETTINGS.NOT_VALID'| translate }}\n                        </mat-error>\n                        <mat-error *ngIf=\"ecmHost.hasError('required')\">\n                            {{ 'CORE.HOST_SETTINGS.REQUIRED'| translate }}\n                        </mat-error>\n                    </mat-form-field>\n                    <p>\n                </mat-card-content>\n            </ng-container>\n\n            <ng-container *ngIf=\"isALL() || isBPM()\">\n                <mat-card-content>\n                    <mat-form-field class=\"adf-full-width\">\n                        <mat-label>{{'CORE.HOST_SETTINGS.BP-HOST' | translate }}</mat-label>\n                        <input matInput [formControl]=\"bpmHost\" data-automation-id=\"bpmHost\" type=\"text\"\n                               id=\"bpmHost\" placeholder=\"http(s)://host|ip:port(/path)\">\n                        <mat-error *ngIf=\"bpmHost.hasError('pattern')\">\n                            {{ 'CORE.HOST_SETTINGS.NOT_VALID'| translate }}\n                        </mat-error>\n                        <mat-error *ngIf=\"bpmHost.hasError('required')\">\n                            {{ 'CORE.HOST_SETTINGS.REQUIRED'| translate }}\n                        </mat-error>\n                    </mat-form-field>\n                </mat-card-content>\n            </ng-container>\n\n            <ng-container *ngIf=\"isOAUTH()\">\n                <mat-card-content>\n                    <mat-form-field class=\"adf-full-width\">\n                        <mat-label>Identity Host</mat-label>\n                        <input matInput name=\"identityHost\" id=\"identityHost\" formControlName=\"identityHost\"\n                                placeholder=\"http(s)://host|ip:port(/path)\">\n                        <mat-error *ngIf=\"identityHost.hasError('pattern')\">\n                            {{ 'CORE.HOST_SETTINGS.NOT_VALID'| translate }}\n                        </mat-error>\n                        <mat-error *ngIf=\"identityHost.hasError('required')\">\n                            {{ 'CORE.HOST_SETTINGS.REQUIRED'| translate }}\n                        </mat-error>\n                    </mat-form-field>\n                </mat-card-content>\n            </ng-container>\n\n            <ng-container *ngIf=\"isOAUTH()\">\n                <div formGroupName=\"oauthConfig\">\n                    <mat-form-field class=\"adf-full-width\">\n                        <mat-label>Auth Host</mat-label>\n                        <input matInput name=\"host\" id=\"oauthHost\" formControlName=\"host\"\n                               placeholder=\"http(s)://host|ip:port(/path)\">\n                        <mat-error *ngIf=\"host.hasError('pattern')\">\n                            {{ 'CORE.HOST_SETTINGS.NOT_VALID'| translate }}\n                        </mat-error>\n                        <mat-error *ngIf=\"host.hasError('required')\">\n                            {{ 'CORE.HOST_SETTINGS.REQUIRED'| translate }}\n                        </mat-error>\n                    </mat-form-field>\n                    <mat-form-field class=\"adf-full-width\">\n                        <mat-label>{{ 'CORE.HOST_SETTINGS.CLIENT'| translate }}</mat-label>\n                        <input matInput name=\"clientId\" id=\"clientId\" formControlName=\"clientId\"\n                               placeholder=\"Client Id\">\n                        <mat-error *ngIf=\"clientId.hasError('required')\">\n                            {{ 'CORE.HOST_SETTINGS.REQUIRED'| translate }}\n                        </mat-error>\n                    </mat-form-field>\n\n                    <mat-form-field class=\"adf-full-width\">\n                        <mat-label>{{ 'CORE.HOST_SETTINGS.SCOPE'| translate }}</mat-label>\n                        <input matInput name=\"{{ 'CORE.HOST_SETTINGS.SCOPE'| translate }}\"\n                               formControlName=\"scope\" placeholder=\"Scope Id\">\n                        <mat-error *ngIf=\"scope.hasError('required')\">\n                            {{ 'CORE.HOST_SETTINGS.REQUIRED'| translate }}\n                        </mat-error>\n                    </mat-form-field>\n\n                    <label for=\"silentLogin\">{{ 'CORE.HOST_SETTINGS.SILENT'| translate }}</label>\n                    <mat-slide-toggle class=\"adf-full-width\" name=\"silentLogin\" [color]=\"'primary'\"\n                                      formControlName=\"silentLogin\">\n                    </mat-slide-toggle>\n\n                    <label for=\"implicitFlow\">{{ 'CORE.HOST_SETTINGS.IMPLICIT-FLOW'| translate }}</label>\n                    <mat-slide-toggle class=\"adf-full-width\" name=\"implicitFlow\" [color]=\"'primary'\"\n                                      formControlName=\"implicitFlow\">\n                    </mat-slide-toggle>\n\n\n                    <mat-form-field class=\"adf-full-width\">\n                        <mat-label>{{ 'CORE.HOST_SETTINGS.REDIRECT'| translate }}</mat-label>\n                        <input matInput placeholder=\"{{ 'CORE.HOST_SETTINGS.REDIRECT'| translate }}\"\n                               name=\"redirectUri\" formControlName=\"redirectUri\">\n                        <mat-error *ngIf=\"redirectUri.hasError('required')\">\n                            {{ 'CORE.HOST_SETTINGS.REQUIRED'| translate }}\n                        </mat-error>\n                    </mat-form-field>\n\n                    <mat-form-field class=\"adf-full-width\">\n                        <mat-label>{{ 'CORE.HOST_SETTINGS.REDIRECT_LOGOUT'| translate }}</mat-label>\n                        <input id=\"logout-url\" matInput placeholder=\"{{ 'CORE.HOST_SETTINGS.REDIRECT_LOGOUT'| translate }}\"\n                               name=\"redirectUriLogout\" formControlName=\"redirectUriLogout\">\n                    </mat-form-field>\n\n                    <mat-form-field class=\"adf-full-width\">\n                        <mat-label>{{ 'CORE.HOST_SETTINGS.PUBLIC_URLS'| translate }}</mat-label>\n                        <input id=\"public-url\" matInput placeholder=\"{{ 'CORE.HOST_SETTINGS.PUBLIC_URLS'| translate }}\"\n                               name=\"publicUrls\" formControlName=\"publicUrls\">\n                    </mat-form-field>\n\n                </div>\n            </ng-container>\n            <mat-card-actions class=\"adf-actions\">\n                <button mat-button (click)=\"onCancel()\" color=\"primary\">\n                    {{'CORE.HOST_SETTINGS.BACK' | translate }}\n                </button>\n                <button type=\"submit\" id=\"host-button\" class=\"adf-login-button\" mat-raised-button\n                        color=\"primary\" data-automation-id=\"host-button\"\n                        [disabled]=\"!form.valid\">\n                    {{'CORE.HOST_SETTINGS.APPLY' | translate }}\n                </button>\n            </mat-card-actions>\n        </form>\n    </mat-card>\n</div>\n",
                host: {
                    'class': 'adf-host-settings'
                },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-host-settings{align-items:center;display:flex;min-height:100%}.adf-host-settings .adf-authentication-type{margin-bottom:20px;margin-top:10px}.adf-host-settings .adf-setting-container{border-collapse:collapse;border-spacing:0;display:table;margin:0 auto;width:800px}.adf-host-settings .adf-setting-card-padding{display:table-cell;margin:0;vertical-align:middle;width:50%}.adf-host-settings .adf-settings-link-icon{margin-right:10px;position:relative;top:6px}.adf-host-settings .adf-settings-actions{display:flex;justify-content:flex-end}.adf-host-settings .adf-full-width{width:100%}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormBuilder }, { type: ɵngcc2.StorageService }, { type: ɵngcc3.AlfrescoApiService }, { type: ɵngcc4.AppConfigService }]; }, { providers: [{
            type: Input
        }], error: [{
            type: Output
        }], cancel: [{
            type: Output
        }], success: [{
            type: Output
        }] }); })();
