// @alfresco/adf-core/esm2015/settings/host-settings.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { TranslateModule } from '@ngx-translate/core';
import { MaterialModule } from '../material.module';
import { HostSettingsComponent } from './host-settings.component';
import * as ɵngcc0 from '@angular/core';
export class HostSettingsModule {
}
HostSettingsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: HostSettingsModule });
HostSettingsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function HostSettingsModule_Factory(t) { return new (t || HostSettingsModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            TranslateModule,
            FormsModule,
            ReactiveFormsModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(HostSettingsModule, { declarations: function () { return [HostSettingsComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        TranslateModule,
        FormsModule,
        ReactiveFormsModule]; }, exports: function () { return [HostSettingsComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HostSettingsModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    TranslateModule,
                    FormsModule,
                    ReactiveFormsModule
                ],
                declarations: [
                    HostSettingsComponent
                ],
                exports: [
                    HostSettingsComponent
                ]
            }]
    }], null, null); })();
