// @alfresco/adf-content-services/esm2015/lib/search/components/search-filter-chips/search-filter-menu-card/search-filter-menu-card.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Output, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/icon';
import * as ɵngcc2 from '@angular/material/tooltip';
import * as ɵngcc3 from '@angular/material/divider';
import * as ɵngcc4 from '@ngx-translate/core';

const _c0 = [[["filter-title"]], [["filter-content"]], [["filter-actions"]]];
const _c1 = ["filter-title", "filter-content", "filter-actions"];
export class SearchFilterMenuCardComponent {
    constructor() {
        this.close = new EventEmitter();
    }
    onClose() {
        this.close.emit();
    }
}
SearchFilterMenuCardComponent.ɵfac = function SearchFilterMenuCardComponent_Factory(t) { return new (t || SearchFilterMenuCardComponent)(); };
SearchFilterMenuCardComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SearchFilterMenuCardComponent, selectors: [["adf-search-filter-menu-card"]], outputs: { close: "close" }, ngContentSelectors: _c1, decls: 12, vars: 3, consts: [[1, "adf-search-filter-menu-card"], [1, "adf-search-filter-title"], [1, "adf-search-filter-title-action", 3, "matTooltip", "click"], [1, "adf-search-filter-content"], [1, "adf-search-filter-actions"]], template: function SearchFilterMenuCardComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵprojection(2);
        ɵngcc0.ɵɵelementStart(3, "mat-icon", 2);
        ɵngcc0.ɵɵlistener("click", function SearchFilterMenuCardComponent_Template_mat_icon_click_3_listener() { return ctx.onClose(); });
        ɵngcc0.ɵɵpipe(4, "translate");
        ɵngcc0.ɵɵtext(5, " close ");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(6, "mat-divider");
        ɵngcc0.ɵɵelementStart(7, "div", 3);
        ɵngcc0.ɵɵprojection(8, 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(9, "mat-divider");
        ɵngcc0.ɵɵelementStart(10, "div", 4);
        ɵngcc0.ɵɵprojection(11, 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("matTooltip", ɵngcc0.ɵɵpipeBind1(4, 1, "SEARCH.FILTER.BUTTONS.CLOSE"));
    } }, directives: [ɵngcc1.MatIcon, ɵngcc2.MatTooltip, ɵngcc3.MatDivider], pipes: [ɵngcc4.TranslatePipe], styles: [".adf-search-filter-menu-card{background:var(--theme-card-bg-color);color:var(--theme-text-fg-color)}.adf-search-filter-menu-card .adf-search-filter-title{flex:1 1 auto;font-size:14px;font-style:inherit;font-weight:700;height:32px;letter-spacing:.15px;line-height:24px;padding:16px 12px}.adf-search-filter-menu-card .adf-search-filter-title-action{float:right}.adf-search-filter-menu-card .adf-search-filter-content{overflow:auto;padding:16px 12px}.adf-search-filter-menu-card .adf-search-filter-actions{display:flex;justify-content:space-between;padding:16px 12px}.adf-search-filter-menu-card .adf-search-filter-actions .adf-search-action-button{border-radius:6px}"], encapsulation: 2 });
SearchFilterMenuCardComponent.propDecorators = {
    close: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SearchFilterMenuCardComponent, [{
        type: Component,
        args: [{
                selector: 'adf-search-filter-menu-card',
                template: "<div class=\"adf-search-filter-menu-card\">\n    <div class=\"adf-search-filter-title\">\n        <ng-content select=\"filter-title\"></ng-content>\n        <mat-icon class=\"adf-search-filter-title-action\"\n                  (click)=\"onClose()\"\n                  [matTooltip]=\"'SEARCH.FILTER.BUTTONS.CLOSE' | translate\">\n            close\n        </mat-icon>\n    </div>\n\n    <mat-divider></mat-divider>\n\n    <div class=\"adf-search-filter-content\">\n        <ng-content select=\"filter-content\"></ng-content>\n    </div>\n\n    <mat-divider></mat-divider>\n\n    <div class=\"adf-search-filter-actions\">\n        <ng-content select=\"filter-actions\"></ng-content>\n    </div>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-search-filter-menu-card{background:var(--theme-card-bg-color);color:var(--theme-text-fg-color)}.adf-search-filter-menu-card .adf-search-filter-title{flex:1 1 auto;font-size:14px;font-style:inherit;font-weight:700;height:32px;letter-spacing:.15px;line-height:24px;padding:16px 12px}.adf-search-filter-menu-card .adf-search-filter-title-action{float:right}.adf-search-filter-menu-card .adf-search-filter-content{overflow:auto;padding:16px 12px}.adf-search-filter-menu-card .adf-search-filter-actions{display:flex;justify-content:space-between;padding:16px 12px}.adf-search-filter-menu-card .adf-search-filter-actions .adf-search-action-button{border-radius:6px}"]
            }]
    }], function () { return []; }, { close: [{
            type: Output
        }] }); })();
