// @alfresco/adf-core/esm2015/templates/template.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TranslateModule } from '@ngx-translate/core';
import { MaterialModule } from '../material.module';
import { ErrorContentComponent } from './error-content/error-content.component';
import { EmptyContentComponent } from './empty-content/empty-content.component';
import { IconModule } from '../icon/icon.module';
import * as ɵngcc0 from '@angular/core';
export class TemplateModule {
}
TemplateModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TemplateModule });
TemplateModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function TemplateModule_Factory(t) { return new (t || TemplateModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            TranslateModule,
            IconModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TemplateModule, { declarations: function () { return [ErrorContentComponent,
        EmptyContentComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        TranslateModule,
        IconModule]; }, exports: function () { return [ErrorContentComponent,
        EmptyContentComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TemplateModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    TranslateModule,
                    IconModule
                ],
                declarations: [
                    ErrorContentComponent,
                    EmptyContentComponent
                ],
                exports: [
                    ErrorContentComponent,
                    EmptyContentComponent
                ]
            }]
    }], null, null); })();
