// @alfresco/adf-core/esm2015/templates/empty-content/empty-content.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ChangeDetectionStrategy, ViewEncapsulation, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../icon/icon.component';
import * as ɵngcc2 from '@ngx-translate/core';

const _c0 = ["*"];
export class EmptyContentComponent {
    constructor() {
        this.icon = 'cake';
        this.title = '';
        this.subtitle = '';
    }
}
EmptyContentComponent.ɵfac = function EmptyContentComponent_Factory(t) { return new (t || EmptyContentComponent)(); };
EmptyContentComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: EmptyContentComponent, selectors: [["adf-empty-content"]], hostAttrs: [1, "adf-empty-content"], inputs: { icon: "icon", title: "title", subtitle: "subtitle" }, ngContentSelectors: _c0, decls: 10, vars: 10, consts: [[1, "adf-empty-content", 3, "title"], [1, "adf-empty-content__icon", 3, "value"], [1, "adf-empty-content__title"], [1, "adf-empty-content__subtitle"]], template: function EmptyContentComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵpipe(1, "translate");
        ɵngcc0.ɵɵelement(2, "adf-icon", 1);
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵtext(4);
        ɵngcc0.ɵɵpipe(5, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(6, "div", 3);
        ɵngcc0.ɵɵtext(7);
        ɵngcc0.ɵɵpipe(8, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵprojection(9);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("title", ɵngcc0.ɵɵpipeBind1(1, 4, ctx.title));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("value", ctx.icon);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(5, 6, ctx.title));
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(8, 8, ctx.subtitle));
    } }, directives: [ɵngcc1.IconComponent], pipes: [ɵngcc2.TranslatePipe], styles: [".adf-empty-content{align-items:center;color:var(--theme-text-color);display:flex;flex-direction:column;height:100%;justify-content:center;width:100%}.adf-empty-content__icon .mat-icon{font-size:var(--theme-display-3-font-size);height:var(--theme-display-3-font-size)!important;width:var(--theme-display-3-font-size)!important}.adf-empty-content__title{font-size:18px;font-weight:600;line-height:0;margin:10px 0;text-align:center;white-space:normal}.adf-empty-content__subtitle,.adf-empty-content__text{font-size:14px;font-weight:300;line-height:inherit;text-align:center;white-space:normal}.adf-empty-content .adf-icon{opacity:.6}"], encapsulation: 2, changeDetection: 0 });
EmptyContentComponent.propDecorators = {
    icon: [{ type: Input }],
    title: [{ type: Input }],
    subtitle: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EmptyContentComponent, [{
        type: Component,
        args: [{
                selector: 'adf-empty-content',
                template: "<div class=\"adf-empty-content\" [title]=\"title | translate\">\n    <adf-icon class=\"adf-empty-content__icon\" [value]=\"icon\"></adf-icon>\n    <div class=\"adf-empty-content__title\">{{ title | translate }}</div>\n    <div class=\"adf-empty-content__subtitle\">{{ subtitle | translate }}</div>\n    <ng-content></ng-content>\n</div>\n",
                changeDetection: ChangeDetectionStrategy.OnPush,
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-empty-content' },
                styles: [".adf-empty-content{align-items:center;color:var(--theme-text-color);display:flex;flex-direction:column;height:100%;justify-content:center;width:100%}.adf-empty-content__icon .mat-icon{font-size:var(--theme-display-3-font-size);height:var(--theme-display-3-font-size)!important;width:var(--theme-display-3-font-size)!important}.adf-empty-content__title{font-size:18px;font-weight:600;line-height:0;margin:10px 0;text-align:center;white-space:normal}.adf-empty-content__subtitle,.adf-empty-content__text{font-size:14px;font-weight:300;line-height:inherit;text-align:center;white-space:normal}.adf-empty-content .adf-icon{opacity:.6}"]
            }]
    }], function () { return []; }, { icon: [{
            type: Input
        }], title: [{
            type: Input
        }], subtitle: [{
            type: Input
        }] }); })();
