// @alfresco/adf-process-services-cloud/esm2015/lib/group/pipe/group-initial.pipe.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Pipe } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class InitialGroupNamePipe {
    transform(group) {
        let result = '';
        if (group) {
            result = this.getInitialGroupName(group.name).toUpperCase();
        }
        return result;
    }
    getInitialGroupName(groupName) {
        groupName = (groupName ? groupName[0] : '');
        return groupName;
    }
}
InitialGroupNamePipe.ɵfac = function InitialGroupNamePipe_Factory(t) { return new (t || InitialGroupNamePipe)(); };
InitialGroupNamePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "groupNameInitial", type: InitialGroupNamePipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InitialGroupNamePipe, [{
        type: Pipe,
        args: [{
                name: 'groupNameInitial'
            }]
    }], null, null); })();
