// @alfresco/adf-process-services-cloud/esm2015/lib/form/components/widgets/attach-file/file-properties-table-cloud.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { LocalizedDatePipe, ThumbnailService } from '@alfresco/adf-core';
import { Node } from '@alfresco/js-api';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/material/table';
import * as ɵngcc4 from '@angular/material/icon';
import * as ɵngcc5 from '@angular/material/list';
import * as ɵngcc6 from '@angular/material/core';
import * as ɵngcc7 from '@angular/material/menu';
import * as ɵngcc8 from '@angular/material/button';
import * as ɵngcc9 from '@ngx-translate/core';

function FilePropertiesTableCloudComponent_table_1_th_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "th", 11);
} }
function FilePropertiesTableCloudComponent_table_1_td_3_mat_icon_1_Template(rf, ctx) { if (rf & 1) {
    const _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "mat-icon", 15);
    ɵngcc0.ɵɵlistener("click", function FilePropertiesTableCloudComponent_table_1_td_3_mat_icon_1_Template_mat_icon_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r15); const element_r10 = ɵngcc0.ɵɵnextContext().$implicit; const ctx_r13 = ɵngcc0.ɵɵnextContext(2); return ctx_r13.onRowClicked(element_r10); });
    ɵngcc0.ɵɵtext(1, " check_circle ");
    ɵngcc0.ɵɵelementEnd();
} }
function FilePropertiesTableCloudComponent_table_1_td_3_img_2_Template(rf, ctx) { if (rf & 1) {
    const _r18 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "img", 16);
    ɵngcc0.ɵɵlistener("click", function FilePropertiesTableCloudComponent_table_1_td_3_img_2_Template_img_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r18); const element_r10 = ɵngcc0.ɵɵnextContext().$implicit; const ctx_r16 = ɵngcc0.ɵɵnextContext(2); return ctx_r16.onRowClicked(element_r10); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r10 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r12 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("id", "file-" + (element_r10 == null ? null : element_r10.id) + "-icon")("src", element_r10.content ? ctx_r12.getIcon(element_r10.content.mimeType) : ctx_r12.getIcon(element_r10["mimeType"]), ɵngcc0.ɵɵsanitizeUrl)("alt", ctx_r12.mimeTypeIcon);
} }
function FilePropertiesTableCloudComponent_table_1_td_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 12);
    ɵngcc0.ɵɵtemplate(1, FilePropertiesTableCloudComponent_table_1_td_3_mat_icon_1_Template, 2, 0, "mat-icon", 13);
    ɵngcc0.ɵɵtemplate(2, FilePropertiesTableCloudComponent_table_1_td_3_img_2_Template, 1, 3, "img", 14);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r10 = ctx.$implicit;
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r2.selectedNode && element_r10.id === ctx_r2.selectedNode.id);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r2.selectedNode || element_r10.id !== ctx_r2.selectedNode.id);
} }
function FilePropertiesTableCloudComponent_table_1_th_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th", 11);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, "FORM.FIELD.FILE_NAME"));
} }
function FilePropertiesTableCloudComponent_table_1_td_6_Template(rf, ctx) { if (rf & 1) {
    const _r22 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "td", 12);
    ɵngcc0.ɵɵelementStart(1, "span", 17);
    ɵngcc0.ɵɵlistener("click", function FilePropertiesTableCloudComponent_table_1_td_6_Template_span_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r22); const element_r20 = ctx.$implicit; const ctx_r21 = ɵngcc0.ɵɵnextContext(2); return ctx_r21.onRowClicked(element_r20); });
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r20 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("id", "file-" + (element_r20 == null ? null : element_r20.id));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(element_r20.name);
} }
function FilePropertiesTableCloudComponent_table_1_ng_container_7_th_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th", 11);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "titlecase");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const columnName_r23 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(columnName_r23.title ? columnName_r23.title : ɵngcc0.ɵɵpipeBind1(2, 1, columnName_r23.name));
} }
function FilePropertiesTableCloudComponent_table_1_ng_container_7_td_2_Template(rf, ctx) { if (rf & 1) {
    const _r29 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "td", 12);
    ɵngcc0.ɵɵelementStart(1, "span", 19);
    ɵngcc0.ɵɵlistener("click", function FilePropertiesTableCloudComponent_table_1_ng_container_7_td_2_Template_span_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r29); const row_r27 = ctx.$implicit; const ctx_r28 = ɵngcc0.ɵɵnextContext(3); return ctx_r28.onRowClicked(row_r27); });
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r27 = ctx.$implicit;
    const columnName_r23 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r25 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("id", "fileProperty-" + (row_r27 == null ? null : row_r27.id) + "-" + (columnName_r23 == null ? null : columnName_r23.name));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r25.getColumnValue(row_r27, columnName_r23));
} }
function FilePropertiesTableCloudComponent_table_1_ng_container_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0, 18);
    ɵngcc0.ɵɵtemplate(1, FilePropertiesTableCloudComponent_table_1_ng_container_7_th_1_Template, 3, 3, "th", 4);
    ɵngcc0.ɵɵtemplate(2, FilePropertiesTableCloudComponent_table_1_ng_container_7_td_2_Template, 3, 2, "td", 5);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const columnName_r23 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("matColumnDef", columnName_r23.name);
} }
function FilePropertiesTableCloudComponent_table_1_th_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "th", 11);
} }
function FilePropertiesTableCloudComponent_table_1_td_10_button_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "button", 25);
    ɵngcc0.ɵɵelementStart(1, "mat-icon");
    ɵngcc0.ɵɵtext(2, "more_vert");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r31 = ɵngcc0.ɵɵnextContext().$implicit;
    const _r33 = ɵngcc0.ɵɵreference(3);
    ɵngcc0.ɵɵpropertyInterpolate("id", "file-" + (element_r31 == null ? null : element_r31.id) + "-option-menu");
    ɵngcc0.ɵɵproperty("matMenuTriggerFor", _r33);
} }
function FilePropertiesTableCloudComponent_table_1_td_10_button_4_Template(rf, ctx) { if (rf & 1) {
    const _r41 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 26);
    ɵngcc0.ɵɵlistener("click", function FilePropertiesTableCloudComponent_table_1_td_10_button_4_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r41); const element_r31 = ɵngcc0.ɵɵnextContext().$implicit; const ctx_r39 = ɵngcc0.ɵɵnextContext(2); return ctx_r39.onAttachFileClicked(element_r31); });
    ɵngcc0.ɵɵelementStart(1, "mat-icon");
    ɵngcc0.ɵɵtext(2, "visibility");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r31 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵpropertyInterpolate("id", "file-" + (element_r31 == null ? null : element_r31.id) + "-show-file");
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(5, 2, "FORM.FIELD.VIEW_FILE"));
} }
function FilePropertiesTableCloudComponent_table_1_td_10_button_5_Template(rf, ctx) { if (rf & 1) {
    const _r45 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 26);
    ɵngcc0.ɵɵlistener("click", function FilePropertiesTableCloudComponent_table_1_td_10_button_5_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r45); const element_r31 = ɵngcc0.ɵɵnextContext().$implicit; const ctx_r43 = ɵngcc0.ɵɵnextContext(2); return ctx_r43.downloadContent(element_r31); });
    ɵngcc0.ɵɵelementStart(1, "mat-icon");
    ɵngcc0.ɵɵtext(2, "file_download");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r31 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵpropertyInterpolate("id", "file-" + (element_r31 == null ? null : element_r31.id) + "-download-file");
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(5, 2, "FORM.FIELD.DOWNLOAD_FILE"));
} }
function FilePropertiesTableCloudComponent_table_1_td_10_button_6_Template(rf, ctx) { if (rf & 1) {
    const _r49 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 26);
    ɵngcc0.ɵɵlistener("click", function FilePropertiesTableCloudComponent_table_1_td_10_button_6_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r49); const element_r31 = ɵngcc0.ɵɵnextContext().$implicit; const ctx_r47 = ɵngcc0.ɵɵnextContext(2); return ctx_r47.contentModelFormFileHandler(element_r31); });
    ɵngcc0.ɵɵelementStart(1, "mat-icon", 27);
    ɵngcc0.ɵɵtext(2, "low_priority");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r31 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵpropertyInterpolate("id", "file-" + (element_r31 == null ? null : element_r31.id) + "-retrieve-file-metadata");
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(5, 2, "ADF_CLOUD_FORM_COMPONENT.RETRIEVE_METADATA"));
} }
function FilePropertiesTableCloudComponent_table_1_td_10_button_7_Template(rf, ctx) { if (rf & 1) {
    const _r53 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 28);
    ɵngcc0.ɵɵlistener("click", function FilePropertiesTableCloudComponent_table_1_td_10_button_7_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r53); const element_r31 = ɵngcc0.ɵɵnextContext().$implicit; const ctx_r51 = ɵngcc0.ɵɵnextContext(2); return ctx_r51.onRemoveAttachFile(element_r31); })("keyup.enter", function FilePropertiesTableCloudComponent_table_1_td_10_button_7_Template_button_keyup_enter_0_listener() { ɵngcc0.ɵɵrestoreView(_r53); const element_r31 = ɵngcc0.ɵɵnextContext().$implicit; const ctx_r54 = ɵngcc0.ɵɵnextContext(2); return ctx_r54.onRemoveAttachFile(element_r31); });
    ɵngcc0.ɵɵelementStart(1, "mat-icon", 27);
    ɵngcc0.ɵɵtext(2, "highlight_off");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r31 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵpropertyInterpolate("id", "file-" + (element_r31 == null ? null : element_r31.id) + "-remove-file");
    ɵngcc0.ɵɵproperty("id", "file-" + (element_r31 == null ? null : element_r31.id) + "-remove");
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(5, 3, "FORM.FIELD.REMOVE_FILE"));
} }
function FilePropertiesTableCloudComponent_table_1_td_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 12);
    ɵngcc0.ɵɵtemplate(1, FilePropertiesTableCloudComponent_table_1_td_10_button_1_Template, 3, 2, "button", 20);
    ɵngcc0.ɵɵelementStart(2, "mat-menu", 21, 22);
    ɵngcc0.ɵɵtemplate(4, FilePropertiesTableCloudComponent_table_1_td_10_button_4_Template, 6, 4, "button", 23);
    ɵngcc0.ɵɵtemplate(5, FilePropertiesTableCloudComponent_table_1_td_10_button_5_Template, 6, 4, "button", 23);
    ɵngcc0.ɵɵtemplate(6, FilePropertiesTableCloudComponent_table_1_td_10_button_6_Template, 6, 4, "button", 23);
    ɵngcc0.ɵɵtemplate(7, FilePropertiesTableCloudComponent_table_1_td_10_button_7_Template, 6, 5, "button", 24);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r31 = ctx.$implicit;
    const ctx_r7 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !!(element_r31.content == null ? null : element_r31.content.mimeType));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r7.displayMenuOption("show") && !!(element_r31.content == null ? null : element_r31.content.mimeType));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r7.displayMenuOption("download") && !!(element_r31.content == null ? null : element_r31.content.mimeType));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r7.displayMenuOption("retrieveMetadata") && !!(element_r31.content == null ? null : element_r31.content.mimeType));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r7.field.readOnly && ctx_r7.displayMenuOption("remove"));
} }
function FilePropertiesTableCloudComponent_table_1_tr_11_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "tr", 29);
} }
function FilePropertiesTableCloudComponent_table_1_tr_12_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "tr", 30);
} }
function FilePropertiesTableCloudComponent_table_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "table", 2);
    ɵngcc0.ɵɵelementContainerStart(1, 3);
    ɵngcc0.ɵɵtemplate(2, FilePropertiesTableCloudComponent_table_1_th_2_Template, 1, 0, "th", 4);
    ɵngcc0.ɵɵtemplate(3, FilePropertiesTableCloudComponent_table_1_td_3_Template, 3, 2, "td", 5);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵelementContainerStart(4, 6);
    ɵngcc0.ɵɵtemplate(5, FilePropertiesTableCloudComponent_table_1_th_5_Template, 3, 3, "th", 4);
    ɵngcc0.ɵɵtemplate(6, FilePropertiesTableCloudComponent_table_1_td_6_Template, 3, 2, "td", 5);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵtemplate(7, FilePropertiesTableCloudComponent_table_1_ng_container_7_Template, 3, 1, "ng-container", 7);
    ɵngcc0.ɵɵelementContainerStart(8, 8);
    ɵngcc0.ɵɵtemplate(9, FilePropertiesTableCloudComponent_table_1_th_9_Template, 1, 0, "th", 4);
    ɵngcc0.ɵɵtemplate(10, FilePropertiesTableCloudComponent_table_1_td_10_Template, 8, 5, "td", 5);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵtemplate(11, FilePropertiesTableCloudComponent_table_1_tr_11_Template, 1, 0, "tr", 9);
    ɵngcc0.ɵɵtemplate(12, FilePropertiesTableCloudComponent_table_1_tr_12_Template, 1, 0, "tr", 10);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("dataSource", ctx_r0.uploadedFiles);
    ɵngcc0.ɵɵadvance(7);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.field == null ? null : ctx_r0.field.params == null ? null : ctx_r0.field.params.displayableCMProperties);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵproperty("matHeaderRowDef", ctx_r0.displayedColumns);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("matRowDefColumns", ctx_r0.displayedColumns);
} }
export class FilePropertiesTableCloudComponent {
    constructor(localizedDatePipe, thumbnailService) {
        this.localizedDatePipe = localizedDatePipe;
        this.thumbnailService = thumbnailService;
        this.rowClick = new EventEmitter();
        this.attachFileClick = new EventEmitter();
        this.downloadFile = new EventEmitter();
        this.contentModelFileHandler = new EventEmitter();
        this.removeAttachFile = new EventEmitter();
    }
    onRowClicked(file) {
        this.rowClick.emit(file);
    }
    onAttachFileClicked(nodeSelector) {
        this.attachFileClick.emit(nodeSelector);
    }
    downloadContent(file) {
        this.downloadFile.emit(file);
    }
    contentModelFormFileHandler(file) {
        this.contentModelFileHandler.emit(file);
    }
    onRemoveAttachFile(file) {
        this.removeAttachFile.emit(file);
    }
    getIcon(mimeType) {
        return this.thumbnailService.getMimeTypeIcon(mimeType);
    }
    getColumnValue(file, displayableCMProperty) {
        var _a;
        if (!file.properties[displayableCMProperty.prefixedName]) {
            const fieldProperty = (_a = this.field.params.displayableCMProperties) === null || _a === void 0 ? void 0 : _a.find(property => property.name === displayableCMProperty.name);
            return fieldProperty.defaultValue ? this.checkDateTypeAndTransform(displayableCMProperty.dataType, fieldProperty.defaultValue) : '--';
        }
        return file.properties[displayableCMProperty.prefixedName] ?
            this.checkDateTypeAndTransform(displayableCMProperty.dataType, file.properties[displayableCMProperty.prefixedName]) :
            '--';
    }
    checkDateTypeAndTransform(dataType, value) {
        if (dataType === 'd:date') {
            return this.localizedDatePipe.transform(value);
        }
        else if (dataType === 'd:datetime') {
            return this.localizedDatePipe.transform(value, 'medium');
        }
        return value;
    }
    displayMenuOption(option) {
        var _a, _b;
        return ((_b = (_a = this.field) === null || _a === void 0 ? void 0 : _a.params) === null || _b === void 0 ? void 0 : _b.menuOptions) ? this.field.params.menuOptions[option] : option !== FilePropertiesTableCloudComponent.RETRIEVE_METADATA_OPTION;
    }
}
FilePropertiesTableCloudComponent.ɵfac = function FilePropertiesTableCloudComponent_Factory(t) { return new (t || FilePropertiesTableCloudComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.LocalizedDatePipe), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ThumbnailService)); };
FilePropertiesTableCloudComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FilePropertiesTableCloudComponent, selectors: [["adf-cloud-file-properties-table"]], inputs: { uploadedFiles: "uploadedFiles", hasFile: "hasFile", selectedNode: "selectedNode", field: "field", displayedColumns: "displayedColumns", mimeTypeIcon: "mimeTypeIcon" }, outputs: { rowClick: "rowClick", attachFileClick: "attachFileClick", downloadFile: "downloadFile", contentModelFileHandler: "contentModelFileHandler", removeAttachFile: "removeAttachFile" }, decls: 2, vars: 1, consts: [[1, "adf-file-properties-table"], ["mat-table", "", "class", "mat-elevation-z0", 3, "dataSource", 4, "ngIf"], ["mat-table", "", 1, "mat-elevation-z0", 3, "dataSource"], ["matColumnDef", "icon"], ["mat-header-cell", "", 4, "matHeaderCellDef"], ["mat-cell", "", 4, "matCellDef"], ["matColumnDef", "fileName"], [3, "matColumnDef", 4, "ngFor", "ngForOf"], ["matColumnDef", "action"], ["mat-header-row", "", 4, "matHeaderRowDef"], ["mat-row", "", 4, "matRowDef", "matRowDefColumns"], ["mat-header-cell", ""], ["mat-cell", ""], ["mat-list-icon", "", "class", "adf-datatable-selected", 3, "click", 4, "ngIf"], ["mat-list-icon", "", "class", "adf-attach-widget__icon", "role", "button", "tabindex", "0", 3, "id", "src", "alt", "click", 4, "ngIf"], ["mat-list-icon", "", 1, "adf-datatable-selected", 3, "click"], ["mat-list-icon", "", "role", "button", "tabindex", "0", 1, "adf-attach-widget__icon", 3, "id", "src", "alt", "click"], ["matLine", "", "role", "button", "tabindex", "0", 1, "adf-file", 3, "id", "click"], [3, "matColumnDef"], ["matLine", "", "role", "button", "tabindex", "0", 3, "id", "click"], ["mat-icon-button", "", 3, "id", "matMenuTriggerFor", 4, "ngIf"], ["xPosition", "before"], ["fileActionMenu", "matMenu"], ["mat-menu-item", "", 3, "id", "click", 4, "ngIf"], ["mat-menu-item", "", 3, "id", "click", "keyup.enter", 4, "ngIf"], ["mat-icon-button", "", 3, "id", "matMenuTriggerFor"], ["mat-menu-item", "", 3, "id", "click"], [1, "mat-24"], ["mat-menu-item", "", 3, "id", "click", "keyup.enter"], ["mat-header-row", ""], ["mat-row", ""]], template: function FilePropertiesTableCloudComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, FilePropertiesTableCloudComponent_table_1_Template, 13, 4, "table", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasFile);
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.MatTable, ɵngcc3.MatColumnDef, ɵngcc3.MatHeaderCellDef, ɵngcc3.MatCellDef, ɵngcc2.NgForOf, ɵngcc3.MatHeaderRowDef, ɵngcc3.MatRowDef, ɵngcc3.MatHeaderCell, ɵngcc3.MatCell, ɵngcc4.MatIcon, ɵngcc5.MatListIconCssMatStyler, ɵngcc6.MatLine, ɵngcc7._MatMenu, ɵngcc8.MatButton, ɵngcc7.MatMenuTrigger, ɵngcc7.MatMenuItem, ɵngcc3.MatHeaderRow, ɵngcc3.MatRow], pipes: [ɵngcc9.TranslatePipe, ɵngcc2.TitleCasePipe], styles: [".adf-file-properties-table[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]{border:1px solid var(--theme-border-color);box-shadow:none;width:100%}.adf-file-properties-table[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   .adf-datatable-selected[_ngcontent-%COMP%]{padding:6px}.adf-file-properties-table[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tr.mat-header-row[_ngcontent-%COMP%]{height:32px}.adf-file-properties-table[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   th[_ngcontent-%COMP%]{font-weight:700;padding-right:0!important}.adf-file-properties-table[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]{max-width:50px;overflow:hidden;padding-right:0!important;text-overflow:ellipsis;white-space:nowrap}.adf-file-properties-table[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   td.mat-cell[_ngcontent-%COMP%]:first-of-type, .adf-file-properties-table[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   td.mat-cell[_ngcontent-%COMP%]:last-of-type, .adf-file-properties-table[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   td.mat-footer-cell[_ngcontent-%COMP%]:first-of-type, .adf-file-properties-table[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   td.mat-footer-cell[_ngcontent-%COMP%]:last-of-type{width:18%!important}"] });
FilePropertiesTableCloudComponent.RETRIEVE_METADATA_OPTION = 'retrieveMetadata';
FilePropertiesTableCloudComponent.ctorParameters = () => [
    { type: LocalizedDatePipe },
    { type: ThumbnailService }
];
FilePropertiesTableCloudComponent.propDecorators = {
    uploadedFiles: [{ type: Input }],
    hasFile: [{ type: Input }],
    selectedNode: [{ type: Input }],
    field: [{ type: Input }],
    displayedColumns: [{ type: Input }],
    mimeTypeIcon: [{ type: Input }],
    rowClick: [{ type: Output }],
    attachFileClick: [{ type: Output }],
    downloadFile: [{ type: Output }],
    contentModelFileHandler: [{ type: Output }],
    removeAttachFile: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FilePropertiesTableCloudComponent, [{
        type: Component,
        args: [{
                selector: 'adf-cloud-file-properties-table',
                template: "<div class=\"adf-file-properties-table\">\n    <table mat-table [dataSource]=\"uploadedFiles\" class=\"mat-elevation-z0\" *ngIf=\"hasFile\">\n        <ng-container matColumnDef=\"icon\">\n            <th mat-header-cell *matHeaderCellDef></th>\n            <td mat-cell *matCellDef=\"let element\">\n                <mat-icon mat-list-icon class=\"adf-datatable-selected\"\n                    *ngIf=\"selectedNode && element.id === selectedNode.id\" (click)=\"onRowClicked(element)\">\n                    check_circle\n                </mat-icon>\n                <img mat-list-icon class=\"adf-attach-widget__icon\"\n                    *ngIf=\"!selectedNode || element.id !== selectedNode.id\" [id]=\"'file-'+element?.id+'-icon'\"\n                    (click)=\"onRowClicked(element)\"\n                    [src]=\"element.content ? getIcon(element.content.mimeType) : getIcon(element['mimeType'])\"\n                    [alt]=\"mimeTypeIcon\" role=\"button\" tabindex=\"0\" />\n            </td>\n        </ng-container>\n    \n        <ng-container matColumnDef=\"fileName\">\n            <th mat-header-cell *matHeaderCellDef>{{ 'FORM.FIELD.FILE_NAME' | translate }}</th>\n            <td mat-cell *matCellDef=\"let element\">\n                <span matLine id=\"{{'file-'+element?.id}}\" role=\"button\" tabindex=\"0\" class=\"adf-file\"\n                    (click)=\"onRowClicked(element)\">{{element.name}}</span>\n            </td>\n        </ng-container>\n    \n        <ng-container *ngFor=\"let columnName of field?.params?.displayableCMProperties\" [matColumnDef]=\"columnName.name\">\n            <th mat-header-cell *matHeaderCellDef>{{ columnName.title ? columnName.title : columnName.name | titlecase\n                }}</th>\n            <td mat-cell *matCellDef=\"let row\">\n                <span matLine id=\"{{'fileProperty-'+row?.id+'-'+columnName?.name}}\" role=\"button\" tabindex=\"0\"\n                    (click)=\"onRowClicked(row)\">{{ getColumnValue(row, columnName) }}</span>\n            </td>\n        </ng-container>\n    \n        <ng-container matColumnDef=\"action\">\n            <th mat-header-cell *matHeaderCellDef></th>\n            <td mat-cell *matCellDef=\"let element\">\n                <button id=\"{{'file-'+element?.id+'-option-menu'}}\" mat-icon-button [matMenuTriggerFor]=\"fileActionMenu\"\n                    *ngIf=\"!!element.content?.mimeType\">\n                    <mat-icon>more_vert</mat-icon>\n                </button>\n                <mat-menu #fileActionMenu=\"matMenu\" xPosition=\"before\">\n                    <button *ngIf=\"displayMenuOption('show') && !!element.content?.mimeType\"\n                        id=\"{{'file-'+element?.id+'-show-file'}}\" mat-menu-item (click)=\"onAttachFileClicked(element)\">\n                        <mat-icon>visibility</mat-icon>\n                        <span>{{ 'FORM.FIELD.VIEW_FILE' | translate }}</span>\n                    </button>\n                    <button *ngIf=\"displayMenuOption('download') && !!element.content?.mimeType\"\n                        id=\"{{'file-'+element?.id+'-download-file'}}\" mat-menu-item (click)=\"downloadContent(element)\">\n                        <mat-icon>file_download</mat-icon>\n                        <span>{{ 'FORM.FIELD.DOWNLOAD_FILE' | translate }}</span>\n                    </button>\n                    <button *ngIf=\"displayMenuOption('retrieveMetadata') && !!element.content?.mimeType\"\n                        id=\"{{'file-'+element?.id+'-retrieve-file-metadata'}}\" mat-menu-item\n                        (click)=\"contentModelFormFileHandler(element)\">\n                        <mat-icon class=\"mat-24\">low_priority</mat-icon>\n                        <span>{{ 'ADF_CLOUD_FORM_COMPONENT.RETRIEVE_METADATA' | translate }}</span>\n                    </button>\n                    <button *ngIf=\"!field.readOnly && displayMenuOption('remove')\"\n                        id=\"{{'file-'+element?.id+'-remove-file'}}\" mat-menu-item [id]=\"'file-'+element?.id+'-remove'\"\n                        (click)=\"onRemoveAttachFile(element);\" (keyup.enter)=\"onRemoveAttachFile(element);\">\n                        <mat-icon class=\"mat-24\">highlight_off</mat-icon>\n                        <span>{{ 'FORM.FIELD.REMOVE_FILE' | translate }}</span>\n                    </button>\n                </mat-menu>\n            </td>\n        </ng-container>\n        <tr mat-header-row *matHeaderRowDef=\"displayedColumns\"></tr>\n        <tr mat-row *matRowDef=\"let row; columns: displayedColumns\"></tr>\n    \n    </table>\n</div>\n",
                styles: [".adf-file-properties-table table{border:1px solid var(--theme-border-color);box-shadow:none;width:100%}.adf-file-properties-table table .adf-datatable-selected{padding:6px}.adf-file-properties-table table tr.mat-header-row{height:32px}.adf-file-properties-table table th{font-weight:700;padding-right:0!important}.adf-file-properties-table table td{max-width:50px;overflow:hidden;padding-right:0!important;text-overflow:ellipsis;white-space:nowrap}.adf-file-properties-table table td.mat-cell:first-of-type,.adf-file-properties-table table td.mat-cell:last-of-type,.adf-file-properties-table table td.mat-footer-cell:first-of-type,.adf-file-properties-table table td.mat-footer-cell:last-of-type{width:18%!important}"]
            }]
    }], function () { return [{ type: ɵngcc1.LocalizedDatePipe }, { type: ɵngcc1.ThumbnailService }]; }, { rowClick: [{
            type: Output
        }], attachFileClick: [{
            type: Output
        }], downloadFile: [{
            type: Output
        }], contentModelFileHandler: [{
            type: Output
        }], removeAttachFile: [{
            type: Output
        }], uploadedFiles: [{
            type: Input
        }], hasFile: [{
            type: Input
        }], selectedNode: [{
            type: Input
        }], field: [{
            type: Input
        }], displayedColumns: [{
            type: Input
        }], mimeTypeIcon: [{
            type: Input
        }] }); })();
