// @alfresco/adf-core/esm2015/testing/core.testing.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule, APP_INITIALIZER } from '@angular/core';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateModule } from '@ngx-translate/core';
import { CoreModule } from '../core.module';
import { AlfrescoApiService } from '../services/alfresco-api.service';
import { AlfrescoApiServiceMock } from '../mock/alfresco-api.service.mock';
import { AppConfigService } from '../app-config/app-config.service';
import { AppConfigServiceMock } from '../mock/app-config.service.mock';
import { TranslationService } from '../services/translation.service';
import { TranslationMock } from '../mock/translation.service.mock';
import { DatePipe } from '@angular/common';
import { CookieService } from '../services/cookie.service';
import { CookieServiceMock } from '../mock/cookie.service.mock';
import { HttpClientModule } from '@angular/common/http';
import { directionalityConfigFactory } from '../services/directionality-config-factory';
import { DirectionalityConfigService } from '../services/directionality-config.service';
import { versionCompatibilityFactory } from '../services/version-compatibility-factory';
import { VersionCompatibilityService } from '../services/version-compatibility.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@ngx-translate/core';
import * as ɵngcc2 from '../core.module';
const ɵ0 = directionalityConfigFactory, ɵ1 = versionCompatibilityFactory;
export class CoreTestingModule {
}
CoreTestingModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CoreTestingModule });
CoreTestingModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CoreTestingModule_Factory(t) { return new (t || CoreTestingModule)(); }, providers: [
        DatePipe,
        { provide: AlfrescoApiService, useClass: AlfrescoApiServiceMock },
        { provide: AppConfigService, useClass: AppConfigServiceMock },
        { provide: TranslationService, useClass: TranslationMock },
        { provide: CookieService, useClass: CookieServiceMock },
        {
            provide: APP_INITIALIZER,
            useFactory: ɵ0,
            deps: [DirectionalityConfigService],
            multi: true
        },
        {
            provide: APP_INITIALIZER,
            useFactory: ɵ1,
            deps: [VersionCompatibilityService],
            multi: true
        }
    ], imports: [[
            NoopAnimationsModule,
            RouterTestingModule,
            HttpClientModule,
            TranslateModule.forRoot(),
            CoreModule.forRoot()
        ], NoopAnimationsModule,
        CoreModule,
        TranslateModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CoreTestingModule, { imports: function () { return [NoopAnimationsModule,
        RouterTestingModule,
        HttpClientModule, ɵngcc1.TranslateModule, ɵngcc2.CoreModule]; }, exports: function () { return [NoopAnimationsModule,
        CoreModule,
        TranslateModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CoreTestingModule, [{
        type: NgModule,
        args: [{
                imports: [
                    NoopAnimationsModule,
                    RouterTestingModule,
                    HttpClientModule,
                    TranslateModule.forRoot(),
                    CoreModule.forRoot()
                ],
                providers: [
                    DatePipe,
                    { provide: AlfrescoApiService, useClass: AlfrescoApiServiceMock },
                    { provide: AppConfigService, useClass: AppConfigServiceMock },
                    { provide: TranslationService, useClass: TranslationMock },
                    { provide: CookieService, useClass: CookieServiceMock },
                    {
                        provide: APP_INITIALIZER,
                        useFactory: ɵ0,
                        deps: [DirectionalityConfigService],
                        multi: true
                    },
                    {
                        provide: APP_INITIALIZER,
                        useFactory: ɵ1,
                        deps: [VersionCompatibilityService],
                        multi: true
                    }
                ],
                exports: [
                    NoopAnimationsModule,
                    CoreModule,
                    TranslateModule
                ]
            }]
    }], null, null); })();
export { ɵ0, ɵ1 };
