// @alfresco/adf-core/esm2015/datatable/components/filesize-cell/filesize-cell.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { DataTableCellComponent } from '../datatable-cell/datatable-cell.component';
import { AlfrescoApiService } from '../../../services/alfresco-api.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../../services/alfresco-api.service';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../../../pipes/file-size.pipe';

function FileSizeCellComponent_ng_container_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "span", 1);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const fileSize_r1 = ctx.ngIf;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("title", ctx_r0.tooltip);
    ɵngcc0.ɵɵattribute("aria-label", fileSize_r1);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(fileSize_r1);
} }
export class FileSizeCellComponent extends DataTableCellComponent {
    constructor(alfrescoApiService) {
        super(alfrescoApiService);
    }
}
FileSizeCellComponent.ɵfac = function FileSizeCellComponent_Factory(t) { return new (t || FileSizeCellComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.AlfrescoApiService)); };
FileSizeCellComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FileSizeCellComponent, selectors: [["adf-filesize-cell"]], hostAttrs: [1, "adf-filesize-cell"], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 3, vars: 5, consts: [[4, "ngIf"], [3, "title"]], template: function FileSizeCellComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, FileSizeCellComponent_ng_container_0_Template, 3, 3, "ng-container", 0);
        ɵngcc0.ɵɵpipe(1, "adfFileSize");
        ɵngcc0.ɵɵpipe(2, "async");
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(1, 1, ɵngcc0.ɵɵpipeBind1(2, 3, ctx.value$)));
    } }, directives: [ɵngcc2.NgIf], pipes: [ɵngcc3.FileSizePipe, ɵngcc2.AsyncPipe], encapsulation: 2 });
FileSizeCellComponent.ctorParameters = () => [
    { type: AlfrescoApiService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FileSizeCellComponent, [{
        type: Component,
        args: [{
                selector: 'adf-filesize-cell',
                template: `
        <ng-container *ngIf="(value$ | async | adfFileSize) as fileSize">
            <span
                [title]="tooltip"
                [attr.aria-label]="fileSize"
                >{{ fileSize }}</span
            >
        </ng-container>
    `,
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-filesize-cell' }
            }]
    }], function () { return [{ type: ɵngcc1.AlfrescoApiService }]; }, null); })();
