// @alfresco/adf-process-services/esm2015/lib/people/components/people-selector/people-selector.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewChild, ViewEncapsulation, EventEmitter, Input, Output } from '@angular/core';
import { PeopleProcessService, UserProcessModel, LogService, TranslationService } from '@alfresco/adf-core';
import { PeopleSearchFieldComponent } from '../people-search-field/people-search-field.component';
import { getDisplayUser } from '../../helpers/get-display-user';
import { of } from 'rxjs';
import { catchError } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
import * as ɵngcc2 from '../people-search-field/people-search-field.component';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@angular/material/button';
import * as ɵngcc5 from '@angular/material/icon';

const _c0 = ["peopleSearchField"];
function PeopleSelectorComponent_button_2_Template(rf, ctx) { if (rf & 1) {
    const _r3 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 3);
    ɵngcc0.ɵɵlistener("click", function PeopleSelectorComponent_button_2_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r3); const ctx_r2 = ɵngcc0.ɵɵnextContext(); return ctx_r2.userDeselected(); });
    ɵngcc0.ɵɵelementStart(1, "mat-icon");
    ɵngcc0.ɵɵtext(2, "cancel");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} }
const DEFAULT_ASSIGNEE_PLACEHOLDER = 'ADF_TASK_LIST.PEOPLE.ASSIGNEE';
export class PeopleSelectorComponent {
    constructor(peopleProcessService, logService, translationService) {
        this.peopleProcessService = peopleProcessService;
        this.logService = logService;
        this.translationService = translationService;
        this.peopleIdChange = new EventEmitter();
        this.performSearch = this.searchUser.bind(this);
        this.defaultPlaceholder = this.translationService.instant(DEFAULT_ASSIGNEE_PLACEHOLDER);
    }
    searchUser(searchWord) {
        return this.peopleProcessService.getWorkflowUsers(undefined, searchWord)
            .pipe(catchError(this.onSearchUserError.bind(this)));
    }
    onSearchUserError() {
        this.logService.error('getWorkflowUsers threw error');
        return of([]);
    }
    userSelected(user) {
        this.updateUserSelection(user);
    }
    userDeselected() {
        this.updateUserSelection(undefined);
    }
    updateUserSelection(user) {
        this.selectedUser = user;
        this.peopleIdChange.emit(user && user.id || undefined);
        this.searchFieldComponent.reset();
    }
    get placeholder() {
        if (!this.selectedUser) {
            return this.defaultPlaceholder;
        }
        return getDisplayUser(this.selectedUser.firstName, this.selectedUser.lastName, ' ');
    }
}
PeopleSelectorComponent.ɵfac = function PeopleSelectorComponent_Factory(t) { return new (t || PeopleSelectorComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.PeopleProcessService), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.LogService), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TranslationService)); };
PeopleSelectorComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PeopleSelectorComponent, selectors: [["adf-people-selector"]], viewQuery: function PeopleSelectorComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.searchFieldComponent = _t.first);
    } }, hostAttrs: [1, "adf-people-selector"], inputs: { peopleId: "peopleId" }, outputs: { peopleIdChange: "peopleIdChange" }, decls: 3, vars: 3, consts: [[1, "adf-people-selector-field", 3, "performSearch", "placeholder", "rowClick"], ["peopleSearchField", ""], ["mat-icon-button", "", "class", "adf-people-selector-deselect", "data-automation-id", "adf-people-selector-deselect", 3, "click", 4, "ngIf"], ["mat-icon-button", "", "data-automation-id", "adf-people-selector-deselect", 1, "adf-people-selector-deselect", 3, "click"]], template: function PeopleSelectorComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "adf-people-search-field", 0, 1);
        ɵngcc0.ɵɵlistener("rowClick", function PeopleSelectorComponent_Template_adf_people_search_field_rowClick_0_listener($event) { return ctx.userSelected($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(2, PeopleSelectorComponent_button_2_Template, 3, 0, "button", 2);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("performSearch", ctx.performSearch)("placeholder", ctx.placeholder);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.selectedUser);
    } }, directives: [ɵngcc2.PeopleSearchFieldComponent, ɵngcc3.NgIf, ɵngcc4.MatButton, ɵngcc5.MatIcon], styles: [".adf-people-selector{display:flex;flex-flow:row;justify-content:space-between}.adf-people-selector-field{flex:1 1 auto}.adf-people-selector-deselect{flex:0 0 auto;right:5px;top:5px}"], encapsulation: 2 });
PeopleSelectorComponent.ctorParameters = () => [
    { type: PeopleProcessService },
    { type: LogService },
    { type: TranslationService }
];
PeopleSelectorComponent.propDecorators = {
    peopleId: [{ type: Input }],
    peopleIdChange: [{ type: Output }],
    searchFieldComponent: [{ type: ViewChild, args: ['peopleSearchField', { static: true },] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PeopleSelectorComponent, [{
        type: Component,
        args: [{
                selector: 'adf-people-selector',
                template: "<adf-people-search-field\n    #peopleSearchField\n    class=\"adf-people-selector-field\"\n    [performSearch]=\"performSearch\"\n    [placeholder]=\"placeholder\"\n    (rowClick)=\"userSelected($event)\">\n</adf-people-search-field>\n<button\n    *ngIf=\"selectedUser\"\n    mat-icon-button\n    class=\"adf-people-selector-deselect\"\n    data-automation-id=\"adf-people-selector-deselect\"\n    (click)=\"userDeselected()\">\n        <mat-icon>cancel</mat-icon>\n</button>\n",
                host: { 'class': 'adf-people-selector' },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-people-selector{display:flex;flex-flow:row;justify-content:space-between}.adf-people-selector-field{flex:1 1 auto}.adf-people-selector-deselect{flex:0 0 auto;right:5px;top:5px}"]
            }]
    }], function () { return [{ type: ɵngcc1.PeopleProcessService }, { type: ɵngcc1.LogService }, { type: ɵngcc1.TranslationService }]; }, { peopleIdChange: [{
            type: Output
        }], peopleId: [{
            type: Input
        }], searchFieldComponent: [{
            type: ViewChild,
            args: ['peopleSearchField', { static: true }]
        }] }); })();
