// @alfresco/adf-process-services-cloud/esm2015/lib/models/date-cloud-filter.model.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export var DateCloudFilterType;
(function (DateCloudFilterType) {
    DateCloudFilterType["NO_DATE"] = "NO_DATE";
    DateCloudFilterType["TODAY"] = "TODAY";
    DateCloudFilterType["TOMORROW"] = "TOMORROW";
    DateCloudFilterType["NEXT_7_DAYS"] = "NEXT_7_DAYS";
    DateCloudFilterType["WEEK"] = "WEEK";
    DateCloudFilterType["MONTH"] = "MONTH";
    DateCloudFilterType["QUARTER"] = "QUARTER";
    DateCloudFilterType["YEAR"] = "YEAR";
    DateCloudFilterType["RANGE"] = "RANGE";
})(DateCloudFilterType || (DateCloudFilterType = {}));
