// @alfresco/adf-core/esm2015/datatable/directives/header-filter-template.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ContentChild, Directive, TemplateRef } from '@angular/core';
import { DataTableComponent } from '../components/datatable/datatable.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../components/datatable/datatable.component';
export class HeaderFilterTemplateDirective {
    constructor(dataTable) {
        this.dataTable = dataTable;
    }
    ngAfterContentInit() {
        if (this.dataTable) {
            this.dataTable.headerFilterTemplate = this.template;
        }
    }
}
HeaderFilterTemplateDirective.ɵfac = function HeaderFilterTemplateDirective_Factory(t) { return new (t || HeaderFilterTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DataTableComponent)); };
HeaderFilterTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: HeaderFilterTemplateDirective, selectors: [["adf-header-filter-template"]], contentQueries: function HeaderFilterTemplateDirective_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, TemplateRef, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.template = _t.first);
    } } });
HeaderFilterTemplateDirective.ctorParameters = () => [
    { type: DataTableComponent }
];
HeaderFilterTemplateDirective.propDecorators = {
    template: [{ type: ContentChild, args: [TemplateRef,] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HeaderFilterTemplateDirective, [{
        type: Directive,
        args: [{
                selector: 'adf-header-filter-template'
            }]
    }], function () { return [{ type: ɵngcc1.DataTableComponent }]; }, { template: [{
            type: ContentChild,
            args: [TemplateRef]
        }] }); })();
