// @alfresco/adf-core/esm2015/datatable/directives/loading-template.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ContentChild, Directive, TemplateRef } from '@angular/core';
import { DataTableComponent } from '../components/datatable/datatable.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../components/datatable/datatable.component';
export class LoadingContentTemplateDirective {
    constructor(dataTable) {
        this.dataTable = dataTable;
    }
    ngAfterContentInit() {
        if (this.dataTable) {
            this.dataTable.loadingTemplate = this.template;
        }
    }
}
LoadingContentTemplateDirective.ɵfac = function LoadingContentTemplateDirective_Factory(t) { return new (t || LoadingContentTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DataTableComponent)); };
LoadingContentTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: LoadingContentTemplateDirective, selectors: [["adf-loading-content-template"], ["loading-content-template"]], contentQueries: function LoadingContentTemplateDirective_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, TemplateRef, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.template = _t.first);
    } } });
LoadingContentTemplateDirective.ctorParameters = () => [
    { type: DataTableComponent }
];
LoadingContentTemplateDirective.propDecorators = {
    template: [{ type: ContentChild, args: [TemplateRef,] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LoadingContentTemplateDirective, [{
        type: Directive,
        args: [{
                selector: 'adf-loading-content-template, loading-content-template'
            }]
    }], function () { return [{ type: ɵngcc1.DataTableComponent }]; }, { template: [{
            type: ContentChild,
            args: [TemplateRef]
        }] }); })();
