// @alfresco/adf-process-services/esm2015/lib/app-list/apps-list.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { FlexLayoutModule } from '@angular/flex-layout';
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { CoreModule } from '@alfresco/adf-core';
import { AppsListComponent } from './apps-list.component';
import { SelectAppsDialogComponent } from './select-apps-dialog.component';
import * as ɵngcc0 from '@angular/core';
export class AppsListModule {
}
AppsListModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AppsListModule });
AppsListModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AppsListModule_Factory(t) { return new (t || AppsListModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            FlexLayoutModule,
            CoreModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AppsListModule, { declarations: function () { return [AppsListComponent,
        SelectAppsDialogComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        FlexLayoutModule,
        CoreModule]; }, exports: function () { return [AppsListComponent,
        SelectAppsDialogComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppsListModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    FlexLayoutModule,
                    CoreModule
                ],
                declarations: [
                    AppsListComponent,
                    SelectAppsDialogComponent
                ],
                exports: [
                    AppsListComponent,
                    SelectAppsDialogComponent
                ]
            }]
    }], null, null); })();
