// @alfresco/adf-content-services/esm2015/lib/search/components/search-panel/search-panel.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Inject, ViewEncapsulation } from '@angular/core';
import { ContentNodeSelectorPanelService } from '../../../content-node-selector/content-node-selector-panel.service';
import { SearchQueryBuilderService } from '../../services/search-query-builder.service';
import { SEARCH_QUERY_SERVICE_TOKEN } from '../../search-query-service.token';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../../content-node-selector/content-node-selector-panel.service';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../search-filter/search-filter.component';
import * as ɵngcc4 from '../../services/search-query-builder.service';

function SearchPanelComponent_adf_search_filter_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-search-filter", 2);
} if (rf & 2) {
    ɵngcc0.ɵɵproperty("showContextFacets", false);
} }
export class SearchPanelComponent {
    constructor(contentNodeSelectorPanelService, queryBuilderService) {
        this.contentNodeSelectorPanelService = contentNodeSelectorPanelService;
        this.queryBuilderService = queryBuilderService;
    }
    ngOnInit() {
        this.queryBuilderService.categories = this.contentNodeSelectorPanelService.convertCustomModelPropertiesToSearchCategories();
    }
    hasCustomModels() {
        var _a, _b;
        return ((_b = (_a = this.contentNodeSelectorPanelService) === null || _a === void 0 ? void 0 : _a.customModels) === null || _b === void 0 ? void 0 : _b.length) > 0;
    }
}
SearchPanelComponent.ɵfac = function SearchPanelComponent_Factory(t) { return new (t || SearchPanelComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ContentNodeSelectorPanelService), ɵngcc0.ɵɵdirectiveInject(SEARCH_QUERY_SERVICE_TOKEN)); };
SearchPanelComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SearchPanelComponent, selectors: [["adf-search-panel"]], hostAttrs: [1, "adf-search-panel"], decls: 2, vars: 1, consts: [["data-automation-id", "adf-search-panel-container", 1, "adf-search-panel-scrollable"], ["class", "app-search-settings", 3, "showContextFacets", 4, "ngIf"], [1, "app-search-settings", 3, "showContextFacets"]], template: function SearchPanelComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, SearchPanelComponent_adf_search_filter_1_Template, 1, 1, "adf-search-filter", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasCustomModels());
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.SearchFilterComponent], styles: [".adf-search-panel-scrollable{max-height:349px;overflow:scroll}"], encapsulation: 2 });
SearchPanelComponent.ctorParameters = () => [
    { type: ContentNodeSelectorPanelService },
    { type: SearchQueryBuilderService, decorators: [{ type: Inject, args: [SEARCH_QUERY_SERVICE_TOKEN,] }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SearchPanelComponent, [{
        type: Component,
        args: [{
                selector: 'adf-search-panel',
                template: "<div class=\"adf-search-panel-scrollable\" data-automation-id=\"adf-search-panel-container\">\n    <adf-search-filter *ngIf=\"hasCustomModels()\"\n                       class=\"app-search-settings\"\n                       [showContextFacets]=\"false\">\n    </adf-search-filter>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-search-panel' },
                styles: [".adf-search-panel-scrollable{max-height:349px;overflow:scroll}"]
            }]
    }], function () { return [{ type: ɵngcc1.ContentNodeSelectorPanelService }, { type: ɵngcc4.SearchQueryBuilderService, decorators: [{
                type: Inject,
                args: [SEARCH_QUERY_SERVICE_TOKEN]
            }] }]; }, null); })();
