// @alfresco/adf-content-services/esm2015/lib/directives/content-directive.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { TranslateModule } from '@ngx-translate/core';
import { NodeLockDirective } from './node-lock.directive';
import { NodeCounterComponent, NodeCounterDirective } from './node-counter.directive';
import * as ɵngcc0 from '@angular/core';
export class ContentDirectiveModule {
}
ContentDirectiveModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ContentDirectiveModule });
ContentDirectiveModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ContentDirectiveModule_Factory(t) { return new (t || ContentDirectiveModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            TranslateModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ContentDirectiveModule, { declarations: function () { return [NodeLockDirective,
        NodeCounterDirective,
        NodeCounterComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        TranslateModule]; }, exports: function () { return [NodeLockDirective,
        NodeCounterDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContentDirectiveModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    TranslateModule
                ],
                declarations: [
                    NodeLockDirective,
                    NodeCounterDirective,
                    NodeCounterComponent
                ],
                exports: [
                    NodeLockDirective,
                    NodeCounterDirective
                ]
            }]
    }], null, null); })();
