// @alfresco/adf-content-services/esm2015/lib/directives/node-counter.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, Input, Component, ComponentFactoryResolver, ViewContainerRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@ngx-translate/core';

const _c0 = function (a0) { return { count: a0 }; };
export class NodeCounterDirective {
    constructor(resolver, viewContainerRef) {
        this.resolver = resolver;
        this.viewContainerRef = viewContainerRef;
    }
    ngOnInit() {
        const componentFactory = this.resolver.resolveComponentFactory(NodeCounterComponent);
        this.componentRef = this.viewContainerRef.createComponent(componentFactory).instance;
        this.componentRef.counter = this.counter;
    }
    ngOnChanges() {
        if (this.componentRef) {
            this.componentRef.counter = this.counter;
        }
    }
}
NodeCounterDirective.ɵfac = function NodeCounterDirective_Factory(t) { return new (t || NodeCounterDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ComponentFactoryResolver), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef)); };
NodeCounterDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NodeCounterDirective, selectors: [["", "adf-node-counter", ""]], inputs: { counter: ["adf-node-counter", "counter"] }, features: [ɵngcc0.ɵɵNgOnChangesFeature] });
NodeCounterDirective.ctorParameters = () => [
    { type: ComponentFactoryResolver },
    { type: ViewContainerRef }
];
NodeCounterDirective.propDecorators = {
    counter: [{ type: Input, args: ['adf-node-counter',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NodeCounterDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-node-counter]'
            }]
    }], function () { return [{ type: ɵngcc0.ComponentFactoryResolver }, { type: ɵngcc0.ViewContainerRef }]; }, { counter: [{
            type: Input,
            args: ['adf-node-counter']
        }] }); })();
export class NodeCounterComponent {
}
NodeCounterComponent.ɵfac = function NodeCounterComponent_Factory(t) { return new (t || NodeCounterComponent)(); };
NodeCounterComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NodeCounterComponent, selectors: [["adf-node-counter"]], decls: 3, vars: 6, template: function NodeCounterComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵtext(1);
        ɵngcc0.ɵɵpipe(2, "translate");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind2(2, 1, "NODE_COUNTER.SELECTED_COUNT", ɵngcc0.ɵɵpureFunction1(4, _c0, ctx.counter)));
    } }, pipes: [ɵngcc1.TranslatePipe], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NodeCounterComponent, [{
        type: Component,
        args: [{
                selector: 'adf-node-counter',
                template: `
        <div>{{ 'NODE_COUNTER.SELECTED_COUNT' | translate: { count: counter } }}</div>
    `
            }]
    }], null, null); })();
