// @alfresco/adf-core/esm2015/form/components/widgets/hyperlink/hyperlink.widget.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { WidgetComponent } from './../widget.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../../../services/form.service';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/material/tooltip';
import * as ɵngcc4 from '@ngx-translate/core';

function HyperlinkWidgetComponent_span_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1, "*");
    ɵngcc0.ɵɵelementEnd();
} }
export class HyperlinkWidgetComponent extends WidgetComponent {
    constructor(formService) {
        super(formService);
        this.formService = formService;
        this.linkUrl = '#';
        this.linkText = null;
    }
    ngOnInit() {
        if (this.field) {
            this.linkUrl = this.getHyperlinkUrl(this.field);
            this.linkText = this.getHyperlinkText(this.field);
        }
    }
    getHyperlinkUrl(field) {
        let value = field.value || field.hyperlinkUrl;
        if (value && !/^https?:\/\//i.test(value)) {
            value = `${HyperlinkWidgetComponent.DEFAULT_HYPERLINK_SCHEME}${value}`;
        }
        return value || HyperlinkWidgetComponent.DEFAULT_HYPERLINK_URL;
    }
    getHyperlinkText(field) {
        if (field) {
            return field.displayText || field.hyperlinkUrl || field.value;
        }
        return null;
    }
}
HyperlinkWidgetComponent.ɵfac = function HyperlinkWidgetComponent_Factory(t) { return new (t || HyperlinkWidgetComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService)); };
HyperlinkWidgetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: HyperlinkWidgetComponent, selectors: [["hyperlink-widget"]], hostBindings: function HyperlinkWidgetComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function HyperlinkWidgetComponent_click_HostBindingHandler($event) { return ctx.event($event); })("blur", function HyperlinkWidgetComponent_blur_HostBindingHandler($event) { return ctx.event($event); })("change", function HyperlinkWidgetComponent_change_HostBindingHandler($event) { return ctx.event($event); })("focus", function HyperlinkWidgetComponent_focus_HostBindingHandler($event) { return ctx.event($event); })("focusin", function HyperlinkWidgetComponent_focusin_HostBindingHandler($event) { return ctx.event($event); })("focusout", function HyperlinkWidgetComponent_focusout_HostBindingHandler($event) { return ctx.event($event); })("input", function HyperlinkWidgetComponent_input_HostBindingHandler($event) { return ctx.event($event); })("invalid", function HyperlinkWidgetComponent_invalid_HostBindingHandler($event) { return ctx.event($event); })("select", function HyperlinkWidgetComponent_select_HostBindingHandler($event) { return ctx.event($event); });
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 8, vars: 11, consts: [[1, "adf-label"], [4, "ngIf"], ["matTooltipPosition", "above", "matTooltipShowDelay", "1000", 3, "matTooltip"], ["target", "_blank", "rel", "nofollow", 3, "href"]], template: function HyperlinkWidgetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "label", 0);
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵpipe(3, "translate");
        ɵngcc0.ɵɵtemplate(4, HyperlinkWidgetComponent_span_4_Template, 2, 0, "span", 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 2);
        ɵngcc0.ɵɵelementStart(6, "a", 3);
        ɵngcc0.ɵɵtext(7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMapInterpolate1("adf-hyperlink-widget ", ctx.field.className, "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵattribute("for", ctx.field.id);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 9, ctx.field.name));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isRequired());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("matTooltip", ctx.field.tooltip);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("href", ctx.linkUrl, ɵngcc0.ɵɵsanitizeUrl);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ctx.linkText);
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.MatTooltip], pipes: [ɵngcc4.TranslatePipe], styles: [".adf-hyperlink-widget{border-top:.84375em solid transparent;padding:.4375em 0}.adf-hyperlink-widget a{color:var(--theme-primary-color)}"], encapsulation: 2 });
HyperlinkWidgetComponent.DEFAULT_HYPERLINK_URL = '#';
HyperlinkWidgetComponent.DEFAULT_HYPERLINK_SCHEME = 'http://';
HyperlinkWidgetComponent.ctorParameters = () => [
    { type: FormService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HyperlinkWidgetComponent, [{
        type: Component,
        args: [{
                selector: 'hyperlink-widget',
                template: "<div class=\"adf-hyperlink-widget {{field.className}}\">\n    <label class=\"adf-label\" [attr.for]=\"field.id\">{{field.name | translate }}<span\n            *ngIf=\"isRequired()\">*</span></label>\n    <div [matTooltip]=\"field.tooltip\" matTooltipPosition=\"above\" matTooltipShowDelay=\"1000\">\n        <a [href]=\"linkUrl\" target=\"_blank\" rel=\"nofollow\">{{linkText}}</a>\n    </div>\n</div>\n",
                host: {
                    '(click)': 'event($event)',
                    '(blur)': 'event($event)',
                    '(change)': 'event($event)',
                    '(focus)': 'event($event)',
                    '(focusin)': 'event($event)',
                    '(focusout)': 'event($event)',
                    '(input)': 'event($event)',
                    '(invalid)': 'event($event)',
                    '(select)': 'event($event)'
                },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-hyperlink-widget{border-top:.84375em solid transparent;padding:.4375em 0}.adf-hyperlink-widget a{color:var(--theme-primary-color)}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormService }]; }, null); })();
