// @alfresco/adf-process-services-cloud/esm2015/lib/task/task-header/task-header-cloud.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MaterialModule } from '../../material.module';
import { CoreModule } from '@alfresco/adf-core';
import { TaskHeaderCloudComponent } from './components/task-header-cloud.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
export class TaskHeaderCloudModule {
}
TaskHeaderCloudModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TaskHeaderCloudModule });
TaskHeaderCloudModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function TaskHeaderCloudModule_Factory(t) { return new (t || TaskHeaderCloudModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            CoreModule.forChild()
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TaskHeaderCloudModule, { declarations: function () { return [TaskHeaderCloudComponent]; }, imports: function () { return [CommonModule,
        MaterialModule, ɵngcc1.CoreModule]; }, exports: function () { return [TaskHeaderCloudComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TaskHeaderCloudModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    CoreModule.forChild()
                ],
                declarations: [
                    TaskHeaderCloudComponent
                ],
                exports: [
                    TaskHeaderCloudComponent
                ]
            }]
    }], null, null); })();
