// @alfresco/adf-content-services/esm2015/lib/social/social.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { LikeComponent } from './like.component';
import { RatingComponent } from './rating.component';
import * as ɵngcc0 from '@angular/core';
export class SocialModule {
}
SocialModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SocialModule });
SocialModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SocialModule_Factory(t) { return new (t || SocialModule)(); }, imports: [[
            CommonModule,
            MaterialModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SocialModule, { declarations: function () { return [RatingComponent,
        LikeComponent]; }, imports: function () { return [CommonModule,
        MaterialModule]; }, exports: function () { return [RatingComponent,
        LikeComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SocialModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule
                ],
                exports: [
                    RatingComponent,
                    LikeComponent
                ],
                declarations: [
                    RatingComponent,
                    LikeComponent
                ]
            }]
    }], null, null); })();
