// @alfresco/adf-process-services-cloud/esm2015/lib/form/models/form-cloud-representation.model.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export class FormCloudRepresentation {
    constructor(obj) {
        this.id = obj.id || null;
        this.name = obj.name || null;
        this.description = obj.description || null;
        this.version = obj.version || null;
        this.tabs = obj.tabs || null;
        this.fields = obj.fields || null;
        this.outcomes = obj.outcomes || null;
        this.metadata = obj.metadata || null;
        this.variables = obj.variables || null;
        this.taskId = obj.taskId || null;
        this.taskName = obj.taskName || null;
        this.processDefinitionId = obj.processDefinitionId || null;
        this.processInstanceId = obj.processInstanceId || null;
        this.selectedOutcome = obj.selectedOutcome || null;
    }
}
export var FileSourceTypes;
(function (FileSourceTypes) {
    FileSourceTypes["ALL_FILE_SOURCES_SERVICE_ID"] = "all-file-sources";
    FileSourceTypes["ALFRESCO_CONTENT_SOURCES_SERVICE_ID"] = "alfresco-content";
})(FileSourceTypes || (FileSourceTypes = {}));
export var DestinationFolderPathType;
(function (DestinationFolderPathType) {
    DestinationFolderPathType["STATIC_TYPE"] = "value";
    DestinationFolderPathType["STRING_TYPE"] = "string";
    DestinationFolderPathType["FOLDER_TYPE"] = "folder";
})(DestinationFolderPathType || (DestinationFolderPathType = {}));
