// @alfresco/adf-process-services-cloud/esm2015/lib/process/start-process/start-process-cloud.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { FlexLayoutModule } from '@angular/flex-layout';
import { MaterialModule } from '../../material.module';
import { FormCloudModule } from '../../form/form-cloud.module';
import { StartProcessCloudComponent } from './components/start-process-cloud.component';
import { CoreModule } from '@alfresco/adf-core';
import * as ɵngcc0 from '@angular/core';
export class StartProcessCloudModule {
}
StartProcessCloudModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: StartProcessCloudModule });
StartProcessCloudModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function StartProcessCloudModule_Factory(t) { return new (t || StartProcessCloudModule)(); }, imports: [[
            FormsModule,
            CommonModule,
            FormCloudModule,
            MaterialModule,
            FlexLayoutModule,
            ReactiveFormsModule,
            CoreModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(StartProcessCloudModule, { declarations: function () { return [StartProcessCloudComponent]; }, imports: function () { return [FormsModule,
        CommonModule,
        FormCloudModule,
        MaterialModule,
        FlexLayoutModule,
        ReactiveFormsModule,
        CoreModule]; }, exports: function () { return [StartProcessCloudComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(StartProcessCloudModule, [{
        type: NgModule,
        args: [{
                imports: [
                    FormsModule,
                    CommonModule,
                    FormCloudModule,
                    MaterialModule,
                    FlexLayoutModule,
                    ReactiveFormsModule,
                    CoreModule
                ],
                declarations: [
                    StartProcessCloudComponent
                ],
                exports: [
                    StartProcessCloudComponent
                ]
            }]
    }], null, null); })();
