// @alfresco/adf-process-services-cloud/esm2015/lib/process/directives/cancel-process.directive.js
import { Directive, HostListener, Output, EventEmitter, ElementRef } from '@angular/core';
import { IdentityUserService } from '@alfresco/adf-core';
import { ProcessCloudService } from '../services/process-cloud.service';
import { takeUntil } from 'rxjs/operators';
import { Subject } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/process-cloud.service';
import * as ɵngcc2 from '@alfresco/adf-core';
export class CancelProcessDirective {
    constructor(elementRef, processCloudService, identityUserService) {
        this.elementRef = elementRef;
        this.processCloudService = processCloudService;
        this.identityUserService = identityUserService;
        this.success = new EventEmitter();
        this.error = new EventEmitter();
        this.canCancelProcess = false;
        this.onDestroy$ = new Subject();
    }
    ngOnInit() {
        this.processCloudService.dataChangesDetected
            .pipe(takeUntil(this.onDestroy$))
            .subscribe((processDetails) => {
            this.processInstanceDetails = processDetails;
            this.canCancelProcess = this.checkCanCancelProcess();
            this.setElementVisibility();
        });
    }
    onClick() {
        this.cancelProcess();
    }
    setElementVisibility() {
        this.elementRef.nativeElement.disabled = !this.canCancelProcess;
    }
    checkCanCancelProcess() {
        const currentUser = this.identityUserService.getCurrentUserInfo().username;
        return this.processInstanceDetails.initiator === currentUser && this.processInstanceDetails.status === 'RUNNING';
    }
    cancelProcess() {
        if (this.canCancelProcess) {
            this.processCloudService.cancelProcess(this.processInstanceDetails.appName, this.processInstanceDetails.id)
                .subscribe((response) => this.success.emit(response), (error) => this.error.emit(error));
        }
        else {
            this.error.emit('Permission denied, only process initiator can cancel the process');
        }
    }
    ngOnDestroy() {
        this.onDestroy$.next(true);
        this.onDestroy$.complete();
    }
}
CancelProcessDirective.ɵfac = function CancelProcessDirective_Factory(t) { return new (t || CancelProcessDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ProcessCloudService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.IdentityUserService)); };
CancelProcessDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CancelProcessDirective, selectors: [["", "adf-cloud-cancel-process", ""]], hostBindings: function CancelProcessDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function CancelProcessDirective_click_HostBindingHandler() { return ctx.onClick(); });
    } }, outputs: { success: "success", error: "error" } });
CancelProcessDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: ProcessCloudService },
    { type: IdentityUserService }
];
CancelProcessDirective.propDecorators = {
    success: [{ type: Output }],
    error: [{ type: Output }],
    onClick: [{ type: HostListener, args: ['click',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CancelProcessDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-cloud-cancel-process]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc1.ProcessCloudService }, { type: ɵngcc2.IdentityUserService }]; }, { success: [{
            type: Output
        }], error: [{
            type: Output
        }], onClick: [{
            type: HostListener,
            args: ['click']
        }] }); })();
