// @alfresco/adf-process-services-cloud/esm2015/lib/process/directives/process-directive.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { CancelProcessDirective } from './cancel-process.directive';
import * as ɵngcc0 from '@angular/core';
export class ProcessDirectiveModule {
}
ProcessDirectiveModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ProcessDirectiveModule });
ProcessDirectiveModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ProcessDirectiveModule_Factory(t) { return new (t || ProcessDirectiveModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ProcessDirectiveModule, { declarations: function () { return [CancelProcessDirective]; }, exports: function () { return [CancelProcessDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ProcessDirectiveModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    CancelProcessDirective
                ],
                exports: [
                    CancelProcessDirective
                ]
            }]
    }], null, null); })();
