// @alfresco/adf-core/esm2015/services/auth-guard-bpm.service.js
import { __awaiter } from "tslib";
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { AppConfigService } from '../app-config/app-config.service';
import { AuthenticationService } from './authentication.service';
import { AuthGuardBase } from './auth-guard-base';
import { MatDialog } from '@angular/material/dialog';
import { StorageService } from './storage.service';
import * as i0 from "@angular/core";
import * as i1 from "./authentication.service";
import * as i2 from "@angular/router";
import * as i3 from "../app-config/app-config.service";
import * as i4 from "@angular/material/dialog";
import * as i5 from "./storage.service";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './authentication.service';
import * as ɵngcc2 from '@angular/router';
import * as ɵngcc3 from '../app-config/app-config.service';
import * as ɵngcc4 from '@angular/material/dialog';
import * as ɵngcc5 from './storage.service';
export class AuthGuardBpm extends AuthGuardBase {
    constructor(authenticationService, router, appConfigService, dialog, storageService) {
        super(authenticationService, router, appConfigService, dialog, storageService);
    }
    checkLogin(_, redirectUrl) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.authenticationService.isBpmLoggedIn() || this.withCredentials) {
                return true;
            }
            return this.redirectToUrl(redirectUrl);
        });
    }
}
AuthGuardBpm.ɵfac = function AuthGuardBpm_Factory(t) { return new (t || AuthGuardBpm)(ɵngcc0.ɵɵinject(ɵngcc1.AuthenticationService), ɵngcc0.ɵɵinject(ɵngcc2.Router), ɵngcc0.ɵɵinject(ɵngcc3.AppConfigService), ɵngcc0.ɵɵinject(ɵngcc4.MatDialog), ɵngcc0.ɵɵinject(ɵngcc5.StorageService)); };
AuthGuardBpm.ɵprov = i0.ɵɵdefineInjectable({ factory: function AuthGuardBpm_Factory() { return new AuthGuardBpm(i0.ɵɵinject(i1.AuthenticationService), i0.ɵɵinject(i2.Router), i0.ɵɵinject(i3.AppConfigService), i0.ɵɵinject(i4.MatDialog), i0.ɵɵinject(i5.StorageService)); }, token: AuthGuardBpm, providedIn: "root" });
AuthGuardBpm.ctorParameters = () => [
    { type: AuthenticationService },
    { type: Router },
    { type: AppConfigService },
    { type: MatDialog },
    { type: StorageService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AuthGuardBpm, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.AuthenticationService }, { type: ɵngcc2.Router }, { type: ɵngcc3.AppConfigService }, { type: ɵngcc4.MatDialog }, { type: ɵngcc5.StorageService }]; }, null); })();
