// @alfresco/adf-core/esm2015/services/auth-bearer.interceptor.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { throwError as observableThrowError } from 'rxjs';
import { Injectable, Injector } from '@angular/core';
import { AuthenticationService } from './authentication.service';
import { catchError, mergeMap } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
export class AuthBearerInterceptor {
    constructor(injector) {
        this.injector = injector;
    }
    loadExcludedUrlsRegex() {
        const excludedUrls = this.authService.getBearerExcludedUrls();
        this.excludedUrlsRegex = excludedUrls.map((urlPattern) => new RegExp(urlPattern, 'gi')) || [];
    }
    intercept(req, next) {
        this.authService = this.injector.get(AuthenticationService);
        if (!this.authService || !this.authService.getBearerExcludedUrls()) {
            return next.handle(req);
        }
        if (!this.excludedUrlsRegex) {
            this.loadExcludedUrlsRegex();
        }
        const urlRequest = req.url;
        const shallPass = !!this.excludedUrlsRegex.find((regex) => regex.test(urlRequest));
        if (shallPass) {
            return next.handle(req)
                .pipe(catchError((error) => {
                return observableThrowError(error);
            }));
        }
        return this.authService.addTokenToHeader(req.headers)
            .pipe(mergeMap((headersWithBearer) => {
            const headerWithContentType = this.appendJsonContentType(headersWithBearer);
            const kcReq = req.clone({ headers: headerWithContentType });
            return next.handle(kcReq)
                .pipe(catchError((error) => {
                return observableThrowError(error);
            }));
        }));
    }
    appendJsonContentType(headers) {
        return headers.set('Content-Type', 'application/json;charset=UTF-8');
    }
}
AuthBearerInterceptor.ɵfac = function AuthBearerInterceptor_Factory(t) { return new (t || AuthBearerInterceptor)(ɵngcc0.ɵɵinject(ɵngcc0.Injector)); };
AuthBearerInterceptor.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: AuthBearerInterceptor, factory: AuthBearerInterceptor.ɵfac });
AuthBearerInterceptor.ctorParameters = () => [
    { type: Injector }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AuthBearerInterceptor, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc0.Injector }]; }, null); })();
