// @alfresco/adf-core/esm2015/services/directionality-config.service.js
import { Injectable, Inject, RendererFactory2 } from '@angular/core';
import { Directionality } from '@angular/cdk/bidi';
import { UserPreferencesService } from '../services/user-preferences.service';
import { DOCUMENT } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "./user-preferences.service";
import * as i3 from "@angular/cdk/bidi";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/user-preferences.service';
import * as ɵngcc2 from '@angular/cdk/bidi';
export class DirectionalityConfigService {
    constructor(document, rendererFactory, userPreferencesService, directionality) {
        this.document = document;
        this.rendererFactory = rendererFactory;
        this.userPreferencesService = userPreferencesService;
        this.directionality = directionality;
        const renderer = this.rendererFactory.createRenderer(null, null);
        this.userPreferencesService
            .select('textOrientation')
            .subscribe((direction) => {
            renderer.setAttribute(this.document.body, 'dir', direction);
            this.directionality.value = direction;
        });
    }
}
DirectionalityConfigService.ɵfac = function DirectionalityConfigService_Factory(t) { return new (t || DirectionalityConfigService)(ɵngcc0.ɵɵinject(DOCUMENT), ɵngcc0.ɵɵinject(ɵngcc0.RendererFactory2), ɵngcc0.ɵɵinject(ɵngcc1.UserPreferencesService), ɵngcc0.ɵɵinject(ɵngcc2.Directionality)); };
DirectionalityConfigService.ɵprov = i0.ɵɵdefineInjectable({ factory: function DirectionalityConfigService_Factory() { return new DirectionalityConfigService(i0.ɵɵinject(i1.DOCUMENT), i0.ɵɵinject(i0.RendererFactory2), i0.ɵɵinject(i2.UserPreferencesService), i0.ɵɵinject(i3.Directionality)); }, token: DirectionalityConfigService, providedIn: "root" });
DirectionalityConfigService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] },
    { type: RendererFactory2 },
    { type: UserPreferencesService },
    { type: Directionality }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DirectionalityConfigService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [DOCUMENT]
            }] }, { type: ɵngcc0.RendererFactory2 }, { type: ɵngcc1.UserPreferencesService }, { type: ɵngcc2.Directionality }]; }, null); })();
