// @alfresco/adf-core/esm2015/services/ecm-user.service.js
import { Injectable } from '@angular/core';
import { from } from 'rxjs';
import { map } from 'rxjs/operators';
import { ContentService } from './content.service';
import { AlfrescoApiService } from './alfresco-api.service';
import { EcmUserModel } from '../models/ecm-user.model';
import { PeopleApi } from '@alfresco/js-api';
import * as i0 from "@angular/core";
import * as i1 from "./alfresco-api.service";
import * as i2 from "./content.service";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './alfresco-api.service';
import * as ɵngcc2 from './content.service';
export class EcmUserService {
    constructor(apiService, contentService) {
        this.apiService = apiService;
        this.contentService = contentService;
    }
    get peopleApi() {
        var _a;
        this._peopleApi = (_a = this._peopleApi) !== null && _a !== void 0 ? _a : new PeopleApi(this.apiService.getInstance());
        return this._peopleApi;
    }
    getUserInfo(userName) {
        return from(this.peopleApi.getPerson(userName))
            .pipe(map((personEntry) => new EcmUserModel(personEntry.entry)));
    }
    getCurrentUserInfo() {
        return this.getUserInfo('-me-');
    }
    getUserProfileImage(avatarId) {
        return this.contentService.getContentUrl(avatarId);
    }
}
EcmUserService.ɵfac = function EcmUserService_Factory(t) { return new (t || EcmUserService)(ɵngcc0.ɵɵinject(ɵngcc1.AlfrescoApiService), ɵngcc0.ɵɵinject(ɵngcc2.ContentService)); };
EcmUserService.ɵprov = i0.ɵɵdefineInjectable({ factory: function EcmUserService_Factory() { return new EcmUserService(i0.ɵɵinject(i1.AlfrescoApiService), i0.ɵɵinject(i2.ContentService)); }, token: EcmUserService, providedIn: "root" });
EcmUserService.ctorParameters = () => [
    { type: AlfrescoApiService },
    { type: ContentService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EcmUserService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.AlfrescoApiService }, { type: ɵngcc2.ContentService }]; }, null); })();
