// @alfresco/adf-core/esm2015/services/language.service.js
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { AppConfigService, AppConfigValues } from '../app-config/app-config.service';
import { UserPreferencesService } from './user-preferences.service';
import * as i0 from "@angular/core";
import * as i1 from "../app-config/app-config.service";
import * as i2 from "./user-preferences.service";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../app-config/app-config.service';
import * as ɵngcc2 from './user-preferences.service';
export class LanguageService {
    constructor(appConfigService, userPreferencesService) {
        this.userPreferencesService = userPreferencesService;
        this.languages = new BehaviorSubject([
            { 'key': 'de', 'label': 'Deutsch' },
            { 'key': 'en', 'label': 'English' },
            { 'key': 'es', 'label': 'Español' },
            { 'key': 'fr', 'label': 'Français' },
            { 'key': 'it', 'label': 'Italiano' },
            { 'key': 'ja', 'label': '日本語' },
            { 'key': 'nb', 'label': 'Bokmål' },
            { 'key': 'nl', 'label': 'Nederlands' },
            { 'key': 'pt-BR', 'label': 'Português (Brasil)' },
            { 'key': 'ru', 'label': 'Русский' },
            { 'key': 'zh-CN', 'label': '中文简体' },
            { 'key': 'cs', 'label': 'Čeština' },
            { 'key': 'da', 'label': 'Dansk' },
            { 'key': 'fi', 'label': 'Suomi' },
            { 'key': 'pl', 'label': 'Polski' },
            { 'key': 'sv', 'label': 'Svenska' },
            { 'key': 'ar', 'label': 'العربية', direction: 'rtl' }
        ]);
        this.languages$ = this.languages.asObservable();
        const customLanguages = appConfigService.get(AppConfigValues.APP_CONFIG_LANGUAGES_KEY);
        this.setLanguages(customLanguages);
    }
    changeLanguage(language) {
        this.userPreferencesService.locale = language.key;
        this.userPreferencesService.set('textOrientation', language.direction || 'ltr');
    }
    setLanguages(items) {
        if (items && items.length > 0) {
            this.languages.next(items);
        }
    }
}
LanguageService.ɵfac = function LanguageService_Factory(t) { return new (t || LanguageService)(ɵngcc0.ɵɵinject(ɵngcc1.AppConfigService), ɵngcc0.ɵɵinject(ɵngcc2.UserPreferencesService)); };
LanguageService.ɵprov = i0.ɵɵdefineInjectable({ factory: function LanguageService_Factory() { return new LanguageService(i0.ɵɵinject(i1.AppConfigService), i0.ɵɵinject(i2.UserPreferencesService)); }, token: LanguageService, providedIn: "root" });
LanguageService.ctorParameters = () => [
    { type: AppConfigService },
    { type: UserPreferencesService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LanguageService, [{
        type: Injectable,
        args: [{ providedIn: 'root' }]
    }], function () { return [{ type: ɵngcc1.AppConfigService }, { type: ɵngcc2.UserPreferencesService }]; }, null); })();
