// @alfresco/adf-core/esm2015/services/oauth2.service.js
import { Injectable } from '@angular/core';
import { AlfrescoApiService } from './alfresco-api.service';
import { from } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "./alfresco-api.service";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './alfresco-api.service';
export const JSON_TYPE = ['application/json'];
export class OAuth2Service {
    constructor(alfrescoApiService) {
        this.alfrescoApiService = alfrescoApiService;
    }
    get apiClient() {
        return this.alfrescoApiService.getInstance().oauth2Auth;
    }
    request(opts) {
        return from(this.apiClient.callCustomApi(opts.url, opts.httpMethod, opts.pathParams, opts.queryParams, {}, {}, opts.bodyParam, JSON_TYPE, JSON_TYPE, Object));
    }
    get(opts) {
        return this.request(Object.assign(Object.assign({}, opts), { httpMethod: 'GET' }));
    }
    put(opts) {
        return this.request(Object.assign(Object.assign({}, opts), { httpMethod: 'PUT' }));
    }
    post(opts) {
        return this.request(Object.assign(Object.assign({}, opts), { httpMethod: 'POST' }));
    }
    delete(opts) {
        return this.request(Object.assign(Object.assign({}, opts), { httpMethod: 'DELETE' }));
    }
}
OAuth2Service.ɵfac = function OAuth2Service_Factory(t) { return new (t || OAuth2Service)(ɵngcc0.ɵɵinject(ɵngcc1.AlfrescoApiService)); };
OAuth2Service.ɵprov = i0.ɵɵdefineInjectable({ factory: function OAuth2Service_Factory() { return new OAuth2Service(i0.ɵɵinject(i1.AlfrescoApiService)); }, token: OAuth2Service, providedIn: "root" });
OAuth2Service.ctorParameters = () => [
    { type: AlfrescoApiService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OAuth2Service, [{
        type: Injectable,
        args: [{ providedIn: 'root' }]
    }], function () { return [{ type: ɵngcc1.AlfrescoApiService }]; }, null); })();
