// @alfresco/adf-core/esm2015/services/version-compatibility.service.js
import { Injectable } from '@angular/core';
import { DiscoveryApiService } from './discovery-api.service';
import { filter } from 'rxjs/operators';
import { ReplaySubject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "./discovery-api.service";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './discovery-api.service';
export class VersionCompatibilityService {
    constructor(discoveryApiService) {
        this.discoveryApiService = discoveryApiService;
        this.acsVersionInitialized$ = new ReplaySubject();
        this.discoveryApiService.ecmProductInfo$
            .pipe(filter(acsInfo => !!acsInfo))
            .subscribe((acsInfo) => this.initializeAcsVersion(acsInfo.version));
    }
    initializeAcsVersion(acsVersion) {
        this.acsVersion = acsVersion;
        this.acsVersionInitialized$.next();
    }
    getAcsVersion() {
        return this.acsVersion;
    }
    isVersionSupported(requiredVersion) {
        const parsedRequiredVersion = this.parseVersion(requiredVersion);
        const currentVersion = this.getAcsVersion();
        let versionSupported = false;
        if (currentVersion) {
            if (+currentVersion.major > +parsedRequiredVersion.major) {
                versionSupported = true;
            }
            else if (currentVersion.major === parsedRequiredVersion.major &&
                +currentVersion.minor > +parsedRequiredVersion.minor) {
                versionSupported = true;
            }
            else if (currentVersion.major === parsedRequiredVersion.major &&
                currentVersion.minor === parsedRequiredVersion.minor &&
                +currentVersion.patch >= +parsedRequiredVersion.patch) {
                versionSupported = true;
            }
        }
        return versionSupported;
    }
    parseVersion(version) {
        const major = version.split('.')[0];
        const minor = version.split('.')[1] || '0';
        const patch = version.split('.')[2] || '0';
        return {
            major: major,
            minor: minor,
            patch: patch
        };
    }
}
VersionCompatibilityService.ɵfac = function VersionCompatibilityService_Factory(t) { return new (t || VersionCompatibilityService)(ɵngcc0.ɵɵinject(ɵngcc1.DiscoveryApiService)); };
VersionCompatibilityService.ɵprov = i0.ɵɵdefineInjectable({ factory: function VersionCompatibilityService_Factory() { return new VersionCompatibilityService(i0.ɵɵinject(i1.DiscoveryApiService)); }, token: VersionCompatibilityService, providedIn: "root" });
VersionCompatibilityService.ctorParameters = () => [
    { type: DiscoveryApiService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(VersionCompatibilityService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.DiscoveryApiService }]; }, null); })();
