// @alfresco/adf-core/esm2015/form/components/widgets/dynamic-table/editors/datetime/datetime.editor.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { UserPreferencesService, UserPreferenceValues } from '../../../../../../services/user-preferences.service';
import { MomentDateAdapter } from '../../../../../../utils/moment-date-adapter';
import { MOMENT_DATE_FORMATS } from '../../../../../../utils/moment-date-formats.model';
import { Component, Input } from '@angular/core';
import { DateAdapter, MAT_DATE_FORMATS } from '@angular/material/core';
import moment from 'moment-es6';
import { DynamicTableModel } from './../../dynamic-table.widget.model';
import { DatetimeAdapter, MAT_DATETIME_FORMATS } from '@mat-datetimepicker/core';
import { MomentDatetimeAdapter, MAT_MOMENT_DATETIME_FORMATS } from '@mat-datetimepicker/moment';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/core';
import * as ɵngcc2 from '../../../../../../services/user-preferences.service';
import * as ɵngcc3 from '@angular/material/form-field';
import * as ɵngcc4 from '@angular/material/input';
import * as ɵngcc5 from '@mat-datetimepicker/core';
import * as ɵngcc6 from '@angular/forms';
const ɵ0 = MOMENT_DATE_FORMATS, ɵ1 = MAT_MOMENT_DATETIME_FORMATS;
export class DateTimeEditorComponent {
    constructor(dateAdapter, userPreferencesService) {
        this.dateAdapter = dateAdapter;
        this.userPreferencesService = userPreferencesService;
        this.DATE_TIME_FORMAT = 'DD/MM/YYYY HH:mm';
        this.onDestroy$ = new Subject();
    }
    ngOnInit() {
        this.userPreferencesService
            .select(UserPreferenceValues.Locale)
            .pipe(takeUntil(this.onDestroy$))
            .subscribe(locale => this.dateAdapter.setLocale(locale));
        const momentDateAdapter = this.dateAdapter;
        momentDateAdapter.overrideDisplayFormat = this.DATE_TIME_FORMAT;
        this.value = moment(this.table.getCellValue(this.row, this.column), this.DATE_TIME_FORMAT);
    }
    ngOnDestroy() {
        this.onDestroy$.next(true);
        this.onDestroy$.complete();
    }
    onDateChanged(newDateValue) {
        if (newDateValue && newDateValue.value) {
            const newValue = moment(newDateValue.value, this.DATE_TIME_FORMAT);
            this.row.value[this.column.id] = newDateValue.value.format(this.DATE_TIME_FORMAT);
            this.value = newValue;
            this.table.flushValue();
        }
        else if (newDateValue) {
            const newValue = moment(newDateValue, this.DATE_TIME_FORMAT);
            this.value = newValue;
            this.row.value[this.column.id] = newDateValue;
            this.table.flushValue();
        }
        else {
            this.row.value[this.column.id] = '';
        }
    }
}
DateTimeEditorComponent.ɵfac = function DateTimeEditorComponent_Factory(t) { return new (t || DateTimeEditorComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DateAdapter), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.UserPreferencesService)); };
DateTimeEditorComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DateTimeEditorComponent, selectors: [["adf-datetime-editor"]], inputs: { table: "table", row: "row", column: "column" }, features: [ɵngcc0.ɵɵProvidersFeature([
            { provide: DateAdapter, useClass: MomentDateAdapter },
            { provide: MAT_DATE_FORMATS, useValue: ɵ0 },
            { provide: DatetimeAdapter, useClass: MomentDatetimeAdapter },
            { provide: MAT_DATETIME_FORMATS, useValue: ɵ1 }
        ])], decls: 8, vars: 11, consts: [[1, "adf-date-editor"], ["matInput", "", 3, "matDatetimepicker", "ngModel", "id", "required", "disabled", "ngModelChange", "focusout", "dateChange"], ["matSuffix", "", 1, "adf-date-editor-button", 3, "for"], ["type", "datetime", 3, "openOnFocus", "timeInterval"], ["datetimePicker", ""]], template: function DateTimeEditorComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "mat-form-field", 0);
        ɵngcc0.ɵɵelementStart(2, "label");
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "input", 1);
        ɵngcc0.ɵɵlistener("ngModelChange", function DateTimeEditorComponent_Template_input_ngModelChange_4_listener($event) { return ctx.value = $event; })("focusout", function DateTimeEditorComponent_Template_input_focusout_4_listener($event) { return ctx.onDateChanged($event.srcElement.value); })("dateChange", function DateTimeEditorComponent_Template_input_dateChange_4_listener($event) { return ctx.onDateChanged($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(5, "mat-datetimepicker-toggle", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(6, "mat-datetimepicker", 3, 4);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        const _r0 = ɵngcc0.ɵɵreference(7);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵattribute("for", ctx.column.id);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate2("", ctx.column.name, " ", ctx.DATE_TIME_FORMAT, "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("matDatetimepicker", _r0)("ngModel", ctx.value)("id", ctx.column.id)("required", ctx.column.required)("disabled", !ctx.column.editable);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("for", _r0);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("openOnFocus", true)("timeInterval", 5);
    } }, directives: [ɵngcc3.MatFormField, ɵngcc4.MatInput, ɵngcc5.MatDatetimepickerInput, ɵngcc6.DefaultValueAccessor, ɵngcc6.NgControlStatus, ɵngcc6.NgModel, ɵngcc6.RequiredValidator, ɵngcc5.MatDatetimepickerToggle, ɵngcc3.MatSuffix, ɵngcc5.MatDatetimepicker], styles: [".adf-date-editor[_ngcontent-%COMP%]{width:100%}.adf-date-editor-button[_ngcontent-%COMP%]{position:relative;top:25px}"] });
DateTimeEditorComponent.ctorParameters = () => [
    { type: DateAdapter },
    { type: UserPreferencesService }
];
DateTimeEditorComponent.propDecorators = {
    table: [{ type: Input }],
    row: [{ type: Input }],
    column: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DateTimeEditorComponent, [{
        type: Component,
        args: [{
                selector: 'adf-datetime-editor',
                template: "<div>\n    <mat-form-field class=\"adf-date-editor\">\n        <label [attr.for]=\"column.id\">{{column.name}} {{DATE_TIME_FORMAT}}</label>\n        <input matInput\n            [matDatetimepicker]=\"datetimePicker\"\n            [(ngModel)]=\"value\"\n            [id]=\"column.id\"\n            [required]=\"column.required\"\n            [disabled]=\"!column.editable\"\n            (focusout)=\"onDateChanged($any($event).srcElement.value)\"\n            (dateChange)=\"onDateChanged($event)\">\n            <mat-datetimepicker-toggle\n                matSuffix\n                [for]=\"datetimePicker\"\n                class=\"adf-date-editor-button\">\n            </mat-datetimepicker-toggle>\n    </mat-form-field>\n    <mat-datetimepicker\n        #datetimePicker\n        type=\"datetime\"\n        [openOnFocus]=\"true\"\n        [timeInterval]=\"5\">\n    </mat-datetimepicker>\n</div>\n",
                providers: [
                    { provide: DateAdapter, useClass: MomentDateAdapter },
                    { provide: MAT_DATE_FORMATS, useValue: ɵ0 },
                    { provide: DatetimeAdapter, useClass: MomentDatetimeAdapter },
                    { provide: MAT_DATETIME_FORMATS, useValue: ɵ1 }
                ],
                styles: [".adf-date-editor{width:100%}.adf-date-editor-button{position:relative;top:25px}"]
            }]
    }], function () { return [{ type: ɵngcc1.DateAdapter }, { type: ɵngcc2.UserPreferencesService }]; }, { table: [{
            type: Input
        }], row: [{
            type: Input
        }], column: [{
            type: Input
        }] }); })();
export { ɵ0, ɵ1 };
