// @alfresco/adf-content-services/esm2015/lib/permission-manager/components/add-permission/add-permission-dialog.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Inject, ViewEncapsulation } from '@angular/core';
import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
import { MemberModel } from '../../models/member.model';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/dialog';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/material/button';
import * as ɵngcc4 from '@angular/material/icon';
import * as ɵngcc5 from '@alfresco/adf-core';
import * as ɵngcc6 from '../user-icon-column/user-icon-column.component';
import * as ɵngcc7 from '../user-name-column/user-name-column.component';
import * as ɵngcc8 from '../user-role-column/user-role-column.component';
import * as ɵngcc9 from './add-permission-panel.component';
import * as ɵngcc10 from '@ngx-translate/core';

function AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-user-icon-column", 13);
} if (rf & 2) {
    const context_r8 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("context", context_r8);
} }
function AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-user-name-column", 13);
} if (rf & 2) {
    const context_r9 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("context", context_r9);
} }
function AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_8_Template(rf, ctx) { if (rf & 1) {
    const _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "adf-user-role-column", 14);
    ɵngcc0.ɵɵlistener("roleChanged", function AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_8_Template_adf_user_role_column_roleChanged_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r12); const entry_r10 = ctx.$implicit; const ctx_r11 = ɵngcc0.ɵɵnextContext(3); return ctx_r11.onMemberUpdate($event, entry_r10.row.obj); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const entry_r10 = ctx.$implicit;
    const ctx_r5 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("readonly", entry_r10.row.obj.readonly)("value", entry_r10.data.getValue(entry_r10.row, entry_r10.col))("roles", ctx_r5.data.roles);
} }
function AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_10_Template(rf, ctx) { if (rf & 1) {
    const _r14 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "adf-user-role-column", 15);
    ɵngcc0.ɵɵlistener("roleChanged", function AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_10_Template_adf_user_role_column_roleChanged_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r14); const ctx_r13 = ɵngcc0.ɵɵnextContext(3); return ctx_r13.onBulkUpdate($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r6 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("roles", ctx_r6.data.roles);
} }
function AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_12_Template(rf, ctx) { if (rf & 1) {
    const _r17 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 16);
    ɵngcc0.ɵɵlistener("click", function AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_12_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r17); const entry_r15 = ctx.$implicit; const ctx_r16 = ɵngcc0.ɵɵnextContext(3); return ctx_r16.onMemberDelete(entry_r15.row.obj); });
    ɵngcc0.ɵɵelementStart(1, "mat-icon");
    ɵngcc0.ɵɵtext(2, "highlight_off");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const entry_r15 = ctx.$implicit;
    ɵngcc0.ɵɵstyleProp("display", entry_r15.row.obj.readonly ? "none" : "block");
} }
const _c0 = function (a0) { return { count: a0 }; };
function AddPermissionDialogComponent_ng_container_3_adf_datatable_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "adf-datatable", 8);
    ɵngcc0.ɵɵelementStart(1, "data-columns");
    ɵngcc0.ɵɵelementStart(2, "data-column", 9);
    ɵngcc0.ɵɵtemplate(3, AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_3_Template, 1, 1, "ng-template");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "data-column", 10);
    ɵngcc0.ɵɵpipe(5, "translate");
    ɵngcc0.ɵɵtemplate(6, AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_6_Template, 1, 1, "ng-template");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(7, "data-column", 11);
    ɵngcc0.ɵɵtemplate(8, AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_8_Template, 1, 3, "ng-template");
    ɵngcc0.ɵɵelementStart(9, "adf-data-column-header");
    ɵngcc0.ɵɵtemplate(10, AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_10_Template, 1, 1, "ng-template");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(11, "data-column", 12);
    ɵngcc0.ɵɵtemplate(12, AddPermissionDialogComponent_ng_container_3_adf_datatable_9_ng_template_12_Template, 3, 2, "ng-template");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("rows", ctx_r2.selectedMembers)("stickyHeader", true);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("sortable", false);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("title", ɵngcc0.ɵɵpipeBind2(5, 5, "PERMISSION_MANAGER.COLUMN.NAME", ɵngcc0.ɵɵpureFunction1(8, _c0, ctx_r2.selectedMembers.length)));
    ɵngcc0.ɵɵadvance(7);
    ɵngcc0.ɵɵproperty("sortable", false);
} }
function AddPermissionDialogComponent_ng_container_3_Template(rf, ctx) { if (rf & 1) {
    const _r19 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "mat-dialog-content");
    ɵngcc0.ɵɵelementStart(2, "button", 2);
    ɵngcc0.ɵɵlistener("click", function AddPermissionDialogComponent_ng_container_3_Template_button_click_2_listener() { ɵngcc0.ɵɵrestoreView(_r19); const ctx_r18 = ɵngcc0.ɵɵnextContext(); return ctx_r18.enableSearch(); });
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵpipe(4, "translate");
    ɵngcc0.ɵɵelement(5, "span", 3);
    ɵngcc0.ɵɵelementStart(6, "mat-icon");
    ɵngcc0.ɵɵtext(7, "search");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(8, "div", 4);
    ɵngcc0.ɵɵtemplate(9, AddPermissionDialogComponent_ng_container_3_adf_datatable_9_Template, 13, 10, "adf-datatable", 5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(10, "mat-dialog-actions");
    ɵngcc0.ɵɵelementStart(11, "button", 6);
    ɵngcc0.ɵɵtext(12);
    ɵngcc0.ɵɵpipe(13, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(14, "button", 7);
    ɵngcc0.ɵɵlistener("click", function AddPermissionDialogComponent_ng_container_3_Template_button_click_14_listener() { ɵngcc0.ɵɵrestoreView(_r19); const ctx_r20 = ɵngcc0.ɵɵnextContext(); return ctx_r20.onAddClicked(); });
    ɵngcc0.ɵɵtext(15);
    ɵngcc0.ɵɵpipe(16, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(4, 6, "PERMISSION_MANAGER.ADD-PERMISSION.SEARCH"), " ");
    ɵngcc0.ɵɵadvance(6);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.selectedMembers.length);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(13, 8, "PERMISSION_MANAGER.ADD-PERMISSION.CLOSE-ACTION"), " ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("mat-dialog-close", true)("disabled", !ctx_r0.isValid());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(16, 10, "PERMISSION_MANAGER.ADD-PERMISSION.ADD-ACTION"), " ");
} }
function AddPermissionDialogComponent_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    const _r22 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "mat-dialog-content");
    ɵngcc0.ɵɵelementStart(2, "adf-add-permission-panel", 17);
    ɵngcc0.ɵɵlistener("select", function AddPermissionDialogComponent_ng_container_4_Template_adf_add_permission_panel_select_2_listener($event) { ɵngcc0.ɵɵrestoreView(_r22); const ctx_r21 = ɵngcc0.ɵɵnextContext(); return ctx_r21.onSelect($event); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "mat-dialog-actions");
    ɵngcc0.ɵɵelementStart(4, "button", 18);
    ɵngcc0.ɵɵlistener("click", function AddPermissionDialogComponent_ng_container_4_Template_button_click_4_listener() { ɵngcc0.ɵɵrestoreView(_r22); const ctx_r23 = ɵngcc0.ɵɵnextContext(); return ctx_r23.canCloseDialog(); });
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵpipe(6, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(7, "button", 19);
    ɵngcc0.ɵɵlistener("click", function AddPermissionDialogComponent_ng_container_4_Template_button_click_7_listener() { ɵngcc0.ɵɵrestoreView(_r22); const ctx_r24 = ɵngcc0.ɵɵnextContext(); return ctx_r24.onSearchAddClicked(); });
    ɵngcc0.ɵɵtext(8);
    ɵngcc0.ɵɵpipe(9, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(6, 3, "PERMISSION_MANAGER.ADD-PERMISSION.CLOSE-ACTION"), " ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("disabled", !ctx_r1.currentSelection.length);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(9, 5, "PERMISSION_MANAGER.ADD-PERMISSION.SELECT-ACTION"), " ");
} }
export class AddPermissionDialogComponent {
    constructor(data, dialogRef) {
        this.data = data;
        this.dialogRef = dialogRef;
        this.isSearchActive = true;
        this.selectedMembers = [];
        this.existingMembers = [];
        this.currentSelection = [];
        this.existingMembers = this.data.node.permissions.locallySet || [];
    }
    onSelect(items) {
        this.currentSelection = items;
    }
    onAddClicked() {
        const selection = this.selectedMembers.filter(member => !member.readonly).map(member => member.toPermissionElement());
        this.data.confirm.next(selection);
        this.data.confirm.complete();
    }
    onSearchAddClicked() {
        const newMembers = this.currentSelection.map(item => MemberModel.parseFromSearchResult(item))
            .filter(({ id }) => !this.selectedMembers.find((member) => member.id === id));
        this.selectedMembers = this.selectedMembers.concat(newMembers);
        this.selectedMembers.forEach((member) => {
            const existingMember = this.existingMembers.find(({ authorityId }) => authorityId === member.id);
            if (!!existingMember) {
                member.role = existingMember.name;
                member.accessStatus = existingMember.accessStatus;
                member.readonly = true;
            }
        });
        this.disableSearch();
    }
    canCloseDialog() {
        if (!!this.selectedMembers.length) {
            this.disableSearch();
        }
        else {
            this.dialogRef.close();
        }
    }
    enableSearch() {
        this.isSearchActive = true;
    }
    disableSearch() {
        this.isSearchActive = false;
    }
    onBulkUpdate(role) {
        this.selectedMembers.filter(member => !member.readonly)
            .forEach(member => (member.role = role));
    }
    onMemberDelete({ id }) {
        const index = this.selectedMembers.findIndex((member) => member.id === id);
        this.selectedMembers.splice(index, 1);
        if (this.selectedMembers.length === 0) {
            this.enableSearch();
            this.currentSelection = [];
        }
    }
    onMemberUpdate(role, member) {
        const _member = this.selectedMembers.find(({ id }) => id === member.id);
        _member.role = role;
    }
    isValid() {
        return this.selectedMembers.filter(({ readonly }) => !readonly).length && this.selectedMembers.every(({ role }) => !!role);
    }
}
AddPermissionDialogComponent.ɵfac = function AddPermissionDialogComponent_Factory(t) { return new (t || AddPermissionDialogComponent)(ɵngcc0.ɵɵdirectiveInject(MAT_DIALOG_DATA), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.MatDialogRef)); };
AddPermissionDialogComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AddPermissionDialogComponent, selectors: [["adf-add-permission-dialog"]], decls: 5, vars: 5, consts: [["mat-dialog-title", "", "id", "add-permission-dialog-title"], [4, "ngIf"], ["mat-button", "", 1, "adf-search-user-button", 3, "click"], [1, "adf-toolbar--spacer"], [1, "adf-new-permission-table"], ["class", "adf-datatable-permission", "selectionMode", "none", "data-automation-id", "adf-user-role-selection-table", 3, "rows", "stickyHeader", 4, "ngIf"], ["mat-button", "", "mat-dialog-close", "", "data-automation-id", "add-permission-dialog-close-button"], ["mat-button", "", "data-automation-id", "add-permission-dialog-confirm-button", 1, "adf-choose-action", 3, "mat-dialog-close", "disabled", "click"], ["selectionMode", "none", "data-automation-id", "adf-user-role-selection-table", 1, "adf-datatable-permission", 3, "rows", "stickyHeader"], ["key", "$thumbunail", 1, "adf-datatable-cell--image", "adf-authority-icon-column", 3, "sortable"], ["key", "id", 1, "adf-ellipsis-cell", "adf-expand-cell-5", "adf-authorityId-column", 3, "title"], ["title", "PERMISSION_MANAGER.PERMISSION_DISPLAY.ROLE", "key", "role", 1, "adf-ellipsis-cell", "adf-expand-cell-4"], ["key", "", 1, "adf-datatable-cell", "adf-delete-permission-column", 3, "sortable"], [3, "context"], ["id", "adf-select-role-permission", 3, "readonly", "value", "roles", "roleChanged"], ["placeholder", "PERMISSION_MANAGER.COLUMN.BULK-ROLE", "id", "adf-bulk-select-role-permission", 1, "adf-permission-role-column-header", 3, "roles", "roleChanged"], ["mat-icon-button", "", "data-automation-id", "adf-delete-permission-button", 1, "adf-add-member-action", 3, "click"], [1, "adf-search-container", 3, "select"], ["mat-button", "", "data-automation-id", "add-permission-dialog-close-button", 3, "click"], ["mat-button", "", "data-automation-id", "add-permission-dialog-confirm-button", 3, "disabled", "click"]], template: function AddPermissionDialogComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "h2", 0);
        ɵngcc0.ɵɵtext(1);
        ɵngcc0.ɵɵpipe(2, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(3, AddPermissionDialogComponent_ng_container_3_Template, 17, 12, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(4, AddPermissionDialogComponent_ng_container_4_Template, 10, 7, "ng-container", 1);
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 3, (ctx.data == null ? null : ctx.data.title) ? ctx.data == null ? null : ctx.data.title : "PERMISSION_MANAGER.ADD-PERMISSION.BASE-DIALOG-TITLE"), "\n");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.isSearchActive);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isSearchActive);
    } }, directives: [ɵngcc1.MatDialogTitle, ɵngcc2.NgIf, ɵngcc1.MatDialogContent, ɵngcc3.MatButton, ɵngcc4.MatIcon, ɵngcc1.MatDialogActions, ɵngcc1.MatDialogClose, ɵngcc5.DataTableComponent, ɵngcc5.DataColumnListComponent, ɵngcc5.DataColumnComponent, ɵngcc5.DateColumnHeaderComponent, ɵngcc6.UserIconColumnComponent, ɵngcc7.UserNameColumnComponent, ɵngcc8.UserRoleColumnComponent, ɵngcc9.AddPermissionPanelComponent], pipes: [ɵngcc10.TranslatePipe], styles: [".adf-add-permission-dialog .mat-dialog-title{color:var(--theme-text-bold-color);font-size:20px;font-stretch:normal;font-style:normal;font-weight:600;letter-spacing:-.5px;line-height:1.6;margin-left:24px;margin-right:24px}.adf-add-permission-dialog .mat-dialog-container{padding-left:0;padding-right:0}.adf-add-permission-dialog .mat-dialog-content{flex-grow:1;height:80vh;margin:0;overflow:hidden}.adf-add-permission-dialog .mat-dialog-content .adf-new-permission-table{height:90%}.adf-add-permission-dialog .mat-dialog-content .adf-search-container{height:100%}.adf-add-permission-dialog .mat-dialog-actions{background-color:var(--theme-background-color);color:var(--theme-foreground-text-color);display:flex;justify-content:flex-end;padding:0 24px}.adf-add-permission-dialog .mat-dialog-actions button{font-weight:400;text-transform:uppercase}.adf-add-permission-dialog .mat-dialog-actions .adf-choose-action[disabled]{color:var(--theme-secondary-text-color)}.adf-add-permission-dialog .mat-dialog-actions .adf-choose-action:enabled{color:var(--theme-primary-color)}.adf-add-permission-dialog .adf-search-user-button{width:100%}.adf-add-permission-dialog .adf-search-user-button .mat-button-wrapper{align-items:center;display:flex}.adf-add-permission-dialog .adf-add-member-action{padding:0 15px}"], encapsulation: 2 });
AddPermissionDialogComponent.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] }] },
    { type: MatDialogRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AddPermissionDialogComponent, [{
        type: Component,
        args: [{
                selector: 'adf-add-permission-dialog',
                template: "<h2 mat-dialog-title id=\"add-permission-dialog-title\">\n    {{ (data?.title ? data?.title : \"PERMISSION_MANAGER.ADD-PERMISSION.BASE-DIALOG-TITLE\") | translate }}\n</h2>\n\n<ng-container *ngIf=\"!isSearchActive\">\n    <mat-dialog-content>\n        <button mat-button (click)=\"enableSearch()\" class=\"adf-search-user-button\">\n            {{ \"PERMISSION_MANAGER.ADD-PERMISSION.SEARCH\" | translate }}\n            <span class=\"adf-toolbar--spacer\"></span>\n            <mat-icon>search</mat-icon>\n        </button>\n\n        <div class=\"adf-new-permission-table\">\n            <adf-datatable [rows]=\"selectedMembers\"\n                           class=\"adf-datatable-permission\"\n                           selectionMode=\"none\"\n                           [stickyHeader]=\"true\"\n                           data-automation-id=\"adf-user-role-selection-table\"\n                           *ngIf=\"selectedMembers.length\">\n                <data-columns>\n                    <data-column class=\"adf-datatable-cell--image adf-authority-icon-column\" key=\"$thumbunail\" [sortable]=\"false\">\n                        <ng-template let-context>\n                            <adf-user-icon-column [context]=\"context\"></adf-user-icon-column>\n                        </ng-template>\n                    </data-column>\n\n                    <data-column class=\"adf-ellipsis-cell adf-expand-cell-5 adf-authorityId-column\"\n                                 [title]=\"'PERMISSION_MANAGER.COLUMN.NAME' | translate:{count:selectedMembers.length}\"\n                                 key=\"id\">\n                        <ng-template let-context>\n                            <adf-user-name-column [context]=\"context\"></adf-user-name-column>\n                        </ng-template>\n                    </data-column>\n\n                    <data-column class=\"adf-ellipsis-cell adf-expand-cell-4\"\n                                 title=\"PERMISSION_MANAGER.PERMISSION_DISPLAY.ROLE\"\n                                 key=\"role\">\n                        <ng-template let-entry=\"$implicit\">\n                            <adf-user-role-column [readonly]=\"entry.row.obj.readonly\"\n                                                  [value]=\"entry.data.getValue(entry.row, entry.col)\"\n                                                  [roles]=\"data.roles\"\n                                                  id=\"adf-select-role-permission\"\n                                                  (roleChanged)=\"onMemberUpdate($event, entry.row.obj)\">\n                            </adf-user-role-column>\n                        </ng-template>\n\n                        <adf-data-column-header>\n                            <ng-template>\n                                <adf-user-role-column  class=\"adf-permission-role-column-header\"\n                                                       placeholder=\"PERMISSION_MANAGER.COLUMN.BULK-ROLE\"\n                                                       [roles]=\"data.roles\"\n                                                       id=\"adf-bulk-select-role-permission\"\n                                                       (roleChanged)=\"onBulkUpdate($event)\">\n                                </adf-user-role-column>\n                            </ng-template>\n                        </adf-data-column-header>\n                    </data-column>\n\n                    <data-column class=\"adf-datatable-cell adf-delete-permission-column\" key=\"\" [sortable]=\"false\">\n                        <ng-template let-entry=\"$implicit\">\n                            <button mat-icon-button\n                                    class=\"adf-add-member-action\"\n                                    [style.display]=\"entry.row.obj.readonly ? 'none': 'block'\"\n                                    (click)=\"onMemberDelete(entry.row.obj)\"\n                                    data-automation-id=\"adf-delete-permission-button\">\n                                <mat-icon>highlight_off</mat-icon>\n                            </button>\n                        </ng-template>\n                    </data-column>\n                </data-columns>\n            </adf-datatable>\n        </div>\n\n    </mat-dialog-content>\n\n    <mat-dialog-actions>\n        <button mat-button\n                mat-dialog-close\n                data-automation-id=\"add-permission-dialog-close-button\">\n            {{ \"PERMISSION_MANAGER.ADD-PERMISSION.CLOSE-ACTION\" | translate }}\n        </button>\n        <button mat-button\n                data-automation-id=\"add-permission-dialog-confirm-button\"\n                [mat-dialog-close]=\"true\"\n                class=\"adf-choose-action\"\n                [disabled]=\"!isValid()\"\n                (click)=\"onAddClicked()\">\n            {{ \"PERMISSION_MANAGER.ADD-PERMISSION.ADD-ACTION\" | translate }}\n        </button>\n    </mat-dialog-actions>\n</ng-container>\n\n<ng-container *ngIf=\"isSearchActive\">\n    <mat-dialog-content>\n        <adf-add-permission-panel class=\"adf-search-container\" (select)=\"onSelect($event)\"></adf-add-permission-panel>\n    </mat-dialog-content>\n\n    <mat-dialog-actions>\n            <button mat-button\n                    (click)=\"canCloseDialog()\"\n                    data-automation-id=\"add-permission-dialog-close-button\">\n                {{ \"PERMISSION_MANAGER.ADD-PERMISSION.CLOSE-ACTION\" | translate }}\n            </button>\n            <button mat-button\n                    data-automation-id=\"add-permission-dialog-confirm-button\"\n                    [disabled]=\"!currentSelection.length\"\n                    (click)=\"onSearchAddClicked()\">\n                {{ \"PERMISSION_MANAGER.ADD-PERMISSION.SELECT-ACTION\" | translate }}\n            </button>\n    </mat-dialog-actions>\n</ng-container>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-add-permission-dialog .mat-dialog-title{color:var(--theme-text-bold-color);font-size:20px;font-stretch:normal;font-style:normal;font-weight:600;letter-spacing:-.5px;line-height:1.6;margin-left:24px;margin-right:24px}.adf-add-permission-dialog .mat-dialog-container{padding-left:0;padding-right:0}.adf-add-permission-dialog .mat-dialog-content{flex-grow:1;height:80vh;margin:0;overflow:hidden}.adf-add-permission-dialog .mat-dialog-content .adf-new-permission-table{height:90%}.adf-add-permission-dialog .mat-dialog-content .adf-search-container{height:100%}.adf-add-permission-dialog .mat-dialog-actions{background-color:var(--theme-background-color);color:var(--theme-foreground-text-color);display:flex;justify-content:flex-end;padding:0 24px}.adf-add-permission-dialog .mat-dialog-actions button{font-weight:400;text-transform:uppercase}.adf-add-permission-dialog .mat-dialog-actions .adf-choose-action[disabled]{color:var(--theme-secondary-text-color)}.adf-add-permission-dialog .mat-dialog-actions .adf-choose-action:enabled{color:var(--theme-primary-color)}.adf-add-permission-dialog .adf-search-user-button{width:100%}.adf-add-permission-dialog .adf-search-user-button .mat-button-wrapper{align-items:center;display:flex}.adf-add-permission-dialog .adf-add-member-action{padding:0 15px}"]
            }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [MAT_DIALOG_DATA]
            }] }, { type: ɵngcc1.MatDialogRef }]; }, null); })();
