// @alfresco/adf-content-services/esm2015/lib/permission-manager/components/add-permission/search-config-permission.service.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable, Optional, Inject, InjectionToken } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export const SEARCH_QUERY_TOKEN = new InjectionToken('Alfresco Search Query Token');
export class SearchPermissionConfigurationService {
    constructor(queryProvider) {
        this.queryProvider = queryProvider;
    }
    generateQueryBody(searchTerm, maxResults, skipCount) {
        const defaultQueryBody = {
            query: {
                query: this.getQuery(searchTerm)
            },
            include: ['properties', 'aspectNames'],
            paging: {
                maxItems: maxResults,
                skipCount: skipCount
            },
            filterQueries: [
                { query: "TYPE:'cm:authority'" }
            ]
        };
        return defaultQueryBody;
    }
    getQuery(searchTerm) {
        let query;
        if (this.queryProvider && this.queryProvider.query) {
            query = this.queryProvider.query.replace(new RegExp(/\${([^}]+)}/g), searchTerm);
        }
        else {
            query = `(email:*${searchTerm}* OR firstName:*${searchTerm}* OR lastName:*${searchTerm}* OR displayName:*${searchTerm}* OR authorityName:*${searchTerm}* OR authorityDisplayName:*${searchTerm}*) AND ANAME:(\"0/APP.DEFAULT\")`;
        }
        return query;
    }
}
SearchPermissionConfigurationService.ɵfac = function SearchPermissionConfigurationService_Factory(t) { return new (t || SearchPermissionConfigurationService)(ɵngcc0.ɵɵinject(SEARCH_QUERY_TOKEN, 8)); };
SearchPermissionConfigurationService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: SearchPermissionConfigurationService, factory: SearchPermissionConfigurationService.ɵfac });
SearchPermissionConfigurationService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [SEARCH_QUERY_TOKEN,] }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SearchPermissionConfigurationService, [{
        type: Injectable
    }], function () { return [{ type: undefined, decorators: [{
                type: Optional
            }, {
                type: Inject,
                args: [SEARCH_QUERY_TOKEN]
            }] }]; }, null); })();
