// @alfresco/adf-content-services/esm2015/lib/group/services/group.service.js
import { __awaiter } from "tslib";
import { Injectable } from '@angular/core';
import { GroupsApi } from '@alfresco/js-api';
import { AlfrescoApiService } from '@alfresco/adf-core';
import * as i0 from "@angular/core";
import * as i1 from "@alfresco/adf-core";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
export class GroupService {
    constructor(alfrescoApiService) {
        this.alfrescoApiService = alfrescoApiService;
    }
    get groupsApi() {
        var _a;
        this._groupsApi = (_a = this._groupsApi) !== null && _a !== void 0 ? _a : new GroupsApi(this.alfrescoApiService.getInstance());
        return this._groupsApi;
    }
    listAllGroupMembershipsForPerson(personId, opts, accumulator = []) {
        return __awaiter(this, void 0, void 0, function* () {
            const groupsPaginated = yield this.groupsApi.listGroupMembershipsForPerson(personId, opts);
            accumulator = [...accumulator, ...groupsPaginated.list.entries];
            if (groupsPaginated.list.pagination.hasMoreItems) {
                const skip = groupsPaginated.list.pagination.skipCount + groupsPaginated.list.pagination.count;
                return this.listAllGroupMembershipsForPerson(personId, {
                    maxItems: opts.maxItems,
                    skipCount: skip
                }, accumulator);
            }
            else {
                return accumulator;
            }
        });
    }
}
GroupService.ɵfac = function GroupService_Factory(t) { return new (t || GroupService)(ɵngcc0.ɵɵinject(ɵngcc1.AlfrescoApiService)); };
GroupService.ɵprov = i0.ɵɵdefineInjectable({ factory: function GroupService_Factory() { return new GroupService(i0.ɵɵinject(i1.AlfrescoApiService)); }, token: GroupService, providedIn: "root" });
GroupService.ctorParameters = () => [
    { type: AlfrescoApiService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GroupService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.AlfrescoApiService }]; }, null); })();
