// @alfresco/adf-core/esm2015/form/components/widgets/display-text/display-text.widget.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { WidgetComponent } from './../widget.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../../../services/form.service';
import * as ɵngcc2 from '@angular/material/tooltip';
import * as ɵngcc3 from '@ngx-translate/core';
export class DisplayTextWidgetComponent extends WidgetComponent {
    constructor(formService) {
        super(formService);
        this.formService = formService;
    }
}
DisplayTextWidgetComponent.ɵfac = function DisplayTextWidgetComponent_Factory(t) { return new (t || DisplayTextWidgetComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService)); };
DisplayTextWidgetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DisplayTextWidgetComponent, selectors: [["display-text-widget"]], hostBindings: function DisplayTextWidgetComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function DisplayTextWidgetComponent_click_HostBindingHandler($event) { return ctx.event($event); })("blur", function DisplayTextWidgetComponent_blur_HostBindingHandler($event) { return ctx.event($event); })("change", function DisplayTextWidgetComponent_change_HostBindingHandler($event) { return ctx.event($event); })("focus", function DisplayTextWidgetComponent_focus_HostBindingHandler($event) { return ctx.event($event); })("focusin", function DisplayTextWidgetComponent_focusin_HostBindingHandler($event) { return ctx.event($event); })("focusout", function DisplayTextWidgetComponent_focusout_HostBindingHandler($event) { return ctx.event($event); })("input", function DisplayTextWidgetComponent_input_HostBindingHandler($event) { return ctx.event($event); })("invalid", function DisplayTextWidgetComponent_invalid_HostBindingHandler($event) { return ctx.event($event); })("select", function DisplayTextWidgetComponent_select_HostBindingHandler($event) { return ctx.event($event); });
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 3, vars: 7, consts: [["matTooltipPosition", "above", "matTooltipShowDelay", "1000", 3, "matTooltip"]], template: function DisplayTextWidgetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtext(1);
        ɵngcc0.ɵɵpipe(2, "translate");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMapInterpolate1("adf-display-text-widget ", ctx.field.className, "");
        ɵngcc0.ɵɵproperty("matTooltip", ctx.field.tooltip);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 5, ctx.field.value));
    } }, directives: [ɵngcc2.MatTooltip], pipes: [ɵngcc3.TranslatePipe], styles: [".adf-display-text-widget{white-space:pre-wrap}"], encapsulation: 2 });
DisplayTextWidgetComponent.ctorParameters = () => [
    { type: FormService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DisplayTextWidgetComponent, [{
        type: Component,
        args: [{
                selector: 'display-text-widget',
                template: "<div [matTooltip]=\"field.tooltip\" matTooltipPosition=\"above\" matTooltipShowDelay=\"1000\"\n    class=\"adf-display-text-widget {{field.className}}\">{{field.value | translate}}</div>\n",
                host: {
                    '(click)': 'event($event)',
                    '(blur)': 'event($event)',
                    '(change)': 'event($event)',
                    '(focus)': 'event($event)',
                    '(focusin)': 'event($event)',
                    '(focusout)': 'event($event)',
                    '(input)': 'event($event)',
                    '(invalid)': 'event($event)',
                    '(select)': 'event($event)'
                },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-display-text-widget{white-space:pre-wrap}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormService }]; }, null); })();
