// @alfresco/adf-content-services/esm2015/lib/tree-view/services/tree-view.service.js
import { NodesApiService } from '@alfresco/adf-core';
import { Injectable } from '@angular/core';
import { TreeBaseNode } from '../models/tree-view.model';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@alfresco/adf-core";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
export class TreeViewService {
    constructor(nodeApi) {
        this.nodeApi = nodeApi;
    }
    getTreeNodes(nodeId) {
        return this.nodeApi.getNodeChildren(nodeId)
            .pipe(map((nodePage) => {
            return nodePage.list.entries.filter((node) => node.entry.isFolder ? node : null);
        }), map((nodes) => nodes.map((node) => new TreeBaseNode(node))));
    }
}
TreeViewService.ɵfac = function TreeViewService_Factory(t) { return new (t || TreeViewService)(ɵngcc0.ɵɵinject(ɵngcc1.NodesApiService)); };
TreeViewService.ɵprov = i0.ɵɵdefineInjectable({ factory: function TreeViewService_Factory() { return new TreeViewService(i0.ɵɵinject(i1.NodesApiService)); }, token: TreeViewService, providedIn: "root" });
TreeViewService.ctorParameters = () => [
    { type: NodesApiService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TreeViewService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.NodesApiService }]; }, null); })();
