// @alfresco/adf-content-services/esm2015/lib/permission-manager/components/user-name-column/user-name-column.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { NodeEntry } from '@alfresco/js-api';
import { NodePermissionService } from '../../services/node-permission.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/node-permission.service';
import * as ɵngcc2 from '@angular/common';

function UserNameColumnComponent_span_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 3);
    ɵngcc0.ɵɵpipe(1, "async");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind1(1, 2, ctx_r0.subTitleText$));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(3, 4, ctx_r0.subTitleText$), " ");
} }
export class UserNameColumnComponent {
    constructor(nodePermissionService) {
        this.nodePermissionService = nodePermissionService;
        this.displayText$ = new BehaviorSubject('');
        this.subTitleText$ = new BehaviorSubject('');
    }
    ngOnInit() {
        var _a, _b;
        if (this.context != null) {
            const { person, group, authorityId } = (_b = (_a = this.context.row.obj) === null || _a === void 0 ? void 0 : _a.entry) !== null && _b !== void 0 ? _b : this.context.row.obj;
            const permissionGroup = authorityId ? { displayName: authorityId } : null;
            this.updatePerson(person);
            this.updateGroup(group || permissionGroup);
        }
        if (this.node) {
            const { person, group } = this.nodePermissionService.transformNodeToUserPerson(this.node.entry);
            this.updatePerson(person);
            this.updateGroup(group);
        }
    }
    updatePerson(person) {
        var _a, _b, _c;
        if (person) {
            this.displayText$.next(`${(_a = person.firstName) !== null && _a !== void 0 ? _a : ''} ${(_b = person.lastName) !== null && _b !== void 0 ? _b : ''}`);
            this.subTitleText$.next((_c = person.email) !== null && _c !== void 0 ? _c : '');
        }
    }
    updateGroup(group) {
        if (group) {
            this.displayText$.next(group.displayName);
        }
    }
}
UserNameColumnComponent.ɵfac = function UserNameColumnComponent_Factory(t) { return new (t || UserNameColumnComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NodePermissionService)); };
UserNameColumnComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: UserNameColumnComponent, selectors: [["adf-user-name-column"]], hostAttrs: [1, "adf-user-name-column", "adf-datatable-content-cell", "adf-expand-cell-5", "adf-ellipsis-cell"], inputs: { context: "context", node: "node" }, decls: 9, vars: 12, consts: [[1, "adf-ellipsis-cell"], [1, "adf-user-name-column", 3, "title"], ["class", "adf-user-email-column", 3, "title", 4, "ngIf"], [1, "adf-user-email-column", 3, "title"]], template: function UserNameColumnComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵpipe(1, "async");
        ɵngcc0.ɵɵelementStart(2, "span", 1);
        ɵngcc0.ɵɵpipe(3, "async");
        ɵngcc0.ɵɵtext(4);
        ɵngcc0.ɵɵpipe(5, "async");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(6, "br");
        ɵngcc0.ɵɵtemplate(7, UserNameColumnComponent_span_7_Template, 4, 6, "span", 2);
        ɵngcc0.ɵɵpipe(8, "async");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵattribute("data-automation-id", ɵngcc0.ɵɵpipeBind1(1, 4, ctx.displayText$));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind1(3, 6, ctx.displayText$));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(5, 8, ctx.displayText$), "");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(8, 10, ctx.subTitleText$));
    } }, directives: [ɵngcc2.NgIf], pipes: [ɵngcc2.AsyncPipe], styles: [".adf-user-name-column{font-size:14px;font-weight:600}.adf-user-email-column{color:var(--theme-foreground-text-color);font-size:14px;letter-spacing:-.2px;line-height:1.43}"], encapsulation: 2 });
UserNameColumnComponent.ctorParameters = () => [
    { type: NodePermissionService }
];
UserNameColumnComponent.propDecorators = {
    context: [{ type: Input }],
    node: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UserNameColumnComponent, [{
        type: Component,
        args: [{
                selector: 'adf-user-name-column',
                template: `
        <div class="adf-ellipsis-cell" [attr.data-automation-id]="displayText$ | async">
            <span class="adf-user-name-column" title="{{ displayText$ | async }}"> {{ displayText$ | async }}</span>
            <br/>
            <span class="adf-user-email-column" title="{{ subTitleText$ | async }}" *ngIf="subTitleText$ | async">
                {{ subTitleText$ | async }}
            </span>
        </div>
    `,
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-user-name-column adf-datatable-content-cell adf-expand-cell-5 adf-ellipsis-cell' },
                styles: [".adf-user-name-column{font-size:14px;font-weight:600}.adf-user-email-column{color:var(--theme-foreground-text-color);font-size:14px;letter-spacing:-.2px;line-height:1.43}"]
            }]
    }], function () { return [{ type: ɵngcc1.NodePermissionService }]; }, { context: [{
            type: Input
        }], node: [{
            type: Input
        }] }); })();
