// @alfresco/adf-core/esm2015/card-view/components/card-view-boolitem/card-view-boolitem.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input } from '@angular/core';
import { CardViewUpdateService } from '../../services/card-view-update.service';
import { BaseCardView } from '../base-card-view';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/card-view-update.service';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/material/checkbox';
import * as ɵngcc4 from '@ngx-translate/core';

function CardViewBoolItemComponent_ng_container_0_Template(rf, ctx) { if (rf & 1) {
    const _r2 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "div", 1);
    ɵngcc0.ɵɵelementStart(2, "mat-checkbox", 2);
    ɵngcc0.ɵɵlistener("change", function CardViewBoolItemComponent_ng_container_0_Template_mat_checkbox_change_2_listener($event) { ɵngcc0.ɵɵrestoreView(_r2); const ctx_r1 = ɵngcc0.ɵɵnextContext(); return ctx_r1.changed($event); });
    ɵngcc0.ɵɵpipe(3, "translate");
    ɵngcc0.ɵɵelementStart(4, "div", 3);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵpipe(6, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("checked", ctx_r0.property.displayValue)("disabled", !ctx_r0.isEditable());
    ɵngcc0.ɵɵattribute("data-automation-id", "card-boolean-" + ctx_r0.property.key)("title", ɵngcc0.ɵɵpipeBind1(3, 6, "CORE.METADATA.ACTIONS.TOGGLE"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵattribute("data-automation-id", "card-boolean-label-" + ctx_r0.property.key);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(6, 8, ctx_r0.property.label));
} }
export class CardViewBoolItemComponent extends BaseCardView {
    constructor(cardViewUpdateService) {
        super(cardViewUpdateService);
    }
    isEditable() {
        return this.editable && this.property.editable;
    }
    changed(change) {
        this.cardViewUpdateService.update(Object.assign({}, this.property), change.checked);
        this.property.value = change.checked;
    }
}
CardViewBoolItemComponent.ɵfac = function CardViewBoolItemComponent_Factory(t) { return new (t || CardViewBoolItemComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CardViewUpdateService)); };
CardViewBoolItemComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CardViewBoolItemComponent, selectors: [["adf-card-view-boolitem"]], inputs: { editable: "editable" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [[4, "ngIf"], [1, "adf-property-value"], ["color", "primary", 3, "checked", "disabled", "change"], [1, "adf-property-label"]], template: function CardViewBoolItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, CardViewBoolItemComponent_ng_container_0_Template, 7, 10, "ng-container", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", !ctx.property.isEmpty() || ctx.isEditable());
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.MatCheckbox], pipes: [ɵngcc4.TranslatePipe], encapsulation: 2 });
CardViewBoolItemComponent.ctorParameters = () => [
    { type: CardViewUpdateService }
];
CardViewBoolItemComponent.propDecorators = {
    editable: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CardViewBoolItemComponent, [{
        type: Component,
        args: [{
                selector: 'adf-card-view-boolitem',
                template: "<ng-container *ngIf=\"!property.isEmpty() || isEditable()\">\n    <div class=\"adf-property-value\">\n        <mat-checkbox [attr.data-automation-id]=\"'card-boolean-' + property.key\"\n                      [attr.title]=\"'CORE.METADATA.ACTIONS.TOGGLE' | translate\"\n                      [checked]=\"property.displayValue\"\n                      [disabled]=\"!isEditable()\"\n                      color=\"primary\"\n                      (change)=\"changed($event)\">\n            <div [attr.data-automation-id]=\"'card-boolean-label-' + property.key\"\n                 class=\"adf-property-label\">{{ property.label | translate }}</div>\n        </mat-checkbox>\n    </div>\n</ng-container>\n"
            }]
    }], function () { return [{ type: ɵngcc1.CardViewUpdateService }]; }, { editable: [{
            type: Input
        }] }); })();
