// @alfresco/adf-content-services/esm2015/lib/tag/tag-node-list.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { TagService } from './services/tag.service';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './services/tag.service';
import * as ɵngcc2 from '@angular/material/chips';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@angular/material/icon';

function TagNodeListComponent_mat_chip_1_mat_icon_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-icon", 4);
    ɵngcc0.ɵɵtext(1, "cancel ");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const currentEntry_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵpropertyInterpolate1("id", "tag_chips_delete_", currentEntry_r1.entry.tag, "");
} }
function TagNodeListComponent_mat_chip_1_Template(rf, ctx) { if (rf & 1) {
    const _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "mat-chip", 1);
    ɵngcc0.ɵɵlistener("removed", function TagNodeListComponent_mat_chip_1_Template_mat_chip_removed_0_listener() { ɵngcc0.ɵɵrestoreView(_r6); const currentEntry_r1 = ctx.$implicit; const ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.removeTag(currentEntry_r1.entry.id); });
    ɵngcc0.ɵɵelementStart(1, "span", 2);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(3, TagNodeListComponent_mat_chip_1_mat_icon_3_Template, 2, 1, "mat-icon", 3);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const currentEntry_r1 = ctx.$implicit;
    const idx_r2 = ctx.index;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate1("id", "tag_name_", idx_r2, "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(currentEntry_r1.entry.tag);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.showDelete);
} }
export class TagNodeListComponent {
    constructor(tagService) {
        this.tagService = tagService;
        this.showDelete = true;
        this.results = new EventEmitter();
        this.onDestroy$ = new Subject();
    }
    ngOnChanges() {
        this.refreshTag();
    }
    ngOnInit() {
        this.tagService.refresh
            .pipe(takeUntil(this.onDestroy$))
            .subscribe(() => this.refreshTag());
    }
    ngOnDestroy() {
        this.onDestroy$.next(true);
        this.onDestroy$.complete();
    }
    refreshTag() {
        if (this.nodeId) {
            this.tagService.getTagsByNodeId(this.nodeId).subscribe((tagPaging) => {
                this.tagsEntries = tagPaging.list.entries;
                this.results.emit(this.tagsEntries);
            });
        }
    }
    removeTag(tag) {
        this.tagService.removeTag(this.nodeId, tag).subscribe(() => {
            this.refreshTag();
        });
    }
}
TagNodeListComponent.ɵfac = function TagNodeListComponent_Factory(t) { return new (t || TagNodeListComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TagService)); };
TagNodeListComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TagNodeListComponent, selectors: [["adf-tag-node-list"]], hostAttrs: [1, "adf-tag-node-list"], inputs: { showDelete: "showDelete", nodeId: "nodeId" }, outputs: { results: "results" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 2, vars: 1, consts: [["class", "adf-tag-chips", 3, "removed", 4, "ngFor", "ngForOf"], [1, "adf-tag-chips", 3, "removed"], [3, "id"], ["class", "adf-tag-chips-delete-icon", "matChipRemove", "", 3, "id", 4, "ngIf"], ["matChipRemove", "", 1, "adf-tag-chips-delete-icon", 3, "id"]], template: function TagNodeListComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "mat-chip-list");
        ɵngcc0.ɵɵtemplate(1, TagNodeListComponent_mat_chip_1_Template, 4, 3, "mat-chip", 0);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.tagsEntries);
    } }, directives: [ɵngcc2.MatChipList, ɵngcc3.NgForOf, ɵngcc2.MatChip, ɵngcc3.NgIf, ɵngcc4.MatIcon, ɵngcc2.MatChipRemove], styles: [".adf-tag-node-list .adf-tag-chips{background-color:var(--theme-primary-color);color:var(--theme-primary-color-default-contrast)}.adf-tag-node-list .adf-tag-chips-delete{background:none;border:0;cursor:pointer;float:right;height:17px;margin:-1px 0 0 10px;overflow:visible;padding:0;width:20px}.adf-tag-node-list .adf-tag-chips-delete-icon{background-repeat:no-repeat;color:var(--theme-primary-color-default-contrast)!important;display:inline-block;fill:currentColor;font-size:20px;height:20px;width:20px}"], encapsulation: 2 });
TagNodeListComponent.ctorParameters = () => [
    { type: TagService }
];
TagNodeListComponent.propDecorators = {
    nodeId: [{ type: Input }],
    showDelete: [{ type: Input }],
    results: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TagNodeListComponent, [{
        type: Component,
        args: [{
                selector: 'adf-tag-node-list',
                template: "<mat-chip-list>\n    <mat-chip class=\"adf-tag-chips\"\n              *ngFor=\"let currentEntry of tagsEntries; let idx = index\" (removed)=\"removeTag(currentEntry.entry.id)\">\n        <span id=\"tag_name_{{idx}}\">{{currentEntry.entry.tag}}</span>\n        <mat-icon *ngIf=\"showDelete\" id=\"tag_chips_delete_{{currentEntry.entry.tag}}\"\n                  class=\"adf-tag-chips-delete-icon\" matChipRemove>cancel\n        </mat-icon>\n    </mat-chip>\n</mat-chip-list>\n",
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-tag-node-list' },
                styles: [".adf-tag-node-list .adf-tag-chips{background-color:var(--theme-primary-color);color:var(--theme-primary-color-default-contrast)}.adf-tag-node-list .adf-tag-chips-delete{background:none;border:0;cursor:pointer;float:right;height:17px;margin:-1px 0 0 10px;overflow:visible;padding:0;width:20px}.adf-tag-node-list .adf-tag-chips-delete-icon{background-repeat:no-repeat;color:var(--theme-primary-color-default-contrast)!important;display:inline-block;fill:currentColor;font-size:20px;height:20px;width:20px}"]
            }]
    }], function () { return [{ type: ɵngcc1.TagService }]; }, { showDelete: [{
            type: Input
        }], results: [{
            type: Output
        }], nodeId: [{
            type: Input
        }] }); })();
