// @alfresco/adf-core/esm2015/form/components/widgets/error/error.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { animate, state, style, transition, trigger } from '@angular/animations';
import { Component, Input, ViewEncapsulation } from '@angular/core';
import { ErrorMessageModel } from '../core/index';
import { FormService } from './../../../services/form.service';
import { WidgetComponent } from './../widget.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../../../services/form.service';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/material/icon';
import * as ɵngcc4 from '@ngx-translate/core';

function ErrorWidgetComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelementStart(1, "div", 2);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "mat-icon", 3);
    ɵngcc0.ɵɵtext(5, "warning");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("@transitionMessages", ctx_r0._subscriptAnimationState);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind2(3, 2, ctx_r0.error.message, ctx_r0.translateParameters));
} }
function ErrorWidgetComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelementStart(1, "div", 2);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("@transitionMessages", ctx_r1._subscriptAnimationState);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.required);
} }
export class ErrorWidgetComponent extends WidgetComponent {
    constructor(formService) {
        super(formService);
        this.formService = formService;
        this.translateParameters = null;
        this._subscriptAnimationState = '';
    }
    ngOnChanges(changes) {
        if (changes['required']) {
            this.required = changes.required.currentValue;
            this._subscriptAnimationState = 'enter';
        }
        if (changes['error'] && changes['error'].currentValue) {
            if (changes.error.currentValue.isActive()) {
                this.error = changes.error.currentValue;
                this.translateParameters = this.error.getAttributesAsJsonObj();
                this._subscriptAnimationState = 'enter';
            }
        }
    }
}
ErrorWidgetComponent.ɵfac = function ErrorWidgetComponent_Factory(t) { return new (t || ErrorWidgetComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService)); };
ErrorWidgetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ErrorWidgetComponent, selectors: [["error-widget"]], hostBindings: function ErrorWidgetComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function ErrorWidgetComponent_click_HostBindingHandler($event) { return ctx.event($event); })("blur", function ErrorWidgetComponent_blur_HostBindingHandler($event) { return ctx.event($event); })("change", function ErrorWidgetComponent_change_HostBindingHandler($event) { return ctx.event($event); })("focus", function ErrorWidgetComponent_focus_HostBindingHandler($event) { return ctx.event($event); })("focusin", function ErrorWidgetComponent_focusin_HostBindingHandler($event) { return ctx.event($event); })("focusout", function ErrorWidgetComponent_focusout_HostBindingHandler($event) { return ctx.event($event); })("input", function ErrorWidgetComponent_input_HostBindingHandler($event) { return ctx.event($event); })("invalid", function ErrorWidgetComponent_invalid_HostBindingHandler($event) { return ctx.event($event); })("select", function ErrorWidgetComponent_select_HostBindingHandler($event) { return ctx.event($event); });
    } }, inputs: { required: "required", error: "error" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature, ɵngcc0.ɵɵNgOnChangesFeature], decls: 3, vars: 2, consts: [[1, "adf-error-text-container"], [4, "ngIf"], [1, "adf-error-text"], [1, "adf-error-icon"]], template: function ErrorWidgetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, ErrorWidgetComponent_div_1_Template, 6, 5, "div", 1);
        ɵngcc0.ɵɵtemplate(2, ErrorWidgetComponent_div_2_Template, 3, 2, "div", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.error == null ? null : ctx.error.isActive());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.required);
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.MatIcon], pipes: [ɵngcc4.TranslatePipe], styles: [".adf-error-text{width:85%}"], encapsulation: 2, data: { animation: [
            trigger('transitionMessages', [
                state('enter', style({ opacity: 1, transform: 'translateY(0%)' })),
                transition('void => enter', [
                    style({ opacity: 0, transform: 'translateY(-100%)' }),
                    animate('300ms cubic-bezier(0.55, 0, 0.55, 0.2)')
                ])
            ])
        ] } });
ErrorWidgetComponent.ctorParameters = () => [
    { type: FormService }
];
ErrorWidgetComponent.propDecorators = {
    error: [{ type: Input }],
    required: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ErrorWidgetComponent, [{
        type: Component,
        args: [{
                selector: 'error-widget',
                template: "<div class=\"adf-error-text-container\">\n    <div *ngIf=\"error?.isActive()\" [@transitionMessages]=\"_subscriptAnimationState\">\n        <div class=\"adf-error-text\">{{error.message | translate:translateParameters}}</div>\n        <mat-icon class=\"adf-error-icon\">warning</mat-icon>\n    </div>\n    <div *ngIf=\"required\" [@transitionMessages]=\"_subscriptAnimationState\">\n            <div class=\"adf-error-text\">{{required}}</div>\n    </div>\n</div>\n",
                animations: [
                    trigger('transitionMessages', [
                        state('enter', style({ opacity: 1, transform: 'translateY(0%)' })),
                        transition('void => enter', [
                            style({ opacity: 0, transform: 'translateY(-100%)' }),
                            animate('300ms cubic-bezier(0.55, 0, 0.55, 0.2)')
                        ])
                    ])
                ],
                host: {
                    '(click)': 'event($event)',
                    '(blur)': 'event($event)',
                    '(change)': 'event($event)',
                    '(focus)': 'event($event)',
                    '(focusin)': 'event($event)',
                    '(focusout)': 'event($event)',
                    '(input)': 'event($event)',
                    '(invalid)': 'event($event)',
                    '(select)': 'event($event)'
                },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-error-text{width:85%}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormService }]; }, { required: [{
            type: Input
        }], error: [{
            type: Input
        }] }); })();
