// apollo-angular/esm2015/testing/module.js
import { Apollo } from 'apollo-angular';
import { ApolloLink, InMemoryCache, ApolloCache, } from '@apollo/client/core';
import { NgModule, InjectionToken, Inject, Optional } from '@angular/core';
import { ApolloTestingController } from './controller';
import { ApolloTestingBackend } from './backend';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from 'apollo-angular';
import * as ɵngcc2 from './backend';
import * as ɵngcc3 from '@apollo/client/core';
export const APOLLO_TESTING_CACHE = new InjectionToken('apollo-angular/testing cache');
export const APOLLO_TESTING_NAMED_CACHE = new InjectionToken('apollo-angular/testing named cache');
export const APOLLO_TESTING_CLIENTS = new InjectionToken('apollo-angular/testing named clients');
function addClient(name, op) {
    op.clientName = name;
    return op;
}
export class ApolloTestingModuleCore {
    constructor(apollo, backend, namedClients, cache, namedCaches) {
        function createOptions(name, c) {
            return {
                link: new ApolloLink((operation) => backend.handle(addClient(name, operation))),
                cache: c ||
                    new InMemoryCache({
                        addTypename: false,
                    }),
            };
        }
        apollo.create(createOptions('default', cache));
        if (namedClients && namedClients.length) {
            namedClients.forEach((name) => {
                const caches = namedCaches && typeof namedCaches === 'object' ? namedCaches : {};
                apollo.createNamed(name, createOptions(name, caches[name]));
            });
        }
    }
}
ApolloTestingModuleCore.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ApolloTestingModuleCore });
ApolloTestingModuleCore.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ApolloTestingModuleCore_Factory(t) { return new (t || ApolloTestingModuleCore)(ɵngcc0.ɵɵinject(ɵngcc1.Apollo), ɵngcc0.ɵɵinject(ɵngcc2.ApolloTestingBackend), ɵngcc0.ɵɵinject(APOLLO_TESTING_CLIENTS, 8), ɵngcc0.ɵɵinject(APOLLO_TESTING_CACHE, 8), ɵngcc0.ɵɵinject(APOLLO_TESTING_NAMED_CACHE, 8)); }, providers: [
        ApolloTestingBackend,
        { provide: ApolloTestingController, useExisting: ApolloTestingBackend },
    ] });
ApolloTestingModuleCore.ctorParameters = () => [
    { type: Apollo },
    { type: ApolloTestingBackend },
    { type: Array, decorators: [{ type: Optional }, { type: Inject, args: [APOLLO_TESTING_CLIENTS,] }] },
    { type: ApolloCache, decorators: [{ type: Optional }, { type: Inject, args: [APOLLO_TESTING_CACHE,] }] },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [APOLLO_TESTING_NAMED_CACHE,] }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ApolloTestingModuleCore, [{
        type: NgModule,
        args: [{
                providers: [
                    ApolloTestingBackend,
                    { provide: ApolloTestingController, useExisting: ApolloTestingBackend },
                ]
            }]
    }], function () { return [{ type: ɵngcc1.Apollo }, { type: ɵngcc2.ApolloTestingBackend }, { type: Array, decorators: [{
                type: Optional
            }, {
                type: Inject,
                args: [APOLLO_TESTING_CLIENTS]
            }] }, { type: ɵngcc3.ApolloCache, decorators: [{
                type: Optional
            }, {
                type: Inject,
                args: [APOLLO_TESTING_CACHE]
            }] }, { type: undefined, decorators: [{
                type: Optional
            }, {
                type: Inject,
                args: [APOLLO_TESTING_NAMED_CACHE]
            }] }]; }, null); })();
export class ApolloTestingModule {
    static withClients(names) {
        return {
            ngModule: ApolloTestingModuleCore,
            providers: [
                {
                    provide: APOLLO_TESTING_CLIENTS,
                    useValue: names,
                },
            ],
        };
    }
}
ApolloTestingModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ApolloTestingModule });
ApolloTestingModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ApolloTestingModule_Factory(t) { return new (t || ApolloTestingModule)(); }, imports: [[ApolloTestingModuleCore]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ApolloTestingModule, { imports: [ApolloTestingModuleCore] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ApolloTestingModule, [{
        type: NgModule,
        args: [{
                imports: [ApolloTestingModuleCore]
            }]
    }], null, null); })();
