// @alfresco/adf-content-services/esm2015/lib/content-node-selector/name-location-cell/name-location-cell.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Input, ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class NameLocationCellComponent {
    constructor() {
        this.name = '';
        this.path = '';
    }
    ngOnInit() {
        if (this.row) {
            this.name = this.row.getValue('name');
            const fullPath = this.row.getValue('path');
            if (fullPath) {
                this.path = fullPath.name || '';
            }
        }
    }
}
NameLocationCellComponent.ɵfac = function NameLocationCellComponent_Factory(t) { return new (t || NameLocationCellComponent)(); };
NameLocationCellComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NameLocationCellComponent, selectors: [["adf-name-location-cell"]], hostAttrs: [1, "adf-name-location-cell", "adf-datatable-content-cell"], inputs: { row: "row" }, decls: 4, vars: 4, consts: [[1, "adf-name-location-cell-name", "adf-datatable-cell-value", 3, "title"], [1, "adf-name-location-cell-location", "adf-datatable-cell-value", 3, "title"]], template: function NameLocationCellComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtext(1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(2, "div", 1);
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("title", ctx.name);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ctx.name);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("title", ctx.path);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ctx.path);
    } }, styles: [".adf-name-location-cell{display:grid}.adf-name-location-cell-location{color:var(--theme-text-color);font-size:12px}"], encapsulation: 2, changeDetection: 0 });
NameLocationCellComponent.propDecorators = {
    row: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NameLocationCellComponent, [{
        type: Component,
        args: [{
                selector: 'adf-name-location-cell',
                template: `
        <div class="adf-name-location-cell-name adf-datatable-cell-value" [title]="name">{{ name }}</div>
        <div class="adf-name-location-cell-location adf-datatable-cell-value" [title]="path">{{ path }}</div>
    `,
                encapsulation: ViewEncapsulation.None,
                changeDetection: ChangeDetectionStrategy.OnPush,
                host: { class: 'adf-name-location-cell adf-datatable-content-cell' },
                styles: [".adf-name-location-cell{display:grid}.adf-name-location-cell-location{color:var(--theme-text-color);font-size:12px}"]
            }]
    }], function () { return []; }, { row: [{
            type: Input
        }] }); })();
