// @alfresco/adf-process-services/esm2015/lib/people/components/people-search/people-search.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../people-search-field/people-search-field.component';
import * as ɵngcc2 from '@angular/material/button';
import * as ɵngcc3 from '@ngx-translate/core';

const _c0 = [[["", "adf-people-search-title", ""], ["", "people-search-title", ""]], [["", "adf-people-search-action-label", ""], ["", "people-search-action-label", ""]]];
const _c1 = ["[adf-people-search-title], [people-search-title]", "[adf-people-search-action-label], [people-search-action-label]"];
export class PeopleSearchComponent {
    constructor() {
        this.searchPeople = new EventEmitter();
        this.success = new EventEmitter();
        this.closeSearch = new EventEmitter();
        this.selectedUser = {};
    }
    ngOnInit() {
        this.filteredResults$ = this.results
            .pipe(map((users) => {
            return users.filter((user) => user.id !== this.selectedUser.id);
        }));
        this.performSearch = this.performSearchCallback.bind(this);
    }
    performSearchCallback(event) {
        this.searchPeople.emit(event);
        return this.filteredResults$;
    }
    onRowClick(user) {
        this.selectedUser = user;
    }
    closeSearchList() {
        this.closeSearch.emit();
    }
    involveUserAndClose() {
        this.involveUser();
        this.closeSearchList();
    }
    involveUser() {
        if (this.selectedUser === undefined) {
            return;
        }
        this.success.emit(this.selectedUser);
    }
}
PeopleSearchComponent.ɵfac = function PeopleSearchComponent_Factory(t) { return new (t || PeopleSearchComponent)(); };
PeopleSearchComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PeopleSearchComponent, selectors: [["adf-people-search"]], hostAttrs: [1, "adf-people-search"], inputs: { results: "results" }, outputs: { searchPeople: "searchPeople", success: "success", closeSearch: "closeSearch" }, ngContentSelectors: _c1, decls: 9, vars: 4, consts: [[1, "adf-search-text-header"], [3, "performSearch", "rowClick"], [1, "search-list-action-container"], ["mat-button", "", "type", "button", "id", "close-people-search", 3, "click"], ["mat-button", "", "type", "button", "id", "add-people", 3, "click"]], template: function PeopleSearchComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(2, "adf-people-search-field", 1);
        ɵngcc0.ɵɵlistener("rowClick", function PeopleSearchComponent_Template_adf_people_search_field_rowClick_2_listener($event) { return ctx.onRowClick($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelementStart(4, "button", 3);
        ɵngcc0.ɵɵlistener("click", function PeopleSearchComponent_Template_button_click_4_listener() { return ctx.closeSearchList(); });
        ɵngcc0.ɵɵtext(5);
        ɵngcc0.ɵɵpipe(6, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "button", 4);
        ɵngcc0.ɵɵlistener("click", function PeopleSearchComponent_Template_button_click_7_listener() { return ctx.involveUserAndClose(); });
        ɵngcc0.ɵɵprojection(8, 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("performSearch", ctx.performSearch);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(6, 2, "ADF_TASK_LIST.PEOPLE.DIALOG_CLOSE"), " ");
    } }, directives: [ɵngcc1.PeopleSearchFieldComponent, ɵngcc2.MatButton], pipes: [ɵngcc3.TranslatePipe], styles: [".adf-people-search{width:100%}.adf-people-search .adf-activiti-label{font-weight:bolder}.adf-people-search .adf-fix-element-user-list{padding:0}.adf-people-search .adf-search-text-header{font-weight:700;opacity:.54}.adf-people-search .adf-search-list-action-container{border-top:1px solid #eee;margin-top:5px;padding:5px 0;text-align:right}.adf-people-search .adf-search-list-action-container>button{font-weight:bolder;opacity:.54}.adf-people-search .adf-search-list-action-container>button:hover{color:var(--theme-primary-color)}"], encapsulation: 2 });
PeopleSearchComponent.ctorParameters = () => [];
PeopleSearchComponent.propDecorators = {
    results: [{ type: Input }],
    searchPeople: [{ type: Output }],
    success: [{ type: Output }],
    closeSearch: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PeopleSearchComponent, [{
        type: Component,
        args: [{
                selector: 'adf-people-search',
                template: "<div class=\"adf-search-text-header\">\n    <ng-content select=\"[adf-people-search-title], [people-search-title]\"></ng-content>\n</div>\n\n<adf-people-search-field [performSearch]=\"performSearch\" (rowClick)=\"onRowClick($event)\"></adf-people-search-field>\n\n<div class=\"search-list-action-container\">\n    <button mat-button type=\"button\" id=\"close-people-search\" (click)=\"closeSearchList()\">\n        {{'ADF_TASK_LIST.PEOPLE.DIALOG_CLOSE' | translate }}\n    </button>\n    <button mat-button type=\"button\" id=\"add-people\" (click)=\"involveUserAndClose()\">\n        <ng-content select=\"[adf-people-search-action-label], [people-search-action-label]\"></ng-content>\n    </button>\n</div>\n",
                host: {
                    'class': 'adf-people-search'
                },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-people-search{width:100%}.adf-people-search .adf-activiti-label{font-weight:bolder}.adf-people-search .adf-fix-element-user-list{padding:0}.adf-people-search .adf-search-text-header{font-weight:700;opacity:.54}.adf-people-search .adf-search-list-action-container{border-top:1px solid #eee;margin-top:5px;padding:5px 0;text-align:right}.adf-people-search .adf-search-list-action-container>button{font-weight:bolder;opacity:.54}.adf-people-search .adf-search-list-action-container>button:hover{color:var(--theme-primary-color)}"]
            }]
    }], function () { return []; }, { searchPeople: [{
            type: Output
        }], success: [{
            type: Output
        }], closeSearch: [{
            type: Output
        }], results: [{
            type: Input
        }] }); })();
