// @alfresco/adf-process-services-cloud/esm2015/lib/common/process-common.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { CoreModule } from '@alfresco/adf-core';
import { DateRangeFilterComponent } from './date-range-filter/date-range-filter.component';
import { MaterialModule } from '../material.module';
import { CommonModule } from '@angular/common';
import { DateRangeFilterService } from './date-range-filter/date-range-filter.service';
import * as ɵngcc0 from '@angular/core';
export class ProcessCommonModule {
}
ProcessCommonModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ProcessCommonModule });
ProcessCommonModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ProcessCommonModule_Factory(t) { return new (t || ProcessCommonModule)(); }, providers: [
        DateRangeFilterService
    ], imports: [[
            CommonModule,
            CoreModule,
            MaterialModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ProcessCommonModule, { declarations: function () { return [DateRangeFilterComponent]; }, imports: function () { return [CommonModule,
        CoreModule,
        MaterialModule]; }, exports: function () { return [DateRangeFilterComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ProcessCommonModule, [{
        type: NgModule,
        args: [{
                declarations: [DateRangeFilterComponent],
                imports: [
                    CommonModule,
                    CoreModule,
                    MaterialModule
                ],
                exports: [
                    DateRangeFilterComponent
                ],
                providers: [
                    DateRangeFilterService
                ]
            }]
    }], null, null); })();
