// @alfresco/adf-process-services-cloud/esm2015/lib/app/app-list-cloud.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { FlexLayoutModule } from '@angular/flex-layout';
import { MaterialModule } from '../material.module';
import { CommonModule } from '@angular/common';
import { AppDetailsCloudComponent } from './components/app-details-cloud.component';
import { AppListCloudComponent } from './components/app-list-cloud.component';
import { TemplateModule, CoreModule } from '@alfresco/adf-core';
import * as ɵngcc0 from '@angular/core';
export class AppListCloudModule {
}
AppListCloudModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AppListCloudModule });
AppListCloudModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AppListCloudModule_Factory(t) { return new (t || AppListCloudModule)(); }, imports: [[
            CommonModule,
            TemplateModule,
            MaterialModule,
            FlexLayoutModule,
            CoreModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AppListCloudModule, { declarations: function () { return [AppListCloudComponent,
        AppDetailsCloudComponent]; }, imports: function () { return [CommonModule,
        TemplateModule,
        MaterialModule,
        FlexLayoutModule,
        CoreModule]; }, exports: function () { return [AppListCloudComponent,
        AppDetailsCloudComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppListCloudModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    TemplateModule,
                    MaterialModule,
                    FlexLayoutModule,
                    CoreModule
                ],
                declarations: [
                    AppListCloudComponent,
                    AppDetailsCloudComponent
                ],
                exports: [
                    AppListCloudComponent,
                    AppDetailsCloudComponent
                ]
            }]
    }], null, null); })();
