// @alfresco/adf-core/esm2015/sorting-picker/sorting-picker.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, Input, EventEmitter, Output } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/form-field';
import * as ɵngcc2 from '@angular/material/select';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@angular/material/core';
import * as ɵngcc5 from '@angular/material/button';
import * as ɵngcc6 from '@angular/material/icon';
import * as ɵngcc7 from '@ngx-translate/core';

function SortingPickerComponent_mat_option_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-option", 4);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const option_r2 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("value", option_r2.key);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 2, option_r2.label), " ");
} }
function SortingPickerComponent_button_6_mat_icon_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-icon");
    ɵngcc0.ɵɵtext(1, "arrow_upward");
    ɵngcc0.ɵɵelementEnd();
} }
function SortingPickerComponent_button_6_mat_icon_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-icon");
    ɵngcc0.ɵɵtext(1, "arrow_downward");
    ɵngcc0.ɵɵelementEnd();
} }
function SortingPickerComponent_button_6_Template(rf, ctx) { if (rf & 1) {
    const _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 5);
    ɵngcc0.ɵɵlistener("click", function SortingPickerComponent_button_6_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r6); const ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.toggleSortDirection(); });
    ɵngcc0.ɵɵtemplate(1, SortingPickerComponent_button_6_mat_icon_1_Template, 2, 0, "mat-icon", 6);
    ɵngcc0.ɵɵtemplate(2, SortingPickerComponent_button_6_mat_icon_2_Template, 2, 0, "mat-icon", 6);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.ascending);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r1.ascending);
} }
export class SortingPickerComponent {
    constructor() {
        this.options = [];
        this.ascending = true;
        this.valueChange = new EventEmitter();
        this.sortingChange = new EventEmitter();
    }
    onOptionChanged(event) {
        this.selected = event.value;
        this.valueChange.emit(this.selected);
    }
    toggleSortDirection() {
        this.ascending = !this.ascending;
        this.sortingChange.emit(this.ascending);
    }
}
SortingPickerComponent.ɵfac = function SortingPickerComponent_Factory(t) { return new (t || SortingPickerComponent)(); };
SortingPickerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SortingPickerComponent, selectors: [["adf-sorting-picker"]], hostAttrs: [1, "adf-sorting-picker"], inputs: { options: "options", ascending: "ascending", selected: "selected" }, outputs: { valueChange: "valueChange", sortingChange: "sortingChange" }, decls: 7, vars: 6, consts: [["floatLabel", "always"], [3, "value", "valueChange", "selectionChange"], [3, "value", 4, "ngFor", "ngForOf"], ["mat-icon-button", "", "aria-label", "'CORE.SEARCH.TOGGLE_ASC_DESC_ORDER' | translate", 3, "click", 4, "ngIf"], [3, "value"], ["mat-icon-button", "", "aria-label", "'CORE.SEARCH.TOGGLE_ASC_DESC_ORDER' | translate", 3, "click"], [4, "ngIf"]], template: function SortingPickerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "mat-form-field", 0);
        ɵngcc0.ɵɵelementStart(1, "mat-label");
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵpipe(3, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "mat-select", 1);
        ɵngcc0.ɵɵlistener("valueChange", function SortingPickerComponent_Template_mat_select_valueChange_4_listener($event) { return ctx.selected = $event; })("selectionChange", function SortingPickerComponent_Template_mat_select_selectionChange_4_listener($event) { return ctx.onOptionChanged($event); });
        ɵngcc0.ɵɵtemplate(5, SortingPickerComponent_mat_option_5_Template, 3, 4, "mat-option", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(6, SortingPickerComponent_button_6_Template, 3, 2, "button", 3);
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 4, "CORE.SEARCH.SORT_BY"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("value", ctx.selected);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.options);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.selected);
    } }, directives: [ɵngcc1.MatFormField, ɵngcc1.MatLabel, ɵngcc2.MatSelect, ɵngcc3.NgForOf, ɵngcc3.NgIf, ɵngcc4.MatOption, ɵngcc5.MatButton, ɵngcc6.MatIcon], pipes: [ɵngcc7.TranslatePipe], encapsulation: 2 });
SortingPickerComponent.propDecorators = {
    options: [{ type: Input }],
    selected: [{ type: Input }],
    ascending: [{ type: Input }],
    valueChange: [{ type: Output }],
    sortingChange: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SortingPickerComponent, [{
        type: Component,
        args: [{
                selector: 'adf-sorting-picker',
                template: "<mat-form-field floatLabel=\"always\">\n    <mat-label>{{'CORE.SEARCH.SORT_BY' | translate}}</mat-label>\n    <mat-select [(value)]=\"selected\" (selectionChange)=\"onOptionChanged($event)\">\n        <mat-option *ngFor=\"let option of options\" [value]=\"option.key\">\n            {{ option.label | translate }}\n        </mat-option>\n    </mat-select>\n</mat-form-field>\n\n<button *ngIf=\"selected\" mat-icon-button (click)=\"toggleSortDirection()\" aria-label=\"'CORE.SEARCH.TOGGLE_ASC_DESC_ORDER' | translate\">\n    <mat-icon *ngIf=\"ascending\">arrow_upward</mat-icon>\n    <mat-icon *ngIf=\"!ascending\">arrow_downward</mat-icon>\n</button>\n",
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-sorting-picker' }
            }]
    }], function () { return []; }, { options: [{
            type: Input
        }], ascending: [{
            type: Input
        }], valueChange: [{
            type: Output
        }], sortingChange: [{
            type: Output
        }], selected: [{
            type: Input
        }] }); })();
