// @alfresco/adf-core/esm2015/sorting-picker/sorting-picker.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TranslateModule } from '@ngx-translate/core';
import { MaterialModule } from '../material.module';
import { SortingPickerComponent } from './sorting-picker.component';
import * as ɵngcc0 from '@angular/core';
export class SortingPickerModule {
}
SortingPickerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SortingPickerModule });
SortingPickerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SortingPickerModule_Factory(t) { return new (t || SortingPickerModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            TranslateModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SortingPickerModule, { declarations: function () { return [SortingPickerComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        TranslateModule]; }, exports: function () { return [SortingPickerComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SortingPickerModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    TranslateModule
                ],
                declarations: [
                    SortingPickerComponent
                ],
                exports: [
                    SortingPickerComponent
                ]
            }]
    }], null, null); })();
