// @alfresco/adf-content-services/esm2015/lib/content-metadata/content-metadata.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { FlexLayoutModule } from '@angular/flex-layout';
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { CoreModule } from '@alfresco/adf-core';
import { ContentMetadataComponent } from './components/content-metadata/content-metadata.component';
import { ContentMetadataCardComponent } from './components/content-metadata-card/content-metadata-card.component';
import * as ɵngcc0 from '@angular/core';
export class ContentMetadataModule {
}
ContentMetadataModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ContentMetadataModule });
ContentMetadataModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ContentMetadataModule_Factory(t) { return new (t || ContentMetadataModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            FlexLayoutModule,
            CoreModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ContentMetadataModule, { declarations: function () { return [ContentMetadataComponent,
        ContentMetadataCardComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        FlexLayoutModule,
        CoreModule]; }, exports: function () { return [ContentMetadataComponent,
        ContentMetadataCardComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContentMetadataModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    FlexLayoutModule,
                    CoreModule
                ],
                exports: [
                    ContentMetadataComponent,
                    ContentMetadataCardComponent
                ],
                declarations: [
                    ContentMetadataComponent,
                    ContentMetadataCardComponent
                ]
            }]
    }], null, null); })();
