// @alfresco/adf-process-services-cloud/esm2015/lib/task/start-task/start-task-cloud.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FlexLayoutModule } from '@angular/flex-layout';
import { MaterialModule } from '../../material.module';
import { CoreModule } from '@alfresco/adf-core';
import { StartTaskCloudComponent } from './components/start-task-cloud.component';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { GroupCloudModule } from '../../group/group-cloud.module';
import { FormCloudModule } from '../../form/form-cloud.module';
import { PeopleCloudModule } from '../../people/people-cloud.module';
import * as ɵngcc0 from '@angular/core';
export class StartTaskCloudModule {
}
StartTaskCloudModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: StartTaskCloudModule });
StartTaskCloudModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function StartTaskCloudModule_Factory(t) { return new (t || StartTaskCloudModule)(); }, imports: [[
            CommonModule,
            FlexLayoutModule,
            MaterialModule,
            FormsModule,
            ReactiveFormsModule,
            GroupCloudModule,
            CoreModule,
            FormCloudModule,
            PeopleCloudModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(StartTaskCloudModule, { declarations: function () { return [StartTaskCloudComponent]; }, imports: function () { return [CommonModule,
        FlexLayoutModule,
        MaterialModule,
        FormsModule,
        ReactiveFormsModule,
        GroupCloudModule,
        CoreModule,
        FormCloudModule,
        PeopleCloudModule]; }, exports: function () { return [StartTaskCloudComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(StartTaskCloudModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    FlexLayoutModule,
                    MaterialModule,
                    FormsModule,
                    ReactiveFormsModule,
                    GroupCloudModule,
                    CoreModule,
                    FormCloudModule,
                    PeopleCloudModule
                ],
                declarations: [StartTaskCloudComponent],
                exports: [
                    StartTaskCloudComponent
                ]
            }]
    }], null, null); })();
