// @alfresco/adf-content-services/esm2015/lib/content-node-share/content-node-share.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CoreModule } from '@alfresco/adf-core';
import { MaterialModule } from '../material.module';
import { ShareDialogComponent } from './content-node-share.dialog';
import { NodeSharedDirective } from './content-node-share.directive';
import * as ɵngcc0 from '@angular/core';
export class ContentNodeShareModule {
    static forRoot() {
        return {
            ngModule: ContentNodeShareModule
        };
    }
    static forChild() {
        return {
            ngModule: ContentNodeShareModule
        };
    }
}
ContentNodeShareModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ContentNodeShareModule });
ContentNodeShareModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ContentNodeShareModule_Factory(t) { return new (t || ContentNodeShareModule)(); }, imports: [[
            CoreModule,
            CommonModule,
            MaterialModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ContentNodeShareModule, { declarations: function () { return [ShareDialogComponent,
        NodeSharedDirective]; }, imports: function () { return [CoreModule,
        CommonModule,
        MaterialModule]; }, exports: function () { return [ShareDialogComponent,
        NodeSharedDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContentNodeShareModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CoreModule,
                    CommonModule,
                    MaterialModule
                ],
                declarations: [
                    ShareDialogComponent,
                    NodeSharedDirective
                ],
                exports: [
                    ShareDialogComponent,
                    NodeSharedDirective
                ]
            }]
    }], null, null); })();
