// @alfresco/adf-core/esm2015/layout/components/sidebar-action/sidebar-action-menu.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ChangeDetectionStrategy, Component, Directive, Input, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/material/menu';
import * as ɵngcc3 from '@angular/material/button';

function SidebarActionMenuComponent_button_1_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 8);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r3.title);
} }
function SidebarActionMenuComponent_button_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "button", 6);
    ɵngcc0.ɵɵtemplate(1, SidebarActionMenuComponent_button_1_span_1_Template, 2, 1, "span", 7);
    ɵngcc0.ɵɵprojection(2, 1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    const _r2 = ɵngcc0.ɵɵreference(4);
    ɵngcc0.ɵɵproperty("matMenuTriggerFor", _r2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.title);
} }
function SidebarActionMenuComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 9);
    ɵngcc0.ɵɵprojection(1, 2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext();
    const _r2 = ɵngcc0.ɵɵreference(4);
    ɵngcc0.ɵɵproperty("matMenuTriggerFor", _r2);
} }
const _c0 = [[["", "adf-sidebar-menu-options", ""], ["", "sidebar-menu-options", ""]], [["", "adf-sidebar-menu-title-icon", ""], ["", "sidebar-menu-title-icon", ""]], [["", "adf-sidebar-menu-expand-icon", ""], ["", "sidebar-menu-expand-icon", ""]]];
const _c1 = ["[adf-sidebar-menu-options], [sidebar-menu-options]", "[adf-sidebar-menu-title-icon], [sidebar-menu-title-icon]", "[adf-sidebar-menu-expand-icon], [sidebar-menu-expand-icon]"];
export class SidebarActionMenuComponent {
    constructor() {
        this.width = 272;
    }
    isExpanded() {
        return this.expanded;
    }
}
SidebarActionMenuComponent.ɵfac = function SidebarActionMenuComponent_Factory(t) { return new (t || SidebarActionMenuComponent)(); };
SidebarActionMenuComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SidebarActionMenuComponent, selectors: [["adf-sidebar-action-menu"]], hostAttrs: [1, "adf-sidebar-action-menu"], inputs: { width: "width", title: "title", expanded: "expanded" }, ngContentSelectors: _c1, decls: 7, vars: 5, consts: [[1, "adf-sidebar-action-menu"], ["mat-raised-button", "", "class", "adf-sidebar-action-menu-button", "data-automation-id", "create-button", 3, "matMenuTriggerFor", 4, "ngIf"], ["class", "adf-sidebar-action-menu-icon", 3, "matMenuTriggerFor", 4, "ngIf"], ["yPosition", "below", 1, "adf-sidebar-action-menu-panel", 3, "overlapTrigger"], ["adfSidebarMenu", "matMenu"], [1, "adf-sidebar-action-menu-options"], ["mat-raised-button", "", "data-automation-id", "create-button", 1, "adf-sidebar-action-menu-button", 3, "matMenuTriggerFor"], ["class", "adf-sidebar-action-menu-text", 4, "ngIf"], [1, "adf-sidebar-action-menu-text"], [1, "adf-sidebar-action-menu-icon", 3, "matMenuTriggerFor"]], template: function SidebarActionMenuComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, SidebarActionMenuComponent_button_1_Template, 3, 2, "button", 1);
        ɵngcc0.ɵɵtemplate(2, SidebarActionMenuComponent_div_2_Template, 2, 1, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "mat-menu", 3, 4);
        ɵngcc0.ɵɵelementStart(5, "div", 5);
        ɵngcc0.ɵɵprojection(6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isExpanded());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.isExpanded());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("overlapTrigger", false);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("width", ctx.width, "px");
    } }, directives: [ɵngcc1.NgIf, ɵngcc2._MatMenu, ɵngcc3.MatButton, ɵngcc2.MatMenuTrigger], styles: [".adf-sidebar-action-menu .mat-raised-button{background-color:var(--theme-primary-color);border-radius:4px;box-shadow:none!important;color:var(--theme-primary-color-default-contrast)!important;display:block;font-weight:700;height:37.5px;width:100%}.adf-sidebar-action-menu .mat-raised-button mat-icon{color:var(--theme-primary-color-default-contrast)!important;height:25px;width:24px}.adf-sidebar-action-menu-text{height:20px;text-align:left;width:100%}.adf-sidebar-action-menu-icon{color:var(--theme-text-color);cursor:pointer;margin:18px 0 0 20px}.adf-sidebar-action-menu-icon:hover{color:var(--theme-primary-color)}.adf-sidebar-action-menu-options{letter-spacing:-.4px;text-align:left}.adf-sidebar-action-menu-options .mat-menu-item{color:var(--theme-text-bold-color);font-size:14px;letter-spacing:-.4px;line-height:1.5;text-align:left}.adf-sidebar-action-menu-options .mat-menu-item:hover{color:var(--theme-primary-color);opacity:inherit}.adf-sidebar-action-menu-panel{border-radius:2px;box-shadow:0 3px 1px -2px rgba(0,0,0,.2),0 2px 2px 0 rgba(0,0,0,.14),0 1px 5px 0 rgba(0,0,0,.12);margin-top:7.5px}"], encapsulation: 2, changeDetection: 0 });
SidebarActionMenuComponent.propDecorators = {
    title: [{ type: Input }],
    expanded: [{ type: Input }],
    width: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidebarActionMenuComponent, [{
        type: Component,
        args: [{
                selector: 'adf-sidebar-action-menu',
                template: "<div class=\"adf-sidebar-action-menu\">\n    <button *ngIf=\"isExpanded()\" mat-raised-button class=\"adf-sidebar-action-menu-button\" data-automation-id=\"create-button\" [matMenuTriggerFor]=\"adfSidebarMenu\">\n        <span *ngIf=\"title\" class=\"adf-sidebar-action-menu-text\">{{ title }}</span>\n        <ng-content select=\"[adf-sidebar-menu-title-icon], [sidebar-menu-title-icon]\"></ng-content>\n    </button>\n\n    <div *ngIf=\"!isExpanded()\" class=\"adf-sidebar-action-menu-icon\" [matMenuTriggerFor]=\"adfSidebarMenu\">\n        <ng-content select=\"[adf-sidebar-menu-expand-icon], [sidebar-menu-expand-icon]\"></ng-content>\n    </div>\n\n    <mat-menu #adfSidebarMenu=\"matMenu\" class=\"adf-sidebar-action-menu-panel\" [overlapTrigger]=\"false\" yPosition=\"below\">\n        <div class=\"adf-sidebar-action-menu-options\" [style.width.px]=\"width\">\n            <ng-content select=\"[adf-sidebar-menu-options], [sidebar-menu-options]\"></ng-content>\n        </div>\n    </mat-menu>\n</div>\n",
                changeDetection: ChangeDetectionStrategy.OnPush,
                encapsulation: ViewEncapsulation.None,
                host: { 'class': 'adf-sidebar-action-menu' },
                styles: [".adf-sidebar-action-menu .mat-raised-button{background-color:var(--theme-primary-color);border-radius:4px;box-shadow:none!important;color:var(--theme-primary-color-default-contrast)!important;display:block;font-weight:700;height:37.5px;width:100%}.adf-sidebar-action-menu .mat-raised-button mat-icon{color:var(--theme-primary-color-default-contrast)!important;height:25px;width:24px}.adf-sidebar-action-menu-text{height:20px;text-align:left;width:100%}.adf-sidebar-action-menu-icon{color:var(--theme-text-color);cursor:pointer;margin:18px 0 0 20px}.adf-sidebar-action-menu-icon:hover{color:var(--theme-primary-color)}.adf-sidebar-action-menu-options{letter-spacing:-.4px;text-align:left}.adf-sidebar-action-menu-options .mat-menu-item{color:var(--theme-text-bold-color);font-size:14px;letter-spacing:-.4px;line-height:1.5;text-align:left}.adf-sidebar-action-menu-options .mat-menu-item:hover{color:var(--theme-primary-color);opacity:inherit}.adf-sidebar-action-menu-panel{border-radius:2px;box-shadow:0 3px 1px -2px rgba(0,0,0,.2),0 2px 2px 0 rgba(0,0,0,.14),0 1px 5px 0 rgba(0,0,0,.12);margin-top:7.5px}"]
            }]
    }], function () { return []; }, { width: [{
            type: Input
        }], title: [{
            type: Input
        }], expanded: [{
            type: Input
        }] }); })();
export class SidebarMenuDirective {
}
SidebarMenuDirective.ɵfac = function SidebarMenuDirective_Factory(t) { return new (t || SidebarMenuDirective)(); };
SidebarMenuDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SidebarMenuDirective, selectors: [["", "adf-sidebar-menu-options", ""], ["", "sidebar-menu-options", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidebarMenuDirective, [{
        type: Directive,
        args: [{ selector: '[adf-sidebar-menu-options], [sidebar-menu-options]' }]
    }], null, null); })();
export class SidebarMenuTitleIconDirective {
}
SidebarMenuTitleIconDirective.ɵfac = function SidebarMenuTitleIconDirective_Factory(t) { return new (t || SidebarMenuTitleIconDirective)(); };
SidebarMenuTitleIconDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SidebarMenuTitleIconDirective, selectors: [["", "adf-sidebar-menu-title-icon", ""], ["", "sidebar-menu-title-icon", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidebarMenuTitleIconDirective, [{
        type: Directive,
        args: [{ selector: '[adf-sidebar-menu-title-icon], [sidebar-menu-title-icon]' }]
    }], null, null); })();
export class SidebarMenuExpandIconDirective {
}
SidebarMenuExpandIconDirective.ɵfac = function SidebarMenuExpandIconDirective_Factory(t) { return new (t || SidebarMenuExpandIconDirective)(); };
SidebarMenuExpandIconDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SidebarMenuExpandIconDirective, selectors: [["", "adf-sidebar-menu-expand-icon", ""], ["", "sidebar-menu-expand-icon", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidebarMenuExpandIconDirective, [{
        type: Directive,
        args: [{ selector: '[adf-sidebar-menu-expand-icon], [sidebar-menu-expand-icon]' }]
    }], null, null); })();
