// @alfresco/adf-content-services/esm2015/lib/dialogs/dialog.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CoreModule } from '@alfresco/adf-core';
import { MaterialModule } from '../material.module';
import { FolderDialogComponent } from './folder.dialog';
import { NodeLockDialogComponent } from './node-lock.dialog';
import { ConfirmDialogComponent } from './confirm.dialog';
import { MatDatetimepickerModule } from '@mat-datetimepicker/core';
import { MatMomentDatetimeModule } from '@mat-datetimepicker/moment';
import { LibraryDialogComponent } from './library/library.dialog';
import * as ɵngcc0 from '@angular/core';
export class DialogModule {
}
DialogModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DialogModule });
DialogModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DialogModule_Factory(t) { return new (t || DialogModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            CoreModule,
            FormsModule,
            ReactiveFormsModule,
            MatMomentDatetimeModule,
            MatDatetimepickerModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DialogModule, { declarations: function () { return [FolderDialogComponent,
        NodeLockDialogComponent,
        ConfirmDialogComponent,
        LibraryDialogComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        CoreModule,
        FormsModule,
        ReactiveFormsModule,
        MatMomentDatetimeModule,
        MatDatetimepickerModule]; }, exports: function () { return [FolderDialogComponent,
        NodeLockDialogComponent,
        ConfirmDialogComponent,
        LibraryDialogComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DialogModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    CoreModule,
                    FormsModule,
                    ReactiveFormsModule,
                    MatMomentDatetimeModule,
                    MatDatetimepickerModule
                ],
                declarations: [
                    FolderDialogComponent,
                    NodeLockDialogComponent,
                    ConfirmDialogComponent,
                    LibraryDialogComponent
                ],
                exports: [
                    FolderDialogComponent,
                    NodeLockDialogComponent,
                    ConfirmDialogComponent,
                    LibraryDialogComponent
                ]
            }]
    }], null, null); })();
