// @alfresco/adf-core/esm2015/card-view/components/card-view-selectitem/select-filter-input/select-filter-input.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, ViewChild, ElementRef, Inject, Output, EventEmitter } from '@angular/core';
import { ESCAPE, TAB } from '@angular/cdk/keycodes';
import { MatSelect } from '@angular/material/select';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/form-field';
import * as ɵngcc2 from '@angular/material/input';
import * as ɵngcc3 from '@angular/forms';
import * as ɵngcc4 from '@angular/common';
import * as ɵngcc5 from '@angular/material/button';
import * as ɵngcc6 from '@angular/material/icon';
import * as ɵngcc7 from '@ngx-translate/core';
import * as ɵngcc8 from '@angular/material/select';

const _c0 = ["selectFilterInput"];
function SelectFilterInputComponent_button_6_Template(rf, ctx) { if (rf & 1) {
    const _r3 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 4);
    ɵngcc0.ɵɵlistener("keydown.enter", function SelectFilterInputComponent_button_6_Template_button_keydown_enter_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r3); const ctx_r2 = ɵngcc0.ɵɵnextContext(); return ctx_r2.reset($event); })("click", function SelectFilterInputComponent_button_6_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r3); const ctx_r4 = ɵngcc0.ɵɵnextContext(); return ctx_r4.reset(); });
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵelementStart(2, "mat-icon");
    ɵngcc0.ɵɵtext(3, "close");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind1(1, 1, "SELECT_FILTER.BUTTON.ARIA_LABEL"));
} }
export class SelectFilterInputComponent {
    constructor(matSelect) {
        this.matSelect = matSelect;
        this.change = new EventEmitter();
        this.term = '';
        this.onDestroy$ = new Subject();
    }
    onModelChange(value) {
        this.change.next(value);
    }
    ngOnInit() {
        this.change
            .pipe(takeUntil(this.onDestroy$))
            .subscribe((val) => this.term = val);
        this.matSelect.openedChange
            .pipe(takeUntil(this.onDestroy$))
            .subscribe((isOpened) => {
            if (isOpened) {
                this.selectFilterInput.nativeElement.focus();
            }
            else {
                this.change.next('');
            }
        });
    }
    reset(event) {
        if (event) {
            event.stopPropagation();
        }
        this.change.next('');
        this.selectFilterInput.nativeElement.focus();
    }
    handleKeydown($event) {
        if (this.term) {
            if ($event.keyCode === ESCAPE) {
                event.stopPropagation();
                this.change.next('');
            }
            if ($event.target.tagName === 'INPUT' && $event.keyCode === TAB) {
                event.stopPropagation();
            }
        }
    }
    ngOnDestroy() {
        this.onDestroy$.next();
        this.onDestroy$.complete();
    }
}
SelectFilterInputComponent.ɵfac = function SelectFilterInputComponent_Factory(t) { return new (t || SelectFilterInputComponent)(ɵngcc0.ɵɵdirectiveInject(MatSelect)); };
SelectFilterInputComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SelectFilterInputComponent, selectors: [["adf-select-filter-input"]], viewQuery: function SelectFilterInputComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true, ElementRef);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.selectFilterInput = _t.first);
    } }, hostAttrs: [1, "adf-select-filter-input"], outputs: { change: "change" }, decls: 7, vars: 8, consts: [[1, "adf-select-filter-input-container"], ["matInput", "", "autocomplete", "off", 3, "placeholder", "ngModel", "keydown", "ngModelChange", "change"], ["selectFilterInput", ""], ["mat-button", "", "matSuffix", "", "mat-icon-button", "", 3, "keydown.enter", "click", 4, "ngIf"], ["mat-button", "", "matSuffix", "", "mat-icon-button", "", 3, "keydown.enter", "click"]], template: function SelectFilterInputComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "mat-form-field");
        ɵngcc0.ɵɵelementStart(2, "input", 1, 2);
        ɵngcc0.ɵɵlistener("keydown", function SelectFilterInputComponent_Template_input_keydown_2_listener($event) { return ctx.handleKeydown($event); })("ngModelChange", function SelectFilterInputComponent_Template_input_ngModelChange_2_listener($event) { return ctx.onModelChange($event); })("change", function SelectFilterInputComponent_Template_input_change_2_listener($event) { return $event.stopPropagation(); });
        ɵngcc0.ɵɵpipe(4, "translate");
        ɵngcc0.ɵɵpipe(5, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(6, SelectFilterInputComponent_button_6_Template, 4, 3, "button", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("placeholder", ɵngcc0.ɵɵpipeBind1(4, 4, "SELECT_FILTER.INPUT.PLACEHOLDER"))("ngModel", ctx.term);
        ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind1(5, 6, "SELECT_FILTER.INPUT.ARIA_LABEL"));
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("ngIf", ctx.term);
    } }, directives: [ɵngcc1.MatFormField, ɵngcc2.MatInput, ɵngcc3.DefaultValueAccessor, ɵngcc3.NgControlStatus, ɵngcc3.NgModel, ɵngcc4.NgIf, ɵngcc5.MatButton, ɵngcc1.MatSuffix, ɵngcc6.MatIcon], pipes: [ɵngcc7.TranslatePipe], styles: [".adf-select-filter-input{display:flex;height:4em}.adf-select-filter-input .adf-select-filter-input-container{background:var(--theme-card-bg-color);color:var(--theme-text-bold-color);display:flex;font-size:14px;height:4em;line-height:3em;padding:5px 16px 0;position:absolute;top:0;width:100%;z-index:100}.adf-select-filter-input .mat-form-field{width:100%}.mat-select-panel.adf-select-filter{max-height:calc(256px + 4em);overflow-x:hidden!important;transform:none!important}"], encapsulation: 2 });
SelectFilterInputComponent.ctorParameters = () => [
    { type: MatSelect, decorators: [{ type: Inject, args: [MatSelect,] }] }
];
SelectFilterInputComponent.propDecorators = {
    selectFilterInput: [{ type: ViewChild, args: ['selectFilterInput', { read: ElementRef, static: false },] }],
    change: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SelectFilterInputComponent, [{
        type: Component,
        args: [{
                selector: 'adf-select-filter-input',
                template: "\n<div class=\"adf-select-filter-input-container\">\n    <mat-form-field>\n        <input matInput\n            autocomplete=\"off\"\n            (keydown)=\"handleKeydown($event)\"\n            [placeholder]=\"'SELECT_FILTER.INPUT.PLACEHOLDER' | translate\"\n            #selectFilterInput\n            [ngModel]=\"term\"\n            (ngModelChange)=\"onModelChange($event)\"\n            [attr.aria-label]=\"'SELECT_FILTER.INPUT.ARIA_LABEL' | translate\"\n            (change)=\"$event.stopPropagation()\"\n        />\n\n        <button mat-button\n            matSuffix\n            mat-icon-button\n            [attr.aria-label]=\"'SELECT_FILTER.BUTTON.ARIA_LABEL' | translate\"\n            *ngIf=\"term\"\n            (keydown.enter)=\"reset($event)\"\n            (click)=\"reset()\">\n            <mat-icon>close</mat-icon>\n        </button>\n    </mat-form-field>\n</div>\n",
                host: { 'class': 'adf-select-filter-input' },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-select-filter-input{display:flex;height:4em}.adf-select-filter-input .adf-select-filter-input-container{background:var(--theme-card-bg-color);color:var(--theme-text-bold-color);display:flex;font-size:14px;height:4em;line-height:3em;padding:5px 16px 0;position:absolute;top:0;width:100%;z-index:100}.adf-select-filter-input .mat-form-field{width:100%}.mat-select-panel.adf-select-filter{max-height:calc(256px + 4em);overflow-x:hidden!important;transform:none!important}"]
            }]
    }], function () { return [{ type: ɵngcc8.MatSelect, decorators: [{
                type: Inject,
                args: [MatSelect]
            }] }]; }, { change: [{
            type: Output
        }], selectFilterInput: [{
            type: ViewChild,
            args: ['selectFilterInput', { read: ElementRef, static: false }]
        }] }); })();
