// @alfresco/adf-extensions/esm2015/lib/services/app-extension.service.js
import { __awaiter } from "tslib";
import { Injectable } from '@angular/core';
import { ExtensionService } from '../services/extension.service';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "./extension.service";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/extension.service';
export class AppExtensionService {
    constructor(extensionService) {
        this.extensionService = extensionService;
        this._references = new BehaviorSubject([]);
        this.references$ = this._references.asObservable();
    }
    load() {
        return __awaiter(this, void 0, void 0, function* () {
            const config = yield this.extensionService.load();
            this.setup(config);
        });
    }
    setup(config) {
        if (!config) {
            return;
        }
        const references = (config.$references || [])
            .filter((entry) => typeof entry === 'object')
            .map((entry) => entry);
        this._references.next(references);
    }
    getDocumentListPreset(key) {
        return this.extensionService
            .getElements(`features.documentList.${key}`)
            .filter((entry) => !entry.disabled);
    }
    getViewerExtensions() {
        return this.extensionService
            .getElements('features.viewer.content')
            .filter((extension) => !this.isViewerExtensionDisabled(extension));
    }
    isViewerExtensionDisabled(extension) {
        if (extension) {
            if (extension.disabled) {
                return true;
            }
            if (extension.rules && extension.rules.disabled) {
                return this.extensionService.evaluateRule(extension.rules.disabled);
            }
        }
        return false;
    }
}
AppExtensionService.ɵfac = function AppExtensionService_Factory(t) { return new (t || AppExtensionService)(ɵngcc0.ɵɵinject(ɵngcc1.ExtensionService)); };
AppExtensionService.ɵprov = i0.ɵɵdefineInjectable({ factory: function AppExtensionService_Factory() { return new AppExtensionService(i0.ɵɵinject(i1.ExtensionService)); }, token: AppExtensionService, providedIn: "root" });
AppExtensionService.ctorParameters = () => [
    { type: ExtensionService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppExtensionService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.ExtensionService }]; }, null); })();
