// @alfresco/adf-extensions/esm2015/lib/services/component-register.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as ɵngcc0 from '@angular/core';
export class ComponentRegisterService {
    constructor() {
        this.components = {};
    }
    setComponents(values) {
        if (values) {
            this.components = Object.assign({}, this.components, values);
        }
    }
    getComponentById(id) {
        return this.components[id];
    }
    hasComponentById(id) {
        return !!this.getComponentById(id);
    }
}
ComponentRegisterService.ɵfac = function ComponentRegisterService_Factory(t) { return new (t || ComponentRegisterService)(); };
ComponentRegisterService.ɵprov = i0.ɵɵdefineInjectable({ factory: function ComponentRegisterService_Factory() { return new ComponentRegisterService(); }, token: ComponentRegisterService, providedIn: "root" });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ComponentRegisterService, [{
        type: Injectable,
        args: [{ providedIn: 'root' }]
    }], function () { return []; }, null); })();
