// @alfresco/adf-extensions/esm2015/lib/services/rule.service.js
import { Injectable } from '@angular/core';
import { ExtensionLoaderService } from './extension-loader.service';
import * as i0 from "@angular/core";
import * as i1 from "./extension-loader.service";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './extension-loader.service';
export class RuleService {
    constructor(loader) {
        this.loader = loader;
        this.context = null;
        this.rules = [];
        this.evaluators = {};
    }
    setup(config) {
        this.rules = this.loader.getRules(config);
    }
    setEvaluators(values) {
        if (values) {
            this.evaluators = Object.assign({}, this.evaluators, values);
        }
    }
    getRuleById(id) {
        return this.rules.find((ref) => ref.id === id);
    }
    getEvaluator(key) {
        if (key && key.startsWith('!')) {
            const fn = this.evaluators[key.substring(1)];
            return (context, ...args) => {
                return !fn(context, ...args);
            };
        }
        return this.evaluators[key];
    }
    evaluateRule(ruleId, context) {
        const ruleRef = this.getRuleById(ruleId);
        context = context || this.context;
        if (ruleRef) {
            const evaluator = this.getEvaluator(ruleRef.type);
            if (evaluator) {
                return evaluator(context, ...ruleRef.parameters);
            }
        }
        else {
            const evaluator = this.getEvaluator(ruleId);
            if (evaluator) {
                return evaluator(context);
            }
        }
        return false;
    }
}
RuleService.ɵfac = function RuleService_Factory(t) { return new (t || RuleService)(ɵngcc0.ɵɵinject(ɵngcc1.ExtensionLoaderService)); };
RuleService.ɵprov = i0.ɵɵdefineInjectable({ factory: function RuleService_Factory() { return new RuleService(i0.ɵɵinject(i1.ExtensionLoaderService)); }, token: RuleService, providedIn: "root" });
RuleService.ctorParameters = () => [
    { type: ExtensionLoaderService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RuleService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.ExtensionLoaderService }]; }, null); })();
