// @alfresco/adf-core/esm2015/data-column/data-column.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { DataColumnListComponent } from './data-column-list.component';
import { DataColumnComponent } from './data-column.component';
import { DateColumnHeaderComponent } from './data-column-header.component';
import * as ɵngcc0 from '@angular/core';
export class DataColumnModule {
}
DataColumnModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DataColumnModule });
DataColumnModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DataColumnModule_Factory(t) { return new (t || DataColumnModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DataColumnModule, { declarations: function () { return [DataColumnComponent,
        DataColumnListComponent,
        DateColumnHeaderComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [DataColumnComponent,
        DataColumnListComponent,
        DateColumnHeaderComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataColumnModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    DataColumnComponent,
                    DataColumnListComponent,
                    DateColumnHeaderComponent
                ],
                exports: [
                    DataColumnComponent,
                    DataColumnListComponent,
                    DateColumnHeaderComponent
                ]
            }]
    }], null, null); })();
