// @alfresco/adf-core/esm2015/data-column/data-column-header.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ContentChild, TemplateRef } from '@angular/core';
import { DataColumnComponent } from './data-column.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './data-column.component';
export class DateColumnHeaderComponent {
    constructor(columnComponent) {
        this.columnComponent = columnComponent;
    }
    ngAfterContentInit() {
        if (this.columnComponent) {
            this.columnComponent.header = this.header;
        }
    }
}
DateColumnHeaderComponent.ɵfac = function DateColumnHeaderComponent_Factory(t) { return new (t || DateColumnHeaderComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DataColumnComponent)); };
DateColumnHeaderComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DateColumnHeaderComponent, selectors: [["adf-data-column-header"]], contentQueries: function DateColumnHeaderComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, TemplateRef, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.header = _t.first);
    } }, decls: 0, vars: 0, template: function DateColumnHeaderComponent_Template(rf, ctx) { }, encapsulation: 2 });
DateColumnHeaderComponent.ctorParameters = () => [
    { type: DataColumnComponent }
];
DateColumnHeaderComponent.propDecorators = {
    header: [{ type: ContentChild, args: [TemplateRef,] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DateColumnHeaderComponent, [{
        type: Component,
        args: [{
                selector: 'adf-data-column-header',
                template: ''
            }]
    }], function () { return [{ type: ɵngcc1.DataColumnComponent }]; }, { header: [{
            type: ContentChild,
            args: [TemplateRef]
        }] }); })();
