// @alfresco/adf-core/esm2015/data-column/data-column-list.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ContentChildren, QueryList } from '@angular/core';
import { DataColumnComponent } from './data-column.component';
import * as ɵngcc0 from '@angular/core';
export class DataColumnListComponent {
}
DataColumnListComponent.ɵfac = function DataColumnListComponent_Factory(t) { return new (t || DataColumnListComponent)(); };
DataColumnListComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataColumnListComponent, selectors: [["data-columns"]], contentQueries: function DataColumnListComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, DataColumnComponent, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.columns = _t);
    } }, decls: 0, vars: 0, template: function DataColumnListComponent_Template(rf, ctx) { }, encapsulation: 2 });
DataColumnListComponent.propDecorators = {
    columns: [{ type: ContentChildren, args: [DataColumnComponent,] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataColumnListComponent, [{
        type: Component,
        args: [{
                selector: 'data-columns',
                template: ''
            }]
    }], null, { columns: [{
            type: ContentChildren,
            args: [DataColumnComponent]
        }] }); })();
