// @alfresco/adf-core/esm2015/data-column/data-column.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ContentChild, Input, TemplateRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class DataColumnComponent {
    constructor() {
        this.type = 'text';
        this.sortable = true;
        this.title = '';
        this.editable = false;
        this.focus = true;
    }
    ngOnInit() {
        if (!this.srTitle && this.key === '$thumbnail') {
            this.srTitle = 'Thumbnail';
        }
    }
}
DataColumnComponent.ɵfac = function DataColumnComponent_Factory(t) { return new (t || DataColumnComponent)(); };
DataColumnComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataColumnComponent, selectors: [["data-column"]], contentQueries: function DataColumnComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, TemplateRef, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.template = _t.first);
    } }, inputs: { type: "type", sortable: "sortable", title: "title", editable: "editable", focus: "focus", srTitle: ["sr-title", "srTitle"], key: "key", format: "format", formatTooltip: "formatTooltip", cssClass: ["class", "cssClass"], copyContent: "copyContent", sortingKey: "sortingKey" }, decls: 0, vars: 0, template: function DataColumnComponent_Template(rf, ctx) { }, encapsulation: 2 });
DataColumnComponent.propDecorators = {
    key: [{ type: Input }],
    type: [{ type: Input }],
    format: [{ type: Input }],
    sortable: [{ type: Input }],
    title: [{ type: Input }],
    template: [{ type: ContentChild, args: [TemplateRef,] }],
    formatTooltip: [{ type: Input }],
    srTitle: [{ type: Input, args: ['sr-title',] }],
    cssClass: [{ type: Input, args: ['class',] }],
    copyContent: [{ type: Input }],
    editable: [{ type: Input }],
    focus: [{ type: Input }],
    sortingKey: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataColumnComponent, [{
        type: Component,
        args: [{
                selector: 'data-column',
                template: ''
            }]
    }], function () { return []; }, { type: [{
            type: Input
        }], sortable: [{
            type: Input
        }], title: [{
            type: Input
        }], editable: [{
            type: Input
        }], focus: [{
            type: Input
        }], srTitle: [{
            type: Input,
            args: ['sr-title']
        }], key: [{
            type: Input
        }], format: [{
            type: Input
        }], template: [{
            type: ContentChild,
            args: [TemplateRef]
        }], formatTooltip: [{
            type: Input
        }], cssClass: [{
            type: Input,
            args: ['class']
        }], copyContent: [{
            type: Input
        }], sortingKey: [{
            type: Input
        }] }); })();
