// @alfresco/adf-content-services/esm2015/lib/folder-directive/folder-directive.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { FolderCreateDirective } from './folder-create.directive';
import { FolderEditDirective } from './folder-edit.directive';
import * as ɵngcc0 from '@angular/core';
export class FolderDirectiveModule {
}
FolderDirectiveModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FolderDirectiveModule });
FolderDirectiveModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FolderDirectiveModule_Factory(t) { return new (t || FolderDirectiveModule)(); }, imports: [[
            CommonModule,
            MaterialModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FolderDirectiveModule, { declarations: function () { return [FolderCreateDirective,
        FolderEditDirective]; }, imports: function () { return [CommonModule,
        MaterialModule]; }, exports: function () { return [FolderCreateDirective,
        FolderEditDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FolderDirectiveModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule
                ],
                declarations: [
                    FolderCreateDirective,
                    FolderEditDirective
                ],
                exports: [
                    FolderCreateDirective,
                    FolderEditDirective
                ]
            }]
    }], null, null); })();
