// @alfresco/adf-core/esm2015/layout/directives/sidenav-layout-navigation.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ContentChild, Directive, TemplateRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class SidenavLayoutNavigationDirective {
}
SidenavLayoutNavigationDirective.ɵfac = function SidenavLayoutNavigationDirective_Factory(t) { return new (t || SidenavLayoutNavigationDirective)(); };
SidenavLayoutNavigationDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SidenavLayoutNavigationDirective, selectors: [["adf-sidenav-layout-navigation"]], contentQueries: function SidenavLayoutNavigationDirective_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, TemplateRef, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.template = _t.first);
    } } });
SidenavLayoutNavigationDirective.propDecorators = {
    template: [{ type: ContentChild, args: [TemplateRef,] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidenavLayoutNavigationDirective, [{
        type: Directive,
        args: [{
                selector: 'adf-sidenav-layout-navigation'
            }]
    }], null, { template: [{
            type: ContentChild,
            args: [TemplateRef]
        }] }); })();
