// @alfresco/adf-process-services/esm2015/lib/task-list/components/attach-form.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { FormService, LogService } from '@alfresco/adf-core';
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { TaskListService } from './../services/tasklist.service';
import { FormControl, Validators } from '@angular/forms';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../services/tasklist.service';
import * as ɵngcc2 from '@alfresco/adf-core';
import * as ɵngcc3 from '@angular/material/card';
import * as ɵngcc4 from '@angular/material/form-field';
import * as ɵngcc5 from '@angular/material/select';
import * as ɵngcc6 from '@angular/forms';
import * as ɵngcc7 from '@angular/common';
import * as ɵngcc8 from '@angular/material/button';
import * as ɵngcc9 from '@angular/material/core';
import * as ɵngcc10 from '../../form/form.component';
import * as ɵngcc11 from '@ngx-translate/core';

function AttachFormComponent_mat_option_12_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-option", 13);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const form_r3 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("value", form_r3.id);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(form_r3.name);
} }
function AttachFormComponent_adf_form_13_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-form", 14);
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("formId", ctx_r1.selectedFormId)("readOnly", true)("showCompleteButton", false)("showRefreshButton", false)("showValidationIcon", false);
} }
function AttachFormComponent_button_16_Template(rf, ctx) { if (rf & 1) {
    const _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 15);
    ɵngcc0.ɵɵlistener("click", function AttachFormComponent_button_16_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r5); const ctx_r4 = ɵngcc0.ɵɵnextContext(); return ctx_r4.onRemoveButtonClick(); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, "ADF_TASK_LIST.ATTACH_FORM.REMOVE_FORM"));
} }
export class AttachFormComponent {
    constructor(taskService, logService, formService) {
        this.taskService = taskService;
        this.logService = logService;
        this.formService = formService;
        this.cancelAttachForm = new EventEmitter();
        this.success = new EventEmitter();
        this.error = new EventEmitter();
        this.disableSubmit = true;
    }
    ngOnInit() {
        this.attachFormControl = new FormControl('', Validators.required);
        this.attachFormControl.valueChanges.subscribe((currentValue) => {
            if (this.attachFormControl.valid) {
                this.disableSubmit = this.formId === currentValue;
            }
        });
    }
    ngOnChanges() {
        this.formId = undefined;
        this.disableSubmit = true;
        this.loadFormsTask();
        if (this.formKey) {
            this.onFormAttached();
        }
    }
    onCancelButtonClick() {
        this.selectedFormId = this.formId;
        this.cancelAttachForm.emit();
    }
    onRemoveButtonClick() {
        this.taskService.deleteForm(this.taskId).subscribe(() => {
            this.formId = this.selectedFormId = null;
            this.success.emit();
        }, (err) => {
            this.error.emit(err);
            this.logService.error('An error occurred while trying to delete the form');
        });
    }
    onAttachFormButtonClick() {
        this.attachForm(this.taskId, this.selectedFormId);
    }
    loadFormsTask() {
        this.taskService.getFormList().subscribe((form) => {
            this.forms = form;
        }, (err) => {
            this.error.emit(err);
            this.logService.error('An error occurred while trying to get the forms');
        });
    }
    onFormAttached() {
        this.formService.getTaskForm(this.taskId)
            .subscribe((res) => {
            this.formService.getFormDefinitionByName(res.name).subscribe((formDef) => {
                this.formId = this.selectedFormId = formDef;
            });
        }, (err) => {
            this.error.emit(err);
            this.logService.error('Could not load forms');
        });
    }
    attachForm(taskId, formId) {
        if (taskId && formId) {
            this.taskService.attachFormToATask(taskId, formId)
                .subscribe(() => {
                this.success.emit();
            }, (err) => {
                this.error.emit(err);
                this.logService.error('Could not attach form');
            });
        }
    }
}
AttachFormComponent.ɵfac = function AttachFormComponent_Factory(t) { return new (t || AttachFormComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TaskListService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.LogService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.FormService)); };
AttachFormComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AttachFormComponent, selectors: [["adf-attach-form"]], inputs: { taskId: "taskId", formKey: "formKey" }, outputs: { cancelAttachForm: "cancelAttachForm", success: "success", error: "error" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 24, vars: 18, consts: [[1, "adf-attach-form"], [1, "adf-attache-form-message-container"], [1, "mat-card-title"], [1, "adf-form-title"], [1, "adf-attach-form-row"], [1, "adf-grid-full-width"], ["id", "form_id", 3, "formControl", "placeholder", "ngModel", "ngModelChange"], [3, "value", 4, "ngFor", "ngForOf"], [3, "formId", "readOnly", "showCompleteButton", "showRefreshButton", "showValidationIcon", 4, "ngIf"], [1, "adf-attach-form-mat-card-actions"], ["mat-button", "", "id", "adf-attach-form-remove-button", "color", "warn", 3, "click", 4, "ngIf"], ["mat-button", "", "id", "adf-attach-form-cancel-button", 3, "click"], ["mat-button", "", "id", "adf-attach-form-attach-button", "color", "primary", 3, "disabled", "click"], [3, "value"], [3, "formId", "readOnly", "showCompleteButton", "showRefreshButton", "showValidationIcon"], ["mat-button", "", "id", "adf-attach-form-remove-button", "color", "warn", 3, "click"]], template: function AttachFormComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "mat-card");
        ɵngcc0.ɵɵelementStart(2, "mat-card-content");
        ɵngcc0.ɵɵelementStart(3, "div", 1);
        ɵngcc0.ɵɵelementStart(4, "mat-card-title", 2);
        ɵngcc0.ɵɵelementStart(5, "h4", 3);
        ɵngcc0.ɵɵtext(6);
        ɵngcc0.ɵɵpipe(7, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(8, "div", 4);
        ɵngcc0.ɵɵelementStart(9, "mat-form-field", 5);
        ɵngcc0.ɵɵelementStart(10, "mat-select", 6);
        ɵngcc0.ɵɵlistener("ngModelChange", function AttachFormComponent_Template_mat_select_ngModelChange_10_listener($event) { return ctx.selectedFormId = $event; });
        ɵngcc0.ɵɵpipe(11, "translate");
        ɵngcc0.ɵɵtemplate(12, AttachFormComponent_mat_option_12_Template, 2, 2, "mat-option", 7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(13, AttachFormComponent_adf_form_13_Template, 1, 5, "adf-form", 8);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(14, "mat-card-actions", 9);
        ɵngcc0.ɵɵelementStart(15, "div");
        ɵngcc0.ɵɵtemplate(16, AttachFormComponent_button_16_Template, 3, 3, "button", 10);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(17, "div");
        ɵngcc0.ɵɵelementStart(18, "button", 11);
        ɵngcc0.ɵɵlistener("click", function AttachFormComponent_Template_button_click_18_listener() { return ctx.onCancelButtonClick(); });
        ɵngcc0.ɵɵtext(19);
        ɵngcc0.ɵɵpipe(20, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "button", 12);
        ɵngcc0.ɵɵlistener("click", function AttachFormComponent_Template_button_click_21_listener() { return ctx.onAttachFormButtonClick(); });
        ɵngcc0.ɵɵtext(22);
        ɵngcc0.ɵɵpipe(23, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(6);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(7, 10, "ADF_TASK_LIST.ATTACH_FORM.SELECT_FORM"));
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵpropertyInterpolate("placeholder", ɵngcc0.ɵɵpipeBind1(11, 12, "ADF_TASK_LIST.ATTACH_FORM.SELECT_OPTION"));
        ɵngcc0.ɵɵproperty("formControl", ctx.attachFormControl)("ngModel", ctx.selectedFormId);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.forms);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.attachFormControl.valid);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngIf", ctx.formKey);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(20, 14, "ADF_TASK_LIST.START_TASK.FORM.ACTION.CANCEL"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("disabled", ctx.disableSubmit);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(23, 16, "ADF_TASK_LIST.START_TASK.FORM.LABEL.ATTACHFORM"));
    } }, directives: [ɵngcc3.MatCard, ɵngcc3.MatCardContent, ɵngcc3.MatCardTitle, ɵngcc4.MatFormField, ɵngcc5.MatSelect, ɵngcc6.NgControlStatus, ɵngcc6.FormControlDirective, ɵngcc7.NgForOf, ɵngcc7.NgIf, ɵngcc3.MatCardActions, ɵngcc8.MatButton, ɵngcc9.MatOption, ɵngcc10.FormComponent], pipes: [ɵngcc11.TranslatePipe], styles: [".adf-attach-form[_ngcontent-%COMP%]   .mat-form-field[_ngcontent-%COMP%]{width:100%}.adf-attach-form-row[_ngcontent-%COMP%]{display:flex;justify-content:space-between;margin:20px 0}.adf-attach-form[_ngcontent-%COMP%]   .adf-attach-form-mat-card-actions[_ngcontent-%COMP%]{justify-content:space-between;margin-top:30px;text-align:right}"] });
AttachFormComponent.ctorParameters = () => [
    { type: TaskListService },
    { type: LogService },
    { type: FormService }
];
AttachFormComponent.propDecorators = {
    taskId: [{ type: Input }],
    formKey: [{ type: Input }],
    cancelAttachForm: [{ type: Output }],
    success: [{ type: Output }],
    error: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AttachFormComponent, [{
        type: Component,
        args: [{
                selector: 'adf-attach-form',
                template: "<div class=\"adf-attach-form\">\n    <mat-card>\n        <mat-card-content>\n            <div class=\"adf-attache-form-message-container\">\n                <mat-card-title class=\"mat-card-title\">\n                    <h4 class=\"adf-form-title\">{{ 'ADF_TASK_LIST.ATTACH_FORM.SELECT_FORM' | translate }}</h4>\n                </mat-card-title>\n                <div class=\"adf-attach-form-row\">\n                    <mat-form-field class=\"adf-grid-full-width\">\n                        <mat-select [formControl]=\"attachFormControl\" placeholder=\"{{ 'ADF_TASK_LIST.ATTACH_FORM.SELECT_OPTION' | translate }}\" id=\"form_id\" [(ngModel)]=\"selectedFormId\">\n                            <mat-option *ngFor=\"let form of forms\" [value]=\"form.id\">{{ form.name }}</mat-option>\n                        </mat-select>\n                    </mat-form-field>\n                </div>\n\n                <adf-form *ngIf=\"this.attachFormControl.valid\"\n                    [formId]=\"selectedFormId\"\n                    [readOnly]=\"true\"\n                    [showCompleteButton]=\"false\"\n                    [showRefreshButton]=\"false\"\n                    [showValidationIcon]=\"false\">\n                </adf-form>\n            </div>\n        </mat-card-content>\n\n        <mat-card-actions class=\"adf-attach-form-mat-card-actions\">\n            <div>\n                <button mat-button id=\"adf-attach-form-remove-button\" color=\"warn\" *ngIf=\"formKey\" (click)=\"onRemoveButtonClick()\">{{ 'ADF_TASK_LIST.ATTACH_FORM.REMOVE_FORM' | translate }}</button>\n            </div>\n            <div>\n                <button mat-button id=\"adf-attach-form-cancel-button\" (click)=\"onCancelButtonClick()\">{{ 'ADF_TASK_LIST.START_TASK.FORM.ACTION.CANCEL' | translate }}</button>\n                <button mat-button id=\"adf-attach-form-attach-button\" [disabled]=\"disableSubmit\" color=\"primary\" (click)=\"onAttachFormButtonClick()\">{{ 'ADF_TASK_LIST.START_TASK.FORM.LABEL.ATTACHFORM' | translate }}</button>\n            </div>\n        </mat-card-actions>\n    </mat-card>\n</div>\n",
                styles: [".adf-attach-form .mat-form-field{width:100%}.adf-attach-form-row{display:flex;justify-content:space-between;margin:20px 0}.adf-attach-form .adf-attach-form-mat-card-actions{justify-content:space-between;margin-top:30px;text-align:right}"]
            }]
    }], function () { return [{ type: ɵngcc1.TaskListService }, { type: ɵngcc2.LogService }, { type: ɵngcc2.FormService }]; }, { cancelAttachForm: [{
            type: Output
        }], success: [{
            type: Output
        }], error: [{
            type: Output
        }], taskId: [{
            type: Input
        }], formKey: [{
            type: Input
        }] }); })();
