// @alfresco/adf-core/esm2015/form/services/form-rendering.service.js
import { DynamicComponentMapper, DynamicComponentResolver } from '../../services/dynamic-component-mapper.service';
import { Injectable } from '@angular/core';
import * as widgets from './../components/widgets/index';
import * as i0 from "@angular/core";
import * as ɵngcc0 from '@angular/core';
export class FormRenderingService extends DynamicComponentMapper {
    constructor() {
        super(...arguments);
        this.defaultValue = widgets.UnknownWidgetComponent;
        this.types = {
            'text': DynamicComponentResolver.fromType(widgets.TextWidgetComponent),
            'string': DynamicComponentResolver.fromType(widgets.TextWidgetComponent),
            'integer': DynamicComponentResolver.fromType(widgets.NumberWidgetComponent),
            'multi-line-text': DynamicComponentResolver.fromType(widgets.MultilineTextWidgetComponentComponent),
            'boolean': DynamicComponentResolver.fromType(widgets.CheckboxWidgetComponent),
            'dropdown': DynamicComponentResolver.fromType(widgets.DropdownWidgetComponent),
            'date': DynamicComponentResolver.fromType(widgets.DateWidgetComponent),
            'amount': DynamicComponentResolver.fromType(widgets.AmountWidgetComponent),
            'radio-buttons': DynamicComponentResolver.fromType(widgets.RadioButtonsWidgetComponent),
            'hyperlink': DynamicComponentResolver.fromType(widgets.HyperlinkWidgetComponent),
            'readonly-text': DynamicComponentResolver.fromType(widgets.DisplayTextWidgetComponent),
            'json': DynamicComponentResolver.fromType(widgets.JsonWidgetComponent),
            'readonly': DynamicComponentResolver.fromType(widgets.TextWidgetComponent),
            'typeahead': DynamicComponentResolver.fromType(widgets.TypeaheadWidgetComponent),
            'people': DynamicComponentResolver.fromType(widgets.PeopleWidgetComponent),
            'functional-group': DynamicComponentResolver.fromType(widgets.FunctionalGroupWidgetComponent),
            'dynamic-table': DynamicComponentResolver.fromType(widgets.DynamicTableWidgetComponent),
            'container': DynamicComponentResolver.fromType(widgets.ContainerWidgetComponent),
            'group': DynamicComponentResolver.fromType(widgets.ContainerWidgetComponent),
            'document': DynamicComponentResolver.fromType(widgets.DocumentWidgetComponent),
            'upload': DynamicComponentResolver.fromType(widgets.UploadWidgetComponent),
            'datetime': DynamicComponentResolver.fromType(widgets.DateTimeWidgetComponent),
            'file-viewer': DynamicComponentResolver.fromType(widgets.FileViewerWidgetComponent)
        };
    }
}
FormRenderingService.ɵfac = function FormRenderingService_Factory(t) { return ɵFormRenderingService_BaseFactory(t || FormRenderingService); };
FormRenderingService.ɵprov = i0.ɵɵdefineInjectable({ factory: function FormRenderingService_Factory() { return new FormRenderingService(); }, token: FormRenderingService, providedIn: "root" });
const ɵFormRenderingService_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(FormRenderingService);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FormRenderingService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], null, null); })();
