// @alfresco/adf-content-services/esm2015/lib/document-list/components/content-action/content-action-list.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component } from '@angular/core';
import { DocumentListComponent } from './../document-list.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../document-list.component';
export class ContentActionListComponent {
    constructor(documentList) {
        this.documentList = documentList;
    }
    registerAction(action) {
        if (this.documentList && action) {
            this.documentList.actions.push(action);
            return true;
        }
        return false;
    }
    unregisterAction(action) {
        if (this.documentList && action) {
            const idx = this.documentList.actions.indexOf(action);
            if (idx >= 0) {
                this.documentList.actions.splice(idx, 1);
                return true;
            }
        }
        return false;
    }
}
ContentActionListComponent.ɵfac = function ContentActionListComponent_Factory(t) { return new (t || ContentActionListComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DocumentListComponent)); };
ContentActionListComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ContentActionListComponent, selectors: [["content-actions"]], decls: 0, vars: 0, template: function ContentActionListComponent_Template(rf, ctx) { }, encapsulation: 2 });
ContentActionListComponent.ctorParameters = () => [
    { type: DocumentListComponent }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContentActionListComponent, [{
        type: Component,
        args: [{
                selector: 'content-actions',
                template: ''
            }]
    }], function () { return [{ type: ɵngcc1.DocumentListComponent }]; }, null); })();
