// @alfresco/adf-core/esm2015/context-menu/context-menu.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, HostListener, Input } from '@angular/core';
import { ContextMenuOverlayService } from './context-menu-overlay.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './context-menu-overlay.service';
export class ContextMenuDirective {
    constructor(contextMenuService) {
        this.contextMenuService = contextMenuService;
        this.enabled = false;
    }
    onShowContextMenu(event) {
        if (this.enabled) {
            if (event) {
                event.preventDefault();
            }
            if (this.links && this.links.length > 0) {
                this.contextMenuService.open({
                    source: event,
                    data: this.links
                });
            }
        }
    }
}
ContextMenuDirective.ɵfac = function ContextMenuDirective_Factory(t) { return new (t || ContextMenuDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ContextMenuOverlayService)); };
ContextMenuDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ContextMenuDirective, selectors: [["", "adf-context-menu", ""]], hostBindings: function ContextMenuDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("contextmenu", function ContextMenuDirective_contextmenu_HostBindingHandler($event) { return ctx.onShowContextMenu($event); });
    } }, inputs: { enabled: ["adf-context-menu-enabled", "enabled"], links: ["adf-context-menu", "links"] } });
ContextMenuDirective.ctorParameters = () => [
    { type: ContextMenuOverlayService }
];
ContextMenuDirective.propDecorators = {
    links: [{ type: Input, args: ['adf-context-menu',] }],
    enabled: [{ type: Input, args: ['adf-context-menu-enabled',] }],
    onShowContextMenu: [{ type: HostListener, args: ['contextmenu', ['$event'],] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContextMenuDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-context-menu]'
            }]
    }], function () { return [{ type: ɵngcc1.ContextMenuOverlayService }]; }, { enabled: [{
            type: Input,
            args: ['adf-context-menu-enabled']
        }], onShowContextMenu: [{
            type: HostListener,
            args: ['contextmenu', ['$event']]
        }], links: [{
            type: Input,
            args: ['adf-context-menu']
        }] }); })();
