// @alfresco/adf-core/esm2015/directives/tooltip-card/tooltip-card.directive.js
import { Directive, ElementRef, HostListener, Input } from '@angular/core';
import { Overlay, OverlayPositionBuilder } from '@angular/cdk/overlay';
import { ComponentPortal } from '@angular/cdk/portal';
import { TooltipCardComponent } from './tooltip-card.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/cdk/overlay';
export class TooltipCardDirective {
    constructor(overlay, overlayPositionBuilder, elementRef) {
        this.overlay = overlay;
        this.overlayPositionBuilder = overlayPositionBuilder;
        this.elementRef = elementRef;
        this.text = '';
        this.image = '';
        this.width = '300';
        this.htmlContent = '';
        this.originX = 'start';
        this.originY = 'top';
        this.overlayX = 'start';
        this.overlayY = 'bottom';
        this.offsetX = 0;
        this.offsetY = -8;
    }
    ngOnDestroy() {
        this.hide();
    }
    ngOnInit() {
        const positionStrategy = this.overlayPositionBuilder
            .flexibleConnectedTo(this.elementRef)
            .withPositions([{
                originX: this.originX,
                originY: this.originY,
                overlayX: this.overlayX,
                overlayY: this.overlayY,
                offsetY: this.offsetY,
                offsetX: this.offsetX
            }]);
        this.overlayRef = this.overlay.create({ positionStrategy });
    }
    show() {
        const tooltipRef = this.overlayRef.attach(new ComponentPortal(TooltipCardComponent));
        tooltipRef.instance.text = this.text;
        tooltipRef.instance.image = this.image;
        tooltipRef.instance.width = this.width;
        tooltipRef.instance.htmlContent = this.htmlContent;
    }
    hide() {
        this.overlayRef.detach();
    }
}
TooltipCardDirective.ɵfac = function TooltipCardDirective_Factory(t) { return new (t || TooltipCardDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Overlay), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.OverlayPositionBuilder), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
TooltipCardDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: TooltipCardDirective, selectors: [["", "adf-tooltip-card", ""]], hostBindings: function TooltipCardDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("mouseenter", function TooltipCardDirective_mouseenter_HostBindingHandler() { return ctx.show(); })("mouseleave", function TooltipCardDirective_mouseleave_HostBindingHandler() { return ctx.hide(); });
    } }, inputs: { text: ["adf-tooltip-card", "text"], image: "image", width: "width", htmlContent: "htmlContent", originX: "originX", originY: "originY", overlayX: "overlayX", overlayY: "overlayY", offsetX: "offsetX", offsetY: "offsetY" } });
TooltipCardDirective.ctorParameters = () => [
    { type: Overlay },
    { type: OverlayPositionBuilder },
    { type: ElementRef }
];
TooltipCardDirective.propDecorators = {
    text: [{ type: Input, args: ['adf-tooltip-card',] }],
    image: [{ type: Input }],
    width: [{ type: Input }],
    htmlContent: [{ type: Input }],
    originX: [{ type: Input }],
    originY: [{ type: Input }],
    overlayX: [{ type: Input }],
    overlayY: [{ type: Input }],
    offsetX: [{ type: Input }],
    offsetY: [{ type: Input }],
    show: [{ type: HostListener, args: ['mouseenter',] }],
    hide: [{ type: HostListener, args: ['mouseleave',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TooltipCardDirective, [{
        type: Directive,
        args: [{ selector: '[adf-tooltip-card]' }]
    }], function () { return [{ type: ɵngcc1.Overlay }, { type: ɵngcc1.OverlayPositionBuilder }, { type: ɵngcc0.ElementRef }]; }, { text: [{
            type: Input,
            args: ['adf-tooltip-card']
        }], image: [{
            type: Input
        }], width: [{
            type: Input
        }], htmlContent: [{
            type: Input
        }], originX: [{
            type: Input
        }], originY: [{
            type: Input
        }], overlayX: [{
            type: Input
        }], overlayY: [{
            type: Input
        }], offsetX: [{
            type: Input
        }], offsetY: [{
            type: Input
        }], show: [{
            type: HostListener,
            args: ['mouseenter']
        }], hide: [{
            type: HostListener,
            args: ['mouseleave']
        }] }); })();
