// @alfresco/adf-core/esm2015/card-view/models/card-view-selectitem.model.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CardViewBaseItemModel } from './card-view-baseitem.model';
import { of } from 'rxjs';
import { switchMap } from 'rxjs/operators';
export class CardViewSelectItemModel extends CardViewBaseItemModel {
    constructor(cardViewSelectItemProperties) {
        super(cardViewSelectItemProperties);
        this.type = 'select';
        this.valueFetch$ = null;
        this.displayNoneOption = cardViewSelectItemProperties.displayNoneOption !== undefined ? cardViewSelectItemProperties.displayNoneOption : true;
        this.options$ = cardViewSelectItemProperties.options$;
        this.valueFetch$ = this.options$.pipe(switchMap((options) => {
            const option = options.find((o) => { var _a; return o.key === ((_a = this.value) === null || _a === void 0 ? void 0 : _a.toString()); });
            return of(option ? option.label : '');
        }));
    }
    get displayValue() {
        return this.valueFetch$;
    }
    setValue(value) {
        this.value = value;
    }
}
