// @alfresco/adf-core/esm2015/info-drawer/info-drawer.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { InfoDrawerLayoutComponent, InfoDrawerTitleDirective, InfoDrawerButtonsDirective, InfoDrawerContentDirective } from './info-drawer-layout.component';
import { InfoDrawerComponent, InfoDrawerTabComponent } from './info-drawer.component';
import { TranslateModule } from '@ngx-translate/core';
import * as ɵngcc0 from '@angular/core';
export class InfoDrawerModule {
}
InfoDrawerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: InfoDrawerModule });
InfoDrawerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function InfoDrawerModule_Factory(t) { return new (t || InfoDrawerModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            TranslateModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(InfoDrawerModule, { declarations: function () { return [InfoDrawerLayoutComponent,
        InfoDrawerTabComponent,
        InfoDrawerComponent,
        InfoDrawerTitleDirective,
        InfoDrawerButtonsDirective,
        InfoDrawerContentDirective]; }, imports: function () { return [CommonModule,
        MaterialModule,
        TranslateModule]; }, exports: function () { return [InfoDrawerLayoutComponent,
        InfoDrawerTabComponent,
        InfoDrawerComponent,
        InfoDrawerTitleDirective,
        InfoDrawerButtonsDirective,
        InfoDrawerContentDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InfoDrawerModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    TranslateModule
                ],
                declarations: [
                    InfoDrawerLayoutComponent,
                    InfoDrawerTabComponent,
                    InfoDrawerComponent,
                    InfoDrawerTitleDirective,
                    InfoDrawerButtonsDirective,
                    InfoDrawerContentDirective
                ],
                exports: [
                    InfoDrawerLayoutComponent,
                    InfoDrawerTabComponent,
                    InfoDrawerComponent,
                    InfoDrawerTitleDirective,
                    InfoDrawerButtonsDirective,
                    InfoDrawerContentDirective
                ]
            }]
    }], null, null); })();
