// @alfresco/adf-core/esm2015/info-drawer/info-drawer.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ContentChildren, EventEmitter, HostListener, Input, Output, QueryList, TemplateRef, ViewChild, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './info-drawer-layout.component';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/material/tabs';
import * as ɵngcc4 from '@angular/material/icon';
import * as ɵngcc5 from '@ngx-translate/core';

function InfoDrawerTabComponent_ng_template_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0);
} }
const _c0 = ["*"];
function InfoDrawerComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, ctx_r0.title));
} }
function InfoDrawerComponent_ng_content_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0, 1, ["*ngIf", "!title", "info-drawer-title", ""]);
} }
function InfoDrawerComponent_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0, 7);
} }
function InfoDrawerComponent_ng_template_5_mat_tab_1_ng_template_2_mat_icon_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-icon");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const contentBlock_r8 = ɵngcc0.ɵɵnextContext(2).$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(contentBlock_r8.icon);
} }
function InfoDrawerComponent_ng_template_5_mat_tab_1_ng_template_2_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const contentBlock_r8 = ɵngcc0.ɵɵnextContext(2).$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, contentBlock_r8.label));
} }
function InfoDrawerComponent_ng_template_5_mat_tab_1_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, InfoDrawerComponent_ng_template_5_mat_tab_1_ng_template_2_mat_icon_0_Template, 2, 1, "mat-icon", 2);
    ɵngcc0.ɵɵtemplate(1, InfoDrawerComponent_ng_template_5_mat_tab_1_ng_template_2_span_1_Template, 3, 3, "span", 2);
} if (rf & 2) {
    const contentBlock_r8 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("ngIf", contentBlock_r8.icon);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", contentBlock_r8.label);
} }
function InfoDrawerComponent_ng_template_5_mat_tab_1_ng_container_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0);
} }
function InfoDrawerComponent_ng_template_5_mat_tab_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-tab", 10);
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵtemplate(2, InfoDrawerComponent_ng_template_5_mat_tab_1_ng_template_2_Template, 2, 2, "ng-template", 11);
    ɵngcc0.ɵɵtemplate(3, InfoDrawerComponent_ng_template_5_mat_tab_1_ng_container_3_Template, 1, 0, "ng-container", 12);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const contentBlock_r8 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("label", ɵngcc0.ɵɵpipeBind1(1, 2, contentBlock_r8.label));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", contentBlock_r8.content);
} }
function InfoDrawerComponent_ng_template_5_Template(rf, ctx) { if (rf & 1) {
    const _r17 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "mat-tab-group", 8);
    ɵngcc0.ɵɵlistener("selectedIndexChange", function InfoDrawerComponent_ng_template_5_Template_mat_tab_group_selectedIndexChange_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r17); const ctx_r16 = ɵngcc0.ɵɵnextContext(); return ctx_r16.selectedIndex = $event; })("selectedTabChange", function InfoDrawerComponent_ng_template_5_Template_mat_tab_group_selectedTabChange_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r17); const ctx_r18 = ɵngcc0.ɵɵnextContext(); return ctx_r18.onTabChange($event); });
    ɵngcc0.ɵɵtemplate(1, InfoDrawerComponent_ng_template_5_mat_tab_1_Template, 4, 4, "mat-tab", 9);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("selectedIndex", ctx_r4.selectedIndex);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r4.contentBlocks);
} }
function InfoDrawerComponent_ng_template_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0, 2);
} }
const _c1 = [[["", "info-drawer-buttons", ""]], [["", "info-drawer-title", ""]], [["", "info-drawer-content", ""]]];
const _c2 = ["[info-drawer-buttons]", "[info-drawer-title]", "[info-drawer-content]"];
export class InfoDrawerTabComponent {
    constructor() {
        this.label = '';
        this.icon = null;
    }
}
InfoDrawerTabComponent.ɵfac = function InfoDrawerTabComponent_Factory(t) { return new (t || InfoDrawerTabComponent)(); };
InfoDrawerTabComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: InfoDrawerTabComponent, selectors: [["adf-info-drawer-tab"]], viewQuery: function InfoDrawerTabComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(TemplateRef, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.content = _t.first);
    } }, inputs: { label: "label", icon: "icon" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function InfoDrawerTabComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵtemplate(0, InfoDrawerTabComponent_ng_template_0_Template, 1, 0, "ng-template");
    } }, encapsulation: 2 });
InfoDrawerTabComponent.propDecorators = {
    label: [{ type: Input }],
    icon: [{ type: Input }],
    content: [{ type: ViewChild, args: [TemplateRef, { static: true },] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InfoDrawerTabComponent, [{
        type: Component,
        args: [{
                selector: 'adf-info-drawer-tab',
                template: '<ng-template><ng-content></ng-content></ng-template>',
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return []; }, { label: [{
            type: Input
        }], icon: [{
            type: Input
        }], content: [{
            type: ViewChild,
            args: [TemplateRef, { static: true }]
        }] }); })();
export class InfoDrawerComponent {
    constructor() {
        this.title = null;
        this.selectedIndex = 0;
        this.showHeader = true;
        this.currentTab = new EventEmitter();
    }
    onKeyDown(event) {
        event.cancelBubble = true;
    }
    onKeyUp(event) {
        event.cancelBubble = true;
    }
    showTabLayout() {
        return this.contentBlocks.length > 0;
    }
    onTabChange(event) {
        this.currentTab.emit(event.index);
    }
}
InfoDrawerComponent.ɵfac = function InfoDrawerComponent_Factory(t) { return new (t || InfoDrawerComponent)(); };
InfoDrawerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: InfoDrawerComponent, selectors: [["adf-info-drawer"]], contentQueries: function InfoDrawerComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, InfoDrawerTabComponent, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.contentBlocks = _t);
    } }, hostAttrs: [1, "adf-info-drawer"], hostBindings: function InfoDrawerComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("keydown", function InfoDrawerComponent_keydown_HostBindingHandler($event) { return ctx.onKeyDown($event); })("keyup", function InfoDrawerComponent_keyup_HostBindingHandler($event) { return ctx.onKeyUp($event); });
    } }, inputs: { title: "title", selectedIndex: "selectedIndex", showHeader: "showHeader" }, outputs: { currentTab: "currentTab" }, ngContentSelectors: _c2, decls: 9, vars: 6, consts: [[3, "showHeader"], ["role", "heading", "aria-level", "1", "info-drawer-title", "", 4, "ngIf"], [4, "ngIf"], ["info-drawer-content", "", 4, "ngIf", "ngIfThen", "ngIfElse"], ["tabLayout", ""], ["singleLayout", ""], ["role", "heading", "aria-level", "1", "info-drawer-title", ""], ["info-drawer-content", ""], [1, "adf-info-drawer-tabs", 3, "selectedIndex", "selectedIndexChange", "selectedTabChange"], ["class", "adf-info-drawer-tab", 3, "label", 4, "ngFor", "ngForOf"], [1, "adf-info-drawer-tab", 3, "label"], ["mat-tab-label", ""], [4, "ngTemplateOutlet"]], template: function InfoDrawerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c1);
        ɵngcc0.ɵɵelementStart(0, "adf-info-drawer-layout", 0);
        ɵngcc0.ɵɵtemplate(1, InfoDrawerComponent_div_1_Template, 3, 3, "div", 1);
        ɵngcc0.ɵɵtemplate(2, InfoDrawerComponent_ng_content_2_Template, 1, 0, "ng-content", 2);
        ɵngcc0.ɵɵprojection(3, 0, ["info-drawer-buttons", ""]);
        ɵngcc0.ɵɵtemplate(4, InfoDrawerComponent_ng_container_4_Template, 1, 0, "ng-container", 3);
        ɵngcc0.ɵɵtemplate(5, InfoDrawerComponent_ng_template_5_Template, 2, 2, "ng-template", null, 4, ɵngcc0.ɵɵtemplateRefExtractor);
        ɵngcc0.ɵɵtemplate(7, InfoDrawerComponent_ng_template_7_Template, 1, 0, "ng-template", null, 5, ɵngcc0.ɵɵtemplateRefExtractor);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        const _r3 = ɵngcc0.ɵɵreference(6);
        const _r5 = ɵngcc0.ɵɵreference(8);
        ɵngcc0.ɵɵproperty("showHeader", ctx.showHeader);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.title);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.title);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showTabLayout())("ngIfThen", _r3)("ngIfElse", _r5);
    } }, directives: [ɵngcc1.InfoDrawerLayoutComponent, ɵngcc2.NgIf, ɵngcc1.InfoDrawerTitleDirective, ɵngcc1.InfoDrawerContentDirective, ɵngcc3.MatTabGroup, ɵngcc2.NgForOf, ɵngcc3.MatTab, ɵngcc3.MatTabLabel, ɵngcc2.NgTemplateOutlet, ɵngcc4.MatIcon], pipes: [ɵngcc5.TranslatePipe], styles: [".adf-info-drawer{display:block}.adf-info-drawer .mat-tab-label{min-width:0}.adf-info-drawer .adf-info-drawer-layout-content{padding:0}.adf-info-drawer .adf-info-drawer-layout-content>:not(.adf-info-drawer-tabs){padding:10px}.adf-info-drawer .adf-info-drawer-layout-content>:not(.adf-info-drawer-tabs)>*{display:block;margin-bottom:20px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body-content>*{display:block;margin-bottom:20px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body-content>:last-child{margin-bottom:0}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-label{flex-grow:1}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-label .mat-icon+span{padding-left:5px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-ink-bar{height:4px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body{padding:10px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body-content{overflow:initial}"], encapsulation: 2 });
InfoDrawerComponent.propDecorators = {
    title: [{ type: Input }],
    selectedIndex: [{ type: Input }],
    showHeader: [{ type: Input }],
    currentTab: [{ type: Output }],
    contentBlocks: [{ type: ContentChildren, args: [InfoDrawerTabComponent,] }],
    onKeyDown: [{ type: HostListener, args: ['keydown', ['$event'],] }],
    onKeyUp: [{ type: HostListener, args: ['keyup', ['$event'],] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InfoDrawerComponent, [{
        type: Component,
        args: [{
                selector: 'adf-info-drawer',
                template: "<adf-info-drawer-layout [showHeader]=\"showHeader\">\n    <div role=\"heading\" aria-level=\"1\" *ngIf=\"title\" info-drawer-title>{{ title | translate }}</div>\n    <ng-content *ngIf=\"!title\" info-drawer-title select=\"[info-drawer-title]\"></ng-content>\n\n    <ng-content info-drawer-buttons select=\"[info-drawer-buttons]\"></ng-content>\n\n    <ng-container info-drawer-content *ngIf=\"showTabLayout(); then tabLayout else singleLayout\"></ng-container>\n\n    <ng-template #tabLayout>\n        <mat-tab-group [(selectedIndex)]=\"selectedIndex\" class=\"adf-info-drawer-tabs\" (selectedTabChange)=\"onTabChange($event)\">\n            <mat-tab *ngFor=\"let contentBlock of contentBlocks\"\n                [label]=\"contentBlock.label | translate\"\n                class=\"adf-info-drawer-tab\">\n\n                <ng-template mat-tab-label>\n                    <mat-icon *ngIf=\"contentBlock.icon\">{{ contentBlock.icon }}</mat-icon>\n                    <span *ngIf=\"contentBlock.label\">{{ contentBlock.label | translate }}</span>\n                </ng-template>\n\n                <ng-container *ngTemplateOutlet=\"contentBlock.content\"></ng-container>\n            </mat-tab>\n        </mat-tab-group>\n    </ng-template>\n\n    <ng-template #singleLayout>\n        <ng-content select=\"[info-drawer-content]\"></ng-content>\n    </ng-template>\n</adf-info-drawer-layout>\n",
                encapsulation: ViewEncapsulation.None,
                host: { 'class': 'adf-info-drawer' },
                styles: [".adf-info-drawer{display:block}.adf-info-drawer .mat-tab-label{min-width:0}.adf-info-drawer .adf-info-drawer-layout-content{padding:0}.adf-info-drawer .adf-info-drawer-layout-content>:not(.adf-info-drawer-tabs){padding:10px}.adf-info-drawer .adf-info-drawer-layout-content>:not(.adf-info-drawer-tabs)>*{display:block;margin-bottom:20px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body-content>*{display:block;margin-bottom:20px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body-content>:last-child{margin-bottom:0}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-label{flex-grow:1}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-label .mat-icon+span{padding-left:5px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-ink-bar{height:4px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body{padding:10px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body-content{overflow:initial}"]
            }]
    }], function () { return []; }, { title: [{
            type: Input
        }], selectedIndex: [{
            type: Input
        }], showHeader: [{
            type: Input
        }], currentTab: [{
            type: Output
        }], onKeyDown: [{
            type: HostListener,
            args: ['keydown', ['$event']]
        }], onKeyUp: [{
            type: HostListener,
            args: ['keyup', ['$event']]
        }], contentBlocks: [{
            type: ContentChildren,
            args: [InfoDrawerTabComponent]
        }] }); })();
