// @alfresco/adf-core/esm2015/login/directives/login-header.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ContentChild, Directive, TemplateRef } from '@angular/core';
import { LoginComponent } from '../components/login.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../components/login.component';
export class LoginHeaderDirective {
    constructor(alfrescoLoginComponent) {
        this.alfrescoLoginComponent = alfrescoLoginComponent;
    }
    ngAfterContentInit() {
        this.alfrescoLoginComponent.headerTemplate = this.template;
    }
}
LoginHeaderDirective.ɵfac = function LoginHeaderDirective_Factory(t) { return new (t || LoginHeaderDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.LoginComponent)); };
LoginHeaderDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: LoginHeaderDirective, selectors: [["adf-login-header"], ["login-header"]], contentQueries: function LoginHeaderDirective_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, TemplateRef, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.template = _t.first);
    } } });
LoginHeaderDirective.ctorParameters = () => [
    { type: LoginComponent }
];
LoginHeaderDirective.propDecorators = {
    template: [{ type: ContentChild, args: [TemplateRef,] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LoginHeaderDirective, [{
        type: Directive,
        args: [{
                selector: 'adf-login-header, login-header'
            }]
    }], function () { return [{ type: ɵngcc1.LoginComponent }]; }, { template: [{
            type: ContentChild,
            args: [TemplateRef]
        }] }); })();
