// @alfresco/adf-content-services/esm2015/lib/permission-manager/components/node-path-column/node-path-column.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Node } from '@alfresco/js-api';
import { Component, Input } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
export class NodePathColumnComponent {
    constructor() {
        this.displayText$ = new BehaviorSubject('');
    }
    ngOnInit() {
        this.updateValue();
    }
    updateValue() {
        this.displayText$.next(this.node.path.name);
    }
}
NodePathColumnComponent.ɵfac = function NodePathColumnComponent_Factory(t) { return new (t || NodePathColumnComponent)(); };
NodePathColumnComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NodePathColumnComponent, selectors: [["adf-node-path-column"]], hostAttrs: [1, "adf-node-path-column", "adf-datatable-content-cell"], inputs: { node: "node" }, decls: 4, vars: 6, consts: [[1, "adf-user-name-column", "adf-datatable-cell-value", 3, "title"]], template: function NodePathColumnComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "span", 0);
        ɵngcc0.ɵɵpipe(1, "async");
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵpipe(3, "async");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind1(1, 2, ctx.displayText$));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(3, 4, ctx.displayText$), " ");
    } }, pipes: [ɵngcc1.AsyncPipe], encapsulation: 2 });
NodePathColumnComponent.propDecorators = {
    node: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NodePathColumnComponent, [{
        type: Component,
        args: [{
                selector: 'adf-node-path-column',
                template: `
        <span class="adf-user-name-column adf-datatable-cell-value"  title="{{ displayText$ | async }}">
            {{ displayText$ | async }}
        </span>
    `,
                host: { class: 'adf-node-path-column adf-datatable-content-cell' }
            }]
    }], function () { return []; }, { node: [{
            type: Input
        }] }); })();
