// @alfresco/adf-core/esm2015/language-menu/language-picker.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/menu';
import * as ɵngcc2 from '@angular/material/icon';
import * as ɵngcc3 from './language-menu.component';
import * as ɵngcc4 from '@ngx-translate/core';
export class LanguagePickerComponent {
}
LanguagePickerComponent.ɵfac = function LanguagePickerComponent_Factory(t) { return new (t || LanguagePickerComponent)(); };
LanguagePickerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: LanguagePickerComponent, selectors: [["adf-picker-button"]], decls: 8, vars: 4, consts: [["mat-menu-item", "", 3, "matMenuTriggerFor"], ["langMenu", "matMenu"]], template: function LanguagePickerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "button", 0);
        ɵngcc0.ɵɵelementStart(1, "mat-icon");
        ɵngcc0.ɵɵtext(2, "language");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵpipe(4, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "mat-menu", null, 1);
        ɵngcc0.ɵɵelement(7, "adf-language-menu");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        const _r0 = ɵngcc0.ɵɵreference(6);
        ɵngcc0.ɵɵproperty("matMenuTriggerFor", _r0);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(4, 2, "ADF.LANGUAGE"), " ");
    } }, directives: [ɵngcc1.MatMenuItem, ɵngcc1.MatMenuTrigger, ɵngcc2.MatIcon, ɵngcc1._MatMenu, ɵngcc3.LanguageMenuComponent], pipes: [ɵngcc4.TranslatePipe], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LanguagePickerComponent, [{
        type: Component,
        args: [{
                selector: 'adf-picker-button',
                template: `
        <button mat-menu-item [matMenuTriggerFor]="langMenu">
            <mat-icon>language</mat-icon>
            {{ 'ADF.LANGUAGE' | translate }}
        </button>
        <mat-menu #langMenu="matMenu">
            <adf-language-menu></adf-language-menu>
        </mat-menu>
    `
            }]
    }], null, null); })();
