// @alfresco/adf-content-services/esm2015/lib/search/components/search-filter-chips/search-widget-chip/search-widget-chip.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { ConfigurableFocusTrapFactory } from '@angular/cdk/a11y';
import { MatMenuTrigger } from '@angular/material/menu';
import { SearchWidgetContainerComponent } from '../../search-widget-container/search-widget-container.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/cdk/a11y';
import * as ɵngcc2 from '@angular/material/chips';
import * as ɵngcc3 from '@angular/material/menu';
import * as ɵngcc4 from '@angular/common';
import * as ɵngcc5 from '@angular/material/icon';
import * as ɵngcc6 from '../search-filter-menu-card/search-filter-menu-card.component';
import * as ɵngcc7 from '../../search-widget-container/search-widget-container.component';
import * as ɵngcc8 from '@angular/material/button';
import * as ɵngcc9 from '@ngx-translate/core';

const _c0 = ["menuContainer"];
const _c1 = ["menuTrigger"];
function SearchWidgetChipComponent_ng_container_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtext(1, ":");
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SearchWidgetChipComponent_span_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 3);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const displayValue_r7 = ctx.ngIf;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" \u00A0", ɵngcc0.ɵɵpipeBind1(2, 1, displayValue_r7), " ");
} }
function SearchWidgetChipComponent_ng_container_22_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("(", ctx_r5.category.component.settings.unit, ")");
} }
export class SearchWidgetChipComponent {
    constructor(focusTrapFactory) {
        this.focusTrapFactory = focusTrapFactory;
    }
    onMenuOpen() {
        if (this.menuContainer && !this.focusTrap) {
            this.focusTrap = this.focusTrapFactory.create(this.menuContainer.nativeElement);
            this.focusTrap.focusInitialElement();
        }
    }
    onClosed() {
        this.focusTrap.destroy();
        this.focusTrap = null;
    }
    onRemove() {
        this.widgetContainerComponent.resetInnerWidget();
        this.menuTrigger.closeMenu();
    }
    onApply() {
        this.widgetContainerComponent.applyInnerWidget();
        this.menuTrigger.closeMenu();
    }
}
SearchWidgetChipComponent.ɵfac = function SearchWidgetChipComponent_Factory(t) { return new (t || SearchWidgetChipComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ConfigurableFocusTrapFactory)); };
SearchWidgetChipComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SearchWidgetChipComponent, selectors: [["adf-search-widget-chip"]], viewQuery: function SearchWidgetChipComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
        ɵngcc0.ɵɵviewQuery(_c1, true);
        ɵngcc0.ɵɵviewQuery(SearchWidgetContainerComponent, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.menuContainer = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.menuTrigger = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.widgetContainerComponent = _t.first);
    } }, inputs: { category: "category" }, decls: 33, vars: 32, consts: [["disableRipple", "", 1, "adf-search-filter-chip", 3, "matMenuTriggerFor", "onMenuOpen"], ["menuTrigger", "matMenuTrigger"], [1, "adf-search-filter-placeholder"], [1, "adf-search-filter-ellipsis"], [4, "ngIf"], ["class", "adf-search-filter-ellipsis", 4, "ngIf"], ["backdropClass", "adf-search-filter-chip-menu", 3, "closed"], ["menu", "matMenu"], ["menuContainer", ""], [3, "click", "keydown.tab", "close"], ["ngProjectAs", "filter-title", 5, ["filter-title"]], ["ngProjectAs", "filter-content", 5, ["filter-content"]], [3, "id", "selector", "settings"], ["widget", ""], ["ngProjectAs", "filter-actions", 5, ["filter-actions"]], ["mat-flat-button", "", "color", "accent", "id", "cancel-filter-button", 1, "adf-search-action-button", 3, "click"], ["mat-flat-button", "", "color", "primary", "id", "apply-filter-button", 1, "adf-search-action-button", 3, "click"]], template: function SearchWidgetChipComponent_Template(rf, ctx) { if (rf & 1) {
        const _r8 = ɵngcc0.ɵɵgetCurrentView();
        ɵngcc0.ɵɵelementStart(0, "mat-chip", 0, 1);
        ɵngcc0.ɵɵlistener("onMenuOpen", function SearchWidgetChipComponent_Template_mat_chip_onMenuOpen_0_listener() { return ctx.onMenuOpen(); });
        ɵngcc0.ɵɵpipe(2, "async");
        ɵngcc0.ɵɵpipe(3, "async");
        ɵngcc0.ɵɵelementStart(4, "span", 2);
        ɵngcc0.ɵɵelementStart(5, "span", 3);
        ɵngcc0.ɵɵtext(6);
        ɵngcc0.ɵɵpipe(7, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(8, SearchWidgetChipComponent_ng_container_8_Template, 2, 0, "ng-container", 4);
        ɵngcc0.ɵɵpipe(9, "async");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(10, SearchWidgetChipComponent_span_10_Template, 3, 3, "span", 5);
        ɵngcc0.ɵɵpipe(11, "async");
        ɵngcc0.ɵɵelementStart(12, "mat-icon");
        ɵngcc0.ɵɵtext(13, "keyboard_arrow_down");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(14, "mat-menu", 6, 7);
        ɵngcc0.ɵɵlistener("closed", function SearchWidgetChipComponent_Template_mat_menu_closed_14_listener() { return ctx.onClosed(); });
        ɵngcc0.ɵɵelementStart(16, "div", null, 8);
        ɵngcc0.ɵɵelementStart(18, "adf-search-filter-menu-card", 9);
        ɵngcc0.ɵɵlistener("click", function SearchWidgetChipComponent_Template_adf_search_filter_menu_card_click_18_listener($event) { return $event.stopPropagation(); })("keydown.tab", function SearchWidgetChipComponent_Template_adf_search_filter_menu_card_keydown_tab_18_listener($event) { return $event.stopPropagation(); })("close", function SearchWidgetChipComponent_Template_adf_search_filter_menu_card_close_18_listener() { ɵngcc0.ɵɵrestoreView(_r8); const _r0 = ɵngcc0.ɵɵreference(1); return _r0.closeMenu(); });
        ɵngcc0.ɵɵelementContainerStart(19, 10);
        ɵngcc0.ɵɵtext(20);
        ɵngcc0.ɵɵpipe(21, "translate");
        ɵngcc0.ɵɵtemplate(22, SearchWidgetChipComponent_ng_container_22_Template, 2, 1, "ng-container", 4);
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵelementContainerStart(23, 11);
        ɵngcc0.ɵɵelement(24, "adf-search-widget-container", 12, 13);
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵelementContainerStart(26, 14);
        ɵngcc0.ɵɵelementStart(27, "button", 15);
        ɵngcc0.ɵɵlistener("click", function SearchWidgetChipComponent_Template_button_click_27_listener() { return ctx.onRemove(); });
        ɵngcc0.ɵɵtext(28);
        ɵngcc0.ɵɵpipe(29, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(30, "button", 16);
        ɵngcc0.ɵɵlistener("click", function SearchWidgetChipComponent_Template_button_click_30_listener() { return ctx.onApply(); });
        ɵngcc0.ɵɵtext(31);
        ɵngcc0.ɵɵpipe(32, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        const _r0 = ɵngcc0.ɵɵreference(1);
        const _r3 = ɵngcc0.ɵɵreference(15);
        const _r6 = ɵngcc0.ɵɵreference(25);
        ɵngcc0.ɵɵclassProp("adf-search-toggle-chip", ɵngcc0.ɵɵpipeBind1(2, 16, _r6.getDisplayValue()) || _r0.menuOpen);
        ɵngcc0.ɵɵproperty("matMenuTriggerFor", _r3);
        ɵngcc0.ɵɵattribute("data-automation-id", "search-filter-chip-" + ctx.category.name)("title", ɵngcc0.ɵɵpipeBind1(3, 18, _r6.getDisplayValue()));
        ɵngcc0.ɵɵadvance(6);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(7, 20, ctx.category.name));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(9, 22, _r6.getDisplayValue()));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(11, 24, _r6.getDisplayValue()));
        ɵngcc0.ɵɵadvance(6);
        ɵngcc0.ɵɵattribute("data-automation-id", "search-field-" + ctx.category.name);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(21, 26, ctx.category.name), " ");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.category.component.settings.unit);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("id", ctx.category.id)("selector", ctx.category.component.selector)("settings", ctx.category.component.settings);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(29, 28, "SEARCH.FILTER.BUTTONS.REMOVE"), " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(32, 30, "SEARCH.FILTER.BUTTONS.APPLY"), " ");
    } }, directives: [ɵngcc2.MatChip, ɵngcc3.MatMenuTrigger, ɵngcc4.NgIf, ɵngcc5.MatIcon, ɵngcc3._MatMenu, ɵngcc6.SearchFilterMenuCardComponent, ɵngcc7.SearchWidgetContainerComponent, ɵngcc8.MatButton], pipes: [ɵngcc4.AsyncPipe, ɵngcc9.TranslatePipe], encapsulation: 2 });
SearchWidgetChipComponent.ctorParameters = () => [
    { type: ConfigurableFocusTrapFactory }
];
SearchWidgetChipComponent.propDecorators = {
    category: [{ type: Input }],
    menuContainer: [{ type: ViewChild, args: ['menuContainer', { static: false },] }],
    menuTrigger: [{ type: ViewChild, args: ['menuTrigger', { static: false },] }],
    widgetContainerComponent: [{ type: ViewChild, args: [SearchWidgetContainerComponent, { static: false },] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SearchWidgetChipComponent, [{
        type: Component,
        args: [{
                selector: 'adf-search-widget-chip',
                template: "<mat-chip [attr.data-automation-id]=\"'search-filter-chip-' + category.name\"\n          disableRipple\n          class=\"adf-search-filter-chip\"\n          [class.adf-search-toggle-chip]=\"(widget.getDisplayValue() | async) || menuTrigger.menuOpen\"\n          [matMenuTriggerFor]=\"menu\"\n          (onMenuOpen)=\"onMenuOpen()\"\n          [attr.title]=\"widget.getDisplayValue() | async\"\n          #menuTrigger=\"matMenuTrigger\">\n    <span class=\"adf-search-filter-placeholder\">\n        <span class=\"adf-search-filter-ellipsis\">{{ category.name | translate }}</span>\n        <ng-container *ngIf=\"widget.getDisplayValue() | async\">:</ng-container>\n    </span>\n    <span class=\"adf-search-filter-ellipsis\" *ngIf=\"widget.getDisplayValue() | async as displayValue\">\n        &nbsp;{{ displayValue | translate }}\n    </span>\n    <mat-icon>keyboard_arrow_down</mat-icon>\n</mat-chip>\n\n<mat-menu #menu=\"matMenu\" backdropClass=\"adf-search-filter-chip-menu\" (closed)=\"onClosed()\">\n    <div #menuContainer [attr.data-automation-id]=\"'search-field-' + category.name\">\n        <adf-search-filter-menu-card (click)=\"$event.stopPropagation()\"\n                                     (keydown.tab)=\"$event.stopPropagation();\"\n                                     (close)=\"menuTrigger.closeMenu()\">\n\n            <ng-container ngProjectAs=\"filter-title\">\n                {{ category.name | translate }} <ng-container *ngIf=\"category.component.settings.unit\">({{category.component.settings.unit}})</ng-container>\n            </ng-container>\n\n            <ng-container ngProjectAs=\"filter-content\">\n                <adf-search-widget-container #widget\n                                             [id]=\"category.id\"\n                                             [selector]=\"category.component.selector\"\n                                             [settings]=\"category.component.settings\">\n                </adf-search-widget-container>\n            </ng-container>\n\n            <ng-container ngProjectAs=\"filter-actions\">\n                <button mat-flat-button class=\"adf-search-action-button\" color=\"accent\" (click)=\"onRemove()\" id=\"cancel-filter-button\">\n                    {{ 'SEARCH.FILTER.BUTTONS.REMOVE' | translate }}\n                </button>\n                <button mat-flat-button class=\"adf-search-action-button\" color=\"primary\" (click)=\"onApply()\" id=\"apply-filter-button\">\n                    {{ 'SEARCH.FILTER.BUTTONS.APPLY' | translate }}\n                </button>\n            </ng-container>\n        </adf-search-filter-menu-card>\n    </div>\n</mat-menu>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.ConfigurableFocusTrapFactory }]; }, { category: [{
            type: Input
        }], menuContainer: [{
            type: ViewChild,
            args: ['menuContainer', { static: false }]
        }], menuTrigger: [{
            type: ViewChild,
            args: ['menuTrigger', { static: false }]
        }], widgetContainerComponent: [{
            type: ViewChild,
            args: [SearchWidgetContainerComponent, { static: false }]
        }] }); })();
