// @alfresco/adf-core/esm2015/about/about-github-link/about-github-link.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/card';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@ngx-translate/core';

function AboutGithubLinkComponent_p_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "p", 5);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate2("", ɵngcc0.ɵɵpipeBind1(2, 2, "ABOUT.VERSION"), ": ", ctx_r0.version, "");
} }
function AboutGithubLinkComponent_div_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelementStart(1, "small");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "div", 6);
    ɵngcc0.ɵɵelementStart(5, "a", 7);
    ɵngcc0.ɵɵtext(6);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 3, "ABOUT.SOURCE_CODE.DESCRIPTION"));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("href", ctx_r1.url, ɵngcc0.ɵɵsanitizeUrl);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.url);
} }
export class AboutGithubLinkComponent {
    constructor() {
        this.url = 'https://github.com/Alfresco/alfresco-ng2-components/commits/';
        this.version = '';
        this.application = '';
    }
}
AboutGithubLinkComponent.ɵfac = function AboutGithubLinkComponent_Factory(t) { return new (t || AboutGithubLinkComponent)(); };
AboutGithubLinkComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AboutGithubLinkComponent, selectors: [["adf-about-github-link"]], inputs: { url: "url", version: "version", application: "application" }, decls: 10, vars: 6, consts: [[1, "adf-github-link-container"], ["data-automation-id", "adf-github-source-code-title"], ["data-automation-id", "adf-github-app-title"], ["data-automation-id", "adf-github-version", 4, "ngIf"], [4, "ngIf"], ["data-automation-id", "adf-github-version"], ["data-automation-id", "adf-github-url"], ["target", "_blank", 3, "href"]], template: function AboutGithubLinkComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div");
        ɵngcc0.ɵɵelementStart(2, "h3", 1);
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵpipe(4, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "mat-card");
        ɵngcc0.ɵɵelementStart(6, "h3", 2);
        ɵngcc0.ɵɵtext(7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(8, AboutGithubLinkComponent_p_8_Template, 3, 4, "p", 3);
        ɵngcc0.ɵɵtemplate(9, AboutGithubLinkComponent_div_9_Template, 7, 5, "div", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(4, 4, "ABOUT.SOURCE_CODE.TITLE"));
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate(ctx.application);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.version);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.url);
    } }, directives: [ɵngcc1.MatCard, ɵngcc2.NgIf], pipes: [ɵngcc3.TranslatePipe], encapsulation: 2 });
AboutGithubLinkComponent.ctorParameters = () => [];
AboutGithubLinkComponent.propDecorators = {
    url: [{ type: Input }],
    version: [{ type: Input }],
    application: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AboutGithubLinkComponent, [{
        type: Component,
        args: [{
                selector: 'adf-about-github-link',
                template: "<div class=\"adf-github-link-container\">\n    <div>\n        <h3 data-automation-id=\"adf-github-source-code-title\">{{ 'ABOUT.SOURCE_CODE.TITLE' | translate }}</h3>\n        <mat-card>\n            <h3 data-automation-id=\"adf-github-app-title\">{{application}}</h3>\n            <p *ngIf=\"version\" data-automation-id=\"adf-github-version\">{{ 'ABOUT.VERSION' | translate }}: {{ version }}</p>\n\n            <div *ngIf=\"url\">\n                <small>{{ 'ABOUT.SOURCE_CODE.DESCRIPTION' | translate }}</small>\n                <div data-automation-id=\"adf-github-url\">\n                    <a [href]=\"url\" target=\"_blank\">{{url}}</a>\n                </div>\n            </div>\n        </mat-card>\n    </div>\n</div>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return []; }, { url: [{
            type: Input
        }], version: [{
            type: Input
        }], application: [{
            type: Input
        }] }); })();
