// @alfresco/adf-process-services/esm2015/lib/people/components/people-list/people-list.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { DataTableComponent, DataColumnListComponent, ShowHeaderMode } from '@alfresco/adf-core';
import { Component, ContentChild, EventEmitter, Input, Output, ViewChild } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';

const _c0 = ["dataTable"];
export class PeopleListComponent {
    constructor() {
        this.actions = false;
        this.clickRow = new EventEmitter();
        this.clickAction = new EventEmitter();
        this.showHeader = ShowHeaderMode.Never;
    }
    ngAfterContentInit() {
        this.peopleDataTable.columnList = this.columnList;
    }
    selectUser(event) {
        this.user = event.value.obj;
        this.clickRow.emit(this.user);
    }
    hasActions() {
        return this.actions;
    }
    onShowRowActionsMenu(event) {
        const removeAction = {
            title: 'Remove',
            name: 'remove'
        };
        event.value.actions = [
            removeAction
        ];
    }
    onExecuteRowAction(event) {
        const args = event.value;
        const action = args.action;
        this.clickAction.emit({ type: action.name, value: args.row.obj });
    }
}
PeopleListComponent.ɵfac = function PeopleListComponent_Factory(t) { return new (t || PeopleListComponent)(); };
PeopleListComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PeopleListComponent, selectors: [["adf-people-list"]], contentQueries: function PeopleListComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, DataColumnListComponent, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.columnList = _t.first);
    } }, viewQuery: function PeopleListComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.peopleDataTable = _t.first);
    } }, inputs: { actions: "actions", users: "users" }, outputs: { clickRow: "clickRow", clickAction: "clickAction" }, decls: 2, vars: 3, consts: [[1, "adf-task-people-list", 3, "rows", "actions", "showHeader", "rowClick", "rowDblClick", "showRowActionsMenu", "executeRowAction"], ["dataTable", ""]], template: function PeopleListComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "adf-datatable", 0, 1);
        ɵngcc0.ɵɵlistener("rowClick", function PeopleListComponent_Template_adf_datatable_rowClick_0_listener($event) { return ctx.selectUser($event); })("rowDblClick", function PeopleListComponent_Template_adf_datatable_rowDblClick_0_listener($event) { return ctx.selectUser($event); })("showRowActionsMenu", function PeopleListComponent_Template_adf_datatable_showRowActionsMenu_0_listener($event) { return ctx.onShowRowActionsMenu($event); })("executeRowAction", function PeopleListComponent_Template_adf_datatable_executeRowAction_0_listener($event) { return ctx.onExecuteRowAction($event); });
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("rows", ctx.users)("actions", ctx.hasActions())("showHeader", ctx.showHeader);
    } }, directives: [ɵngcc1.DataTableComponent], styles: [".adf-task-people-list[_ngcontent-%COMP%]   .adf-datatable-list[_ngcontent-%COMP%]{width:100%}"] });
PeopleListComponent.propDecorators = {
    columnList: [{ type: ContentChild, args: [DataColumnListComponent,] }],
    peopleDataTable: [{ type: ViewChild, args: ['dataTable', { static: true },] }],
    users: [{ type: Input }],
    actions: [{ type: Input }],
    clickRow: [{ type: Output }],
    clickAction: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PeopleListComponent, [{
        type: Component,
        args: [{
                selector: 'adf-people-list',
                template: "<adf-datatable\n    #dataTable\n    class=\"adf-task-people-list\"\n    [rows]=\"users\"\n    [actions]=\"hasActions()\"\n    [showHeader]=\"showHeader\"\n    (rowClick)=\"selectUser($event)\"\n    (rowDblClick)=\"selectUser($event)\"\n    (showRowActionsMenu)=\"onShowRowActionsMenu($event)\"\n    (executeRowAction)=\"onExecuteRowAction($event)\">\n</adf-datatable>\n",
                styles: [".adf-task-people-list .adf-datatable-list{width:100%}"]
            }]
    }], function () { return []; }, { actions: [{
            type: Input
        }], clickRow: [{
            type: Output
        }], clickAction: [{
            type: Output
        }], columnList: [{
            type: ContentChild,
            args: [DataColumnListComponent]
        }], peopleDataTable: [{
            type: ViewChild,
            args: ['dataTable', { static: true }]
        }], users: [{
            type: Input
        }] }); })();
