// @alfresco/adf-content-services/esm2015/lib/search/components/empty-search-result.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class EmptySearchResultComponent {
}
EmptySearchResultComponent.ɵfac = function EmptySearchResultComponent_Factory(t) { return new (t || EmptySearchResultComponent)(); };
EmptySearchResultComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: EmptySearchResultComponent, selectors: [["adf-empty-search-result"]], ngContentSelectors: _c0, decls: 2, vars: 0, consts: [[1, "adf-empty-search-result"]], template: function EmptySearchResultComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EmptySearchResultComponent, [{
        type: Component,
        args: [{
                selector: 'adf-empty-search-result',
                template: "<div class=\"adf-empty-search-result\">\n    <ng-content></ng-content>\n</div>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], null, null); })();
