// @alfresco/adf-content-services/esm2015/lib/search/components/reset-search.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, HostListener } from '@angular/core';
import { SearchFacetFiltersService } from '../services/search-facet-filters.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/search-facet-filters.service';
export class ResetSearchDirective {
    constructor(filterService) {
        this.filterService = filterService;
    }
    onClick() {
        this.filterService.reset();
    }
}
ResetSearchDirective.ɵfac = function ResetSearchDirective_Factory(t) { return new (t || ResetSearchDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.SearchFacetFiltersService)); };
ResetSearchDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ResetSearchDirective, selectors: [["", "adf-reset-search", ""]], hostBindings: function ResetSearchDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function ResetSearchDirective_click_HostBindingHandler() { return ctx.onClick(); });
    } } });
ResetSearchDirective.ctorParameters = () => [
    { type: SearchFacetFiltersService }
];
ResetSearchDirective.propDecorators = {
    onClick: [{ type: HostListener, args: ['click',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ResetSearchDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-reset-search]'
            }]
    }], function () { return [{ type: ɵngcc1.SearchFacetFiltersService }]; }, { onClick: [{
            type: HostListener,
            args: ['click']
        }] }); })();
