// @alfresco/adf-content-services/esm2015/lib/tree-view/components/tree-view.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { FlatTreeControl } from '@angular/cdk/tree';
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { TreeViewDataSource } from '../data/tree-view-datasource';
import { TreeViewService } from '../services/tree-view.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/tree-view.service';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/material/tree';
import * as ɵngcc4 from '@angular/material/button';
import * as ɵngcc5 from '@angular/material/icon';
import * as ɵngcc6 from '@ngx-translate/core';

function TreeViewComponent_mat_tree_0_mat_tree_node_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-tree-node", 5);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const treeNode_r5 = ctx.$implicit;
    ɵngcc0.ɵɵpropertyInterpolate("id", treeNode_r5.name + "-tree-node");
    ɵngcc0.ɵɵproperty("matTreeNodePaddingIndent", 15);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", treeNode_r5.name, " ");
} }
const _c0 = function (a0) { return { name: a0 }; };
function TreeViewComponent_mat_tree_0_mat_tree_node_2_Template(rf, ctx) { if (rf & 1) {
    const _r8 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "mat-tree-node", 5);
    ɵngcc0.ɵɵelementStart(1, "button", 6);
    ɵngcc0.ɵɵlistener("click", function TreeViewComponent_mat_tree_0_mat_tree_node_2_Template_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r8); const treeNode_r6 = ctx.$implicit; const ctx_r7 = ɵngcc0.ɵɵnextContext(2); return ctx_r7.onNodeClicked(treeNode_r6.node); });
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementStart(3, "mat-icon", 7);
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(5, "span", 8);
    ɵngcc0.ɵɵlistener("click", function TreeViewComponent_mat_tree_0_mat_tree_node_2_Template_span_click_5_listener() { ɵngcc0.ɵɵrestoreView(_r8); const treeNode_r6 = ctx.$implicit; const ctx_r9 = ɵngcc0.ɵɵnextContext(2); return ctx_r9.onNodeClicked(treeNode_r6.node); });
    ɵngcc0.ɵɵtext(6);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const treeNode_r6 = ctx.$implicit;
    const ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵpropertyInterpolate("id", treeNode_r6.name + "-tree-child-node");
    ɵngcc0.ɵɵproperty("matTreeNodePaddingIndent", 15);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("id", "button-" + treeNode_r6.name);
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind2(2, 6, "ADF-TREE-VIEW.ACCESSIBILITY.ARIA_LABEL", ɵngcc0.ɵɵpureFunction1(9, _c0, treeNode_r6.name)));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r4.treeControl.isExpanded(treeNode_r6) ? "folder_open" : "folder", " ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(treeNode_r6.name);
} }
function TreeViewComponent_mat_tree_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-tree", 2);
    ɵngcc0.ɵɵtemplate(1, TreeViewComponent_mat_tree_0_mat_tree_node_1_Template, 2, 3, "mat-tree-node", 3);
    ɵngcc0.ɵɵtemplate(2, TreeViewComponent_mat_tree_0_mat_tree_node_2_Template, 7, 11, "mat-tree-node", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("dataSource", ctx_r0.dataSource)("treeControl", ctx_r0.treeControl);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("matTreeNodeDefWhen", ctx_r0.hasChild);
} }
function TreeViewComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 9);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, "ADF-TREE-VIEW.MISSING-ID"), " ");
} }
export class TreeViewComponent {
    constructor(treeViewService) {
        this.treeViewService = treeViewService;
        this.nodeClicked = new EventEmitter();
        this.error = new EventEmitter();
        this.getLevel = (node) => node.level;
        this.isExpandable = (node) => node.expandable;
        this.hasChild = (_, nodeData) => nodeData.expandable;
        this.treeControl = new FlatTreeControl(this.getLevel, this.isExpandable);
        this.dataSource = new TreeViewDataSource(this.treeControl, this.treeViewService);
    }
    ngOnChanges(changes) {
        if (changes['nodeId'] && changes['nodeId'].currentValue &&
            changes['nodeId'].currentValue !== changes['nodeId'].previousValue) {
            this.loadTreeNode();
        }
        else {
            this.dataSource.data = [];
        }
    }
    onNodeClicked(node) {
        this.nodeClicked.emit(node);
    }
    loadTreeNode() {
        this.treeViewService.getTreeNodes(this.nodeId)
            .subscribe((treeNode) => {
            this.dataSource.data = treeNode;
        }, (error) => this.error.emit(error));
    }
}
TreeViewComponent.ɵfac = function TreeViewComponent_Factory(t) { return new (t || TreeViewComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TreeViewService)); };
TreeViewComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TreeViewComponent, selectors: [["adf-tree-view-list"]], inputs: { nodeId: "nodeId" }, outputs: { nodeClicked: "nodeClicked", error: "error" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 3, vars: 2, consts: [["class", "adf-tree-view-main", 3, "dataSource", "treeControl", 4, "ngIf", "ngIfElse"], ["missingNodeId", ""], [1, "adf-tree-view-main", 3, "dataSource", "treeControl"], ["class", "adf-tree-view-node", "matTreeNodePadding", "", 3, "id", "matTreeNodePaddingIndent", 4, "matTreeNodeDef"], ["class", "adf-tree-view-node", "matTreeNodePadding", "", 3, "id", "matTreeNodePaddingIndent", 4, "matTreeNodeDef", "matTreeNodeDefWhen"], ["matTreeNodePadding", "", 1, "adf-tree-view-node", 3, "id", "matTreeNodePaddingIndent"], ["matTreeNodeToggle", "", "mat-icon-button", "", 3, "id", "click"], [1, "adf-tree-view-icon"], ["matTreeNodeToggle", "", 1, "adf-tree-view-label", 3, "click"], ["id", "adf-tree-view-missing-node"]], template: function TreeViewComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, TreeViewComponent_mat_tree_0_Template, 3, 3, "mat-tree", 0);
        ɵngcc0.ɵɵtemplate(1, TreeViewComponent_ng_template_1_Template, 3, 3, "ng-template", null, 1, ɵngcc0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        const _r1 = ɵngcc0.ɵɵreference(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.nodeId)("ngIfElse", _r1);
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.MatTree, ɵngcc3.MatTreeNodeDef, ɵngcc3.MatTreeNode, ɵngcc3.MatTreeNodePadding, ɵngcc4.MatButton, ɵngcc3.MatTreeNodeToggle, ɵngcc5.MatIcon], pipes: [ɵngcc6.TranslatePipe], styles: [".adf-tree-view-icon[_ngcontent-%COMP%]{color:#d9e022}.adf-tree-view-node.mat-tree-node[_ngcontent-%COMP%]{align-items:baseline;font-size:12px;min-height:40px}.adf-tree-view-label[_ngcontent-%COMP%]{cursor:pointer}"] });
TreeViewComponent.ctorParameters = () => [
    { type: TreeViewService }
];
TreeViewComponent.propDecorators = {
    nodeId: [{ type: Input }],
    nodeClicked: [{ type: Output }],
    error: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TreeViewComponent, [{
        type: Component,
        args: [{
                selector: 'adf-tree-view-list',
                template: "<mat-tree class=\"adf-tree-view-main\" [dataSource]=\"dataSource\"\n          [treeControl]=\"treeControl\" *ngIf=\"nodeId; else missingNodeId\">\n    <mat-tree-node class=\"adf-tree-view-node\"\n        *matTreeNodeDef=\"let treeNode\" id=\"{{treeNode.name + '-tree-node'}}\"\n        matTreeNodePadding [matTreeNodePaddingIndent]=\"15\">\n        {{treeNode.name}}\n    </mat-tree-node>\n    <mat-tree-node class=\"adf-tree-view-node\"\n        id=\"{{treeNode.name + '-tree-child-node'}}\" *matTreeNodeDef=\"let treeNode; when: hasChild\"\n        matTreeNodePadding [matTreeNodePaddingIndent]=\"15\">\n        <button matTreeNodeToggle id=\"{{'button-'+treeNode.name}}\" (click)=\"onNodeClicked(treeNode.node)\"\n                mat-icon-button [attr.aria-label]=\"'ADF-TREE-VIEW.ACCESSIBILITY.ARIA_LABEL' | translate: {\n                    name:  treeNode.name\n                }\">\n            <mat-icon class=\"adf-tree-view-icon\">\n                {{treeControl.isExpanded(treeNode) ? 'folder_open' : 'folder'}}\n            </mat-icon>\n        </button>\n        <span class=\"adf-tree-view-label\" matTreeNodeToggle (click)=\"onNodeClicked(treeNode.node)\">{{treeNode.name}}</span>\n    </mat-tree-node>\n</mat-tree>\n<ng-template #missingNodeId>\n    <div id=\"adf-tree-view-missing-node\">\n        {{'ADF-TREE-VIEW.MISSING-ID' | translate}}\n    </div>\n</ng-template>\n",
                styles: [".adf-tree-view-icon{color:#d9e022}.adf-tree-view-node.mat-tree-node{align-items:baseline;font-size:12px;min-height:40px}.adf-tree-view-label{cursor:pointer}"]
            }]
    }], function () { return [{ type: ɵngcc1.TreeViewService }]; }, { nodeClicked: [{
            type: Output
        }], error: [{
            type: Output
        }], nodeId: [{
            type: Input
        }] }); })();
