// @alfresco/adf-content-services/esm2015/lib/permission-manager/components/permission-container/permission-container.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { Node } from '@alfresco/js-api';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../user-icon-column/user-icon-column.component';
import * as ɵngcc4 from '../user-name-column/user-name-column.component';
import * as ɵngcc5 from '../node-path-column/node-path-column.component';
import * as ɵngcc6 from '../user-role-column/user-role-column.component';
import * as ɵngcc7 from '@angular/material/button';
import * as ɵngcc8 from '@angular/material/icon';
import * as ɵngcc9 from '@ngx-translate/core';

function PermissionContainerComponent_ng_template_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-user-icon-column", 7);
} if (rf & 2) {
    const context_r7 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("context", context_r7);
} }
function PermissionContainerComponent_ng_template_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-user-name-column", 7);
} if (rf & 2) {
    const context_r8 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("context", context_r8);
} }
function PermissionContainerComponent_data_column_8_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-node-path-column", 9);
} if (rf & 2) {
    const ctx_r9 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("node", ctx_r9.node);
} }
function PermissionContainerComponent_data_column_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "data-column", 8);
    ɵngcc0.ɵɵtemplate(1, PermissionContainerComponent_data_column_8_ng_template_1_Template, 1, 1, "ng-template");
    ɵngcc0.ɵɵelementEnd();
} }
function PermissionContainerComponent_ng_template_10_Template(rf, ctx) { if (rf & 1) {
    const _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "adf-user-role-column", 10);
    ɵngcc0.ɵɵlistener("roleChanged", function PermissionContainerComponent_ng_template_10_Template_adf_user_role_column_roleChanged_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r12); const entry_r10 = ctx.$implicit; const ctx_r11 = ɵngcc0.ɵɵnextContext(); return ctx_r11.updateRole($event, entry_r10.row.obj); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const entry_r10 = ctx.$implicit;
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("readonly", entry_r10.row.obj.readonly || ctx_r3.isReadOnly)("placeholder", entry_r10.data.getValue(entry_r10.row, entry_r10.col))("value", entry_r10.data.getValue(entry_r10.row, entry_r10.col))("roles", ctx_r3.roles);
} }
function PermissionContainerComponent_adf_data_column_header_11_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    const _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "adf-user-role-column", 11);
    ɵngcc0.ɵɵlistener("roleChanged", function PermissionContainerComponent_adf_data_column_header_11_ng_template_1_Template_adf_user_role_column_roleChanged_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r15); const ctx_r14 = ɵngcc0.ɵɵnextContext(2); return ctx_r14.bulkRoleUpdate($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r13 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("roles", ctx_r13.roles)("value", ctx_r13.bulkSelectionRole);
} }
function PermissionContainerComponent_adf_data_column_header_11_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "adf-data-column-header");
    ɵngcc0.ɵɵtemplate(1, PermissionContainerComponent_adf_data_column_header_11_ng_template_1_Template, 1, 2, "ng-template");
    ɵngcc0.ɵɵelementEnd();
} }
function PermissionContainerComponent_data_column_12_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    const _r19 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 13);
    ɵngcc0.ɵɵlistener("click", function PermissionContainerComponent_data_column_12_ng_template_1_Template_button_click_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r19); const entry_r17 = ctx.$implicit; const ctx_r18 = ɵngcc0.ɵɵnextContext(2); return ctx_r18.removePermission($event, entry_r17.row.obj); });
    ɵngcc0.ɵɵelementStart(1, "mat-icon");
    ɵngcc0.ɵɵtext(2, "delete_outline");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const entry_r17 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("disabled", entry_r17.row.obj.readonly);
    ɵngcc0.ɵɵattribute("data-automation-id", "adf-delete-permission-button-" + entry_r17.row.obj.authorityId);
} }
function PermissionContainerComponent_data_column_12_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "data-column", 12);
    ɵngcc0.ɵɵtemplate(1, PermissionContainerComponent_data_column_12_ng_template_1_Template, 3, 2, "ng-template");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵproperty("sortable", false);
} }
function PermissionContainerComponent_ng_template_14_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-empty-content", 14);
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵpipe(2, "translate");
} if (rf & 2) {
    ɵngcc0.ɵɵproperty("title", ɵngcc0.ɵɵpipeBind1(1, 2, "PERMISSION_MANAGER.MESSAGE.EMPTY-PERMISSION"))("subtitle", ɵngcc0.ɵɵpipeBind1(2, 4, "PERMISSION_MANAGER.MESSAGE.EMPTY-SUBTITLE"));
} }
const _c0 = function (a0) { return { count: a0 }; };
export class PermissionContainerComponent {
    constructor() {
        this.permissions = [];
        this.isReadOnly = false;
        this.showLocation = false;
        this.selectionMode = 'multiple';
        this.update = new EventEmitter();
        this.updateAll = new EventEmitter();
        this.delete = new EventEmitter();
        this.error = new EventEmitter();
    }
    ngOnChanges() {
        this.bulkSelectionRole = '';
    }
    updateRole(role, permission) {
        this.update.emit({ role, permission });
    }
    bulkRoleUpdate(role) {
        this.updateAll.emit(role);
    }
    removePermission(event, permissionRow) {
        event.stopPropagation();
        this.delete.emit(permissionRow);
    }
}
PermissionContainerComponent.ɵfac = function PermissionContainerComponent_Factory(t) { return new (t || PermissionContainerComponent)(); };
PermissionContainerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PermissionContainerComponent, selectors: [["adf-permission-container"]], inputs: { permissions: "permissions", isReadOnly: "isReadOnly", showLocation: "showLocation", selectionMode: "selectionMode", node: "node", roles: "roles" }, outputs: { update: "update", updateAll: "updateAll", delete: "delete", error: "error" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 15, vars: 14, consts: [["id", "adf-permission-display-container", 1, "adf-datatable-permission", 3, "rows", "stickyHeader", "selectionMode"], ["key", "$thumbunail", 1, "adf-datatable-cell--image", "adf-authority-icon-column", 3, "sortable"], ["key", "authorityId", 1, "adf-ellipsis-cell", "adf-expand-cell-5", "adf-authorityId-column", 3, "title"], ["class", "adf-ellipsis-cell adf-expand-cell-5 adf-authorityId-column", "title", "PERMISSION_MANAGER.COLUMN.LOCATION", "key", "location", 4, "ngIf"], ["title", "PERMISSION_MANAGER.PERMISSION_DISPLAY.ROLE", "key", "name", 1, "adf-ellipsis-cell", "adf-expand-cell-4", 3, "sortable"], [4, "ngIf"], ["class", "adf-datatable-cell adf-delete-permission-column", "key", "", 3, "sortable", 4, "ngIf"], [3, "context"], ["title", "PERMISSION_MANAGER.COLUMN.LOCATION", "key", "location", 1, "adf-ellipsis-cell", "adf-expand-cell-5", "adf-authorityId-column"], [3, "node"], ["id", "adf-select-role-permission", 3, "readonly", "placeholder", "value", "roles", "roleChanged"], ["placeholder", "PERMISSION_MANAGER.COLUMN.BULK-ROLE", "id", "adf-bulk-select-role-permission", 1, "adf-permission-role-column-header", 3, "roles", "value", "roleChanged"], ["key", "", 1, "adf-datatable-cell", "adf-delete-permission-column", 3, "sortable"], ["mat-icon-button", "", 3, "disabled", "click"], ["id", "adf-no-permissions-template", "icon", "supervisor_account", 3, "title", "subtitle"]], template: function PermissionContainerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "adf-datatable", 0);
        ɵngcc0.ɵɵelementStart(1, "data-columns");
        ɵngcc0.ɵɵelementStart(2, "data-column", 1);
        ɵngcc0.ɵɵtemplate(3, PermissionContainerComponent_ng_template_3_Template, 1, 1, "ng-template");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "data-column", 2);
        ɵngcc0.ɵɵpipe(5, "translate");
        ɵngcc0.ɵɵtemplate(6, PermissionContainerComponent_ng_template_6_Template, 1, 1, "ng-template");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtext(7, "d ");
        ɵngcc0.ɵɵtemplate(8, PermissionContainerComponent_data_column_8_Template, 2, 0, "data-column", 3);
        ɵngcc0.ɵɵelementStart(9, "data-column", 4);
        ɵngcc0.ɵɵtemplate(10, PermissionContainerComponent_ng_template_10_Template, 1, 4, "ng-template");
        ɵngcc0.ɵɵtemplate(11, PermissionContainerComponent_adf_data_column_header_11_Template, 2, 0, "adf-data-column-header", 5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(12, PermissionContainerComponent_data_column_12_Template, 2, 1, "data-column", 6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(13, "adf-no-content-template");
        ɵngcc0.ɵɵtemplate(14, PermissionContainerComponent_ng_template_14_Template, 3, 6, "ng-template");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("rows", ctx.permissions)("stickyHeader", true)("selectionMode", ctx.selectionMode);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("sortable", false);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("title", ɵngcc0.ɵɵpipeBind2(5, 9, "PERMISSION_MANAGER.COLUMN.NAME", ɵngcc0.ɵɵpureFunction1(12, _c0, ctx.permissions.length)));
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("ngIf", ctx.node && ctx.showLocation);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("sortable", false);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.isReadOnly);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.isReadOnly);
    } }, directives: [ɵngcc1.DataTableComponent, ɵngcc1.DataColumnListComponent, ɵngcc1.DataColumnComponent, ɵngcc2.NgIf, ɵngcc1.NoContentTemplateDirective, ɵngcc3.UserIconColumnComponent, ɵngcc4.UserNameColumnComponent, ɵngcc5.NodePathColumnComponent, ɵngcc6.UserRoleColumnComponent, ɵngcc1.DateColumnHeaderComponent, ɵngcc7.MatButton, ɵngcc8.MatIcon, ɵngcc1.EmptyContentComponent], pipes: [ɵngcc9.TranslatePipe], styles: [".adf-datatable-permission{display:flex;min-width:450px;width:100%}.adf-datatable-permission.adf-datatable{overflow:hidden}.adf-datatable-permission.adf-datatable .adf-delete-permission-column{min-width:80px}.adf-datatable-permission.adf-datatable .adf-delete-permission-column .adf-cell-value{padding-right:10px;place-content:flex-end;width:80px}.adf-datatable-permission.adf-datatable .adf-authorityId-column{flex:40%}.adf-datatable-permission.adf-datatable .adf-authority-icon-column{min-width:40px}.adf-datatable-permission.adf-datatable .adf-datatable-selected>svg{height:40px;width:40px}"], encapsulation: 2 });
PermissionContainerComponent.propDecorators = {
    node: [{ type: Input }],
    permissions: [{ type: Input }],
    roles: [{ type: Input }],
    isReadOnly: [{ type: Input }],
    showLocation: [{ type: Input }],
    selectionMode: [{ type: Input }],
    update: [{ type: Output }],
    updateAll: [{ type: Output }],
    delete: [{ type: Output }],
    error: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PermissionContainerComponent, [{
        type: Component,
        args: [{
                selector: 'adf-permission-container',
                template: "<adf-datatable class=\"adf-datatable-permission\"\n               id=\"adf-permission-display-container\"\n               [rows]=\"permissions\"\n               [stickyHeader]=\"true\"\n               [selectionMode]=\"selectionMode\">\n\n    <data-columns>\n        <data-column class=\"adf-datatable-cell--image adf-authority-icon-column\" key=\"$thumbunail\" [sortable]=\"false\">\n            <ng-template let-context>\n                <adf-user-icon-column [context]=\"context\"></adf-user-icon-column>\n            </ng-template>\n        </data-column>\n\n        <data-column class=\"adf-ellipsis-cell adf-expand-cell-5 adf-authorityId-column\"\n                     [title]=\"'PERMISSION_MANAGER.COLUMN.NAME' | translate:{count:permissions.length}\"\n                     key=\"authorityId\">\n            <ng-template let-context>\n                <adf-user-name-column [context]=\"context\"></adf-user-name-column>\n            </ng-template>\n        </data-column>d\n\n        <data-column class=\"adf-ellipsis-cell adf-expand-cell-5 adf-authorityId-column\"\n                     title=\"PERMISSION_MANAGER.COLUMN.LOCATION\"\n                     key=\"location\"\n                     *ngIf=\"node && showLocation\">\n            <ng-template>\n                <adf-node-path-column [node]=\"node\"></adf-node-path-column>\n            </ng-template>\n        </data-column>\n\n        <data-column\n            class=\"adf-ellipsis-cell adf-expand-cell-4\"\n            title=\"PERMISSION_MANAGER.PERMISSION_DISPLAY.ROLE\"\n            key=\"name\"\n            [sortable]=\"false\">\n            <ng-template let-entry=\"$implicit\">\n                <adf-user-role-column [readonly]=\"entry.row.obj.readonly || isReadOnly\"\n                                      [placeholder]=\"entry.data.getValue(entry.row, entry.col)\"\n                                      [value]=\"entry.data.getValue(entry.row, entry.col)\"\n                                      [roles]=\"roles\"\n                                      id=\"adf-select-role-permission\"\n                                      (roleChanged)=\"updateRole($event, entry.row.obj)\">\n                </adf-user-role-column>\n            </ng-template>\n\n            <adf-data-column-header *ngIf=\"!isReadOnly\">\n                <ng-template>\n                    <adf-user-role-column class=\"adf-permission-role-column-header\"\n                                          placeholder=\"PERMISSION_MANAGER.COLUMN.BULK-ROLE\"\n                                          [roles]=\"roles\"\n                                          [value]=\"bulkSelectionRole\"\n                                          id=\"adf-bulk-select-role-permission\"\n                                          (roleChanged)=\"bulkRoleUpdate($event)\">\n                    </adf-user-role-column>\n                </ng-template>\n            </adf-data-column-header>\n        </data-column>\n\n        <data-column class=\"adf-datatable-cell adf-delete-permission-column\" key=\"\" *ngIf=\"!isReadOnly\" [sortable]=\"false\">\n            <ng-template let-entry=\"$implicit\">\n                <button mat-icon-button\n                        [disabled]=\"entry.row.obj.readonly\"\n                        (click)=\"removePermission($event, entry.row.obj)\"\n                        [attr.data-automation-id]=\"'adf-delete-permission-button-' + entry.row.obj.authorityId\">\n                    <mat-icon>delete_outline</mat-icon>\n                </button>\n            </ng-template>\n        </data-column>\n    </data-columns>\n\n    <adf-no-content-template>\n        <ng-template>\n            <adf-empty-content\n                id=\"adf-no-permissions-template\"\n                icon=\"supervisor_account\"\n                [title]=\"'PERMISSION_MANAGER.MESSAGE.EMPTY-PERMISSION' | translate\"\n                [subtitle]=\"'PERMISSION_MANAGER.MESSAGE.EMPTY-SUBTITLE' | translate\">\n            </adf-empty-content>\n        </ng-template>\n    </adf-no-content-template>\n\n</adf-datatable>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-datatable-permission{display:flex;min-width:450px;width:100%}.adf-datatable-permission.adf-datatable{overflow:hidden}.adf-datatable-permission.adf-datatable .adf-delete-permission-column{min-width:80px}.adf-datatable-permission.adf-datatable .adf-delete-permission-column .adf-cell-value{padding-right:10px;place-content:flex-end;width:80px}.adf-datatable-permission.adf-datatable .adf-authorityId-column{flex:40%}.adf-datatable-permission.adf-datatable .adf-authority-icon-column{min-width:40px}.adf-datatable-permission.adf-datatable .adf-datatable-selected>svg{height:40px;width:40px}"]
            }]
    }], function () { return []; }, { permissions: [{
            type: Input
        }], isReadOnly: [{
            type: Input
        }], showLocation: [{
            type: Input
        }], selectionMode: [{
            type: Input
        }], update: [{
            type: Output
        }], updateAll: [{
            type: Output
        }], delete: [{
            type: Output
        }], error: [{
            type: Output
        }], node: [{
            type: Input
        }], roles: [{
            type: Input
        }] }); })();
