// @alfresco/adf-core/esm2015/card-view/directives/card-view-content-proxy.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ViewContainerRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class CardViewContentProxyDirective {
    constructor(viewContainerRef) {
        this.viewContainerRef = viewContainerRef;
    }
}
CardViewContentProxyDirective.ɵfac = function CardViewContentProxyDirective_Factory(t) { return new (t || CardViewContentProxyDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef)); };
CardViewContentProxyDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CardViewContentProxyDirective, selectors: [["", "adf-card-view-content-proxy", ""]] });
CardViewContentProxyDirective.ctorParameters = () => [
    { type: ViewContainerRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CardViewContentProxyDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-card-view-content-proxy]'
            }]
    }], function () { return [{ type: ɵngcc0.ViewContainerRef }]; }, null); })();
