// @alfresco/adf-core/esm2015/card-view/components/card-view/card-view.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
import { DEFAULT_SEPARATOR } from '../card-view-textitem/card-view-textitem.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../card-view-item-dispatcher/card-view-item-dispatcher.component';

function CardViewComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelementStart(1, "div", 2);
    ɵngcc0.ɵɵelement(2, "adf-card-view-item-dispatcher", 3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const property_r1 = ctx.$implicit;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵattribute("data-automation-id", "header-" + property_r1.key);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("property", property_r1)("editable", ctx_r0.editable)("displayEmpty", ctx_r0.displayEmpty)("displayNoneOption", property_r1["displayNoneOption"] !== undefined ? property_r1["displayNoneOption"] : ctx_r0.displayNoneOption)("displayClearAction", ctx_r0.displayClearAction)("copyToClipboardAction", ctx_r0.copyToClipboardAction)("useChipsForMultiValueProperty", ctx_r0.useChipsForMultiValueProperty)("multiValueSeparator", ctx_r0.multiValueSeparator);
} }
export class CardViewComponent {
    constructor() {
        this.displayEmpty = true;
        this.displayNoneOption = true;
        this.displayClearAction = true;
        this.copyToClipboardAction = true;
        this.useChipsForMultiValueProperty = true;
        this.multiValueSeparator = DEFAULT_SEPARATOR;
    }
}
CardViewComponent.ɵfac = function CardViewComponent_Factory(t) { return new (t || CardViewComponent)(); };
CardViewComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CardViewComponent, selectors: [["adf-card-view"]], inputs: { displayEmpty: "displayEmpty", displayNoneOption: "displayNoneOption", displayClearAction: "displayClearAction", copyToClipboardAction: "copyToClipboardAction", useChipsForMultiValueProperty: "useChipsForMultiValueProperty", multiValueSeparator: "multiValueSeparator", properties: "properties", editable: "editable" }, decls: 2, vars: 1, consts: [[1, "adf-property-list"], [4, "ngFor", "ngForOf"], [1, "adf-property"], [3, "property", "editable", "displayEmpty", "displayNoneOption", "displayClearAction", "copyToClipboardAction", "useChipsForMultiValueProperty", "multiValueSeparator"]], template: function CardViewComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, CardViewComponent_div_1_Template, 3, 9, "div", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.properties);
    } }, directives: [ɵngcc1.NgForOf, ɵngcc2.CardViewItemDispatcherComponent], styles: [".adf-property-list .adf-property{margin-bottom:20px}.adf-property-list .adf-property .adf-property-value-padding-top{margin-top:6px}.adf-property-list .adf-property .adf-property-field{margin-bottom:-25px;width:100%}.adf-property-list .adf-property .adf-property-field .mat-form-field-infix{border-top-width:0}.adf-property-list .adf-property .adf-property-field .mat-form-field-label{margin-top:6px}.adf-property-list .adf-property .adf-property-field .adf-property-read-only .mat-form-field-underline,.adf-property-list .adf-property .adf-property-read-only .mat-form-field-underline{display:none}.adf-property-list .adf-property .adf-property-label{color:var(--theme-text-color);font-size:12px;word-wrap:break-word}.adf-property-list .adf-property .adf-property-value,.adf-property-list .adf-property .mat-form-field-label{color:var(--theme-text-bold-color);font-size:14px}"], encapsulation: 2 });
CardViewComponent.propDecorators = {
    properties: [{ type: Input }],
    editable: [{ type: Input }],
    displayEmpty: [{ type: Input }],
    displayNoneOption: [{ type: Input }],
    displayClearAction: [{ type: Input }],
    copyToClipboardAction: [{ type: Input }],
    useChipsForMultiValueProperty: [{ type: Input }],
    multiValueSeparator: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CardViewComponent, [{
        type: Component,
        args: [{
                selector: 'adf-card-view',
                template: "<div class=\"adf-property-list\">\n    <div *ngFor=\"let property of properties\">\n        <div [attr.data-automation-id]=\"'header-'+property.key\" class=\"adf-property\">\n            <adf-card-view-item-dispatcher\n                [property]=\"property\"\n                [editable]=\"editable\"\n                [displayEmpty]=\"displayEmpty\"\n                [displayNoneOption]=\"property['displayNoneOption'] !== undefined ? property['displayNoneOption'] : displayNoneOption\"\n                [displayClearAction]=\"displayClearAction\"\n                [copyToClipboardAction]=\"copyToClipboardAction\"\n                [useChipsForMultiValueProperty]=\"useChipsForMultiValueProperty\"\n                [multiValueSeparator]=\"multiValueSeparator\">\n            </adf-card-view-item-dispatcher>\n        </div>\n    </div>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-property-list .adf-property{margin-bottom:20px}.adf-property-list .adf-property .adf-property-value-padding-top{margin-top:6px}.adf-property-list .adf-property .adf-property-field{margin-bottom:-25px;width:100%}.adf-property-list .adf-property .adf-property-field .mat-form-field-infix{border-top-width:0}.adf-property-list .adf-property .adf-property-field .mat-form-field-label{margin-top:6px}.adf-property-list .adf-property .adf-property-field .adf-property-read-only .mat-form-field-underline,.adf-property-list .adf-property .adf-property-read-only .mat-form-field-underline{display:none}.adf-property-list .adf-property .adf-property-label{color:var(--theme-text-color);font-size:12px;word-wrap:break-word}.adf-property-list .adf-property .adf-property-value,.adf-property-list .adf-property .mat-form-field-label{color:var(--theme-text-bold-color);font-size:14px}"]
            }]
    }], function () { return []; }, { displayEmpty: [{
            type: Input
        }], displayNoneOption: [{
            type: Input
        }], displayClearAction: [{
            type: Input
        }], copyToClipboardAction: [{
            type: Input
        }], useChipsForMultiValueProperty: [{
            type: Input
        }], multiValueSeparator: [{
            type: Input
        }], properties: [{
            type: Input
        }], editable: [{
            type: Input
        }] }); })();
