// @alfresco/adf-content-services/esm2015/lib/content-metadata/components/content-metadata-card/content-metadata-card.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
import { Node } from '@alfresco/js-api';
import { ContentService, AllowableOperationsEnum, VersionCompatibilityService } from '@alfresco/adf-core';
import { NodeAspectService } from '../../../aspect-list/node-aspect.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
import * as ɵngcc2 from '../../../aspect-list/node-aspect.service';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@angular/material/card';
import * as ɵngcc5 from '../content-metadata/content-metadata.component';
import * as ɵngcc6 from '@angular/flex-layout/flex';
import * as ɵngcc7 from '@angular/material/button';
import * as ɵngcc8 from '@angular/material/icon';
import * as ɵngcc9 from '@ngx-translate/core';

function ContentMetadataCardComponent_mat_card_0_button_5_Template(rf, ctx) { if (rf & 1) {
    const _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 6);
    ɵngcc0.ɵɵlistener("click", function ContentMetadataCardComponent_mat_card_0_button_5_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r5); const ctx_r4 = ɵngcc0.ɵɵnextContext(2); return ctx_r4.openAspectDialog(); });
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementStart(3, "mat-icon");
    ɵngcc0.ɵɵtext(4, "menu");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵattribute("title", ɵngcc0.ɵɵpipeBind1(1, 2, "CORE.METADATA.ACTIONS.EDIT_ASPECTS"))("aria-label", ɵngcc0.ɵɵpipeBind1(2, 4, "CORE.METADATA.ACCESSIBILITY.EDIT_ASPECTS"));
} }
function ContentMetadataCardComponent_mat_card_0_button_6_Template(rf, ctx) { if (rf & 1) {
    const _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 7);
    ɵngcc0.ɵɵlistener("click", function ContentMetadataCardComponent_mat_card_0_button_6_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r7); const ctx_r6 = ɵngcc0.ɵɵnextContext(2); return ctx_r6.toggleEdit(); });
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementStart(3, "mat-icon");
    ɵngcc0.ɵɵtext(4, "mode_edit");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵattribute("title", ɵngcc0.ɵɵpipeBind1(1, 2, "CORE.METADATA.ACTIONS.EDIT"))("aria-label", ɵngcc0.ɵɵpipeBind1(2, 4, "CORE.METADATA.ACCESSIBILITY.EDIT"));
} }
function ContentMetadataCardComponent_mat_card_0_button_7_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "span", 9);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "mat-icon");
    ɵngcc0.ɵɵtext(5, "keyboard_arrow_down");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 1, "ADF_VIEWER.SIDEBAR.METADATA.MORE_INFORMATION"));
} }
function ContentMetadataCardComponent_mat_card_0_button_7_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "span", 9);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "mat-icon");
    ɵngcc0.ɵɵtext(5, "keyboard_arrow_up");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 1, "ADF_VIEWER.SIDEBAR.METADATA.LESS_INFORMATION"));
} }
function ContentMetadataCardComponent_mat_card_0_button_7_Template(rf, ctx) { if (rf & 1) {
    const _r11 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 8);
    ɵngcc0.ɵɵlistener("click", function ContentMetadataCardComponent_mat_card_0_button_7_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r11); const ctx_r10 = ɵngcc0.ɵɵnextContext(2); return ctx_r10.toggleExpanded(); });
    ɵngcc0.ɵɵtemplate(1, ContentMetadataCardComponent_mat_card_0_button_7_ng_container_1_Template, 6, 3, "ng-container", 0);
    ɵngcc0.ɵɵtemplate(2, ContentMetadataCardComponent_mat_card_0_button_7_ng_container_2_Template, 6, 3, "ng-container", 0);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r3.expanded);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r3.expanded);
} }
function ContentMetadataCardComponent_mat_card_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-card");
    ɵngcc0.ɵɵelementStart(1, "mat-card-content");
    ɵngcc0.ɵɵelement(2, "adf-content-metadata", 1);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "mat-card-footer", 2);
    ɵngcc0.ɵɵelementStart(4, "div");
    ɵngcc0.ɵɵtemplate(5, ContentMetadataCardComponent_mat_card_0_button_5_Template, 5, 6, "button", 3);
    ɵngcc0.ɵɵtemplate(6, ContentMetadataCardComponent_mat_card_0_button_6_Template, 5, 6, "button", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(7, ContentMetadataCardComponent_mat_card_0_button_7_Template, 3, 2, "button", 5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("displayDefaultProperties", ctx_r0.displayDefaultProperties)("expanded", ctx_r0.expanded)("node", ctx_r0.node)("displayEmpty", ctx_r0.displayEmpty)("editable", ctx_r0.editable)("multi", ctx_r0.multi)("displayAspect", ctx_r0.displayAspect)("preset", ctx_r0.preset);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.isEditAspectSupported());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.readOnly && ctx_r0.hasAllowableOperations());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.displayDefaultProperties);
} }
export class ContentMetadataCardComponent {
    constructor(contentService, nodeAspectService, versionCompatibilityService) {
        this.contentService = contentService;
        this.nodeAspectService = nodeAspectService;
        this.versionCompatibilityService = versionCompatibilityService;
        this.displayEmpty = false;
        this.displayAspect = null;
        this.readOnly = false;
        this.multi = false;
        this._displayDefaultProperties = true;
        this.editable = false;
        this.editAspectSupported = false;
        this.editAspectSupported = this.versionCompatibilityService.isVersionSupported('7');
    }
    set displayDefaultProperties(value) {
        this._displayDefaultProperties = value;
        this.onDisplayDefaultPropertiesChange();
    }
    get displayDefaultProperties() {
        return this._displayDefaultProperties;
    }
    ngOnChanges(changes) {
        if (changes.displayAspect && changes.displayAspect.currentValue) {
            this.expanded = true;
        }
    }
    onDisplayDefaultPropertiesChange() {
        this.expanded = !this._displayDefaultProperties;
    }
    toggleEdit() {
        this.editable = !this.editable;
    }
    toggleExpanded() {
        this.expanded = !this.expanded;
    }
    hasAllowableOperations() {
        return this.contentService.hasAllowableOperations(this.node, AllowableOperationsEnum.UPDATE);
    }
    openAspectDialog() {
        this.nodeAspectService.updateNodeAspects(this.node.id);
    }
    isEditAspectSupported() {
        return !this.readOnly && this.hasAllowableOperations() && this.editAspectSupported;
    }
}
ContentMetadataCardComponent.ɵfac = function ContentMetadataCardComponent_Factory(t) { return new (t || ContentMetadataCardComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ContentService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.NodeAspectService), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.VersionCompatibilityService)); };
ContentMetadataCardComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ContentMetadataCardComponent, selectors: [["adf-content-metadata-card"]], hostAttrs: [1, "adf-content-metadata-card"], inputs: { displayEmpty: "displayEmpty", displayAspect: "displayAspect", readOnly: "readOnly", multi: "multi", displayDefaultProperties: "displayDefaultProperties", node: "node", preset: "preset" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 1, vars: 1, consts: [[4, "ngIf"], [3, "displayDefaultProperties", "expanded", "node", "displayEmpty", "editable", "multi", "displayAspect", "preset"], ["fxLayout", "row", "fxLayoutAlign", "space-between stretch", 1, "adf-content-metadata-card-footer"], ["mat-icon-button", "", "data-automation-id", "meta-data-card-edit-aspect", 3, "click", 4, "ngIf"], ["mat-icon-button", "", "data-automation-id", "meta-data-card-toggle-edit", 3, "click", 4, "ngIf"], ["mat-button", "", "data-automation-id", "meta-data-card-toggle-expand", 3, "click", 4, "ngIf"], ["mat-icon-button", "", "data-automation-id", "meta-data-card-edit-aspect", 3, "click"], ["mat-icon-button", "", "data-automation-id", "meta-data-card-toggle-edit", 3, "click"], ["mat-button", "", "data-automation-id", "meta-data-card-toggle-expand", 3, "click"], ["data-automation-id", "meta-data-card-toggle-expand-label"]], template: function ContentMetadataCardComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, ContentMetadataCardComponent_mat_card_0_Template, 8, 11, "mat-card", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.node);
    } }, directives: [ɵngcc3.NgIf, ɵngcc4.MatCard, ɵngcc4.MatCardContent, ɵngcc5.ContentMetadataComponent, ɵngcc4.MatCardFooter, ɵngcc6.DefaultLayoutDirective, ɵngcc6.DefaultLayoutAlignDirective, ɵngcc7.MatButton, ɵngcc8.MatIcon], pipes: [ɵngcc9.TranslatePipe], styles: [".adf-content-metadata-card .mat-card{padding:0}.adf-content-metadata-card .mat-card .mat-card-content{margin-bottom:0}.adf-content-metadata-card-footer.mat-card-footer{border-top:1px solid var(--theme-border-color);margin:0;padding:8px 12px}.adf-content-metadata-card-footer.mat-card-footer button{color:var(--theme-text-color)}.adf-content-metadata-card-footer.mat-card-footer button:hover{color:var(--theme-text-bold-color)}"], encapsulation: 2 });
ContentMetadataCardComponent.ctorParameters = () => [
    { type: ContentService },
    { type: NodeAspectService },
    { type: VersionCompatibilityService }
];
ContentMetadataCardComponent.propDecorators = {
    node: [{ type: Input }],
    displayEmpty: [{ type: Input }],
    displayAspect: [{ type: Input }],
    preset: [{ type: Input }],
    readOnly: [{ type: Input }],
    multi: [{ type: Input }],
    displayDefaultProperties: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContentMetadataCardComponent, [{
        type: Component,
        args: [{
                selector: 'adf-content-metadata-card',
                template: "<mat-card *ngIf=\"node\">\n    <mat-card-content>\n        <adf-content-metadata\n            [displayDefaultProperties]=\"displayDefaultProperties\"\n            [expanded]=\"expanded\"\n            [node]=\"node\"\n            [displayEmpty]=\"displayEmpty\"\n            [editable]=\"editable\"\n            [multi]=\"multi\"\n            [displayAspect]=\"displayAspect\"\n            [preset]=\"preset\">\n        </adf-content-metadata>\n    </mat-card-content>\n    <mat-card-footer class=\"adf-content-metadata-card-footer\" fxLayout=\"row\" fxLayoutAlign=\"space-between stretch\">\n        <div>\n            <button *ngIf=\"isEditAspectSupported()\"\n                mat-icon-button\n                (click)=\"openAspectDialog()\"\n                [attr.title]=\"'CORE.METADATA.ACTIONS.EDIT_ASPECTS' | translate\"\n                [attr.aria-label]=\"'CORE.METADATA.ACCESSIBILITY.EDIT_ASPECTS' | translate\"\n                data-automation-id=\"meta-data-card-edit-aspect\">\n                <mat-icon>menu</mat-icon>\n            </button>\n            <button *ngIf=\"!readOnly && hasAllowableOperations()\"\n                mat-icon-button\n                (click)=\"toggleEdit()\"\n                [attr.title]=\"'CORE.METADATA.ACTIONS.EDIT' | translate\"\n                [attr.aria-label]=\"'CORE.METADATA.ACCESSIBILITY.EDIT' | translate\"\n                data-automation-id=\"meta-data-card-toggle-edit\">\n                <mat-icon>mode_edit</mat-icon>\n            </button>\n        </div>\n        <button *ngIf=\"displayDefaultProperties\" mat-button (click)=\"toggleExpanded()\" data-automation-id=\"meta-data-card-toggle-expand\">\n            <ng-container *ngIf=\"!expanded\">\n                <span data-automation-id=\"meta-data-card-toggle-expand-label\">{{ 'ADF_VIEWER.SIDEBAR.METADATA.MORE_INFORMATION' | translate }}</span>\n                <mat-icon>keyboard_arrow_down</mat-icon>\n            </ng-container>\n            <ng-container *ngIf=\"expanded\">\n                <span data-automation-id=\"meta-data-card-toggle-expand-label\">{{ 'ADF_VIEWER.SIDEBAR.METADATA.LESS_INFORMATION' | translate }}</span>\n                <mat-icon>keyboard_arrow_up</mat-icon>\n            </ng-container>\n        </button>\n    </mat-card-footer>\n</mat-card>\n",
                encapsulation: ViewEncapsulation.None,
                host: { 'class': 'adf-content-metadata-card' },
                styles: [".adf-content-metadata-card .mat-card{padding:0}.adf-content-metadata-card .mat-card .mat-card-content{margin-bottom:0}.adf-content-metadata-card-footer.mat-card-footer{border-top:1px solid var(--theme-border-color);margin:0;padding:8px 12px}.adf-content-metadata-card-footer.mat-card-footer button{color:var(--theme-text-color)}.adf-content-metadata-card-footer.mat-card-footer button:hover{color:var(--theme-text-bold-color)}"]
            }]
    }], function () { return [{ type: ɵngcc1.ContentService }, { type: ɵngcc2.NodeAspectService }, { type: ɵngcc1.VersionCompatibilityService }]; }, { displayEmpty: [{
            type: Input
        }], displayAspect: [{
            type: Input
        }], readOnly: [{
            type: Input
        }], multi: [{
            type: Input
        }], displayDefaultProperties: [{
            type: Input
        }], node: [{
            type: Input
        }], preset: [{
            type: Input
        }] }); })();
