// @alfresco/adf-core/esm2015/form/components/widgets/dynamic-table/editors/amount/amount.editor.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input } from '@angular/core';
import { DynamicTableModel } from './../../dynamic-table.widget.model';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/form-field';
import * as ɵngcc2 from '@angular/material/input';
export class AmountEditorComponent {
    ngOnInit() {
        this.displayName = this.table.getDisplayText(this.column);
    }
    onValueChanged(row, column, event) {
        const value = Number(event.target.value);
        row.value[column.id] = value;
    }
}
AmountEditorComponent.ɵfac = function AmountEditorComponent_Factory(t) { return new (t || AmountEditorComponent)(); };
AmountEditorComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AmountEditorComponent, selectors: [["adf-amount-editor"]], inputs: { table: "table", row: "row", column: "column" }, decls: 5, vars: 6, consts: [[1, "adf-amount-editor"], ["matInput", "", "type", "number", 3, "value", "required", "disabled", "id", "keyup"]], template: function AmountEditorComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "mat-form-field");
        ɵngcc0.ɵɵelementStart(2, "label");
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "input", 1);
        ɵngcc0.ɵɵlistener("keyup", function AmountEditorComponent_Template_input_keyup_4_listener($event) { return ctx.onValueChanged(ctx.row, ctx.column, $event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵattribute("for", ctx.column.id);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ctx.displayName);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("value", ctx.table.getCellValue(ctx.row, ctx.column))("required", ctx.column.required)("disabled", !ctx.column.editable)("id", ctx.column.id);
    } }, directives: [ɵngcc1.MatFormField, ɵngcc2.MatInput], styles: [".adf-text-editor[_ngcontent-%COMP%]{width:100%}"] });
AmountEditorComponent.propDecorators = {
    table: [{ type: Input }],
    row: [{ type: Input }],
    column: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AmountEditorComponent, [{
        type: Component,
        args: [{
                selector: 'adf-amount-editor',
                template: "<div class=\"adf-amount-editor\">\n    <mat-form-field>\n    <label [attr.for]=\"column.id\">{{displayName}}</label>\n    <input matInput\n        type=\"number\"\n        [value]=\"table.getCellValue(row, column)\"\n        (keyup)=\"onValueChanged(row, column, $event)\"\n        [required]=\"column.required\"\n        [disabled]=\"!column.editable\"\n        [id]=\"column.id\">\n    </mat-form-field>\n</div>\n",
                styles: [".adf-text-editor{width:100%}"]
            }]
    }], null, { table: [{
            type: Input
        }], row: [{
            type: Input
        }], column: [{
            type: Input
        }] }); })();
