// @alfresco/adf-core/esm2015/pipes/localized-role.pipe.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Pipe } from '@angular/core';
import { TranslationService } from '../services/translation.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/translation.service';
export class LocalizedRolePipe {
    constructor(translationService) {
        this.translationService = translationService;
    }
    transform(value) {
        if (value) {
            const key = `ADF.ROLES.${value.toUpperCase()}`;
            const translation = this.translationService.instant(key);
            if (translation) {
                return translation;
            }
        }
        return value;
    }
}
LocalizedRolePipe.ɵfac = function LocalizedRolePipe_Factory(t) { return new (t || LocalizedRolePipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TranslationService)); };
LocalizedRolePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "adfLocalizedRole", type: LocalizedRolePipe, pure: true });
LocalizedRolePipe.ctorParameters = () => [
    { type: TranslationService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LocalizedRolePipe, [{
        type: Pipe,
        args: [{
                name: 'adfLocalizedRole'
            }]
    }], function () { return [{ type: ɵngcc1.TranslationService }]; }, null); })();
