// @alfresco/adf-core/esm2015/pipes/text-highlight.pipe.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Pipe } from '@angular/core';
import { HighlightTransformService } from '../services/highlight-transform.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/highlight-transform.service';
export class HighlightPipe {
    constructor(highlightTransformService) {
        this.highlightTransformService = highlightTransformService;
    }
    transform(text, search) {
        const highlightTransformResult = this.highlightTransformService.highlight(text, search);
        return highlightTransformResult.text;
    }
}
HighlightPipe.ɵfac = function HighlightPipe_Factory(t) { return new (t || HighlightPipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.HighlightTransformService)); };
HighlightPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "highlight", type: HighlightPipe, pure: true });
HighlightPipe.ctorParameters = () => [
    { type: HighlightTransformService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HighlightPipe, [{
        type: Pipe,
        args: [{
                name: 'highlight'
            }]
    }], function () { return [{ type: ɵngcc1.HighlightTransformService }]; }, null); })();
