// @alfresco/adf-core/esm2015/form/components/widgets/dynamic-table/editors/boolean/boolean.editor.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input } from '@angular/core';
import { DynamicTableModel } from './../../dynamic-table.widget.model';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/checkbox';
export class BooleanEditorComponent {
    onValueChanged(row, column, event) {
        const value = event.checked;
        row.value[column.id] = value;
    }
}
BooleanEditorComponent.ɵfac = function BooleanEditorComponent_Factory(t) { return new (t || BooleanEditorComponent)(); };
BooleanEditorComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: BooleanEditorComponent, selectors: [["adf-boolean-editor"]], inputs: { table: "table", row: "row", column: "column" }, decls: 4, vars: 6, consts: [["color", "primary", 3, "id", "checked", "required", "disabled", "change"], [1, "adf-checkbox-label"]], template: function BooleanEditorComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "label");
        ɵngcc0.ɵɵelementStart(1, "mat-checkbox", 0);
        ɵngcc0.ɵɵlistener("change", function BooleanEditorComponent_Template_mat_checkbox_change_1_listener($event) { return ctx.onValueChanged(ctx.row, ctx.column, $event); });
        ɵngcc0.ɵɵelementStart(2, "span", 1);
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵattribute("for", ctx.column.id);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("id", ctx.column.id)("checked", ctx.table.getCellValue(ctx.row, ctx.column))("required", ctx.column.required)("disabled", !ctx.column.editable);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx.column.name);
    } }, directives: [ɵngcc1.MatCheckbox], styles: [".adf-checkbox-label[_ngcontent-%COMP%]{cursor:pointer;font-size:16px;line-height:24px;margin:0;position:relative}"] });
BooleanEditorComponent.propDecorators = {
    table: [{ type: Input }],
    row: [{ type: Input }],
    column: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BooleanEditorComponent, [{
        type: Component,
        args: [{
                selector: 'adf-boolean-editor',
                template: " <label [attr.for]=\"column.id\">\n    <mat-checkbox\n        color=\"primary\"\n        [id]=\"column.id\"\n        [checked]=\"table.getCellValue(row, column)\"\n        [required]=\"column.required\"\n        [disabled]=\"!column.editable\"\n        (change)=\"onValueChanged(row, column, $event)\">\n    <span class=\"adf-checkbox-label\">{{column.name}}</span>\n    </mat-checkbox>\n</label>\n",
                styles: [".adf-checkbox-label{cursor:pointer;font-size:16px;line-height:24px;margin:0;position:relative}"]
            }]
    }], null, { table: [{
            type: Input
        }], row: [{
            type: Input
        }], column: [{
            type: Input
        }] }); })();
