// @alfresco/adf-process-services-cloud/esm2015/lib/pipes/process-services-cloud-pipe.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { ProcessNameCloudPipe } from './process-name-cloud.pipe';
import * as ɵngcc0 from '@angular/core';
export class ProcessServicesCloudPipeModule {
}
ProcessServicesCloudPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ProcessServicesCloudPipeModule });
ProcessServicesCloudPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ProcessServicesCloudPipeModule_Factory(t) { return new (t || ProcessServicesCloudPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ProcessServicesCloudPipeModule, { declarations: function () { return [ProcessNameCloudPipe]; }, exports: function () { return [ProcessNameCloudPipe]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ProcessServicesCloudPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    ProcessNameCloudPipe
                ],
                exports: [
                    ProcessNameCloudPipe
                ]
            }]
    }], null, null); })();
