// @alfresco/adf-process-services-cloud/esm2015/lib/pipes/process-name-cloud.pipe.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Pipe } from '@angular/core';
import moment from 'moment-es6';
import { LocalizedDatePipe } from '@alfresco/adf-core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
export class ProcessNameCloudPipe {
    constructor(localizedDatePipe) {
        this.localizedDatePipe = localizedDatePipe;
    }
    transform(processNameFormat, processInstance) {
        let processName = processNameFormat;
        if (processName.match(ProcessNameCloudPipe.DATE_TIME_IDENTIFIER_REG_EXP)) {
            const presentDateTime = moment.now();
            processName = processName.replace(ProcessNameCloudPipe.DATE_TIME_IDENTIFIER_REG_EXP, this.localizedDatePipe.transform(presentDateTime, 'medium'));
        }
        if (processName.match(ProcessNameCloudPipe.PROCESS_DEFINITION_IDENTIFIER_REG_EXP)) {
            const selectedProcessDefinitionName = processInstance ? processInstance.processDefinitionName : '';
            processName = processName.replace(ProcessNameCloudPipe.PROCESS_DEFINITION_IDENTIFIER_REG_EXP, selectedProcessDefinitionName);
        }
        return processName;
    }
}
ProcessNameCloudPipe.ɵfac = function ProcessNameCloudPipe_Factory(t) { return new (t || ProcessNameCloudPipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.LocalizedDatePipe)); };
ProcessNameCloudPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "processNameCloud", type: ProcessNameCloudPipe, pure: true });
ProcessNameCloudPipe.DATE_TIME_IDENTIFIER_REG_EXP = new RegExp('%{datetime}', 'i');
ProcessNameCloudPipe.PROCESS_DEFINITION_IDENTIFIER_REG_EXP = new RegExp('%{processdefinition}', 'i');
ProcessNameCloudPipe.ctorParameters = () => [
    { type: LocalizedDatePipe }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ProcessNameCloudPipe, [{
        type: Pipe,
        args: [{ name: 'processNameCloud' }]
    }], function () { return [{ type: ɵngcc1.LocalizedDatePipe }]; }, null); })();
