// @alfresco/adf-process-services-cloud/esm2015/lib/task/task-filters/components/base-task-filters-cloud.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { EventEmitter, Input, Output, Directive } from '@angular/core';
import { Subject } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
export class BaseTaskFiltersCloudComponent {
    constructor() {
        this.appName = '';
        this.showIcons = false;
        this.success = new EventEmitter();
        this.error = new EventEmitter();
        this.counters$ = {};
        this.updatedCounters = [];
        this.onDestroy$ = new Subject();
    }
    ngOnDestroy() {
        this.onDestroy$.next(true);
        this.onDestroy$.complete();
    }
    wasFilterUpdated(filterKey) {
        return this.updatedCounters.includes(filterKey);
    }
    addToUpdatedCounters(filterKey) {
        if (!this.updatedCounters.includes(filterKey)) {
            this.updatedCounters.push(filterKey);
        }
    }
    resetFilterCounter(filterKey) {
        const filterIndex = this.updatedCounters.indexOf(filterKey);
        if (filterIndex > -1) {
            this.updatedCounters.splice(filterIndex, 1);
        }
    }
}
BaseTaskFiltersCloudComponent.ɵfac = function BaseTaskFiltersCloudComponent_Factory(t) { return new (t || BaseTaskFiltersCloudComponent)(); };
BaseTaskFiltersCloudComponent.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BaseTaskFiltersCloudComponent, inputs: { appName: "appName", showIcons: "showIcons", filterParam: "filterParam" }, outputs: { success: "success", error: "error" } });
BaseTaskFiltersCloudComponent.propDecorators = {
    appName: [{ type: Input }],
    filterParam: [{ type: Input }],
    showIcons: [{ type: Input }],
    success: [{ type: Output }],
    error: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BaseTaskFiltersCloudComponent, [{
        type: Directive
    }], function () { return []; }, { appName: [{
            type: Input
        }], showIcons: [{
            type: Input
        }], success: [{
            type: Output
        }], error: [{
            type: Output
        }], filterParam: [{
            type: Input
        }] }); })();
