// @alfresco/adf-core/esm2015/form/components/form-list.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
import { FormService } from './../services/form.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../services/form.service';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../../datatable/components/datatable/datatable.component';
import * as ɵngcc4 from '../../data-column/data-column-list.component';
import * as ɵngcc5 from '../../data-column/data-column.component';

function FormListComponent_adf_datatable_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "adf-datatable", 1);
    ɵngcc0.ɵɵelementStart(1, "data-columns");
    ɵngcc0.ɵɵelement(2, "data-column", 2);
    ɵngcc0.ɵɵelement(3, "data-column", 3);
    ɵngcc0.ɵɵelement(4, "data-column", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("rows", ctx_r0.forms);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("sortable", true);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("sortable", true);
} }
export class FormListComponent {
    constructor(formService) {
        this.formService = formService;
        this.forms = [];
    }
    ngOnChanges() {
        this.getForms();
    }
    isEmpty() {
        return this.forms && this.forms.length === 0;
    }
    getForms() {
        this.formService.getForms().subscribe((forms) => {
            this.forms.push(...forms);
        });
    }
}
FormListComponent.ɵfac = function FormListComponent_Factory(t) { return new (t || FormListComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService)); };
FormListComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FormListComponent, selectors: [["adf-form-list"]], inputs: { forms: "forms" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 1, vars: 1, consts: [[3, "rows", 4, "ngIf"], [3, "rows"], ["key", "name", "type", "text", "title", "Name", 1, "adf-ellipsis-cell", 3, "sortable"], ["key", "lastUpdatedByFullName", "type", "text", "title", "User", 1, "adf-ellipsis-cell", 3, "sortable"], ["key", "lastUpdated", "type", "date", "format", "shortDate", "title", "Date"]], template: function FormListComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, FormListComponent_adf_datatable_0_Template, 5, 3, "adf-datatable", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", !ctx.isEmpty());
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.DataTableComponent, ɵngcc4.DataColumnListComponent, ɵngcc5.DataColumnComponent], encapsulation: 2 });
FormListComponent.ctorParameters = () => [
    { type: FormService }
];
FormListComponent.propDecorators = {
    forms: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FormListComponent, [{
        type: Component,
        args: [{
                selector: 'adf-form-list',
                template: "<adf-datatable *ngIf=\"!isEmpty()\"\n    [rows]=\"forms\">\n    <data-columns>\n        <data-column key=\"name\" type=\"text\" title=\"Name\" class=\"adf-ellipsis-cell\" [sortable]=\"true\"></data-column>\n        <data-column key=\"lastUpdatedByFullName\" type=\"text\" title=\"User\" class=\"adf-ellipsis-cell\" [sortable]=\"true\"></data-column>\n        <data-column key=\"lastUpdated\" type=\"date\" format=\"shortDate\" title=\"Date\"></data-column>\n    </data-columns>\n</adf-datatable>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.FormService }]; }, { forms: [{
            type: Input
        }] }); })();
