// @alfresco/adf-core/esm2015/comments/comment-list.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { EcmUserService } from '../services/ecm-user.service';
import { PeopleProcessService } from '../services/people-process.service';
import { UserPreferencesService, UserPreferenceValues } from '../services/user-preferences.service';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/people-process.service';
import * as ɵngcc2 from '../services/ecm-user.service';
import * as ɵngcc3 from '../services/user-preferences.service';
import * as ɵngcc4 from '@angular/material/list';
import * as ɵngcc5 from '@angular/common';
import * as ɵngcc6 from '@angular/material/core';
import * as ɵngcc7 from '../pipes/time-ago.pipe';

function CommentListComponent_mat_list_item_1_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 10);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const comment_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r2.getUserShortName(comment_r1.createdBy), " ");
} }
function CommentListComponent_mat_list_item_1_img_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 11);
} if (rf & 2) {
    const comment_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("alt", comment_r1.createdBy)("src", ctx_r3.getUserImage(comment_r1.createdBy), ɵngcc0.ɵɵsanitizeUrl);
} }
function CommentListComponent_mat_list_item_1_Template(rf, ctx) { if (rf & 1) {
    const _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "mat-list-item", 2);
    ɵngcc0.ɵɵlistener("click", function CommentListComponent_mat_list_item_1_Template_mat_list_item_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r7); const comment_r1 = ctx.$implicit; const ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.selectComment(comment_r1); });
    ɵngcc0.ɵɵelementStart(1, "div", 3);
    ɵngcc0.ɵɵtemplate(2, CommentListComponent_mat_list_item_1_div_2_Template, 2, 1, "div", 4);
    ɵngcc0.ɵɵelementStart(3, "div");
    ɵngcc0.ɵɵtemplate(4, CommentListComponent_mat_list_item_1_img_4_Template, 1, 2, "img", 5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(5, "div", 6);
    ɵngcc0.ɵɵelementStart(6, "div", 7);
    ɵngcc0.ɵɵtext(7);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(8, "div", 8);
    ɵngcc0.ɵɵelementStart(9, "div", 9);
    ɵngcc0.ɵɵtext(10);
    ɵngcc0.ɵɵpipe(11, "adfTimeAgo");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const comment_r1 = ctx.$implicit;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("adf-is-selected", comment_r1.isSelected);
    ɵngcc0.ɵɵpropertyInterpolate1("id", "adf-comment-", comment_r1 == null ? null : comment_r1.id, "");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.isPictureDefined(comment_r1.createdBy));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.isPictureDefined(comment_r1.createdBy));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate2(" ", comment_r1.createdBy == null ? null : comment_r1.createdBy.firstName, " ", comment_r1.createdBy == null ? null : comment_r1.createdBy.lastName, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("innerHTML", comment_r1.message, ɵngcc0.ɵɵsanitizeHtml);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind2(11, 9, comment_r1.created, ctx_r0.currentLocale), " ");
} }
export class CommentListComponent {
    constructor(peopleProcessService, ecmUserService, userPreferenceService) {
        this.peopleProcessService = peopleProcessService;
        this.ecmUserService = ecmUserService;
        this.userPreferenceService = userPreferenceService;
        this.clickRow = new EventEmitter();
        this.onDestroy$ = new Subject();
    }
    ngOnInit() {
        this.userPreferenceService
            .select(UserPreferenceValues.Locale)
            .pipe(takeUntil(this.onDestroy$))
            .subscribe(locale => this.currentLocale = locale);
    }
    ngOnDestroy() {
        this.onDestroy$.next(true);
        this.onDestroy$.complete();
    }
    selectComment(comment) {
        if (this.selectedComment) {
            this.selectedComment.isSelected = false;
        }
        comment.isSelected = true;
        this.selectedComment = comment;
        this.clickRow.emit(this.selectedComment);
    }
    getUserShortName(user) {
        let shortName = '';
        if (user) {
            if (user.firstName) {
                shortName = user.firstName[0].toUpperCase();
            }
            if (user.lastName) {
                shortName += user.lastName[0].toUpperCase();
            }
        }
        return shortName;
    }
    isPictureDefined(user) {
        return user.pictureId || user.avatarId;
    }
    getUserImage(user) {
        if (this.isAContentUsers(user)) {
            return this.ecmUserService.getUserProfileImage(user.avatarId);
        }
        else {
            return this.peopleProcessService.getUserImage(user);
        }
    }
    isAContentUsers(user) {
        return user.avatarId;
    }
}
CommentListComponent.ɵfac = function CommentListComponent_Factory(t) { return new (t || CommentListComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.PeopleProcessService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.EcmUserService), ɵngcc0.ɵɵdirectiveInject(ɵngcc3.UserPreferencesService)); };
CommentListComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CommentListComponent, selectors: [["adf-comment-list"]], inputs: { comments: "comments" }, outputs: { clickRow: "clickRow" }, decls: 2, vars: 1, consts: [[1, "adf-comment-list"], ["class", "adf-comment-list-item", 3, "adf-is-selected", "id", "click", 4, "ngFor", "ngForOf"], [1, "adf-comment-list-item", 3, "id", "click"], ["id", "comment-user-icon", 1, "adf-comment-img-container"], ["class", "adf-comment-user-icon", 4, "ngIf"], ["class", "adf-people-img", 3, "alt", "src", 4, "ngIf"], [1, "adf-comment-contents"], ["matLine", "", "id", "comment-user", 1, "adf-comment-user-name"], ["matLine", "", "id", "comment-message", 1, "adf-comment-message", 3, "innerHTML"], ["matLine", "", "id", "comment-time", 1, "adf-comment-message-time"], [1, "adf-comment-user-icon"], [1, "adf-people-img", 3, "alt", "src"]], template: function CommentListComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "mat-list", 0);
        ɵngcc0.ɵɵtemplate(1, CommentListComponent_mat_list_item_1_Template, 12, 12, "mat-list-item", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.comments);
    } }, directives: [ɵngcc4.MatList, ɵngcc5.NgForOf, ɵngcc4.MatListItem, ɵngcc5.NgIf, ɵngcc6.MatLine], pipes: [ɵngcc7.TimeAgoPipe], styles: [".adf-is-selected{background:var(--adf-comment-list-primary-color)}.adf-comment-img-container{align-self:flex-start;display:flex;float:left;height:100%;padding-top:18px;width:40px}.adf-comment-list-item{background-position:50%;display:table-row-group;height:100%!important;overflow:hidden;padding-top:12px;transition:background .8s;white-space:normal}.adf-comment-list-item:hover{background:var(--adf-comment-list-primary-color) radial-gradient(circle,transparent 1%,var(--adf-comment-list-primary-color) 0) 50%/15000%}.adf-comment-list-item:active{background-color:var(--adf-comment-list-ripple-color);background-size:100%;transition:background 0s}.adf-comment-user-icon{background-color:var(--theme-primary-color);background-size:cover;border-radius:50%;color:var(--theme-primary-color-default-contrast);font-size:16px;height:20px;padding:10px 5px;text-align:center;width:30px}.adf-comment-user-name{float:left;font-size:14px;font-weight:600;padding:2px 10px;width:90%}.adf-comment-message{font-size:14px;font-style:italic;letter-spacing:-.2px;line-height:1.43;white-space:normal!important;width:calc(100% - 10px)}.adf-comment-message,.adf-comment-message-time{color:var(--theme-foreground-text-color);float:left;padding:2px 10px}.adf-comment-message-time{font-size:12px!important;width:90%}.adf-comment-contents{padding-bottom:12px;padding-left:5px;padding-top:12px;width:calc(100% - 10px)}.adf-people-img{border-radius:90%;height:40px;vertical-align:middle;width:40px}"], encapsulation: 2 });
CommentListComponent.ctorParameters = () => [
    { type: PeopleProcessService },
    { type: EcmUserService },
    { type: UserPreferencesService }
];
CommentListComponent.propDecorators = {
    comments: [{ type: Input }],
    clickRow: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CommentListComponent, [{
        type: Component,
        args: [{
                selector: 'adf-comment-list',
                template: " <mat-list class=\"adf-comment-list\">\n    <mat-list-item *ngFor=\"let comment of comments\"\n                  (click)=\"selectComment(comment)\"\n                  class=\"adf-comment-list-item\"\n                  [class.adf-is-selected]=\"comment.isSelected\"\n                  id=\"adf-comment-{{comment?.id}}\">\n        <div id=\"comment-user-icon\" class=\"adf-comment-img-container\">\n            <div\n                *ngIf=\"!isPictureDefined(comment.createdBy)\"\n                class=\"adf-comment-user-icon\">\n                {{getUserShortName(comment.createdBy)}}\n            </div>\n            <div>\n                <img [alt]=\"comment.createdBy\" *ngIf=\"isPictureDefined(comment.createdBy)\"\n                      class=\"adf-people-img\"\n                     [src]=\"getUserImage(comment.createdBy)\" />\n            </div>\n        </div>\n        <div class=\"adf-comment-contents\">\n            <div matLine id=\"comment-user\" class=\"adf-comment-user-name\">\n                {{comment.createdBy?.firstName}} {{comment.createdBy?.lastName}}\n            </div>\n            <div matLine id=\"comment-message\" class=\"adf-comment-message\" [innerHTML]=\"comment.message\"></div>\n            <div matLine id=\"comment-time\" class=\"adf-comment-message-time\">\n                {{ comment.created | adfTimeAgo: currentLocale }}\n            </div>\n        </div>\n    </mat-list-item>\n</mat-list>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-is-selected{background:var(--adf-comment-list-primary-color)}.adf-comment-img-container{align-self:flex-start;display:flex;float:left;height:100%;padding-top:18px;width:40px}.adf-comment-list-item{background-position:50%;display:table-row-group;height:100%!important;overflow:hidden;padding-top:12px;transition:background .8s;white-space:normal}.adf-comment-list-item:hover{background:var(--adf-comment-list-primary-color) radial-gradient(circle,transparent 1%,var(--adf-comment-list-primary-color) 0) 50%/15000%}.adf-comment-list-item:active{background-color:var(--adf-comment-list-ripple-color);background-size:100%;transition:background 0s}.adf-comment-user-icon{background-color:var(--theme-primary-color);background-size:cover;border-radius:50%;color:var(--theme-primary-color-default-contrast);font-size:16px;height:20px;padding:10px 5px;text-align:center;width:30px}.adf-comment-user-name{float:left;font-size:14px;font-weight:600;padding:2px 10px;width:90%}.adf-comment-message{font-size:14px;font-style:italic;letter-spacing:-.2px;line-height:1.43;white-space:normal!important;width:calc(100% - 10px)}.adf-comment-message,.adf-comment-message-time{color:var(--theme-foreground-text-color);float:left;padding:2px 10px}.adf-comment-message-time{font-size:12px!important;width:90%}.adf-comment-contents{padding-bottom:12px;padding-left:5px;padding-top:12px;width:calc(100% - 10px)}.adf-people-img{border-radius:90%;height:40px;vertical-align:middle;width:40px}"]
            }]
    }], function () { return [{ type: ɵngcc1.PeopleProcessService }, { type: ɵngcc2.EcmUserService }, { type: ɵngcc3.UserPreferencesService }]; }, { clickRow: [{
            type: Output
        }], comments: [{
            type: Input
        }] }); })();
