// @alfresco/adf-core/esm2015/form/components/widgets/multiline-text/multiline-text.widget.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { WidgetComponent } from './../widget.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../../../services/form.service';
import * as ɵngcc2 from '@angular/material/form-field';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@angular/material/input';
import * as ɵngcc5 from '@angular/forms';
import * as ɵngcc6 from '@angular/material/tooltip';
import * as ɵngcc7 from '../error/error.component';
import * as ɵngcc8 from '@ngx-translate/core';

function MultilineTextWidgetComponentComponent_span_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1, "*");
    ɵngcc0.ɵɵelementEnd();
} }
function MultilineTextWidgetComponentComponent_div_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 7);
    ɵngcc0.ɵɵelementStart(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate2("", (ctx_r1.field == null ? null : ctx_r1.field.value == null ? null : ctx_r1.field.value.length) || 0, "/", ctx_r1.field.maxLength, "");
} }
function MultilineTextWidgetComponentComponent_error_widget_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "error-widget", 8);
    ɵngcc0.ɵɵpipe(1, "translate");
} if (rf & 2) {
    ɵngcc0.ɵɵpropertyInterpolate("required", ɵngcc0.ɵɵpipeBind1(1, 1, "FORM.FIELD.REQUIRED"));
} }
export class MultilineTextWidgetComponentComponent extends WidgetComponent {
    constructor(formService) {
        super(formService);
        this.formService = formService;
    }
}
MultilineTextWidgetComponentComponent.ɵfac = function MultilineTextWidgetComponentComponent_Factory(t) { return new (t || MultilineTextWidgetComponentComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService)); };
MultilineTextWidgetComponentComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MultilineTextWidgetComponentComponent, selectors: [["multiline-text-widget"]], hostBindings: function MultilineTextWidgetComponentComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function MultilineTextWidgetComponentComponent_click_HostBindingHandler($event) { return ctx.event($event); })("blur", function MultilineTextWidgetComponentComponent_blur_HostBindingHandler($event) { return ctx.event($event); })("change", function MultilineTextWidgetComponentComponent_change_HostBindingHandler($event) { return ctx.event($event); })("focus", function MultilineTextWidgetComponentComponent_focus_HostBindingHandler($event) { return ctx.event($event); })("focusin", function MultilineTextWidgetComponentComponent_focusin_HostBindingHandler($event) { return ctx.event($event); })("focusout", function MultilineTextWidgetComponentComponent_focusout_HostBindingHandler($event) { return ctx.event($event); })("input", function MultilineTextWidgetComponentComponent_input_HostBindingHandler($event) { return ctx.event($event); })("invalid", function MultilineTextWidgetComponentComponent_invalid_HostBindingHandler($event) { return ctx.event($event); })("select", function MultilineTextWidgetComponentComponent_select_HostBindingHandler($event) { return ctx.event($event); });
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 11, vars: 22, consts: [["floatPlaceholder", "never"], [1, "adf-label"], [4, "ngIf"], ["matInput", "", "type", "text", "rows", "3", "matTooltipPosition", "above", "matTooltipShowDelay", "1000", 1, "adf-input", 3, "matTextareaAutosize", "id", "required", "ngModel", "disabled", "placeholder", "matTooltip", "ngModelChange"], ["class", "adf-multiline-word-counter", 4, "ngIf"], [3, "error"], ["class", "adf-multiline-required-message", 3, "required", 4, "ngIf"], [1, "adf-multiline-word-counter"], [1, "adf-multiline-required-message", 3, "required"]], template: function MultilineTextWidgetComponentComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "mat-form-field", 0);
        ɵngcc0.ɵɵelementStart(2, "label", 1);
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵpipe(4, "translate");
        ɵngcc0.ɵɵtemplate(5, MultilineTextWidgetComponentComponent_span_5_Template, 2, 0, "span", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(6, "textarea", 3);
        ɵngcc0.ɵɵlistener("ngModelChange", function MultilineTextWidgetComponentComponent_Template_textarea_ngModelChange_6_listener($event) { return ctx.field.value = $event; })("ngModelChange", function MultilineTextWidgetComponentComponent_Template_textarea_ngModelChange_6_listener() { return ctx.onFieldChanged(ctx.field); });
        ɵngcc0.ɵɵtext(7, "        ");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(8, MultilineTextWidgetComponentComponent_div_8_Template, 3, 2, "div", 4);
        ɵngcc0.ɵɵelement(9, "error-widget", 5);
        ɵngcc0.ɵɵtemplate(10, MultilineTextWidgetComponentComponent_error_widget_10_Template, 2, 3, "error-widget", 6);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMapInterpolate1("adf-multiline-text-widget ", ctx.field.className, "");
        ɵngcc0.ɵɵclassProp("adf-invalid", !ctx.field.isValid)("adf-readonly", ctx.field.readOnly);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵattribute("for", ctx.field.id);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(4, 20, ctx.field.name));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isRequired());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("matTextareaAutosize", true)("id", ctx.field.id)("required", ctx.isRequired())("ngModel", ctx.field.value)("disabled", ctx.field.readOnly || ctx.readOnly)("placeholder", ctx.field.placeholder)("matTooltip", ctx.field.tooltip);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.field.maxLength > 0);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("error", ctx.field.validationSummary);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isInvalidFieldRequired());
    } }, directives: [ɵngcc2.MatFormField, ɵngcc3.NgIf, ɵngcc4.MatInput, ɵngcc4.MatTextareaAutosize, ɵngcc5.DefaultValueAccessor, ɵngcc5.RequiredValidator, ɵngcc5.NgControlStatus, ɵngcc5.NgModel, ɵngcc6.MatTooltip, ɵngcc7.ErrorWidgetComponent], pipes: [ɵngcc8.TranslatePipe], styles: [".adf-multiline-text-widget{width:100%}.adf-multiline-text-widget .mat-form-field-label-wrapper{top:20px}.adf-multiline-word-counter{float:right;font-size:12px;line-height:14px;margin-top:-20px!important;min-height:24px;min-width:1px;opacity:1;overflow:hidden;padding-left:0;padding-right:2px;padding-top:5px;text-align:right;transition:all .3s cubic-bezier(.55,0,.55,.2)}.adf-multiline-required-message{display:flex}"], encapsulation: 2 });
MultilineTextWidgetComponentComponent.ctorParameters = () => [
    { type: FormService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MultilineTextWidgetComponentComponent, [{
        type: Component,
        args: [{
                selector: 'multiline-text-widget',
                template: "<div class=\"adf-multiline-text-widget {{field.className}}\"\n     [class.adf-invalid]=\"!field.isValid\" [class.adf-readonly]=\"field.readOnly\">\n    <mat-form-field floatPlaceholder=\"never\">\n        <label class=\"adf-label\" [attr.for]=\"field.id\">{{field.name | translate }}<span *ngIf=\"isRequired()\">*</span></label>\n        <textarea matInput class=\"adf-input\"\n                  [matTextareaAutosize]=\"true\"\n                  type=\"text\"\n                  rows=\"3\"\n                  [id]=\"field.id\"\n                  [required]=\"isRequired()\"\n                  [(ngModel)]=\"field.value\"\n                  (ngModelChange)=\"onFieldChanged(field)\"\n                  [disabled]=\"field.readOnly || readOnly\"\n                  [placeholder]=\"field.placeholder\"\n                  [matTooltip]=\"field.tooltip\"\n                  matTooltipPosition=\"above\"\n                  matTooltipShowDelay=\"1000\">\n        </textarea>\n    </mat-form-field>\n    <div *ngIf=\"field.maxLength > 0\" class=\"adf-multiline-word-counter\">\n        <span>{{field?.value?.length || 0}}/{{field.maxLength}}</span>\n    </div>\n    <error-widget [error]=\"field.validationSummary\"></error-widget>\n    <error-widget class=\"adf-multiline-required-message\" *ngIf=\"isInvalidFieldRequired()\" required=\"{{ 'FORM.FIELD.REQUIRED' | translate }}\"></error-widget>\n</div>\n\n",
                host: {
                    '(click)': 'event($event)',
                    '(blur)': 'event($event)',
                    '(change)': 'event($event)',
                    '(focus)': 'event($event)',
                    '(focusin)': 'event($event)',
                    '(focusout)': 'event($event)',
                    '(input)': 'event($event)',
                    '(invalid)': 'event($event)',
                    '(select)': 'event($event)'
                },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-multiline-text-widget{width:100%}.adf-multiline-text-widget .mat-form-field-label-wrapper{top:20px}.adf-multiline-word-counter{float:right;font-size:12px;line-height:14px;margin-top:-20px!important;min-height:24px;min-width:1px;opacity:1;overflow:hidden;padding-left:0;padding-right:2px;padding-top:5px;text-align:right;transition:all .3s cubic-bezier(.55,0,.55,.2)}.adf-multiline-required-message{display:flex}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormService }]; }, null); })();
