// @alfresco/adf-core/esm2015/toolbar/toolbar-title.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class ToolbarTitleComponent {
}
ToolbarTitleComponent.ɵfac = function ToolbarTitleComponent_Factory(t) { return new (t || ToolbarTitleComponent)(); };
ToolbarTitleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ToolbarTitleComponent, selectors: [["adf-toolbar-title"]], hostAttrs: [1, "adf-toolbar-title"], ngContentSelectors: _c0, decls: 1, vars: 0, template: function ToolbarTitleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ToolbarTitleComponent, [{
        type: Component,
        args: [{
                selector: 'adf-toolbar-title',
                template: '<ng-content></ng-content>',
                host: { 'class': 'adf-toolbar-title' }
            }]
    }], null, null); })();
