// @alfresco/adf-core/esm2015/datatable/components/date-cell/date-cell.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { DataTableCellComponent } from '../datatable-cell/datatable-cell.component';
import { UserPreferencesService, UserPreferenceValues } from '../../../services/user-preferences.service';
import { AlfrescoApiService } from '../../../services/alfresco-api.service';
import { AppConfigService } from '../../../app-config/app-config.service';
import { takeUntil } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../../services/user-preferences.service';
import * as ɵngcc2 from '../../../services/alfresco-api.service';
import * as ɵngcc3 from '../../../app-config/app-config.service';
import * as ɵngcc4 from '@angular/common';
import * as ɵngcc5 from '../../../pipes/time-ago.pipe';
import * as ɵngcc6 from '../../../pipes/localized-date.pipe';

function DateCellComponent_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 2);
    ɵngcc0.ɵɵpipe(1, "adfTimeAgo");
    ɵngcc0.ɵɵpipe(2, "async");
    ɵngcc0.ɵɵpipe(3, "adfLocalizedDate");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "adfTimeAgo");
    ɵngcc0.ɵɵpipe(6, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind2(3, 8, ctx_r0.tooltip, "medium"));
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind2(1, 3, ɵngcc0.ɵɵpipeBind1(2, 6, ctx_r0.value$), ctx_r0.currentLocale));
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind2(5, 11, ɵngcc0.ɵɵpipeBind1(6, 14, ctx_r0.value$), ctx_r0.currentLocale), " ");
} }
function DateCellComponent_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 2);
    ɵngcc0.ɵɵpipe(1, "adfLocalizedDate");
    ɵngcc0.ɵɵpipe(2, "adfLocalizedDate");
    ɵngcc0.ɵɵpipe(3, "async");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "adfLocalizedDate");
    ɵngcc0.ɵɵpipe(6, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind2(1, 3, ctx_r2.tooltip, ctx_r2.format));
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind2(2, 6, ɵngcc0.ɵɵpipeBind1(3, 9, ctx_r2.value$), ctx_r2.format));
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind2(5, 11, ɵngcc0.ɵɵpipeBind1(6, 14, ctx_r2.value$), ctx_r2.format), " ");
} }
export class DateCellComponent extends DataTableCellComponent {
    constructor(userPreferenceService, alfrescoApiService, appConfig) {
        super(alfrescoApiService);
        this.dateFormat = appConfig.get('dateValues.defaultDateFormat', DateCellComponent.DATE_FORMAT);
        if (userPreferenceService) {
            userPreferenceService
                .select(UserPreferenceValues.Locale)
                .pipe(takeUntil(this.onDestroy$))
                .subscribe(locale => this.currentLocale = locale);
        }
    }
    get format() {
        if (this.column) {
            return this.column.format || this.dateFormat;
        }
        return this.dateFormat;
    }
}
DateCellComponent.ɵfac = function DateCellComponent_Factory(t) { return new (t || DateCellComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.UserPreferencesService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.AlfrescoApiService), ɵngcc0.ɵɵdirectiveInject(ɵngcc3.AppConfigService)); };
DateCellComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DateCellComponent, selectors: [["adf-date-cell"]], hostAttrs: [1, "adf-date-cell", "adf-datatable-content-cell"], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 4, vars: 2, consts: [["class", "adf-datatable-cell-value", 3, "title", 4, "ngIf", "ngIfElse"], ["standard_date", ""], [1, "adf-datatable-cell-value", 3, "title"]], template: function DateCellComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementContainerStart(0);
        ɵngcc0.ɵɵtemplate(1, DateCellComponent_span_1_Template, 7, 16, "span", 0);
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵtemplate(2, DateCellComponent_ng_template_2_Template, 7, 16, "ng-template", null, 1, ɵngcc0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        const _r1 = ɵngcc0.ɵɵreference(3);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.format === "timeAgo")("ngIfElse", _r1);
    } }, directives: [ɵngcc4.NgIf], pipes: [ɵngcc5.TimeAgoPipe, ɵngcc4.AsyncPipe, ɵngcc6.LocalizedDatePipe], encapsulation: 2 });
DateCellComponent.DATE_FORMAT = 'medium';
DateCellComponent.ctorParameters = () => [
    { type: UserPreferencesService },
    { type: AlfrescoApiService },
    { type: AppConfigService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DateCellComponent, [{
        type: Component,
        args: [{
                selector: 'adf-date-cell',
                template: `
        <ng-container>
            <span
                [attr.aria-label]="value$ | async | adfTimeAgo: currentLocale"
                title="{{ tooltip | adfLocalizedDate: 'medium' }}"
                class="adf-datatable-cell-value"
                *ngIf="format === 'timeAgo'; else standard_date">
                {{ value$ | async | adfTimeAgo: currentLocale }}
            </span>
        </ng-container>
        <ng-template #standard_date>
            <span
                class="adf-datatable-cell-value"
                title="{{ tooltip | adfLocalizedDate: format }}"
                class="adf-datatable-cell-value"
                [attr.aria-label]="value$ | async | adfLocalizedDate: format">
                {{ value$ | async | adfLocalizedDate: format }}
            </span>
        </ng-template>
    `,
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-date-cell adf-datatable-content-cell' }
            }]
    }], function () { return [{ type: ɵngcc1.UserPreferencesService }, { type: ɵngcc2.AlfrescoApiService }, { type: ɵngcc3.AppConfigService }]; }, null); })();
