// @alfresco/adf-process-services-cloud/esm2015/lib/services/base-cloud.service.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { from } from 'rxjs';
export class BaseCloudService {
    constructor(apiService, appConfigService) {
        this.apiService = apiService;
        this.appConfigService = appConfigService;
        this.defaultParams = {
            path: '',
            httpMethod: '',
            contentTypes: ['application/json'],
            accepts: ['application/json'],
            returnType: Object
        };
    }
    getBasePath(appName) {
        return appName
            ? `${this.contextRoot}/${appName}`
            : this.contextRoot;
    }
    post(url, data) {
        return from(this.callApi(Object.assign(Object.assign({}, this.defaultParams), { path: url, httpMethod: 'POST', bodyParam: data })));
    }
    put(url, data) {
        return from(this.callApi(Object.assign(Object.assign({}, this.defaultParams), { path: url, httpMethod: 'PUT', bodyParam: data })));
    }
    delete(url) {
        return from(this.callApi(Object.assign(Object.assign({}, this.defaultParams), { path: url, httpMethod: 'DELETE' })));
    }
    get(url, queryParams) {
        return from(this.callApi(Object.assign(Object.assign({}, this.defaultParams), { path: url, httpMethod: 'GET', queryParams })));
    }
    callApi(params) {
        return this.apiService.getInstance()
            .oauth2Auth.callCustomApi(params.path, params.httpMethod, params.pathParams, params.queryParams, params.headerParams, params.formParams, params.bodyParam, params.contentTypes, params.accepts, params.returnType, params.contextRoot, params.responseType);
    }
    get contextRoot() {
        return this.appConfigService.get('bpmHost', '');
    }
}
