// @alfresco/adf-process-services-cloud/esm2015/lib/services/form-fields.interfaces.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export var PeopleModeOptions;
(function (PeopleModeOptions) {
    PeopleModeOptions["single"] = "single";
    PeopleModeOptions["multiple"] = "multiple";
})(PeopleModeOptions || (PeopleModeOptions = {}));
export var FormFieldType;
(function (FormFieldType) {
    FormFieldType["text"] = "text";
    FormFieldType["multiline"] = "multi-line-text";
    FormFieldType["number"] = "integer";
    FormFieldType["checkbox"] = "boolean";
    FormFieldType["date"] = "date";
    FormFieldType["datetime"] = "datetime";
    FormFieldType["dropdown"] = "dropdown";
    FormFieldType["typeahead"] = "typeahead";
    FormFieldType["amount"] = "amount";
    FormFieldType["radio"] = "radio-buttons";
    FormFieldType["people"] = "people";
    FormFieldType["groupOfPeople"] = "functional-group";
    FormFieldType["dynamicTable"] = "dynamicTable";
    FormFieldType["hyperlink"] = "hyperlink";
    FormFieldType["header"] = "group";
    FormFieldType["uploadFile"] = "upload";
    FormFieldType["uploadFolder"] = "uploadFolder";
    FormFieldType["displayValue"] = "readonly";
    FormFieldType["displayText"] = "readonly-text";
    FormFieldType["fileViewer"] = "file-viewer";
})(FormFieldType || (FormFieldType = {}));
