// @alfresco/adf-process-services-cloud/esm2015/lib/form/services/form-definition-selector-cloud.service.js
import { Injectable } from '@angular/core';
import { AlfrescoApiService, AppConfigService } from '@alfresco/adf-core';
import { map } from 'rxjs/operators';
import { from } from 'rxjs';
import { BaseCloudService } from '../../services/base-cloud.service';
import * as i0 from "@angular/core";
import * as i1 from "@alfresco/adf-core";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
export class FormDefinitionSelectorCloudService extends BaseCloudService {
    constructor(apiService, appConfigService) {
        super(apiService, appConfigService);
    }
    getForms(appName) {
        const url = `${this.getBasePath(appName)}/form/v1/forms`;
        return this.get(url).pipe(map((data) => {
            return data.map((formData) => {
                return formData.formRepresentation;
            });
        }));
    }
    getStandAloneTaskForms(appName) {
        return from(this.getForms(appName)).pipe(map((data) => {
            return data.filter((formData) => formData.standalone || formData.standalone === undefined);
        }));
    }
}
FormDefinitionSelectorCloudService.ɵfac = function FormDefinitionSelectorCloudService_Factory(t) { return new (t || FormDefinitionSelectorCloudService)(ɵngcc0.ɵɵinject(ɵngcc1.AlfrescoApiService), ɵngcc0.ɵɵinject(ɵngcc1.AppConfigService)); };
FormDefinitionSelectorCloudService.ɵprov = i0.ɵɵdefineInjectable({ factory: function FormDefinitionSelectorCloudService_Factory() { return new FormDefinitionSelectorCloudService(i0.ɵɵinject(i1.AlfrescoApiService), i0.ɵɵinject(i1.AppConfigService)); }, token: FormDefinitionSelectorCloudService, providedIn: "root" });
FormDefinitionSelectorCloudService.ctorParameters = () => [
    { type: AlfrescoApiService },
    { type: AppConfigService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FormDefinitionSelectorCloudService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.AlfrescoApiService }, { type: ɵngcc1.AppConfigService }]; }, null); })();
