// @alfresco/adf-process-services-cloud/esm2015/lib/form/services/process-cloud-content.service.js
import { __awaiter } from "tslib";
import { Injectable } from '@angular/core';
import { throwError, from } from 'rxjs';
import { catchError, map } from 'rxjs/operators';
import { AlfrescoApiService, LogService, ContentService, DownloadService } from '@alfresco/adf-core';
import { AuthenticationApi, UploadApi } from '@alfresco/js-api';
import * as i0 from "@angular/core";
import * as i1 from "@alfresco/adf-core";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
export class ProcessCloudContentService {
    constructor(apiService, logService, contentService, downloadService) {
        this.apiService = apiService;
        this.logService = logService;
        this.contentService = contentService;
        this.downloadService = downloadService;
    }
    get uploadApi() {
        var _a;
        this._uploadApi = (_a = this._uploadApi) !== null && _a !== void 0 ? _a : new UploadApi(this.apiService.getInstance());
        return this._uploadApi;
    }
    get authenticationApi() {
        var _a;
        this._authenticationApi = (_a = this._authenticationApi) !== null && _a !== void 0 ? _a : new AuthenticationApi(this.apiService.getInstance());
        return this._authenticationApi;
    }
    createTemporaryRawRelatedContent(file, nodeId) {
        return from(this.uploadApi.uploadFile(file, '', nodeId, '', { overwrite: true })).pipe(map((res) => {
            return Object.assign(Object.assign({}, res.entry), { nodeId: res.entry.id });
        }), catchError(err => this.handleError(err)));
    }
    getRawContentNode(nodeId) {
        return this.contentService.getNodeContent(nodeId);
    }
    downloadNodeContent(blob, fileName) {
        this.contentService.downloadBlob(blob, fileName);
    }
    downloadFile(nodeId) {
        return __awaiter(this, void 0, void 0, function* () {
            const ticket = yield this.getAuthTicket();
            const url = this.contentService.getContentUrl(nodeId, true, ticket);
            this.downloadService.downloadUrl(url, nodeId);
        });
    }
    getAuthTicket() {
        return __awaiter(this, void 0, void 0, function* () {
            const ticket = yield this.authenticationApi.getTicket();
            if (ticket && ticket.entry) {
                return ticket.entry.id || '';
            }
            return '';
        });
    }
    handleError(error) {
        this.logService.error(error);
        return throwError(error || 'Server error');
    }
}
ProcessCloudContentService.ɵfac = function ProcessCloudContentService_Factory(t) { return new (t || ProcessCloudContentService)(ɵngcc0.ɵɵinject(ɵngcc1.AlfrescoApiService), ɵngcc0.ɵɵinject(ɵngcc1.LogService), ɵngcc0.ɵɵinject(ɵngcc1.ContentService), ɵngcc0.ɵɵinject(ɵngcc1.DownloadService)); };
ProcessCloudContentService.ɵprov = i0.ɵɵdefineInjectable({ factory: function ProcessCloudContentService_Factory() { return new ProcessCloudContentService(i0.ɵɵinject(i1.AlfrescoApiService), i0.ɵɵinject(i1.LogService), i0.ɵɵinject(i1.ContentService), i0.ɵɵinject(i1.DownloadService)); }, token: ProcessCloudContentService, providedIn: "root" });
ProcessCloudContentService.ctorParameters = () => [
    { type: AlfrescoApiService },
    { type: LogService },
    { type: ContentService },
    { type: DownloadService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ProcessCloudContentService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.AlfrescoApiService }, { type: ɵngcc1.LogService }, { type: ɵngcc1.ContentService }, { type: ɵngcc1.DownloadService }]; }, null); })();
