// @alfresco/adf-core/esm2015/directives/infinite-select-scroll.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Inject, Directive, EventEmitter, Output } from '@angular/core';
import { MatSelect } from '@angular/material/select';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/select';
const SELECT_ITEM_HEIGHT_EM = 3;
export class InfiniteSelectScrollDirective {
    constructor(matSelect) {
        this.matSelect = matSelect;
        this.scrollEnd = new EventEmitter();
        this.onDestroy$ = new Subject();
        this.itemHeightToWaitBeforeLoadNext = 0;
    }
    ngAfterViewInit() {
        this.matSelect.openedChange
            .pipe(takeUntil(this.onDestroy$))
            .subscribe((opened) => {
            if (opened) {
                this.itemHeightToWaitBeforeLoadNext = (this.getItemHeight() * (InfiniteSelectScrollDirective.MAX_ITEMS / 2));
                this.matSelect.panel.nativeElement.addEventListener('scroll', (event) => this.handleScrollEvent(event));
            }
        });
    }
    ngOnDestroy() {
        this.onDestroy$.next(true);
        this.onDestroy$.complete();
    }
    handleScrollEvent(event) {
        if (this.isScrollInNextFetchArea(event)) {
            this.scrollEnd.emit(event);
        }
    }
    isScrollInNextFetchArea(event) {
        const target = event.target;
        return target.scrollTop >= (target.scrollHeight - target.offsetHeight - this.itemHeightToWaitBeforeLoadNext);
    }
    getItemHeight() {
        return parseFloat(getComputedStyle(this.matSelect.panel.nativeElement).fontSize || '0') * SELECT_ITEM_HEIGHT_EM;
    }
}
InfiniteSelectScrollDirective.ɵfac = function InfiniteSelectScrollDirective_Factory(t) { return new (t || InfiniteSelectScrollDirective)(ɵngcc0.ɵɵdirectiveInject(MatSelect)); };
InfiniteSelectScrollDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: InfiniteSelectScrollDirective, selectors: [["", "adf-infinite-select-scroll", ""]], outputs: { scrollEnd: "scrollEnd" } });
InfiniteSelectScrollDirective.MAX_ITEMS = 50;
InfiniteSelectScrollDirective.ctorParameters = () => [
    { type: MatSelect, decorators: [{ type: Inject, args: [MatSelect,] }] }
];
InfiniteSelectScrollDirective.propDecorators = {
    scrollEnd: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InfiniteSelectScrollDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-infinite-select-scroll]'
            }]
    }], function () { return [{ type: ɵngcc1.MatSelect, decorators: [{
                type: Inject,
                args: [MatSelect]
            }] }]; }, { scrollEnd: [{
            type: Output
        }] }); })();
