// @alfresco/adf-core/esm2015/directives/version-compatibility.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, Input, ViewContainerRef, TemplateRef } from '@angular/core';
import { VersionCompatibilityService } from '../services/version-compatibility.service';
import { take } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/version-compatibility.service';
export class VersionCompatibilityDirective {
    constructor(templateRef, viewContainer, versionCompatibilityService) {
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
        this.versionCompatibilityService = versionCompatibilityService;
    }
    set version(requiredVersion) {
        this.validateAcsVersion(requiredVersion);
    }
    validateAcsVersion(requiredVersion) {
        this.versionCompatibilityService.acsVersionInitialized$.pipe(take(1)).subscribe(() => {
            this.viewContainer.clear();
            if (requiredVersion && this.versionCompatibilityService.isVersionSupported(requiredVersion)) {
                this.viewContainer.createEmbeddedView(this.templateRef);
            }
        });
    }
}
VersionCompatibilityDirective.ɵfac = function VersionCompatibilityDirective_Factory(t) { return new (t || VersionCompatibilityDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.VersionCompatibilityService)); };
VersionCompatibilityDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: VersionCompatibilityDirective, selectors: [["", "adf-acs-version", ""]], inputs: { version: ["adf-acs-version", "version"] } });
VersionCompatibilityDirective.ctorParameters = () => [
    { type: TemplateRef },
    { type: ViewContainerRef },
    { type: VersionCompatibilityService }
];
VersionCompatibilityDirective.propDecorators = {
    version: [{ type: Input, args: ['adf-acs-version',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(VersionCompatibilityDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-acs-version]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }, { type: ɵngcc0.ViewContainerRef }, { type: ɵngcc1.VersionCompatibilityService }]; }, { version: [{
            type: Input,
            args: ['adf-acs-version']
        }] }); })();
