// @alfresco/adf-core/esm2015/card-view/validators/card-view-item-minmax.valiator.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CardViewItemIntValidator } from './card-view-item-int.validator';
export class CardViewItemMinMaxValidator {
    constructor(minValue, maxValue) {
        this.minValue = minValue;
        this.maxValue = maxValue;
        this.message = 'CORE.CARDVIEW.VALIDATORS.MINMAX_VALIDATION_ERROR';
        this.intValidator = new CardViewItemIntValidator();
    }
    isValid(value) {
        return this.intValidator.isValid(value) && (value >= this.minValue && value <= this.maxValue);
    }
}
