// @alfresco/adf-core/esm2015/card-view/validators/validators.map.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CardViewItemMatchValidator } from './card-view-item-match.valiator';
import { CardViewItemMinMaxValidator } from './card-view-item-minmax.valiator';
import { CardViewItemLengthValidator } from './card-view-item-length.valiator';
const ɵ0 = (parameters) => new CardViewItemMinMaxValidator(parameters.minValue, parameters.maxValue), ɵ1 = (parameters) => new CardViewItemMatchValidator(parameters.expression), ɵ2 = (parameters) => new CardViewItemLengthValidator(parameters.minLength, parameters.maxLength);
const validators = {
    minmax: ɵ0,
    regex: ɵ1,
    length: ɵ2
};
export default validators;
export { ɵ0, ɵ1, ɵ2 };
