// @alfresco/adf-core/esm2015/card-view/components/card-view-selectitem/card-view-selectitem.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input } from '@angular/core';
import { CardViewUpdateService } from '../../services/card-view-update.service';
import { BehaviorSubject, combineLatest, Observable, Subject } from 'rxjs';
import { BaseCardView } from '../base-card-view';
import { AppConfigService } from '../../../app-config/app-config.service';
import { takeUntil, map } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/card-view-update.service';
import * as ɵngcc2 from '../../../app-config/app-config.service';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@angular/material/form-field';
import * as ɵngcc5 from '@angular/material/select';
import * as ɵngcc6 from './select-filter-input/select-filter-input.component';
import * as ɵngcc7 from '@angular/material/core';
import * as ɵngcc8 from '@ngx-translate/core';

function CardViewSelectItemComponent_ng_container_0_div_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵpipe(3, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵattribute("data-automation-id", "select-readonly-value-" + ctx_r1.property.key);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 2, ɵngcc0.ɵɵpipeBind1(3, 4, ctx_r1.property.displayValue)));
} }
function CardViewSelectItemComponent_ng_container_0_div_6_adf_select_filter_input_3_Template(rf, ctx) { if (rf & 1) {
    const _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "adf-select-filter-input", 9);
    ɵngcc0.ɵɵlistener("change", function CardViewSelectItemComponent_ng_container_0_div_6_adf_select_filter_input_3_Template_adf_select_filter_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r7); const ctx_r6 = ɵngcc0.ɵɵnextContext(3); return ctx_r6.onFilterInputChange($event); });
    ɵngcc0.ɵɵelementEnd();
} }
function CardViewSelectItemComponent_ng_container_0_div_6_mat_option_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-option");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, "CORE.CARDVIEW.NONE"));
} }
function CardViewSelectItemComponent_ng_container_0_div_6_mat_option_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-option", 10);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const option_r8 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("value", option_r8.key);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 2, option_r8.label), " ");
} }
function CardViewSelectItemComponent_ng_container_0_div_6_Template(rf, ctx) { if (rf & 1) {
    const _r10 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelementStart(1, "mat-form-field", 5);
    ɵngcc0.ɵɵelementStart(2, "mat-select", 6);
    ɵngcc0.ɵɵlistener("valueChange", function CardViewSelectItemComponent_ng_container_0_div_6_Template_mat_select_valueChange_2_listener($event) { ɵngcc0.ɵɵrestoreView(_r10); const ctx_r9 = ɵngcc0.ɵɵnextContext(2); return ctx_r9.value = $event; })("selectionChange", function CardViewSelectItemComponent_ng_container_0_div_6_Template_mat_select_selectionChange_2_listener($event) { ɵngcc0.ɵɵrestoreView(_r10); const ctx_r11 = ɵngcc0.ɵɵnextContext(2); return ctx_r11.onChange($event); });
    ɵngcc0.ɵɵtemplate(3, CardViewSelectItemComponent_ng_container_0_div_6_adf_select_filter_input_3_Template, 1, 0, "adf-select-filter-input", 7);
    ɵngcc0.ɵɵtemplate(4, CardViewSelectItemComponent_ng_container_0_div_6_mat_option_4_Template, 3, 3, "mat-option", 0);
    ɵngcc0.ɵɵtemplate(5, CardViewSelectItemComponent_ng_container_0_div_6_mat_option_5_Template, 3, 4, "mat-option", 8);
    ɵngcc0.ɵɵpipe(6, "async");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("value", ctx_r2.value);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r2.showInputFilter);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r2.showNoneOption());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ɵngcc0.ɵɵpipeBind1(6, 4, ctx_r2.list$));
} }
function CardViewSelectItemComponent_ng_container_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "div", 1);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "div", 2);
    ɵngcc0.ɵɵtemplate(5, CardViewSelectItemComponent_ng_container_0_div_5_Template, 4, 6, "div", 3);
    ɵngcc0.ɵɵtemplate(6, CardViewSelectItemComponent_ng_container_0_div_6_Template, 7, 6, "div", 0);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵattribute("data-automation-id", "card-select-label-" + ctx_r0.property.key);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 4, ctx_r0.property.label));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.isEditable());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.isEditable());
} }
export class CardViewSelectItemComponent extends BaseCardView {
    constructor(cardViewUpdateService, appConfig) {
        super(cardViewUpdateService);
        this.appConfig = appConfig;
        this.editable = false;
        this.displayNoneOption = true;
        this.displayEmpty = true;
        this.filter$ = new BehaviorSubject('');
        this.showInputFilter = false;
        this.onDestroy$ = new Subject();
        this.list$ = null;
    }
    ngOnChanges() {
        this.value = this.property.value;
    }
    ngOnInit() {
        this.getOptions()
            .pipe(takeUntil(this.onDestroy$))
            .subscribe((options) => {
            this.showInputFilter = options.length > this.optionsLimit;
        });
        this.list$ = this.getList();
    }
    onFilterInputChange(value) {
        this.filter$.next(value.toString());
    }
    isEditable() {
        return this.editable && this.property.editable;
    }
    getOptions() {
        return this.options$ || this.property.options$;
    }
    getList() {
        return combineLatest([this.getOptions(), this.filter$])
            .pipe(map(([items, filter]) => items.filter((item) => filter ? item.label.toLowerCase().includes(filter.toLowerCase())
            : true)), takeUntil(this.onDestroy$));
    }
    onChange(event) {
        const selectedOption = event.value !== undefined ? event.value : null;
        this.cardViewUpdateService.update(Object.assign({}, this.property), selectedOption);
        this.property.value = selectedOption;
    }
    showNoneOption() {
        return this.displayNoneOption;
    }
    get showProperty() {
        return this.displayEmpty || !this.property.isEmpty();
    }
    ngOnDestroy() {
        this.onDestroy$.next();
        this.onDestroy$.complete();
    }
    get optionsLimit() {
        return this.appConfig.get('content-metadata.selectFilterLimit', CardViewSelectItemComponent.HIDE_FILTER_LIMIT);
    }
}
CardViewSelectItemComponent.ɵfac = function CardViewSelectItemComponent_Factory(t) { return new (t || CardViewSelectItemComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CardViewUpdateService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.AppConfigService)); };
CardViewSelectItemComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CardViewSelectItemComponent, selectors: [["adf-card-view-selectitem"]], inputs: { editable: "editable", displayNoneOption: "displayNoneOption", displayEmpty: "displayEmpty", options$: "options$" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature, ɵngcc0.ɵɵNgOnChangesFeature], decls: 1, vars: 1, consts: [[4, "ngIf"], [1, "adf-property-label"], [1, "adf-property-field"], ["class", "adf-select-item-padding adf-property-value", "data-automation-class", "read-only-value", 4, "ngIf"], ["data-automation-class", "read-only-value", 1, "adf-select-item-padding", "adf-property-value"], [1, "adf-select-item-padding-editable", "adf-property-value"], ["panelClass", "adf-select-filter", "data-automation-class", "select-box", 3, "value", "valueChange", "selectionChange"], [3, "change", 4, "ngIf"], [3, "value", 4, "ngFor", "ngForOf"], [3, "change"], [3, "value"]], template: function CardViewSelectItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, CardViewSelectItemComponent_ng_container_0_Template, 7, 6, "ng-container", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", !ctx.property.isEmpty() || ctx.isEditable());
    } }, directives: [ɵngcc3.NgIf, ɵngcc4.MatFormField, ɵngcc5.MatSelect, ɵngcc3.NgForOf, ɵngcc6.SelectFilterInputComponent, ɵngcc7.MatOption], pipes: [ɵngcc8.TranslatePipe, ɵngcc3.AsyncPipe], styles: [".mat-form-field-type-mat-select[_ngcontent-%COMP%]{width:100%}.adf-select-item-padding[_ngcontent-%COMP%]{padding-bottom:20px;padding-top:6px}.adf-select-item-padding-editable[_ngcontent-%COMP%]{padding-bottom:6px;padding-top:6px}"] });
CardViewSelectItemComponent.HIDE_FILTER_LIMIT = 5;
CardViewSelectItemComponent.ctorParameters = () => [
    { type: CardViewUpdateService },
    { type: AppConfigService }
];
CardViewSelectItemComponent.propDecorators = {
    editable: [{ type: Input }],
    options$: [{ type: Input }],
    displayNoneOption: [{ type: Input }],
    displayEmpty: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CardViewSelectItemComponent, [{
        type: Component,
        args: [{
                selector: 'adf-card-view-selectitem',
                template: "<ng-container *ngIf=\"!property.isEmpty() || isEditable()\">\n    <div [attr.data-automation-id]=\"'card-select-label-' + property.key\"\n        class=\"adf-property-label\">{{ property.label | translate }}</div>\n    <div class=\"adf-property-field\">\n        <div *ngIf=\"!isEditable()\"\n            class=\"adf-select-item-padding adf-property-value\"\n            [attr.data-automation-id]=\"'select-readonly-value-' + property.key\"\n            data-automation-class=\"read-only-value\">{{ (property.displayValue | async) | translate }}</div>\n        <div *ngIf=\"isEditable()\">\n            <mat-form-field class=\"adf-select-item-padding-editable adf-property-value\">\n                <mat-select [(value)]=\"value\"\n                            panelClass=\"adf-select-filter\"\n                            (selectionChange)=\"onChange($event)\"\n                            data-automation-class=\"select-box\">\n\n                    <adf-select-filter-input *ngIf=\"showInputFilter\" (change)=\"onFilterInputChange($event)\"></adf-select-filter-input>\n\n                    <mat-option *ngIf=\"showNoneOption()\">{{ 'CORE.CARDVIEW.NONE' | translate }}</mat-option>\n                    <mat-option *ngFor=\"let option of list$ | async\"\n                                [value]=\"option.key\">\n                        {{ option.label | translate }}\n                    </mat-option>\n                </mat-select>\n            </mat-form-field>\n        </div>\n    </div>\n</ng-container>\n",
                styles: [".mat-form-field-type-mat-select{width:100%}.adf-select-item-padding{padding-bottom:20px;padding-top:6px}.adf-select-item-padding-editable{padding-bottom:6px;padding-top:6px}"]
            }]
    }], function () { return [{ type: ɵngcc1.CardViewUpdateService }, { type: ɵngcc2.AppConfigService }]; }, { editable: [{
            type: Input
        }], displayNoneOption: [{
            type: Input
        }], displayEmpty: [{
            type: Input
        }], options$: [{
            type: Input
        }] }); })();
