// @alfresco/adf-process-services-cloud/esm2015/lib/task/task-list/task-list-cloud.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MaterialModule } from '../../material.module';
import { TaskListCloudComponent } from './components/task-list-cloud.component';
import { ServiceTaskListCloudComponent } from './components/service-task-list-cloud.component';
import { CoreModule } from '@alfresco/adf-core';
import * as ɵngcc0 from '@angular/core';
export class TaskListCloudModule {
}
TaskListCloudModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TaskListCloudModule });
TaskListCloudModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function TaskListCloudModule_Factory(t) { return new (t || TaskListCloudModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            CoreModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TaskListCloudModule, { declarations: function () { return [TaskListCloudComponent,
        ServiceTaskListCloudComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        CoreModule]; }, exports: function () { return [TaskListCloudComponent,
        ServiceTaskListCloudComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TaskListCloudModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    CoreModule
                ],
                declarations: [
                    TaskListCloudComponent,
                    ServiceTaskListCloudComponent
                ],
                exports: [
                    TaskListCloudComponent,
                    ServiceTaskListCloudComponent
                ]
            }]
    }], null, null); })();
