// @alfresco/adf-core/esm2015/form/components/widgets/text/text.widget.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { WidgetComponent } from './../widget.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../../../services/form.service';
import * as ɵngcc2 from '@angular/material/form-field';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@angular/material/input';
import * as ɵngcc5 from '@angular/forms';
import * as ɵngcc6 from './text-mask.component';
import * as ɵngcc7 from '@angular/material/tooltip';
import * as ɵngcc8 from '../error/error.component';
import * as ɵngcc9 from '@ngx-translate/core';

function TextWidgetComponent_span_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1, "*");
    ɵngcc0.ɵɵelementEnd();
} }
function TextWidgetComponent_error_widget_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "error-widget", 5);
    ɵngcc0.ɵɵpipe(1, "translate");
} if (rf & 2) {
    ɵngcc0.ɵɵpropertyInterpolate("required", ɵngcc0.ɵɵpipeBind1(1, 1, "FORM.FIELD.REQUIRED"));
} }
const _c0 = function (a0, a1) { return { mask: a0, isReversed: a1 }; };
export class TextWidgetComponent extends WidgetComponent {
    constructor(formService) {
        super(formService);
        this.formService = formService;
    }
    ngOnInit() {
        if (this.field.params) {
            this.mask = this.field.params['inputMask'];
            this.placeholder = this.field.params['inputMask'] && this.field.params['inputMaskPlaceholder'] ? this.field.params['inputMaskPlaceholder'] : this.field.placeholder;
            this.isMaskReversed = this.field.params['inputMaskReversed'] ? this.field.params['inputMaskReversed'] : false;
        }
    }
}
TextWidgetComponent.ɵfac = function TextWidgetComponent_Factory(t) { return new (t || TextWidgetComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService)); };
TextWidgetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TextWidgetComponent, selectors: [["text-widget"]], hostBindings: function TextWidgetComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function TextWidgetComponent_click_HostBindingHandler($event) { return ctx.event($event); })("blur", function TextWidgetComponent_blur_HostBindingHandler($event) { return ctx.event($event); })("change", function TextWidgetComponent_change_HostBindingHandler($event) { return ctx.event($event); })("focus", function TextWidgetComponent_focus_HostBindingHandler($event) { return ctx.event($event); })("focusin", function TextWidgetComponent_focusin_HostBindingHandler($event) { return ctx.event($event); })("focusout", function TextWidgetComponent_focusout_HostBindingHandler($event) { return ctx.event($event); })("input", function TextWidgetComponent_input_HostBindingHandler($event) { return ctx.event($event); })("invalid", function TextWidgetComponent_invalid_HostBindingHandler($event) { return ctx.event($event); })("select", function TextWidgetComponent_select_HostBindingHandler($event) { return ctx.event($event); });
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 9, vars: 25, consts: [[1, "adf-label"], [4, "ngIf"], ["matInput", "", "type", "text", "matTooltipPosition", "above", "matTooltipShowDelay", "1000", 1, "adf-input", 3, "id", "required", "value", "ngModel", "disabled", "textMask", "placeholder", "matTooltip", "ngModelChange"], [3, "error"], [3, "required", 4, "ngIf"], [3, "required"]], template: function TextWidgetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "mat-form-field");
        ɵngcc0.ɵɵelementStart(2, "label", 0);
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵpipe(4, "translate");
        ɵngcc0.ɵɵtemplate(5, TextWidgetComponent_span_5_Template, 2, 0, "span", 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(6, "input", 2);
        ɵngcc0.ɵɵlistener("ngModelChange", function TextWidgetComponent_Template_input_ngModelChange_6_listener($event) { return ctx.field.value = $event; })("ngModelChange", function TextWidgetComponent_Template_input_ngModelChange_6_listener() { return ctx.onFieldChanged(ctx.field); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(7, "error-widget", 3);
        ɵngcc0.ɵɵtemplate(8, TextWidgetComponent_error_widget_8_Template, 2, 3, "error-widget", 4);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMapInterpolate1("adf-textfield adf-text-widget ", ctx.field.className, "");
        ɵngcc0.ɵɵclassProp("adf-invalid", !ctx.field.isValid)("adf-readonly", ctx.field.readOnly);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵattribute("for", ctx.field.id);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(4, 20, ctx.field.name));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isRequired());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("id", ctx.field.id)("required", ctx.isRequired())("value", ctx.field.value)("ngModel", ctx.field.value)("disabled", ctx.field.readOnly || ctx.readOnly)("textMask", ɵngcc0.ɵɵpureFunction2(22, _c0, ctx.mask, ctx.isMaskReversed))("placeholder", ctx.placeholder)("matTooltip", ctx.field.tooltip);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("error", ctx.field.validationSummary);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isInvalidFieldRequired());
    } }, directives: [ɵngcc2.MatFormField, ɵngcc3.NgIf, ɵngcc4.MatInput, ɵngcc5.DefaultValueAccessor, ɵngcc5.RequiredValidator, ɵngcc5.NgControlStatus, ɵngcc5.NgModel, ɵngcc6.InputMaskDirective, ɵngcc7.MatTooltip, ɵngcc8.ErrorWidgetComponent], pipes: [ɵngcc9.TranslatePipe], styles: [".adf-text-widget{width:100%}.adf-text-widget .mat-form-field-label-wrapper{top:20px}"], encapsulation: 2 });
TextWidgetComponent.ctorParameters = () => [
    { type: FormService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TextWidgetComponent, [{
        type: Component,
        args: [{
                selector: 'text-widget',
                template: "<div class=\"adf-textfield adf-text-widget {{field.className}}\"\n     [class.adf-invalid]=\"!field.isValid\" [class.adf-readonly]=\"field.readOnly\">\n    <mat-form-field>\n        <label class=\"adf-label\" [attr.for]=\"field.id\">{{field.name | translate }}<span *ngIf=\"isRequired()\">*</span></label>\n        <input matInput\n               class=\"adf-input\"\n               type=\"text\"\n               [id]=\"field.id\"\n               [required]=\"isRequired()\"\n               [value]=\"field.value\"\n               [(ngModel)]=\"field.value\"\n               (ngModelChange)=\"onFieldChanged(field)\"\n               [disabled]=\"field.readOnly || readOnly\"\n               [textMask]=\"{mask: mask, isReversed: isMaskReversed}\"\n               [placeholder]=\"placeholder\"\n               [matTooltip]=\"field.tooltip\"\n                matTooltipPosition=\"above\"\n                matTooltipShowDelay=\"1000\">\n    </mat-form-field>\n    <error-widget [error]=\"field.validationSummary\"></error-widget>\n    <error-widget *ngIf=\"isInvalidFieldRequired()\" required=\"{{ 'FORM.FIELD.REQUIRED' | translate }}\"></error-widget>\n</div>\n",
                host: {
                    '(click)': 'event($event)',
                    '(blur)': 'event($event)',
                    '(change)': 'event($event)',
                    '(focus)': 'event($event)',
                    '(focusin)': 'event($event)',
                    '(focusout)': 'event($event)',
                    '(input)': 'event($event)',
                    '(invalid)': 'event($event)',
                    '(select)': 'event($event)'
                },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-text-widget{width:100%}.adf-text-widget .mat-form-field-label-wrapper{top:20px}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormService }]; }, null); })();
