// @alfresco/adf-content-services/esm2015/lib/site-dropdown/sites-dropdown.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { DropdownSitesComponent } from './sites-dropdown.component';
import { CoreModule } from '@alfresco/adf-core';
import * as ɵngcc0 from '@angular/core';
export class SitesDropdownModule {
}
SitesDropdownModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SitesDropdownModule });
SitesDropdownModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SitesDropdownModule_Factory(t) { return new (t || SitesDropdownModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            FormsModule,
            ReactiveFormsModule,
            CoreModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SitesDropdownModule, { declarations: function () { return [DropdownSitesComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        FormsModule,
        ReactiveFormsModule,
        CoreModule]; }, exports: function () { return [DropdownSitesComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SitesDropdownModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    FormsModule,
                    ReactiveFormsModule,
                    CoreModule
                ],
                exports: [
                    DropdownSitesComponent
                ],
                declarations: [
                    DropdownSitesComponent
                ]
            }]
    }], null, null); })();
