// @alfresco/adf-process-services-cloud/esm2015/lib/task/directives/claim-task-cloud.directive.js
import { __awaiter } from "tslib";
import { Directive, Input, HostListener, Output, EventEmitter } from '@angular/core';
import { IdentityUserService } from '@alfresco/adf-core';
import { TaskCloudService } from '../services/task-cloud.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/task-cloud.service';
import * as ɵngcc2 from '@alfresco/adf-core';
export class ClaimTaskCloudDirective {
    constructor(taskListService, identityUserService) {
        this.taskListService = taskListService;
        this.identityUserService = identityUserService;
        this.appName = '';
        this.success = new EventEmitter();
        this.error = new EventEmitter();
        this.invalidParams = [];
    }
    ngOnInit() {
        this.validateInputs();
    }
    validateInputs() {
        if (!this.isTaskValid()) {
            this.invalidParams.push('taskId');
        }
        if (!this.isAppValid()) {
            this.invalidParams.push('appName');
        }
        if (this.invalidParams.length) {
            throw new Error(`Attribute ${this.invalidParams.join(', ')} is required`);
        }
    }
    isTaskValid() {
        return this.taskId && this.taskId.length > 0;
    }
    isAppValid() {
        return !!this.appName || this.appName === '';
    }
    onClick() {
        return __awaiter(this, void 0, void 0, function* () {
            try {
                yield this.claimTask();
            }
            catch (error) {
                this.error.emit(error);
            }
        });
    }
    claimTask() {
        return __awaiter(this, void 0, void 0, function* () {
            const currentUser = this.identityUserService.getCurrentUserInfo().username;
            try {
                const result = yield this.taskListService.claimTask(this.appName, this.taskId, currentUser).toPromise();
                if (result) {
                    this.success.emit(result);
                }
            }
            catch (error) {
                this.error.emit(error);
            }
        });
    }
}
ClaimTaskCloudDirective.ɵfac = function ClaimTaskCloudDirective_Factory(t) { return new (t || ClaimTaskCloudDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TaskCloudService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.IdentityUserService)); };
ClaimTaskCloudDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ClaimTaskCloudDirective, selectors: [["", "adf-cloud-claim-task", ""]], hostBindings: function ClaimTaskCloudDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function ClaimTaskCloudDirective_click_HostBindingHandler() { return ctx.onClick(); });
    } }, inputs: { appName: "appName", taskId: "taskId" }, outputs: { success: "success", error: "error" } });
ClaimTaskCloudDirective.ctorParameters = () => [
    { type: TaskCloudService },
    { type: IdentityUserService }
];
ClaimTaskCloudDirective.propDecorators = {
    taskId: [{ type: Input }],
    appName: [{ type: Input }],
    success: [{ type: Output }],
    error: [{ type: Output }],
    onClick: [{ type: HostListener, args: ['click',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ClaimTaskCloudDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-cloud-claim-task]'
            }]
    }], function () { return [{ type: ɵngcc1.TaskCloudService }, { type: ɵngcc2.IdentityUserService }]; }, { appName: [{
            type: Input
        }], success: [{
            type: Output
        }], error: [{
            type: Output
        }], onClick: [{
            type: HostListener,
            args: ['click']
        }], taskId: [{
            type: Input
        }] }); })();
