// @alfresco/adf-process-services-cloud/esm2015/lib/task/directives/task-directive.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { CompleteTaskDirective } from './complete-task.directive';
import { ClaimTaskCloudDirective } from './claim-task-cloud.directive';
import { UnClaimTaskCloudDirective } from './unclaim-task-cloud.directive';
import * as ɵngcc0 from '@angular/core';
export class TaskDirectiveModule {
}
TaskDirectiveModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TaskDirectiveModule });
TaskDirectiveModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function TaskDirectiveModule_Factory(t) { return new (t || TaskDirectiveModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TaskDirectiveModule, { declarations: function () { return [CompleteTaskDirective,
        ClaimTaskCloudDirective,
        UnClaimTaskCloudDirective]; }, exports: function () { return [CompleteTaskDirective,
        ClaimTaskCloudDirective,
        UnClaimTaskCloudDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TaskDirectiveModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    CompleteTaskDirective,
                    ClaimTaskCloudDirective,
                    UnClaimTaskCloudDirective
                ],
                exports: [
                    CompleteTaskDirective,
                    ClaimTaskCloudDirective,
                    UnClaimTaskCloudDirective
                ]
            }]
    }], null, null); })();
