// @alfresco/adf-process-services-cloud/esm2015/lib/task/directives/unclaim-task-cloud.directive.js
import { __awaiter } from "tslib";
import { Directive, Input, HostListener, Output, EventEmitter } from '@angular/core';
import { TaskCloudService } from '../services/task-cloud.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/task-cloud.service';
export class UnClaimTaskCloudDirective {
    constructor(taskListService) {
        this.taskListService = taskListService;
        this.appName = '';
        this.success = new EventEmitter();
        this.error = new EventEmitter();
        this.invalidParams = [];
    }
    ngOnInit() {
        this.validateInputs();
    }
    validateInputs() {
        if (!this.isTaskValid()) {
            this.invalidParams.push('taskId');
        }
        if (!this.isAppValid()) {
            this.invalidParams.push('appName');
        }
        if (this.invalidParams.length) {
            throw new Error(`Attribute ${this.invalidParams.join(', ')} is required`);
        }
    }
    isTaskValid() {
        return this.taskId && this.taskId.length > 0;
    }
    isAppValid() {
        return !!this.appName || this.appName === '';
    }
    onClick() {
        return __awaiter(this, void 0, void 0, function* () {
            try {
                yield this.taskListService.unclaimTask(this.appName, this.taskId).toPromise();
                this.success.emit(this.taskId);
            }
            catch (error) {
                this.error.emit(error);
            }
        });
    }
}
UnClaimTaskCloudDirective.ɵfac = function UnClaimTaskCloudDirective_Factory(t) { return new (t || UnClaimTaskCloudDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TaskCloudService)); };
UnClaimTaskCloudDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: UnClaimTaskCloudDirective, selectors: [["", "adf-cloud-unclaim-task", ""]], hostBindings: function UnClaimTaskCloudDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function UnClaimTaskCloudDirective_click_HostBindingHandler() { return ctx.onClick(); });
    } }, inputs: { appName: "appName", taskId: "taskId" }, outputs: { success: "success", error: "error" } });
UnClaimTaskCloudDirective.ctorParameters = () => [
    { type: TaskCloudService }
];
UnClaimTaskCloudDirective.propDecorators = {
    taskId: [{ type: Input }],
    appName: [{ type: Input }],
    success: [{ type: Output }],
    error: [{ type: Output }],
    onClick: [{ type: HostListener, args: ['click',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UnClaimTaskCloudDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-cloud-unclaim-task]'
            }]
    }], function () { return [{ type: ɵngcc1.TaskCloudService }]; }, { appName: [{
            type: Input
        }], success: [{
            type: Output
        }], error: [{
            type: Output
        }], onClick: [{
            type: HostListener,
            args: ['click']
        }], taskId: [{
            type: Input
        }] }); })();
