// @alfresco/adf-process-services-cloud/esm2015/lib/form/components/form-definition-selector-cloud.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { FormDefinitionSelectorCloudService } from '../services/form-definition-selector-cloud.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../services/form-definition-selector-cloud.service';
import * as ɵngcc2 from '@angular/material/form-field';
import * as ɵngcc3 from '@angular/material/select';
import * as ɵngcc4 from '@angular/material/core';
import * as ɵngcc5 from '@angular/common';
import * as ɵngcc6 from '@ngx-translate/core';

function FormDefinitionSelectorCloudComponent_mat_option_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-option", 2);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const form_r1 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("value", form_r1.id);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(form_r1.name);
} }
export class FormDefinitionSelectorCloudComponent {
    constructor(formDefinitionCloudService) {
        this.formDefinitionCloudService = formDefinitionCloudService;
        this.appName = '';
        this.selectForm = new EventEmitter();
    }
    ngOnInit() {
        this.forms$ = this.formDefinitionCloudService.getStandAloneTaskForms(this.appName);
    }
    onSelect(event) {
        this.selectForm.emit(event.value);
    }
}
FormDefinitionSelectorCloudComponent.ɵfac = function FormDefinitionSelectorCloudComponent_Factory(t) { return new (t || FormDefinitionSelectorCloudComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormDefinitionSelectorCloudService)); };
FormDefinitionSelectorCloudComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FormDefinitionSelectorCloudComponent, selectors: [["adf-cloud-form-definition-selector"]], inputs: { appName: "appName" }, outputs: { selectForm: "selectForm" }, decls: 10, vars: 10, consts: [[1, "adf-form-definition-selector"], [1, "adf-form-selector-dropdown", 3, "selectionChange"], [3, "value"], [3, "value", 4, "ngFor", "ngForOf"]], template: function FormDefinitionSelectorCloudComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "mat-form-field", 0);
        ɵngcc0.ɵɵelementStart(1, "mat-label");
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵpipe(3, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "mat-select", 1);
        ɵngcc0.ɵɵlistener("selectionChange", function FormDefinitionSelectorCloudComponent_Template_mat_select_selectionChange_4_listener($event) { return ctx.onSelect($event); });
        ɵngcc0.ɵɵelementStart(5, "mat-option", 2);
        ɵngcc0.ɵɵtext(6);
        ɵngcc0.ɵɵpipe(7, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(8, FormDefinitionSelectorCloudComponent_mat_option_8_Template, 2, 2, "mat-option", 3);
        ɵngcc0.ɵɵpipe(9, "async");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 4, "ADF_CLOUD_TASK_LIST.START_TASK.FORM.LABEL.FORM"));
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("value", "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(7, 6, "ADF_CLOUD_TASK_LIST.START_TASK.FORM.LABEL.NONE"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ɵngcc0.ɵɵpipeBind1(9, 8, ctx.forms$));
    } }, directives: [ɵngcc2.MatFormField, ɵngcc2.MatLabel, ɵngcc3.MatSelect, ɵngcc4.MatOption, ɵngcc5.NgForOf], pipes: [ɵngcc6.TranslatePipe, ɵngcc5.AsyncPipe], styles: [".adf-form-definition-selector[_ngcontent-%COMP%]{width:100%}"] });
FormDefinitionSelectorCloudComponent.ctorParameters = () => [
    { type: FormDefinitionSelectorCloudService }
];
FormDefinitionSelectorCloudComponent.propDecorators = {
    appName: [{ type: Input }],
    selectForm: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FormDefinitionSelectorCloudComponent, [{
        type: Component,
        args: [{
                selector: 'adf-cloud-form-definition-selector',
                template: "<mat-form-field class=\"adf-form-definition-selector\">\n    <mat-label>{{'ADF_CLOUD_TASK_LIST.START_TASK.FORM.LABEL.FORM'|translate}}</mat-label>\n    <mat-select class=\"adf-form-selector-dropdown\" (selectionChange)=\"onSelect($event)\">\n        <mat-option [value]=\"''\">{{'ADF_CLOUD_TASK_LIST.START_TASK.FORM.LABEL.NONE'|translate}}</mat-option>\n        <mat-option *ngFor=\"let form of forms$ | async\" [value]=\"form.id\">{{ form.name }}</mat-option>\n    </mat-select>\n</mat-form-field>\n",
                styles: [".adf-form-definition-selector{width:100%}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormDefinitionSelectorCloudService }]; }, { appName: [{
            type: Input
        }], selectForm: [{
            type: Output
        }] }); })();
