// @alfresco/adf-core/esm2015/buttons-menu/buttons-menu.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TranslateModule } from '@ngx-translate/core';
import { MaterialModule } from '../material.module';
import { ButtonsMenuComponent } from './buttons-menu.component';
import { FlexLayoutModule } from '@angular/flex-layout';
import * as ɵngcc0 from '@angular/core';
export class ButtonsMenuModule {
}
ButtonsMenuModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ButtonsMenuModule });
ButtonsMenuModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ButtonsMenuModule_Factory(t) { return new (t || ButtonsMenuModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            TranslateModule,
            FlexLayoutModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ButtonsMenuModule, { declarations: function () { return [ButtonsMenuComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        TranslateModule,
        FlexLayoutModule]; }, exports: function () { return [ButtonsMenuComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ButtonsMenuModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    TranslateModule,
                    FlexLayoutModule
                ],
                declarations: [
                    ButtonsMenuComponent
                ],
                exports: [
                    ButtonsMenuComponent
                ]
            }]
    }], null, null); })();
