// @alfresco/adf-core/esm2015/buttons-menu/buttons-menu.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ContentChildren, QueryList, ViewEncapsulation } from '@angular/core';
import { MatMenuItem } from '@angular/material/menu';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/material/button';
import * as ɵngcc3 from '@angular/material/menu';
import * as ɵngcc4 from '@angular/material/icon';

function ButtonsMenuComponent_div_0_div_1_ng_content_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0, 0, ["*ngTemplateOutlet", "desktop"]);
} }
function ButtonsMenuComponent_div_0_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelementStart(1, "button", 5);
    ɵngcc0.ɵɵelementStart(2, "mat-icon");
    ɵngcc0.ɵɵtext(3, "more_vert");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "mat-menu", 6, 7);
    ɵngcc0.ɵɵtemplate(6, ButtonsMenuComponent_div_0_div_1_ng_content_6_Template, 1, 0, "ng-content", 8);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const _r5 = ɵngcc0.ɵɵreference(5);
    ɵngcc0.ɵɵnextContext(2);
    const _r1 = ɵngcc0.ɵɵreference(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("matMenuTriggerFor", _r5);
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", _r1);
} }
function ButtonsMenuComponent_div_0_div_2_ng_content_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0, 1, ["*ngTemplateOutlet", "desktop"]);
} }
function ButtonsMenuComponent_div_0_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 9);
    ɵngcc0.ɵɵtemplate(1, ButtonsMenuComponent_div_0_div_2_ng_content_1_Template, 1, 0, "ng-content", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext(2);
    const _r1 = ɵngcc0.ɵɵreference(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", _r1);
} }
function ButtonsMenuComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 2);
    ɵngcc0.ɵɵtemplate(1, ButtonsMenuComponent_div_0_div_1_Template, 7, 2, "div", 3);
    ɵngcc0.ɵɵtemplate(2, ButtonsMenuComponent_div_0_div_2_Template, 2, 1, "div", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.isMobile());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.isMobile());
} }
function ButtonsMenuComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0, 2);
} }
const _c0 = ["*", "*", "*"];
export class ButtonsMenuComponent {
    ngAfterContentInit() {
        this.isMenuEmpty = this.buttons.length <= 0;
    }
    isMobile() {
        return !!/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
    }
}
ButtonsMenuComponent.ɵfac = function ButtonsMenuComponent_Factory(t) { return new (t || ButtonsMenuComponent)(); };
ButtonsMenuComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ButtonsMenuComponent, selectors: [["adf-buttons-action-menu"]], contentQueries: function ButtonsMenuComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, MatMenuItem, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.buttons = _t);
    } }, ngContentSelectors: _c0, decls: 3, vars: 1, consts: [["id", "adf-buttons-menu", "class", "adf-buttons-menu", 4, "ngIf"], ["desktop", ""], ["id", "adf-buttons-menu", 1, "adf-buttons-menu"], [4, "ngIf"], ["class", "adf-buttons-menu-desktop", 4, "ngIf"], ["mat-icon-button", "", 3, "matMenuTriggerFor"], [1, "adf-buttons-menu-mobile"], ["editReportMenu", "matMenu"], [4, "ngTemplateOutlet"], [1, "adf-buttons-menu-desktop"]], template: function ButtonsMenuComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵtemplate(0, ButtonsMenuComponent_div_0_Template, 3, 2, "div", 0);
        ɵngcc0.ɵɵtemplate(1, ButtonsMenuComponent_ng_template_1_Template, 1, 0, "ng-template", null, 1, ɵngcc0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", !ctx.isMenuEmpty);
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.MatButton, ɵngcc3.MatMenuTrigger, ɵngcc4.MatIcon, ɵngcc3._MatMenu, ɵngcc1.NgTemplateOutlet], styles: [".adf-buttons-menu{margin-right:10px}.adf-buttons-menu div{display:flex}.adf-buttons-menu-mobile{margin-right:10px}.adf-buttons-menu-desktop{display:flex}.adf-buttons-menu-desktop button{color:#000;padding:0}.adf-buttons-menu-desktop button>span{display:none}.adf-buttons-menu-desktop button>mat-icon.mat-icon.material-icons{color:#000;margin:0 10px}"], encapsulation: 2 });
ButtonsMenuComponent.propDecorators = {
    buttons: [{ type: ContentChildren, args: [MatMenuItem,] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ButtonsMenuComponent, [{
        type: Component,
        args: [{
                selector: 'adf-buttons-action-menu',
                template: "<div id=\"adf-buttons-menu\" class=\"adf-buttons-menu\" *ngIf=\"!isMenuEmpty\">\n    <div *ngIf=\"isMobile()\">\n        <button mat-icon-button [matMenuTriggerFor]=\"editReportMenu\">\n            <mat-icon>more_vert</mat-icon>\n        </button>\n        <mat-menu #editReportMenu=\"matMenu\" class=\"adf-buttons-menu-mobile\">\n            <ng-content *ngTemplateOutlet=\"desktop\">\n            </ng-content>\n        </mat-menu>\n    </div>\n\n    <div *ngIf=\"!isMobile()\" class=\"adf-buttons-menu-desktop\">\n        <ng-content *ngTemplateOutlet=\"desktop\">\n        </ng-content>\n    </div>\n</div>\n\n<ng-template #desktop>\n    <ng-content></ng-content>\n</ng-template>",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-buttons-menu{margin-right:10px}.adf-buttons-menu div{display:flex}.adf-buttons-menu-mobile{margin-right:10px}.adf-buttons-menu-desktop{display:flex}.adf-buttons-menu-desktop button{color:#000;padding:0}.adf-buttons-menu-desktop button>span{display:none}.adf-buttons-menu-desktop button>mat-icon.mat-icon.material-icons{color:#000;margin:0 10px}"]
            }]
    }], null, { buttons: [{
            type: ContentChildren,
            args: [MatMenuItem]
        }] }); })();
