// @alfresco/adf-core/esm2015/dialogs/edit-json/edit-json.dialog.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatDialogModule } from '@angular/material/dialog';
import { TranslateModule } from '@ngx-translate/core';
import { EditJsonDialogComponent } from './edit-json.dialog';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import * as ɵngcc0 from '@angular/core';
export class EditJsonDialogModule {
}
EditJsonDialogModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: EditJsonDialogModule });
EditJsonDialogModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function EditJsonDialogModule_Factory(t) { return new (t || EditJsonDialogModule)(); }, imports: [[
            CommonModule,
            FormsModule,
            ReactiveFormsModule,
            TranslateModule,
            MatDialogModule,
            MatButtonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(EditJsonDialogModule, { declarations: function () { return [EditJsonDialogComponent]; }, imports: function () { return [CommonModule,
        FormsModule,
        ReactiveFormsModule,
        TranslateModule,
        MatDialogModule,
        MatButtonModule]; }, exports: function () { return [EditJsonDialogComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EditJsonDialogModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    FormsModule,
                    ReactiveFormsModule,
                    TranslateModule,
                    MatDialogModule,
                    MatButtonModule
                ],
                declarations: [
                    EditJsonDialogComponent
                ],
                exports: [
                    EditJsonDialogComponent
                ]
            }]
    }], null, null); })();
