// @alfresco/adf-core/esm2015/dialogs/edit-json/edit-json.dialog.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Inject, Input, ViewEncapsulation } from '@angular/core';
import { MAT_DIALOG_DATA } from '@angular/material/dialog';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/dialog';
import * as ɵngcc2 from '@angular/forms';
import * as ɵngcc3 from '@angular/material/button';
import * as ɵngcc4 from '@angular/common';
import * as ɵngcc5 from '@ngx-translate/core';

function EditJsonDialogComponent_button_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "button", 5);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("mat-dialog-close", ctx_r0.value);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 2, "CORE.DIALOG.EDIT_JSON.UPDATE"), " ");
} }
export class EditJsonDialogComponent {
    constructor(settings) {
        this.settings = settings;
        this.editable = false;
        this.title = 'JSON';
        this.value = '';
    }
    ngOnInit() {
        if (this.settings) {
            this.editable = this.settings.editable;
            this.value = this.settings.value || '';
            this.title = this.settings.title || 'JSON';
        }
    }
}
EditJsonDialogComponent.ɵfac = function EditJsonDialogComponent_Factory(t) { return new (t || EditJsonDialogComponent)(ɵngcc0.ɵɵdirectiveInject(MAT_DIALOG_DATA)); };
EditJsonDialogComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: EditJsonDialogComponent, selectors: [["ng-component"]], hostAttrs: [1, "adf-edit-json-dialog"], inputs: { value: "value" }, decls: 10, vars: 9, consts: [["mat-dialog-title", ""], [3, "ngModel", "ngModelChange"], ["align", "end"], ["mat-button", "", "mat-dialog-close", "", "cdkFocusInitial", ""], ["mat-button", "", 3, "mat-dialog-close", 4, "ngIf"], ["mat-button", "", 3, "mat-dialog-close"]], template: function EditJsonDialogComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "h1", 0);
        ɵngcc0.ɵɵtext(1);
        ɵngcc0.ɵɵpipe(2, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "mat-dialog-content");
        ɵngcc0.ɵɵelementStart(4, "textarea", 1);
        ɵngcc0.ɵɵlistener("ngModelChange", function EditJsonDialogComponent_Template_textarea_ngModelChange_4_listener($event) { return ctx.value = $event; });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "mat-dialog-actions", 2);
        ɵngcc0.ɵɵelementStart(6, "button", 3);
        ɵngcc0.ɵɵtext(7);
        ɵngcc0.ɵɵpipe(8, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(9, EditJsonDialogComponent_button_9_Template, 3, 4, "button", 4);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 5, ctx.title));
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngModel", ctx.value);
        ɵngcc0.ɵɵattribute("readonly", !ctx.editable ? true : null);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(8, 7, "CORE.DIALOG.EDIT_JSON.CLOSE"), " ");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.editable);
    } }, directives: [ɵngcc1.MatDialogTitle, ɵngcc1.MatDialogContent, ɵngcc2.DefaultValueAccessor, ɵngcc2.NgControlStatus, ɵngcc2.NgModel, ɵngcc1.MatDialogActions, ɵngcc3.MatButton, ɵngcc1.MatDialogClose, ɵngcc4.NgIf], pipes: [ɵngcc5.TranslatePipe], styles: [".adf-edit-json-dialog .mat-dialog-content{height:300px;overflow:hidden}.adf-edit-json-dialog textarea{-moz-box-sizing:border-box;-webkit-box-sizing:border-box;box-sizing:border-box;height:100%;margin:0;padding:0;resize:none;width:100%}.adf-edit-json-dialog textarea:focus{outline:none}"], encapsulation: 2 });
EditJsonDialogComponent.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] }] }
];
EditJsonDialogComponent.propDecorators = {
    value: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EditJsonDialogComponent, [{
        type: Component,
        args: [{
                template: "<h1 mat-dialog-title>{{ title | translate }}</h1>\n<mat-dialog-content>\n    <textarea [(ngModel)]=\"value\" [attr.readonly]=\"!editable ? true : null\"></textarea>\n</mat-dialog-content>\n\n<mat-dialog-actions align=\"end\">\n    <button mat-button mat-dialog-close cdkFocusInitial>\n        {{ 'CORE.DIALOG.EDIT_JSON.CLOSE' | translate }}\n    </button>\n    <button *ngIf=\"editable\" mat-button [mat-dialog-close]=\"value\">\n        {{ 'CORE.DIALOG.EDIT_JSON.UPDATE' | translate }}\n    </button>\n</mat-dialog-actions>\n",
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-edit-json-dialog' },
                styles: [".adf-edit-json-dialog .mat-dialog-content{height:300px;overflow:hidden}.adf-edit-json-dialog textarea{-moz-box-sizing:border-box;-webkit-box-sizing:border-box;box-sizing:border-box;height:100%;margin:0;padding:0;resize:none;width:100%}.adf-edit-json-dialog textarea:focus{outline:none}"]
            }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [MAT_DIALOG_DATA]
            }] }]; }, { value: [{
            type: Input
        }] }); })();
