// @alfresco/adf-content-services/esm2015/lib/search/services/search-filter.service.js
import { Injectable } from '@angular/core';
import { SearchTextComponent } from '../components/search-text/search-text.component';
import { SearchRadioComponent } from '../components/search-radio/search-radio.component';
import { SearchSliderComponent } from '../components/search-slider/search-slider.component';
import { SearchNumberRangeComponent } from '../components/search-number-range/search-number-range.component';
import { SearchCheckListComponent } from '../components/search-check-list/search-check-list.component';
import { SearchDateRangeComponent } from '../components/search-date-range/search-date-range.component';
import { SearchDatetimeRangeComponent } from '../components/search-datetime-range/search-datetime-range.component';
import * as i0 from "@angular/core";
import * as ɵngcc0 from '@angular/core';
export class SearchFilterService {
    constructor() {
        this.widgets = {
            'text': SearchTextComponent,
            'radio': SearchRadioComponent,
            'slider': SearchSliderComponent,
            'number-range': SearchNumberRangeComponent,
            'check-list': SearchCheckListComponent,
            'date-range': SearchDateRangeComponent,
            'datetime-range': SearchDatetimeRangeComponent
        };
    }
}
SearchFilterService.ɵfac = function SearchFilterService_Factory(t) { return new (t || SearchFilterService)(); };
SearchFilterService.ɵprov = i0.ɵɵdefineInjectable({ factory: function SearchFilterService_Factory() { return new SearchFilterService(); }, token: SearchFilterService, providedIn: "root" });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SearchFilterService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();
