// @alfresco/adf-core/esm2015/datatable/components/datatable-row/datatable-row.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, ElementRef, Input, HostBinding, HostListener, Output, EventEmitter } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class DataTableRowComponent {
    constructor(element) {
        this.element = element;
        this.disabled = false;
        this.select = new EventEmitter();
    }
    get isSelected() {
        if (!this.row) {
            return false;
        }
        return this.row.isSelected;
    }
    get isAriaSelected() {
        if (!this.row) {
            return false;
        }
        return this.row.isSelected;
    }
    get ariaLabel() {
        if (!this.row) {
            return null;
        }
        if (this.row.isSelected) {
            return this.row.getValue('name') + ' selected' || '';
        }
        else {
            return this.row.getValue('name') || '';
        }
    }
    get tabindex() {
        return this.disabled ? null : 0;
    }
    onKeyDown(event) {
        if (event.target.tagName === this.element.nativeElement.tagName) {
            event.preventDefault();
            this.select.emit(event);
        }
    }
    focus() {
        this.element.nativeElement.focus();
    }
}
DataTableRowComponent.ɵfac = function DataTableRowComponent_Factory(t) { return new (t || DataTableRowComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
DataTableRowComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataTableRowComponent, selectors: [["adf-datatable-row"]], hostAttrs: ["tabindex", "0", "role", "row", 1, "adf-datatable-row"], hostVars: 5, hostBindings: function DataTableRowComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("keydown.space", function DataTableRowComponent_keydown_space_HostBindingHandler($event) { return ctx.onKeyDown($event); });
    } if (rf & 2) {
        ɵngcc0.ɵɵattribute("aria-selected", ctx.isAriaSelected)("aria-label", ctx.ariaLabel)("tabindex", ctx.tabindex);
        ɵngcc0.ɵɵclassProp("adf-is-selected", ctx.isSelected);
    } }, inputs: { disabled: "disabled", row: "row" }, outputs: { select: "select" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function DataTableRowComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
DataTableRowComponent.ctorParameters = () => [
    { type: ElementRef }
];
DataTableRowComponent.propDecorators = {
    row: [{ type: Input }],
    disabled: [{ type: Input }],
    select: [{ type: Output }],
    isSelected: [{ type: HostBinding, args: ['class.adf-is-selected',] }],
    isAriaSelected: [{ type: HostBinding, args: ['attr.aria-selected',] }],
    ariaLabel: [{ type: HostBinding, args: ['attr.aria-label',] }],
    tabindex: [{ type: HostBinding, args: ['attr.tabindex',] }],
    onKeyDown: [{ type: HostListener, args: ['keydown.space', ['$event'],] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataTableRowComponent, [{
        type: Component,
        args: [{
                selector: 'adf-datatable-row',
                template: `<ng-content></ng-content>`,
                encapsulation: ViewEncapsulation.None,
                host: {
                    class: 'adf-datatable-row',
                    tabindex: '0',
                    role: 'row'
                }
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { disabled: [{
            type: Input
        }], select: [{
            type: Output
        }], isSelected: [{
            type: HostBinding,
            args: ['class.adf-is-selected']
        }], isAriaSelected: [{
            type: HostBinding,
            args: ['attr.aria-selected']
        }], ariaLabel: [{
            type: HostBinding,
            args: ['attr.aria-label']
        }], tabindex: [{
            type: HostBinding,
            args: ['attr.tabindex']
        }], onKeyDown: [{
            type: HostListener,
            args: ['keydown.space', ['$event']]
        }], row: [{
            type: Input
        }] }); })();
