// @alfresco/adf-process-services/esm2015/lib/task-list/components/task-form/unclaim-task.directive.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { __awaiter } from "tslib";
import { Directive, HostListener, Input, Output, EventEmitter } from '@angular/core';
import { TaskListService } from '../../services/tasklist.service';
import { LogService } from '@alfresco/adf-core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/tasklist.service';
import * as ɵngcc2 from '@alfresco/adf-core';
export class UnclaimTaskDirective {
    constructor(taskListService, logService) {
        this.taskListService = taskListService;
        this.logService = logService;
        this.success = new EventEmitter();
        this.error = new EventEmitter();
        this.invalidParams = [];
    }
    ngOnInit() {
        this.validateInputs();
    }
    validateInputs() {
        if (!this.isTaskValid()) {
            this.invalidParams.push('taskId');
        }
        if (this.invalidParams.length) {
            throw new Error(`Attribute ${this.invalidParams.join(', ')} is required`);
        }
    }
    isTaskValid() {
        return this.taskId && this.taskId.length > 0;
    }
    onClick() {
        return __awaiter(this, void 0, void 0, function* () {
            try {
                yield this.unclaimTask();
            }
            catch (error) {
                this.error.emit(error);
            }
        });
    }
    unclaimTask() {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.taskListService.unclaimTask(this.taskId).subscribe(() => {
                this.logService.info('Task unclaimed');
                this.success.emit(this.taskId);
            }, error => this.error.emit(error));
        });
    }
}
UnclaimTaskDirective.ɵfac = function UnclaimTaskDirective_Factory(t) { return new (t || UnclaimTaskDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TaskListService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.LogService)); };
UnclaimTaskDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: UnclaimTaskDirective, selectors: [["", "adf-unclaim-task", ""]], hostBindings: function UnclaimTaskDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function UnclaimTaskDirective_click_HostBindingHandler() { return ctx.onClick(); });
    } }, inputs: { taskId: "taskId" }, outputs: { success: "success", error: "error" } });
UnclaimTaskDirective.ctorParameters = () => [
    { type: TaskListService },
    { type: LogService }
];
UnclaimTaskDirective.propDecorators = {
    taskId: [{ type: Input }],
    success: [{ type: Output }],
    error: [{ type: Output }],
    onClick: [{ type: HostListener, args: ['click',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UnclaimTaskDirective, [{
        type: Directive,
        args: [{
                selector: '[adf-unclaim-task]'
            }]
    }], function () { return [{ type: ɵngcc1.TaskListService }, { type: ɵngcc2.LogService }]; }, { success: [{
            type: Output
        }], error: [{
            type: Output
        }], onClick: [{
            type: HostListener,
            args: ['click']
        }], taskId: [{
            type: Input
        }] }); })();
