// @alfresco/adf-content-services/esm2015/lib/version-manager/content-version.service.js
import { Injectable } from '@angular/core';
import { AlfrescoApiService } from '@alfresco/adf-core';
import { ContentApi } from '@alfresco/js-api';
import { of } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@alfresco/adf-core";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
export class ContentVersionService {
    constructor(alfrescoApi) {
        this.alfrescoApi = alfrescoApi;
    }
    get contentApi() {
        if (!this._contentApi) {
            this._contentApi = new ContentApi(this.alfrescoApi.getInstance());
        }
        return this._contentApi;
    }
    getVersionContentUrl(nodeId, versionId, attachment) {
        let nodeDownloadUrl = this.contentApi.getContentUrl(nodeId, attachment);
        nodeDownloadUrl = nodeDownloadUrl.replace('/content', '/versions/' + versionId + '/content');
        return of(nodeDownloadUrl);
    }
}
ContentVersionService.ɵfac = function ContentVersionService_Factory(t) { return new (t || ContentVersionService)(ɵngcc0.ɵɵinject(ɵngcc1.AlfrescoApiService)); };
ContentVersionService.ɵprov = i0.ɵɵdefineInjectable({ factory: function ContentVersionService_Factory() { return new ContentVersionService(i0.ɵɵinject(i1.AlfrescoApiService)); }, token: ContentVersionService, providedIn: "root" });
ContentVersionService.ctorParameters = () => [
    { type: AlfrescoApiService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContentVersionService, [{
        type: Injectable,
        args: [{ providedIn: 'root' }]
    }], function () { return [{ type: ɵngcc1.AlfrescoApiService }]; }, null); })();
