// @alfresco/adf-content-services/esm2015/lib/version-manager/version-manager.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { VersionUploadComponent } from './version-upload.component';
import { VersionManagerComponent } from './version-manager.component';
import { VersionListComponent } from './version-list.component';
import { UploadModule } from '../upload/upload.module';
import { CoreModule } from '@alfresco/adf-core';
import { VersionComparisonComponent } from './version-comparison.component';
import * as ɵngcc0 from '@angular/core';
export class VersionManagerModule {
}
VersionManagerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: VersionManagerModule });
VersionManagerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function VersionManagerModule_Factory(t) { return new (t || VersionManagerModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            CoreModule,
            UploadModule,
            FormsModule
        ], FormsModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(VersionManagerModule, { declarations: function () { return [VersionUploadComponent,
        VersionManagerComponent,
        VersionListComponent,
        VersionComparisonComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        CoreModule,
        UploadModule,
        FormsModule]; }, exports: function () { return [VersionUploadComponent,
        VersionManagerComponent,
        VersionListComponent,
        FormsModule,
        VersionComparisonComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(VersionManagerModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    CoreModule,
                    UploadModule,
                    FormsModule
                ],
                exports: [
                    VersionUploadComponent,
                    VersionManagerComponent,
                    VersionListComponent,
                    FormsModule,
                    VersionComparisonComponent
                ],
                declarations: [
                    VersionUploadComponent,
                    VersionManagerComponent,
                    VersionListComponent,
                    VersionComparisonComponent
                ]
            }]
    }], null, null); })();
