// @alfresco/adf-content-services/esm2015/lib/webscript/webscript.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CoreModule } from '@alfresco/adf-core';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { WebscriptComponent } from './webscript.component';
import * as ɵngcc0 from '@angular/core';
export class WebScriptModule {
}
WebScriptModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WebScriptModule });
WebScriptModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WebScriptModule_Factory(t) { return new (t || WebScriptModule)(); }, imports: [[
            CommonModule,
            MaterialModule,
            CoreModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WebScriptModule, { declarations: function () { return [WebscriptComponent]; }, imports: function () { return [CommonModule,
        MaterialModule,
        CoreModule]; }, exports: function () { return [WebscriptComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WebScriptModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    MaterialModule,
                    CoreModule
                ],
                exports: [
                    WebscriptComponent
                ],
                declarations: [
                    WebscriptComponent
                ]
            }]
    }], null, null); })();
