// @alfresco/adf-process-services/esm2015/lib/attachment/attachment.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { MaterialModule } from '../material.module';
import { TaskAttachmentListComponent } from './task-attachment-list.component';
import { ProcessAttachmentListComponent } from './process-attachment-list.component';
import { CreateProcessAttachmentComponent } from './create-process-attachment.component';
import { AttachmentComponent } from './create-task-attachment.component';
import { CoreModule } from '@alfresco/adf-core';
import * as ɵngcc0 from '@angular/core';
export class AttachmentModule {
}
AttachmentModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AttachmentModule });
AttachmentModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AttachmentModule_Factory(t) { return new (t || AttachmentModule)(); }, imports: [[
            CoreModule,
            MaterialModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AttachmentModule, { declarations: function () { return [TaskAttachmentListComponent,
        ProcessAttachmentListComponent,
        CreateProcessAttachmentComponent,
        CreateProcessAttachmentComponent,
        AttachmentComponent]; }, imports: function () { return [CoreModule,
        MaterialModule]; }, exports: function () { return [TaskAttachmentListComponent,
        ProcessAttachmentListComponent,
        CreateProcessAttachmentComponent,
        CreateProcessAttachmentComponent,
        AttachmentComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AttachmentModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CoreModule,
                    MaterialModule
                ],
                declarations: [
                    TaskAttachmentListComponent,
                    ProcessAttachmentListComponent,
                    CreateProcessAttachmentComponent,
                    CreateProcessAttachmentComponent,
                    AttachmentComponent
                ],
                exports: [
                    TaskAttachmentListComponent,
                    ProcessAttachmentListComponent,
                    CreateProcessAttachmentComponent,
                    CreateProcessAttachmentComponent,
                    AttachmentComponent
                ]
            }]
    }], null, null); })();
