// @alfresco/adf-process-services/esm2015/lib/attachment/create-task-attachment.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { ProcessContentService } from '@alfresco/adf-core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-core';
import * as ɵngcc2 from '@angular/material/button';
import * as ɵngcc3 from '@angular/material/icon';

const _c0 = function () { return ["click"]; };
export class AttachmentComponent {
    constructor(activitiContentService) {
        this.activitiContentService = activitiContentService;
        this.error = new EventEmitter();
        this.success = new EventEmitter();
    }
    ngOnChanges(changes) {
        if (changes['taskId'] && changes['taskId'].currentValue) {
            this.taskId = changes['taskId'].currentValue;
        }
    }
    onFileUpload(event) {
        const filesList = event.detail.files.map((obj) => obj.file);
        for (const fileInfoObj of filesList) {
            const file = fileInfoObj;
            const opts = {
                isRelatedContent: true
            };
            this.activitiContentService.createTaskRelatedContent(this.taskId, file, opts).subscribe((res) => {
                this.success.emit(res);
            }, (err) => {
                this.error.emit(err);
            });
        }
    }
}
AttachmentComponent.ɵfac = function AttachmentComponent_Factory(t) { return new (t || AttachmentComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ProcessContentService)); };
AttachmentComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AttachmentComponent, selectors: [["adf-create-task-attachment"]], inputs: { taskId: "taskId" }, outputs: { error: "error", success: "success" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 3, vars: 4, consts: [["color", "primary", "mat-button", "", "mat-raised-button", "", "mat-icon-button", "", 1, "adf-create-attachment", 3, "adf-upload", "mode", "multiple", "upload-files"]], template: function AttachmentComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "button", 0);
        ɵngcc0.ɵɵlistener("upload-files", function AttachmentComponent_Template_button_upload_files_0_listener($event) { return ctx.onFileUpload($event); });
        ɵngcc0.ɵɵelementStart(1, "mat-icon");
        ɵngcc0.ɵɵtext(2, "add");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("adf-upload", true)("mode", ɵngcc0.ɵɵpureFunction0(3, _c0))("multiple", true);
    } }, directives: [ɵngcc2.MatButton, ɵngcc1.UploadDirective, ɵngcc3.MatIcon], styles: [".adf-create-attachment[_ngcontent-%COMP%]{display:inline-block;line-height:0;vertical-align:middle}"] });
AttachmentComponent.ctorParameters = () => [
    { type: ProcessContentService }
];
AttachmentComponent.propDecorators = {
    taskId: [{ type: Input }],
    error: [{ type: Output }],
    success: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AttachmentComponent, [{
        type: Component,
        args: [{
                selector: 'adf-create-task-attachment',
                template: "<button\n    color=\"primary\"\n    mat-button\n    mat-raised-button\n    mat-icon-button\n    class=\"adf-create-attachment\"\n    [adf-upload]=\"true\"\n    [mode]=\"['click']\"\n    [multiple]=\"true\"\n    (upload-files)=\"onFileUpload($event)\">\n    <mat-icon>add</mat-icon>\n</button>\n",
                styles: [".adf-create-attachment{display:inline-block;line-height:0;vertical-align:middle}"]
            }]
    }], function () { return [{ type: ɵngcc1.ProcessContentService }]; }, { error: [{
            type: Output
        }], success: [{
            type: Output
        }], taskId: [{
            type: Input
        }] }); })();
