// @alfresco/adf-core/esm2015/blank-page/blank-page.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class BlankPageComponent {
}
BlankPageComponent.ɵfac = function BlankPageComponent_Factory(t) { return new (t || BlankPageComponent)(); };
BlankPageComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: BlankPageComponent, selectors: [["adf-blank-page"]], decls: 0, vars: 0, template: function BlankPageComponent_Template(rf, ctx) { }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BlankPageComponent, [{
        type: Component,
        args: [{
                selector: 'adf-blank-page',
                template: ``
            }]
    }], null, null); })();
