// @alfresco/adf-core/esm2015/blank-page/blank-page.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { BlankPageComponent } from './blank-page.component';
import * as ɵngcc0 from '@angular/core';
export class BlankPageModule {
}
BlankPageModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BlankPageModule });
BlankPageModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function BlankPageModule_Factory(t) { return new (t || BlankPageModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BlankPageModule, { declarations: function () { return [BlankPageComponent]; }, exports: function () { return [BlankPageComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BlankPageModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    BlankPageComponent
                ],
                exports: [
                    BlankPageComponent
                ]
            }]
    }], null, null); })();
