// @alfresco/adf-core/esm2015/form/components/widgets/container/container.widget.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { WidgetComponent } from './../widget.component';
import { ContainerWidgetComponentModel } from './container.widget.model';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../../../services/form.service';

function ContainerWidgetComponent_button_2_Template(rf, ctx) { if (rf & 1) {
    const _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 6);
    ɵngcc0.ɵɵlistener("click", function ContainerWidgetComponent_button_2_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r5); const ctx_r4 = ɵngcc0.ɵɵnextContext(); return ctx_r4.onExpanderClicked(); });
    ɵngcc0.ɵɵelementStart(1, "mat-icon");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate((ctx_r0.content == null ? null : ctx_r0.content.isExpanded) ? "expand_more" : "expand_less");
} }
function ContainerWidgetComponent_div_6_div_1_div_1_adf_form_field_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-form-field", 12);
} if (rf & 2) {
    const field_r8 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("field", field_r8);
} }
const _c0 = function (a0) { return { "grid-area": a0 }; };
function ContainerWidgetComponent_div_6_div_1_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 10);
    ɵngcc0.ɵɵtemplate(1, ContainerWidgetComponent_div_6_div_1_div_1_adf_form_field_1_Template, 1, 1, "adf-form-field", 11);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const field_r8 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(2, _c0, "auto / auto / span " + ((field_r8 == null ? null : field_r8.rowspan) || 1) + " / span " + ((field_r8 == null ? null : field_r8.colspan) || 1)));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", field_r8);
} }
const _c1 = function (a0) { return { "grid-template-columns": a0 }; };
function ContainerWidgetComponent_div_6_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 8);
    ɵngcc0.ɵɵtemplate(1, ContainerWidgetComponent_div_6_div_1_div_1_Template, 2, 4, "div", 9);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r6 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(2, _c1, "repeat(" + ctx_r6.numberOfColumns + ", 1fr)"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r6.fields);
} }
function ContainerWidgetComponent_div_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵtemplate(1, ContainerWidgetComponent_div_6_div_1_Template, 2, 4, "div", 7);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.content == null ? null : ctx_r1.content.isExpanded);
} }
function ContainerWidgetComponent_ng_template_7_section_0_div_1_div_1_adf_form_field_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "adf-form-field", 12);
} if (rf & 2) {
    const field_r15 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("field", field_r15);
} }
function ContainerWidgetComponent_ng_template_7_section_0_div_1_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 18);
    ɵngcc0.ɵɵtemplate(1, ContainerWidgetComponent_ng_template_7_section_0_div_1_div_1_adf_form_field_1_Template, 1, 1, "adf-form-field", 11);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const field_r15 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", field_r15);
} }
function ContainerWidgetComponent_ng_template_7_section_0_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 16);
    ɵngcc0.ɵɵtemplate(1, ContainerWidgetComponent_ng_template_7_section_0_div_1_div_1_Template, 2, 1, "div", 17);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r13 = ctx.$implicit;
    const ctx_r12 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵstyleProp("width", ctx_r12.getColumnWith(ctx_r12.field), "%");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", column_r13 == null ? null : column_r13.fields);
} }
function ContainerWidgetComponent_ng_template_7_section_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "section", 14);
    ɵngcc0.ɵɵtemplate(1, ContainerWidgetComponent_ng_template_7_section_0_div_1_Template, 2, 3, "div", 15);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r11 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r11.content == null ? null : ctx_r11.content.columns);
} }
function ContainerWidgetComponent_ng_template_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, ContainerWidgetComponent_ng_template_7_section_0_Template, 2, 1, "section", 13);
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngIf", ctx_r3.content == null ? null : ctx_r3.content.isExpanded);
} }
export class ContainerWidgetComponent extends WidgetComponent {
    constructor(formService) {
        super(formService);
        this.formService = formService;
    }
    onExpanderClicked() {
        if (this.content && this.content.isCollapsible()) {
            this.content.isExpanded = !this.content.isExpanded;
        }
    }
    ngOnInit() {
        if (this.field) {
            this.content = new ContainerWidgetComponentModel(this.field);
            this.getNumberOfColumnsFromTheBiggestBetweenJsonAndColumnsLengthOrOne();
            this.fields = this.getFields();
        }
    }
    getNumberOfColumnsFromTheBiggestBetweenJsonAndColumnsLengthOrOne() {
        var _a, _b, _c, _d;
        this.numberOfColumns = (((_a = this.content.json) === null || _a === void 0 ? void 0 : _a.numberOfColumns) || 1) > (((_b = this.content.columns) === null || _b === void 0 ? void 0 : _b.length) || 1) ?
            (((_c = this.content.json) === null || _c === void 0 ? void 0 : _c.numberOfColumns) || 1) :
            (((_d = this.content.columns) === null || _d === void 0 ? void 0 : _d.length) || 1);
    }
    getFields() {
        var _a;
        const serialisedFormFields = [];
        const maxColumnFieldsSize = this.getMaxColumnFieldSize();
        for (let rowIndex = 0; rowIndex < maxColumnFieldsSize; rowIndex++) {
            (_a = this.content) === null || _a === void 0 ? void 0 : _a.columns.flatMap((currentColumn) => {
                var _a;
                if (!!(currentColumn === null || currentColumn === void 0 ? void 0 : currentColumn.fields[rowIndex])) {
                    serialisedFormFields.push(currentColumn === null || currentColumn === void 0 ? void 0 : currentColumn.fields[rowIndex]);
                }
                else {
                    const firstRowElementColSpan = (_a = currentColumn === null || currentColumn === void 0 ? void 0 : currentColumn.fields[0]) === null || _a === void 0 ? void 0 : _a.colspan;
                    if (!!firstRowElementColSpan && rowIndex > 0) {
                        for (let i = 0; i < firstRowElementColSpan; i++) {
                            serialisedFormFields.push(null);
                        }
                    }
                }
            });
        }
        return serialisedFormFields;
    }
    getMaxColumnFieldSize() {
        var _a, _b, _c, _d, _e, _f;
        let maxFieldSize = 0;
        if (((_b = (_a = this.content) === null || _a === void 0 ? void 0 : _a.columns) === null || _b === void 0 ? void 0 : _b.length) > 0) {
            maxFieldSize = (_f = (_e = (_d = (_c = this.content) === null || _c === void 0 ? void 0 : _c.columns) === null || _d === void 0 ? void 0 : _d.reduce((prevColumn, currentColumn) => {
                var _a;
                return currentColumn.fields.length > ((_a = prevColumn === null || prevColumn === void 0 ? void 0 : prevColumn.fields) === null || _a === void 0 ? void 0 : _a.length) ? currentColumn : prevColumn;
            })) === null || _e === void 0 ? void 0 : _e.fields) === null || _f === void 0 ? void 0 : _f.length;
        }
        return maxFieldSize;
    }
    getColumnWith(field) {
        const colspan = field ? field.colspan : 1;
        return (100 / this.content.json.numberOfColumns) * colspan + '';
    }
}
ContainerWidgetComponent.ɵfac = function ContainerWidgetComponent_Factory(t) { return new (t || ContainerWidgetComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService)); };
ContainerWidgetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ContainerWidgetComponent, selectors: [["container-widget"]], hostBindings: function ContainerWidgetComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function ContainerWidgetComponent_click_HostBindingHandler($event) { return ctx.event($event); })("blur", function ContainerWidgetComponent_blur_HostBindingHandler($event) { return ctx.event($event); })("change", function ContainerWidgetComponent_change_HostBindingHandler($event) { return ctx.event($event); })("focus", function ContainerWidgetComponent_focus_HostBindingHandler($event) { return ctx.event($event); })("focusin", function ContainerWidgetComponent_focusin_HostBindingHandler($event) { return ctx.event($event); })("focusout", function ContainerWidgetComponent_focusout_HostBindingHandler($event) { return ctx.event($event); })("input", function ContainerWidgetComponent_input_HostBindingHandler($event) { return ctx.event($event); })("invalid", function ContainerWidgetComponent_invalid_HostBindingHandler($event) { return ctx.event($event); })("select", function ContainerWidgetComponent_select_HostBindingHandler($event) { return ctx.event($event); });
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 9, vars: 9, consts: [[1, "adf-container-widget__header", 3, "hidden"], ["id", "container-header", 1, "adf-container-widget__header-text"], ["mat-icon-button", "", "class", "mdl-button--icon", 3, "click", 4, "ngIf"], ["id", "container-header-label", 3, "click"], [4, "ngIf", "ngIfElse"], ["fixingTemplate", ""], ["mat-icon-button", "", 1, "mdl-button--icon", 3, "click"], ["class", "adf-grid-list", 3, "ngStyle", 4, "ngIf"], [1, "adf-grid-list", 3, "ngStyle"], ["class", "adf-grid-list-item", 3, "ngStyle", 4, "ngFor", "ngForOf"], [1, "adf-grid-list-item", 3, "ngStyle"], [3, "field", 4, "ngIf"], [3, "field"], ["class", "adf-grid-list-column-view", 4, "ngIf"], [1, "adf-grid-list-column-view"], ["class", "adf-grid-list-single-column", 3, "width", 4, "ngFor", "ngForOf"], [1, "adf-grid-list-single-column"], ["class", "adf-grid-list-column-view-item", 4, "ngFor", "ngForOf"], [1, "adf-grid-list-column-view-item"]], template: function ContainerWidgetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "h4", 1);
        ɵngcc0.ɵɵtemplate(2, ContainerWidgetComponent_button_2_Template, 3, 1, "button", 2);
        ɵngcc0.ɵɵelementStart(3, "span", 3);
        ɵngcc0.ɵɵlistener("click", function ContainerWidgetComponent_Template_span_click_3_listener() { return ctx.onExpanderClicked(); });
        ɵngcc0.ɵɵtext(4);
        ɵngcc0.ɵɵpipe(5, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(6, ContainerWidgetComponent_div_6_Template, 2, 1, "div", 4);
        ɵngcc0.ɵɵtemplate(7, ContainerWidgetComponent_ng_template_7_Template, 1, 1, "ng-template", null, 5, ɵngcc0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        const _r2 = ɵngcc0.ɵɵreference(8);
        ɵngcc0.ɵɵproperty("hidden", !(ctx.content == null ? null : ctx.content.isGroup()));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassProp("adf-collapsible", ctx.content == null ? null : ctx.content.isCollapsible());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.content == null ? null : ctx.content.isCollapsible());
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(5, 7, ctx.content.name));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.field == null ? null : ctx.field.form == null ? null : ctx.field.form.enableFixedSpace)("ngIfElse", _r2);
    } }, styles: [".adf-hidden{display:none}.adf-container-widget__header-text{-moz-user-select:none;-ms-user-select:none;-webkit-touch-callout:none;-webkit-user-select:none;border-bottom:1px solid rgba(0,0,0,.87);cursor:default;padding-bottom:10px;user-select:none}.adf-container-widget__header-text.adf-collapsible{cursor:pointer}.adf-field-list{height:100%;list-style-type:none;padding:0;width:100%}container-widget .adf-grid-list-column-view{align-items:flex-start;display:flex;margin-right:-1%}container-widget .adf-grid-list-single-column{align-items:flex-start;display:flex;flex:1 1 auto;flex-direction:column}container-widget .adf-grid-list-column-view-item{box-sizing:border-box;flex-grow:1;padding-left:1%;padding-right:1%;width:100%}container-widget .adf-grid-list{display:grid}container-widget .adf-grid-list-item{box-sizing:border-box;padding-left:3px;padding-right:3px}@media screen and (max-width:959px){container-widget .adf-grid-list-item{flex:1 0 100%}container-widget .adf-grid-list--column-view{flex-direction:column}container-widget .adf-grid-list-single-column{width:90%!important}container-widget .adf-grid-list-column-view-item{flex:1 0 auto}}container-widget .mat-form-field,container-widget mat-form-field{width:100%}container-widget .mat-input-placeholder{top:1.8em!important}container-widget .mat-focused label{background-color:.3s cubic-bezier(.55,0,.55,.2);color:var(--theme-primary-color);transform:scaleX(1);transition:transform .15s linear}container-widget .mat-focused .mat-form-field-prefix{color:var(--theme-primary-color)}container-widget .mat-grid-tile{overflow:visible;width:80%}container-widget adf-form-field{width:100%}"], encapsulation: 2 });
ContainerWidgetComponent.ctorParameters = () => [
    { type: FormService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContainerWidgetComponent, [{
        type: Component,
        args: [{
                selector: 'container-widget',
                template: "<div [hidden]=\"!content?.isGroup()\" class=\"adf-container-widget__header\">\n    <h4 class=\"adf-container-widget__header-text\" id=\"container-header\"\n        [class.adf-collapsible]=\"content?.isCollapsible()\">\n        <button *ngIf=\"content?.isCollapsible()\"\n                mat-icon-button\n                class=\"mdl-button--icon\"\n                (click)=\"onExpanderClicked()\">\n            <mat-icon>{{ content?.isExpanded ? 'expand_more' : 'expand_less' }}</mat-icon>\n        </button>\n        <span (click)=\"onExpanderClicked()\" id=\"container-header-label\">{{content.name | translate }}</span>\n    </h4>\n</div>\n\n<div *ngIf=\"field?.form?.enableFixedSpace else fixingTemplate\">\n    <div class=\"adf-grid-list\" [ngStyle]=\"{ 'grid-template-columns': 'repeat('+numberOfColumns+', 1fr)'}\"\n        *ngIf=\"content?.isExpanded\">\n        <div class=\"adf-grid-list-item\" *ngFor=\"let field of fields\"\n            [ngStyle]=\"{'grid-area': 'auto / auto / span '+(field?.rowspan || 1)+' / span '+(field?.colspan || 1)}\">\n            <adf-form-field *ngIf=\"field\" [field]=\"field\"></adf-form-field>\n        </div>\n    </div>\n</div>\n\n<ng-template #fixingTemplate>\n    <section class=\"adf-grid-list-column-view\" *ngIf=\"content?.isExpanded\">\n        <div class=\"adf-grid-list-single-column\" *ngFor=\"let column of content?.columns\" [style.width.%]=\"getColumnWith(field)\">\n            <div class=\"adf-grid-list-column-view-item\" *ngFor=\"let field of column?.fields\">\n                <adf-form-field *ngIf=\"field\" [field]=\"field\"></adf-form-field>\n            </div>\n        </div>\n    </section>\n</ng-template>\n",
                host: {
                    '(click)': 'event($event)',
                    '(blur)': 'event($event)',
                    '(change)': 'event($event)',
                    '(focus)': 'event($event)',
                    '(focusin)': 'event($event)',
                    '(focusout)': 'event($event)',
                    '(input)': 'event($event)',
                    '(invalid)': 'event($event)',
                    '(select)': 'event($event)'
                },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-hidden{display:none}.adf-container-widget__header-text{-moz-user-select:none;-ms-user-select:none;-webkit-touch-callout:none;-webkit-user-select:none;border-bottom:1px solid rgba(0,0,0,.87);cursor:default;padding-bottom:10px;user-select:none}.adf-container-widget__header-text.adf-collapsible{cursor:pointer}.adf-field-list{height:100%;list-style-type:none;padding:0;width:100%}container-widget .adf-grid-list-column-view{align-items:flex-start;display:flex;margin-right:-1%}container-widget .adf-grid-list-single-column{align-items:flex-start;display:flex;flex:1 1 auto;flex-direction:column}container-widget .adf-grid-list-column-view-item{box-sizing:border-box;flex-grow:1;padding-left:1%;padding-right:1%;width:100%}container-widget .adf-grid-list{display:grid}container-widget .adf-grid-list-item{box-sizing:border-box;padding-left:3px;padding-right:3px}@media screen and (max-width:959px){container-widget .adf-grid-list-item{flex:1 0 100%}container-widget .adf-grid-list--column-view{flex-direction:column}container-widget .adf-grid-list-single-column{width:90%!important}container-widget .adf-grid-list-column-view-item{flex:1 0 auto}}container-widget .mat-form-field,container-widget mat-form-field{width:100%}container-widget .mat-input-placeholder{top:1.8em!important}container-widget .mat-focused label{background-color:.3s cubic-bezier(.55,0,.55,.2);color:var(--theme-primary-color);transform:scaleX(1);transition:transform .15s linear}container-widget .mat-focused .mat-form-field-prefix{color:var(--theme-primary-color)}container-widget .mat-grid-tile{overflow:visible;width:80%}container-widget adf-form-field{width:100%}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormService }]; }, null); })();
