// @alfresco/adf-core/esm2015/form/components/widgets/dynamic-table/editors/row.editor.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { DynamicRowValidationSummary } from './../dynamic-row-validation-summary.model';
import { DynamicTableModel } from './../dynamic-table.widget.model';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../../error/error.component';
import * as ɵngcc3 from '@angular/material/button';
import * as ɵngcc4 from './dropdown/dropdown.editor';
import * as ɵngcc5 from './date/date.editor';
import * as ɵngcc6 from './datetime/datetime.editor';
import * as ɵngcc7 from './boolean/boolean.editor';
import * as ɵngcc8 from './amount/amount.editor';
import * as ɵngcc9 from './text/text.editor';

function RowEditorComponent_div_1_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelement(1, "adf-dropdown-editor", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("table", ctx_r2.table)("row", ctx_r2.row)("column", column_r1);
} }
function RowEditorComponent_div_1_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelement(1, "adf-date-editor", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("table", ctx_r3.table)("row", ctx_r3.row)("column", column_r1);
} }
function RowEditorComponent_div_1_div_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelement(1, "adf-datetime-editor", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("table", ctx_r4.table)("row", ctx_r4.row)("column", column_r1);
} }
function RowEditorComponent_div_1_div_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelement(1, "adf-boolean-editor", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("table", ctx_r5.table)("row", ctx_r5.row)("column", column_r1);
} }
function RowEditorComponent_div_1_div_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelement(1, "adf-amount-editor", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r6 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("table", ctx_r6.table)("row", ctx_r6.row)("column", column_r1);
} }
function RowEditorComponent_div_1_div_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelement(1, "adf-text-editor", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r7 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("table", ctx_r7.table)("row", ctx_r7.row)("column", column_r1);
} }
function RowEditorComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵelementStart(1, "div", 5);
    ɵngcc0.ɵɵtemplate(2, RowEditorComponent_div_1_div_2_Template, 2, 3, "div", 6);
    ɵngcc0.ɵɵtemplate(3, RowEditorComponent_div_1_div_3_Template, 2, 3, "div", 6);
    ɵngcc0.ɵɵtemplate(4, RowEditorComponent_div_1_div_4_Template, 2, 3, "div", 6);
    ɵngcc0.ɵɵtemplate(5, RowEditorComponent_div_1_div_5_Template, 2, 3, "div", 6);
    ɵngcc0.ɵɵtemplate(6, RowEditorComponent_div_1_div_6_Template, 2, 3, "div", 6);
    ɵngcc0.ɵɵtemplate(7, RowEditorComponent_div_1_div_7_Template, 2, 3, "div", 7);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r1 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitch", column_r1.type);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitchCase", "Dropdown");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitchCase", "Date");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitchCase", "Datetime");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitchCase", "Boolean");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitchCase", "Amount");
} }
export class RowEditorComponent {
    constructor() {
        this.save = new EventEmitter();
        this.cancel = new EventEmitter();
        this.validationSummary = new DynamicRowValidationSummary({ isValid: true, message: '' });
    }
    onCancelChanges() {
        this.cancel.emit({
            table: this.table,
            row: this.row,
            column: this.column
        });
    }
    onSaveChanges() {
        this.validate();
        if (this.isValid()) {
            this.save.emit({
                table: this.table,
                row: this.row,
                column: this.column
            });
        }
    }
    isValid() {
        return this.validationSummary && this.validationSummary.isValid;
    }
    validate() {
        this.validationSummary = this.table.validateRow(this.row);
    }
}
RowEditorComponent.ɵfac = function RowEditorComponent_Factory(t) { return new (t || RowEditorComponent)(); };
RowEditorComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: RowEditorComponent, selectors: [["row-editor"]], inputs: { table: "table", row: "row", column: "column" }, outputs: { save: "save", cancel: "cancel" }, decls: 8, vars: 4, consts: [[1, "row-editor", "mdl-shadow--2dp"], ["class", "mdl-grid", 4, "ngFor", "ngForOf"], [3, "error"], ["mat-button", "", 3, "click"], [1, "mdl-grid"], [1, "mdl-cell", "mdl-cell--6-col", 3, "ngSwitch"], [4, "ngSwitchCase"], [4, "ngSwitchDefault"], [3, "table", "row", "column"]], template: function RowEditorComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, RowEditorComponent_div_1_Template, 8, 6, "div", 1);
        ɵngcc0.ɵɵelement(2, "error-widget", 2);
        ɵngcc0.ɵɵelementStart(3, "div");
        ɵngcc0.ɵɵelementStart(4, "button", 3);
        ɵngcc0.ɵɵlistener("click", function RowEditorComponent_Template_button_click_4_listener() { return ctx.onCancelChanges(); });
        ɵngcc0.ɵɵtext(5, "Cancel");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(6, "button", 3);
        ɵngcc0.ɵɵlistener("click", function RowEditorComponent_Template_button_click_6_listener() { return ctx.onSaveChanges(); });
        ɵngcc0.ɵɵtext(7, "Save");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("row-editor__invalid", !ctx.validationSummary.isValid);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.table.columns);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("error", ctx.validationSummary);
    } }, directives: [ɵngcc1.NgForOf, ɵngcc2.ErrorWidgetComponent, ɵngcc3.MatButton, ɵngcc1.NgSwitch, ɵngcc1.NgSwitchCase, ɵngcc1.NgSwitchDefault, ɵngcc4.DropdownEditorComponent, ɵngcc5.DateEditorComponent, ɵngcc6.DateTimeEditorComponent, ɵngcc7.BooleanEditorComponent, ɵngcc8.AmountEditorComponent, ɵngcc9.TextEditorComponent], styles: [".row-editor[_ngcontent-%COMP%]{padding:8px}.row-editor__validation-summary[_ngcontent-%COMP%]{visibility:hidden}.row-editor__invalid[_ngcontent-%COMP%]   .row-editor__validation-summary[_ngcontent-%COMP%]{color:#d50000;padding:8px 16px;visibility:visible}"] });
RowEditorComponent.ctorParameters = () => [];
RowEditorComponent.propDecorators = {
    table: [{ type: Input }],
    row: [{ type: Input }],
    column: [{ type: Input }],
    save: [{ type: Output }],
    cancel: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RowEditorComponent, [{
        type: Component,
        args: [{
                selector: 'row-editor',
                template: "<div class=\"row-editor mdl-shadow--2dp\"\n     [class.row-editor__invalid]=\"!validationSummary.isValid\">\n    <div class=\"mdl-grid\" *ngFor=\"let column of table.columns\">\n        <div class=\"mdl-cell mdl-cell--6-col\" [ngSwitch]=\"column.type\">\n            <div *ngSwitchCase=\"'Dropdown'\">\n                <adf-dropdown-editor\n                    [table]=\"table\"\n                    [row]=\"row\"\n                    [column]=\"column\">\n                </adf-dropdown-editor>\n            </div>\n            <div *ngSwitchCase=\"'Date'\">\n                <adf-date-editor\n                    [table]=\"table\"\n                    [row]=\"row\"\n                    [column]=\"column\">\n                </adf-date-editor>\n            </div>\n            <div *ngSwitchCase=\"'Datetime'\">\n                <adf-datetime-editor\n                    [table]=\"table\"\n                    [row]=\"row\"\n                    [column]=\"column\">\n                </adf-datetime-editor>\n            </div>\n            <div *ngSwitchCase=\"'Boolean'\">\n                <adf-boolean-editor\n                    [table]=\"table\"\n                    [row]=\"row\"\n                    [column]=\"column\">\n                </adf-boolean-editor>\n            </div>\n            <div *ngSwitchCase=\"'Amount'\">\n                <adf-amount-editor\n                    [table]=\"table\"\n                    [row]=\"row\"\n                    [column]=\"column\">\n                </adf-amount-editor>\n            </div>\n            <div *ngSwitchDefault>\n                <adf-text-editor\n                    [table]=\"table\"\n                    [row]=\"row\"\n                    [column]=\"column\">\n                </adf-text-editor>\n            </div>\n        </div>\n    </div>\n    <error-widget [error]=\"validationSummary\"></error-widget>\n    <div>\n        <button mat-button (click)=\"onCancelChanges()\">Cancel</button>\n        <button mat-button (click)=\"onSaveChanges()\">Save</button>\n    </div>\n</div>\n",
                styles: [".row-editor{padding:8px}.row-editor__validation-summary{visibility:hidden}.row-editor__invalid .row-editor__validation-summary{color:#d50000;padding:8px 16px;visibility:visible}"]
            }]
    }], function () { return []; }, { save: [{
            type: Output
        }], cancel: [{
            type: Output
        }], table: [{
            type: Input
        }], row: [{
            type: Input
        }], column: [{
            type: Input
        }] }); })();
