// @alfresco/adf-core/esm2015/form/components/widgets/tabs/tabs.widget.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

function TabsWidgetComponent_div_0_mat_tab_2_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelement(1, "adf-form-field", 5);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const field_r4 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("field", field_r4.field);
} }
function TabsWidgetComponent_div_0_mat_tab_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-tab", 3);
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵtemplate(2, TabsWidgetComponent_div_0_mat_tab_2_div_2_Template, 2, 1, "div", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const tab_r2 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("label", ɵngcc0.ɵɵpipeBind1(1, 2, tab_r2.title));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngForOf", tab_r2.fields);
} }
function TabsWidgetComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵelementStart(1, "mat-tab-group");
    ɵngcc0.ɵɵtemplate(2, TabsWidgetComponent_div_0_mat_tab_2_Template, 3, 4, "mat-tab", 2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.visibleTabs);
} }
export class TabsWidgetComponent {
    constructor() {
        this.tabs = [];
        this.formTabChanged = new EventEmitter();
        this.visibleTabs = [];
    }
    hasTabs() {
        return this.tabs && this.tabs.length > 0;
    }
    ngAfterContentChecked() {
        this.filterVisibleTabs();
    }
    filterVisibleTabs() {
        this.visibleTabs = this.tabs.filter((tab) => {
            return tab.isVisible;
        });
    }
    tabChanged(field) {
        this.formTabChanged.emit(field);
    }
}
TabsWidgetComponent.ɵfac = function TabsWidgetComponent_Factory(t) { return new (t || TabsWidgetComponent)(); };
TabsWidgetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TabsWidgetComponent, selectors: [["tabs-widget"]], inputs: { tabs: "tabs" }, outputs: { formTabChanged: "formTabChanged" }, decls: 1, vars: 1, consts: [["class", "alfresco-tabs-widget", 4, "ngIf"], [1, "alfresco-tabs-widget"], [3, "label", 4, "ngFor", "ngForOf"], [3, "label"], [4, "ngFor", "ngForOf"], [3, "field"]], template: function TabsWidgetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, TabsWidgetComponent_div_0_Template, 3, 1, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasTabs());
    } }, encapsulation: 2 });
TabsWidgetComponent.propDecorators = {
    tabs: [{ type: Input }],
    formTabChanged: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TabsWidgetComponent, [{
        type: Component,
        args: [{
                selector: 'tabs-widget',
                template: "<div *ngIf=\"hasTabs()\" class=\"alfresco-tabs-widget\">\n    <mat-tab-group>\n        <mat-tab *ngFor=\"let tab of visibleTabs\" [label]=\"tab.title | translate\">\n            <div *ngFor=\"let field of tab.fields\">\n                <adf-form-field [field]=\"field.field\"></adf-form-field>\n             </div>\n        </mat-tab>\n    </mat-tab-group>\n</div>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return []; }, { tabs: [{
            type: Input
        }], formTabChanged: [{
            type: Output
        }] }); })();
