// @alfresco/adf-core/esm2015/datatable/components/empty-list/empty-list.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Directive, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = [[["", "adf-empty-list-header", ""]], [["", "adf-empty-list-body", ""]], [["", "adf-empty-list-footer", ""]], "*"];
const _c1 = ["[adf-empty-list-header]", "[adf-empty-list-body]", "[adf-empty-list-footer]", "*"];
export class EmptyListComponent {
}
EmptyListComponent.ɵfac = function EmptyListComponent_Factory(t) { return new (t || EmptyListComponent)(); };
EmptyListComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: EmptyListComponent, selectors: [["adf-empty-list"]], ngContentSelectors: _c1, decls: 5, vars: 0, consts: [[1, "adf-empty-list_template"]], template: function EmptyListComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵprojection(2, 1);
        ɵngcc0.ɵɵprojection(3, 2);
        ɵngcc0.ɵɵprojection(4, 3);
        ɵngcc0.ɵɵelementEnd();
    } }, styles: [".adf-empty-list_template{margin-bottom:20px;margin-top:20px;text-align:center}"], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EmptyListComponent, [{
        type: Component,
        args: [{
                selector: 'adf-empty-list',
                template: "<div class=\"adf-empty-list_template\">\n    <ng-content select=\"[adf-empty-list-header]\"></ng-content>\n    <ng-content select=\"[adf-empty-list-body]\"></ng-content>\n    <ng-content select=\"[adf-empty-list-footer]\"></ng-content>\n    <ng-content></ng-content>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-empty-list_template{margin-bottom:20px;margin-top:20px;text-align:center}"]
            }]
    }], null, null); })();
export class EmptyListHeaderDirective {
}
EmptyListHeaderDirective.ɵfac = function EmptyListHeaderDirective_Factory(t) { return new (t || EmptyListHeaderDirective)(); };
EmptyListHeaderDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: EmptyListHeaderDirective, selectors: [["", "adf-empty-list-header", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EmptyListHeaderDirective, [{
        type: Directive,
        args: [{ selector: '[adf-empty-list-header]' }]
    }], null, null); })();
export class EmptyListBodyDirective {
}
EmptyListBodyDirective.ɵfac = function EmptyListBodyDirective_Factory(t) { return new (t || EmptyListBodyDirective)(); };
EmptyListBodyDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: EmptyListBodyDirective, selectors: [["", "adf-empty-list-body", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EmptyListBodyDirective, [{
        type: Directive,
        args: [{ selector: '[adf-empty-list-body]' }]
    }], null, null); })();
export class EmptyListFooterDirective {
}
EmptyListFooterDirective.ɵfac = function EmptyListFooterDirective_Factory(t) { return new (t || EmptyListFooterDirective)(); };
EmptyListFooterDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: EmptyListFooterDirective, selectors: [["", "adf-empty-list-footer", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EmptyListFooterDirective, [{
        type: Directive,
        args: [{ selector: '[adf-empty-list-footer]' }]
    }], null, null); })();
