// @alfresco/adf-core/esm2015/search-text/search-text-input.module.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MaterialModule } from '../material.module';
import { FormsModule } from '@angular/forms';
import { SearchTextInputComponent } from './search-text-input.component';
import { TranslateModule } from '@ngx-translate/core';
import { SearchTriggerDirective } from './search-trigger.directive';
import * as ɵngcc0 from '@angular/core';
export class SearchTextModule {
}
SearchTextModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SearchTextModule });
SearchTextModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SearchTextModule_Factory(t) { return new (t || SearchTextModule)(); }, imports: [[
            CommonModule,
            TranslateModule,
            MaterialModule,
            FormsModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SearchTextModule, { declarations: function () { return [SearchTextInputComponent,
        SearchTriggerDirective]; }, imports: function () { return [CommonModule,
        TranslateModule,
        MaterialModule,
        FormsModule]; }, exports: function () { return [SearchTextInputComponent,
        SearchTriggerDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SearchTextModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    SearchTextInputComponent,
                    SearchTriggerDirective
                ],
                imports: [
                    CommonModule,
                    TranslateModule,
                    MaterialModule,
                    FormsModule
                ],
                exports: [
                    SearchTextInputComponent,
                    SearchTriggerDirective
                ]
            }]
    }], null, null); })();
