// @alfresco/adf-core/esm2015/about/about-application-modules/about-application-modules.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
import { ObjectDataTableAdapter } from '../../datatable/data/object-datatable-adapter';
import { AppExtensionService } from '@alfresco/adf-extensions';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@alfresco/adf-extensions';
import * as ɵngcc2 from '../../datatable/components/datatable/datatable.component';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@angular/material/table';
import * as ɵngcc5 from '@ngx-translate/core';

function AboutApplicationModulesComponent_div_8_mat_header_cell_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-header-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, "ABOUT.EXTENSIONS.TABLE_HEADERS.ID"));
} }
function AboutApplicationModulesComponent_div_8_mat_cell_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r17 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(element_r17.$id);
} }
function AboutApplicationModulesComponent_div_8_mat_header_cell_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-header-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, "ABOUT.EXTENSIONS.TABLE_HEADERS.NAME"));
} }
function AboutApplicationModulesComponent_div_8_mat_cell_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r18 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(element_r18.$name);
} }
function AboutApplicationModulesComponent_div_8_mat_header_cell_12_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-header-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, "ABOUT.EXTENSIONS.TABLE_HEADERS.VERSION"));
} }
function AboutApplicationModulesComponent_div_8_mat_cell_13_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r19 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(element_r19.$version);
} }
function AboutApplicationModulesComponent_div_8_mat_header_cell_15_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-header-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, "ABOUT.EXTENSIONS.TABLE_HEADERS.VENDOR"));
} }
function AboutApplicationModulesComponent_div_8_mat_cell_16_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r20 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(element_r20.$vendor);
} }
function AboutApplicationModulesComponent_div_8_mat_header_cell_18_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-header-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, "ABOUT.EXTENSIONS.TABLE_HEADERS.LICENSE"));
} }
function AboutApplicationModulesComponent_div_8_mat_cell_19_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r21 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(element_r21.$license);
} }
function AboutApplicationModulesComponent_div_8_mat_header_cell_21_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-header-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, "ABOUT.EXTENSIONS.TABLE_HEADERS.RUNTIME"));
} }
function AboutApplicationModulesComponent_div_8_mat_cell_22_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r22 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(element_r22.$runtime);
} }
function AboutApplicationModulesComponent_div_8_mat_header_cell_24_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-header-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, "ABOUT.EXTENSIONS.TABLE_HEADERS.DESCRIPTION"));
} }
function AboutApplicationModulesComponent_div_8_mat_cell_25_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-cell");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const element_r23 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(element_r23.$description);
} }
function AboutApplicationModulesComponent_div_8_mat_header_row_26_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "mat-header-row");
} }
function AboutApplicationModulesComponent_div_8_mat_row_27_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "mat-row");
} }
function AboutApplicationModulesComponent_div_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵelementStart(1, "h3");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "mat-table", 5);
    ɵngcc0.ɵɵelementContainerStart(5, 6);
    ɵngcc0.ɵɵtemplate(6, AboutApplicationModulesComponent_div_8_mat_header_cell_6_Template, 3, 3, "mat-header-cell", 7);
    ɵngcc0.ɵɵtemplate(7, AboutApplicationModulesComponent_div_8_mat_cell_7_Template, 2, 1, "mat-cell", 8);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵelementContainerStart(8, 9);
    ɵngcc0.ɵɵtemplate(9, AboutApplicationModulesComponent_div_8_mat_header_cell_9_Template, 3, 3, "mat-header-cell", 7);
    ɵngcc0.ɵɵtemplate(10, AboutApplicationModulesComponent_div_8_mat_cell_10_Template, 2, 1, "mat-cell", 8);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵelementContainerStart(11, 10);
    ɵngcc0.ɵɵtemplate(12, AboutApplicationModulesComponent_div_8_mat_header_cell_12_Template, 3, 3, "mat-header-cell", 7);
    ɵngcc0.ɵɵtemplate(13, AboutApplicationModulesComponent_div_8_mat_cell_13_Template, 2, 1, "mat-cell", 8);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵelementContainerStart(14, 11);
    ɵngcc0.ɵɵtemplate(15, AboutApplicationModulesComponent_div_8_mat_header_cell_15_Template, 3, 3, "mat-header-cell", 7);
    ɵngcc0.ɵɵtemplate(16, AboutApplicationModulesComponent_div_8_mat_cell_16_Template, 2, 1, "mat-cell", 8);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵelementContainerStart(17, 12);
    ɵngcc0.ɵɵtemplate(18, AboutApplicationModulesComponent_div_8_mat_header_cell_18_Template, 3, 3, "mat-header-cell", 7);
    ɵngcc0.ɵɵtemplate(19, AboutApplicationModulesComponent_div_8_mat_cell_19_Template, 2, 1, "mat-cell", 8);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵelementContainerStart(20, 13);
    ɵngcc0.ɵɵtemplate(21, AboutApplicationModulesComponent_div_8_mat_header_cell_21_Template, 3, 3, "mat-header-cell", 7);
    ɵngcc0.ɵɵtemplate(22, AboutApplicationModulesComponent_div_8_mat_cell_22_Template, 2, 1, "mat-cell", 8);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵelementContainerStart(23, 14);
    ɵngcc0.ɵɵtemplate(24, AboutApplicationModulesComponent_div_8_mat_header_cell_24_Template, 3, 3, "mat-header-cell", 7);
    ɵngcc0.ɵɵtemplate(25, AboutApplicationModulesComponent_div_8_mat_cell_25_Template, 2, 1, "mat-cell", 8);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵtemplate(26, AboutApplicationModulesComponent_div_8_mat_header_row_26_Template, 1, 0, "mat-header-row", 15);
    ɵngcc0.ɵɵtemplate(27, AboutApplicationModulesComponent_div_8_mat_row_27_Template, 1, 0, "mat-row", 16);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 4, "ABOUT.EXTENSIONS.TITLE"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("dataSource", ctx_r0.extensions);
    ɵngcc0.ɵɵadvance(22);
    ɵngcc0.ɵɵproperty("matHeaderRowDef", ctx_r0.extensionColumns);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("matRowDefColumns", ctx_r0.extensionColumns);
} }
export class AboutApplicationModulesComponent {
    constructor(appExtensions) {
        this.appExtensions = appExtensions;
        this.extensionColumns = ['$id', '$name', '$version', '$vendor', '$license', '$runtime', '$description'];
        this.showExtensions = true;
        this.regexp = '^(@alfresco)';
        this.onDestroy$ = new Subject();
    }
    ngOnInit() {
        const alfrescoPackages = Object.keys(this.dependencies).filter((val) => {
            return new RegExp(this.regexp).test(val);
        });
        const alfrescoPackagesTableRepresentation = [];
        alfrescoPackages.forEach((val) => {
            alfrescoPackagesTableRepresentation.push({
                name: val,
                version: (this.dependencies[val])
            });
        });
        this.dependencyEntries = new ObjectDataTableAdapter(alfrescoPackagesTableRepresentation, [
            { type: 'text', key: 'name', title: 'Name', sortable: true },
            { type: 'text', key: 'version', title: 'Version', sortable: true }
        ]);
        this.appExtensions.references$
            .pipe(takeUntil(this.onDestroy$))
            .subscribe((extensions) => this.extensions = extensions);
    }
    ngOnDestroy() {
        this.onDestroy$.next(true);
        this.onDestroy$.complete();
    }
}
AboutApplicationModulesComponent.ɵfac = function AboutApplicationModulesComponent_Factory(t) { return new (t || AboutApplicationModulesComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.AppExtensionService)); };
AboutApplicationModulesComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AboutApplicationModulesComponent, selectors: [["adf-about-application-modules"]], inputs: { showExtensions: "showExtensions", regexp: "regexp", dependencies: "dependencies" }, decls: 9, vars: 8, consts: [[1, "adf-about-modules-container"], ["data-automation-id", "adf-about-modules-title"], [3, "data"], ["class", "adf-extension-details-container", 4, "ngIf"], [1, "adf-extension-details-container"], [3, "dataSource"], ["matColumnDef", "$id"], [4, "matHeaderCellDef"], [4, "matCellDef"], ["matColumnDef", "$name"], ["matColumnDef", "$version"], ["matColumnDef", "$vendor"], ["matColumnDef", "$license"], ["matColumnDef", "$runtime"], ["matColumnDef", "$description"], [4, "matHeaderRowDef"], [4, "matRowDef", "matRowDefColumns"]], template: function AboutApplicationModulesComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "h3", 1);
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵpipe(3, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "small");
        ɵngcc0.ɵɵtext(5);
        ɵngcc0.ɵɵpipe(6, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(7, "adf-datatable", 2);
        ɵngcc0.ɵɵtemplate(8, AboutApplicationModulesComponent_div_8_Template, 28, 6, "div", 3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 4, "ABOUT.PACKAGES.TITLE"));
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(6, 6, "ABOUT.PACKAGES.DESCRIPTION"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("data", ctx.dependencyEntries);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showExtensions && ctx.extensions.length);
    } }, directives: [ɵngcc2.DataTableComponent, ɵngcc3.NgIf, ɵngcc4.MatTable, ɵngcc4.MatColumnDef, ɵngcc4.MatHeaderCellDef, ɵngcc4.MatCellDef, ɵngcc4.MatHeaderRowDef, ɵngcc4.MatRowDef, ɵngcc4.MatHeaderCell, ɵngcc4.MatCell, ɵngcc4.MatHeaderRow, ɵngcc4.MatRow], pipes: [ɵngcc5.TranslatePipe], encapsulation: 2 });
AboutApplicationModulesComponent.ctorParameters = () => [
    { type: AppExtensionService }
];
AboutApplicationModulesComponent.propDecorators = {
    showExtensions: [{ type: Input }],
    regexp: [{ type: Input }],
    dependencies: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AboutApplicationModulesComponent, [{
        type: Component,
        args: [{
                selector: 'adf-about-application-modules',
                template: "<div class=\"adf-about-modules-container\">\n    <h3 data-automation-id=\"adf-about-modules-title\">{{ 'ABOUT.PACKAGES.TITLE' | translate }}</h3>\n    <small>{{ 'ABOUT.PACKAGES.DESCRIPTION' | translate }}</small>\n    <adf-datatable [data]=\"dependencyEntries\"></adf-datatable>\n\n    <div class=\"adf-extension-details-container\" *ngIf=\"showExtensions && extensions.length\">\n        <h3>{{ 'ABOUT.EXTENSIONS.TITLE' | translate }}</h3>\n        <mat-table [dataSource]=\"extensions\">\n            <!-- $id Column -->\n            <ng-container matColumnDef=\"$id\">\n                <mat-header-cell\n                    *matHeaderCellDef>{{ 'ABOUT.EXTENSIONS.TABLE_HEADERS.ID' | translate }}</mat-header-cell>\n                <mat-cell *matCellDef=\"let element\">{{element.$id}}</mat-cell>\n            </ng-container>\n\n            <!-- $name Column -->\n            <ng-container matColumnDef=\"$name\">\n                <mat-header-cell\n                    *matHeaderCellDef>{{ 'ABOUT.EXTENSIONS.TABLE_HEADERS.NAME' | translate }}</mat-header-cell>\n                <mat-cell *matCellDef=\"let element\">{{element.$name}}</mat-cell>\n            </ng-container>\n\n            <!-- $version Column -->\n            <ng-container matColumnDef=\"$version\">\n                <mat-header-cell\n                    *matHeaderCellDef>{{ 'ABOUT.EXTENSIONS.TABLE_HEADERS.VERSION' | translate }}</mat-header-cell>\n                <mat-cell *matCellDef=\"let element\">{{element.$version}}</mat-cell>\n            </ng-container>\n\n            <!-- $vendor Column -->\n            <ng-container matColumnDef=\"$vendor\">\n                <mat-header-cell\n                    *matHeaderCellDef>{{ 'ABOUT.EXTENSIONS.TABLE_HEADERS.VENDOR' | translate }}</mat-header-cell>\n                <mat-cell *matCellDef=\"let element\">{{element.$vendor}}</mat-cell>\n            </ng-container>\n\n            <!-- $license Column -->\n            <ng-container matColumnDef=\"$license\">\n                <mat-header-cell\n                    *matHeaderCellDef>{{ 'ABOUT.EXTENSIONS.TABLE_HEADERS.LICENSE' | translate }}</mat-header-cell>\n                <mat-cell *matCellDef=\"let element\">{{element.$license}}</mat-cell>\n            </ng-container>\n\n            <!-- $runtime Column -->\n            <ng-container matColumnDef=\"$runtime\">\n                <mat-header-cell\n                    *matHeaderCellDef>{{ 'ABOUT.EXTENSIONS.TABLE_HEADERS.RUNTIME' | translate }}</mat-header-cell>\n                <mat-cell *matCellDef=\"let element\">{{element.$runtime}}</mat-cell>\n            </ng-container>\n\n            <!-- $description Column -->\n            <ng-container matColumnDef=\"$description\">\n                <mat-header-cell\n                    *matHeaderCellDef>{{ 'ABOUT.EXTENSIONS.TABLE_HEADERS.DESCRIPTION' | translate }}</mat-header-cell>\n                <mat-cell *matCellDef=\"let element\">{{element.$description}}</mat-cell>\n            </ng-container>\n\n            <mat-header-row *matHeaderRowDef=\"extensionColumns\"></mat-header-row>\n            <mat-row *matRowDef=\"let row; columns: extensionColumns;\"></mat-row>\n        </mat-table>\n    </div>\n</div>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.AppExtensionService }]; }, { showExtensions: [{
            type: Input
        }], regexp: [{
            type: Input
        }], dependencies: [{
            type: Input
        }] }); })();
